---
title: "Chapter 8.8: TWAS/PWAS/MWAS (Video Transcript)"
---

# Transcriptome-Wide Association Studies {#sec-video1}

Author: Sasha Gusev (alexander_gusev\@dfci.harvard.edu)

Length: 16:39

I am Sasha Gusev this is my first time at cgsi so thanks everybody for having me and giving me the opportunity to give this tutorial as with the other ones please feel free to interrupt or ask questions throughout and I'll try to sort of break things down in a way that's accessible.

I'm going to be talking about genome-wide Association studies and specifically trying to make sense of genomewide association studies as a way to understand human disease and complex traits and so just to sort of start it at the at a very basic level this is the output of a genomewide association study or GWAS.

The procedure is very straightforward - you collect a lot of genetic data on individuals with the disease and without the disease or with a quantitative trait and then you test each genetic variant (and that's what each of these dots is here) for association with the phenotype. The variants that are significantly associated are above a predefined threshold here, and if they replicate we treat those as genetic variants that are causal for the disease and this is sort of a study design that I think initially almost seemed too simple to work but now over time and with very large sample sizes has produced thousands if not hundreds of thousands of associations for nearly every complex trait that it's been applied to when there were sufficient sample size.

In fact the challenge is now that these Association studies are almost producing too many results and what we would rather have than sort of this figure which is a real plot from geological prostate cancer is something more like this which is a systemic or systematic understanding of the disease of which genes are involved in the disease how they interact what contexts they're relevant in and so forth and so uh whereas initially there is sort of a challenge of just fleshing out this side of the plot getting these associations um I think a key challenge now is in Connect going from this side of the plot over here to an actual understanding of the disease and one of the sort of most basic pieces of getting to that understanding is connecting variants to the genes associated variants to the genes that they likely operate through and then operate on the trait and so we can break it down into this very simple structure we have a variant we want to know its Target Gene and the effect that it has on that disease and so in particular we can break this down even further and first just ask whether we can identify variants that influence the expression of genes in a systematic way and this is something that was observed some time ago is that in fact if you take gene expression and you you basically kind of run a G was but on expression as your outcome gene expression measured in the past through microarrays or now through rna-seq and test variants typically near the gene in CIS with the gene for association with expression across individuals you will find that the expression of many genes is often highly heritable and so there's an estimate here in 2011 that the CIS Locus for an average Gene contributed to between 37 and 24 of the variance of expression and again once you have a heritable phenotype in a population you can sort of apply the GEOS Paradigm to that phenotype and instead we call that an eqtl analysis and I'm sure you folks have seen work from the gtex Consortium over many years applying ettl studies and identifying thousands of variants associated with the expression of many genes in many tissues and in fact again this is one of those cases where as the sample sizes have grown this study design has actually yielded a very large number of associations that are almost like too difficult too many to fully process and I think the most recent Gtech study showed that if you sort of relax the significance threshold for these associations nearly every Gene has at least one eqtl in some tissue and in fact I think that if you continue as the sample sizes have grown even further we see that genes then start to have secondary eqtls and tertiary eqtls and this sort of curve uh does not is not even hitting diminishing returns so that's the piece about identifying genetic variants that influence gene expression and then there's been a lot of work in trying to understand how these eqtls connect to disease and I'll highlight a couple studies in particular which basically asked in a couple different ways whether an eqtl is more likely to be a GEOS variant or is more likely to be associated with a complex trait and so the results on the left show that eqtls specifically as you get more confident about them being the causal eqtl are more enriched for heritability across many complex traits from gwas and then this figure on the right from gamazon at all showing that if you just sort of try to partition the amount of disease heritability that could be explained by eqtls those estimates are also quite High across a large number of complex traits again ranging from maybe 10 percent up to 35 percent so there's this sort of incidental evidence that eqtls are enriched for disease heritability and may therefore give us an instrument to understand um the likely causal genes and eventually go back to that sort of big system-wide understanding of the phenotype so that's the first part of the arrow the other part of this of this network is we want to understand how this genetic mechanism of gene expression actually goes on to influence the trait and for which traits and this is where the approach of a transcriptomide Association study or a tiwas comes in and I'll just start with a very basic sort of thought experiment of what would we um want to do if we had the ideal data set how would we in sort of with infinite resources try to relate gene expression genetics and disease together and I think one way that we could do this is we could estimate expression in the hundreds of thousands of individuals that we have genetics and case control status in here like this represents case control status and then we could ask what genes are genetically correlated meaning the effect sizes on expression are also shared with the effect sizes on disease we could do this for every single Gene across the genome and that would give us an estimate of the genes that in principle could be linked to this phenotype and the the hurdle here is that we we uh very rarely or pretty much never have data at this scale what we typically have is a relatively small study of genotypes and measure gene expression usually as in the case of the G tax in a sort of healthy relatively healthy population that was convenient to sample and then we also have very large disease studies that also have genotypes but no gene expression measured and so the basic Insight of the transcript and mind Association study or tiwas is kind of thinking about the fact that we have what is shared across these two these two studies is the genetics and we know previously that gene expression is itself a heritable trait and if it's a heritable trait then in principle it should be a predictable trait and so what we want to do is use the genetics to predict expression into this study over here where we haven't measured it and then use the predicted expression as a sort of proxy to estimate the relationship between the genetic component or the predicted component of expression um and the uh in the in the phenotype um and again this is all I'm sort of presenting everything in the context of a single Gene but the idea is to use this methodology and scan across every Gene in the genome and identify the set of genes that are significantly uh genetically correlated or uh For Whom the predicted expression is significantly associated with the phenotype uh and so right then we do the test so the first question is can we actually predict gene expression in this way and the fact that we've observed significant eqtls or individual variants that affect expression basically tell us that tells us that that we can and in work that we've done and others have done we've shown using a number of different prediction schemes that I sort of won't go into but that are various forms of penalized or Bayesian regression that you can in fact predict gene expression with a substantial degree of accuracy and in particular when you use models that incorporate all of the genetic variation around the gene you typically have substantial gains in the predictive accuracy so even though the single topic ETL explains a large fraction of the CIS effect or of sort of the total heritability near the gene um there is a very large number of genes for which additional variants contribute substantially to the predictive accuracy and so simply going from a single snip Paradigm to a sort of locus wide Paradigm increases our predictive accuracy and that's going to translate into better Association statistics in the in the eventual guas study um now one additional constraint is that we typically don't really even have this design where there's individual level data in both studies what we actually have more frequently is this design where we have individual level data for the gene expression study and then we have summary statistics for the gys and the summary statistics are basically for every snip the marginal the marginal sorry I think this turned off dead battery I think I know what that means um is there a does anyone know if there's a backup or can you guys just hear me like this I don't know if this can you hear does that does that work okay let me try to decouple one sorry okay all right I'll try to work with this and then let me know if you can get another one for that one cool okay so what we typically have is these summary statistics oh sorry do you need anything else it's perfect okay let's do it great all right that's easier thank you okay so summary statistics uh this is what we actually have and they are the marginal Association statistics for every variant and what we want to know from this kind of data is what would the gene the predicted Gene trade Association have been if we could get to the individual level data and measure it and so this is really where the the uh the tus methodology comes in again because this is the type of data we have most of the time um and I'll just sort of sketch out how this uh this parameter is estimated and the basic idea is that we think about what we would want to do with individual level data and then we kind of move terms around and try to identify pieces that can be estimated from the summary level data and so we start with predicted expression over here x are the genotypes that we use for the prediction W are the weights that we've trained in the gene expression data that gives us this this term G that's the predict expression and then what we want to know is the association between y the phenotype and G the predicted expression so specifically we want to know this orange beta T was so we can kind of plug in the terms into a basic ordinarily squares regression and then decompose these terms and you can start to see pieces here that you can actually estimate from summary level data and in particular you'll see that this covariance between the genotype and the phenotype actually corresponds to this these G was summary statistics that we get the association between each snip and and the phenotype and then this term down here the covariance between the Snips themselves is also something that we in principle can get from reference panels because it doesn't rely on knowing the phenotype and so these two pieces we can get externally we plug them back in and now this is a summary based estimate of the beta tus that only requires the z-scores the reference LD and then these weights which we have we sort of assume that we have a priority and then I won't go into the details of how we derive the variance for this Statistics it's very very similar and the final Association statistic that we get looks like this where again in the numerator you have you can think of this as a weighted sum of the gyc scores that's weighted by the uh the predictors of expression and then in the denominator we have essentially the variance of that predicted expression that accounts for the correlation across these Snips so Snips that are correlated are going to add to the variance and Snips that are independent or not so this is basically the the score and I think this is also kind of a useful framework to think about how you can go from Individual level data to estimates of quantities we're interested in with summary level data uh when we apply this technique to summary based data and individual based data it works really well correlation is nearly perfect and again we didn't really make any assumptions going through it uh going through that previous derivation except for the fact that the LD is well matched to the Target population and also there's sort of a hidden assumption that the effect sizes can't be so enormous that we need to account for changes in the environmental variants and those assumptions are very easily satisfied in most studies so now thinking about when does this approach actually uh lead to associations we ran some simulations where we considered three different study designs under the model where there is a causal Gene and we've observed the predictors of that causal Gene so you could imagine in that scenario just running your standard G was to try to identify the association um you could imagine testing only the top snipped the top eqtl that's associated with expression or you can imagine running a full tus test and when we do that we see that in this scenario because we're testing fewer features we're only testing each gene instead of each snip then the power of the tuos or the eqtl only approach is higher than the guas approach so this is one case where not only are we getting a parameter that we're interested in on its own we also have some increase in power because the multiple testing burden is effectively is effectively lower furthermore if we expand the model and say additionally consider genes with multiple causal variants where now the to us approach of applying a penalized model to the entire Locus is giving us more signals more predictive accuracy than the top eqtl we see that the power of these single snip approaches drops but the power of the tiwas locus-wide approach remains effectively the same and so again this is another scenario where when we have many causal variants for expression that all lead to disease then we can substantially boost power uh and where the truth is is is is sort of in between or maybe a little bit off the page there's going to be some loci where we don't have the measured expression uh at all so these uh expression based approaches will just fail there's going to be some loci where there's only a single variant for the Gene and will be up here and there's going to be some loci where there are many causal variants for the Gene and the tiwas will then maximize power relative to other approaches uh so this is all in simulations under very specific presumed models we can also ask how well does this approach perform in real data and this this has actually been quite a challenging question to answer because as it stands we have very few well-established causal genes for disease so I showed you that plot at the beginning that had over 200 known associations for prostate cancer but the number of well-established really definitively established causal genes for prostate cancer for that study is extremely small and that's sort of the case for most complex traits so we don't actually have kind of working in a regime where we don't really have a ground truth there was a study that was done in this pre-print by weeks at all from The Phoenician lab which I thought was an interesting attempt to to try to get at a ground truth and the basic idea was that if we look at data in their case they looked at data from the UK biobank where you had associations both with common sort of standard gwas and also a rare variant coding variant based set of tests and you identify a Locus where there's both common non-coding associations and rare coding variant associations you can assume maybe it's not a safe assumption but they assume that the rare coding variant is telling you the right causal Gene and so under this model they basically have a kind of ground truth which is what is the rare coding variant tell you the causal Gene is and then they can ask how various other approaches do based on just the blue stuff just the common variant associations for identifying that causal Gene and so now they have a ground truth um they can plot Precision recall curves and they used this approach to evaluate a bunch of different methods listed here and then also to propose their method which is uh conceptually quite different I won't go into it but it's sort of like an ensemble that integrates many many different features at the locus to make the predictions but I think the the what's relevant here is how these other approaches perform and what you can see is that there's quite a lot of heterogeneity in their performance uh the tiwas is here in blue and as one point it has the highest uh recall in this model relative to the other approaches aside from their sort of Ensemble based approach and then additionally they integrated each of these method methods together with their model and in that scenario the tiwas had the highest Precision together with their approach but again I think an important takeaway here is that this is far from a solved model with a clear optimal method the TOs provides you an estimate of a certain statistical quantity but this is biology in biology is complicated and so lots of different approaches have different trade-offs for what they're able to identify and at what levels of precision and recall and then you know one thing I should mention that maybe people are noticing is that if you take a very simple model of just what is the nearest gene or what's the distance or how far away is the potential causal Gene that actually performs really well and in fact it performs about as well as the method that they that they develop developed and also when combined has very good Precision so again I think that there's many explanations for this one is that in fact it may be that the nearest Gene oftentimes is the correct Gene it may also be the case that this specific model tends to emphasize genes that are close to the association statistics but again I think it's also important to keep in mind that probably some hybrid of all of these methods to that also consider proximity is going to eventually be this sort of optimal solution okay uh so that's kind of where we stand with um with with 2s applications um I you know coming back to this figure you can sort of wonder why the Precision uh of tiwas is relatively low compared to these other methods and I think again there's an important set of caveats which were sort of highlighted in this paper from Mike Weinberg at all a couple years ago nature genetics which essentially come down to the fact that tiwas is an association study it's not a causal inference technique and as an association study it's going to be susceptible to tagging and correlation in the same way that genome-wide Association studies are and so in this paper they proposed a number of sort of alternative models which could still identify a significant T was hit one alternative model is that you can have co-regulate Mission at a Locus where the same genetic variant or set of genetic variants Drive the expression of multiple genes both the causal and non-causal genes and this is a real phenomenon that it's not that uncommon that you will identify loci with multiple genes with very high cyst genetic correlation and high genetic correlation to the trait another case is you can imagine some part of the genetic effect on a non-causal gene is tagged because of LD between variants with the effect on the causal Gene and this would produce a false positive Association or it would sort of induce some effect on both the causal the non-causal Gene and the causal Gene and then likewise you can imagine a scenario where the effect on the causal Gene was missed in the gene expression study because it was not sufficiently powered or it didn't get the right context and so this would lead to a false negative association and so again I think this is important to keep in mind that this is a test that is expected to tag the causal mechanism when these assumptions are met but in the real world these assumptions should also be sort of interrogated uh the other I think important limitation and one that's potentially solvable uh is to consider is the fact that as we've seen with other genetic predictors the uh uh these expression predicted expression models do not uh generalize well to other um to different populations and really because most of the data has so far been collected in individuals of European ancestry this is particularly a problem for generalizing to data from non-european populations um or in admix populations with low European ancestry and so this paper showed models that were trained in European individuals that had high accuracy predicted into held out European industry individuals and had significant drops in accuracy when predicting into individuals of African ancestry and again I think that they're potentially interesting ways to address this problem you know probably the the most basic is just to start collecting more data in other populations we should definitely be doing that but also there's methodological approaches that could potentially leverage all of the training data that we have available or think about the differences between populations to improve the prediction of these of these models okay I'm gonna drink so I also wanted to talk a little bit about methods I think these are all methods that we did not develop and had no hand in but that I think are interesting approaches to moving beyond just that beta tus that I described for the association between expression and disease and I'll sort of walk through them briefly you know to to give you guys a flavor of methodologically what else can be done in this space there's a great method called utmost that came out a couple years ago in nature genetics which uh thought about how um gene expression data that's measured in multiple tissues in the same individuals could potentially be used to improve these predictive models so everything that I've been talking about so far sort of assumed that there was a population with some single modality of gene expression but you could imagine and this is exactly what the how the gtex was designed that you've measured multiple tissues for every individual this Y is now a matrix instead of a vector for a given Gene and then the approach that utmost proposes is to actually try to learn the expression for each tissue together with all of the other tissues observed and so again you see some similarities here these bees the what they're learning are sort of the W's that I talked about earlier now are being learned for all tissues at once and they do that by using again a form of penalized regression where they have a penalty within each tissue where they want the weights to generally be sparse and then they also have a penalty across the tissues where they don't want to see a lot of differences between tissues they sort of assume that if a snip is important for one tissue it should also be important for another tissue and this approach particularly for tissues that had relatively small sample sizes substantially increased the prediction accuracy basically by borrowing signal from other tissues that were available and that's sort of shown here in in purple is the increase in prediction accuracy and held out data um another approach thinking about this sort of multi-tissue framework is instead of learning weights using multiple tissues we may be interested in testing multiple tissues where each set of Weights were learned individually and so there was this work a method called multi-skin by barbaridol in 2019 which essentially showed that if what we're interested in is this relationship here between now we have many G's for a single Gene we have the predictive model from one tissue a second tissue a third tissue and we want to know if there's an association for any of these features in a joint model so a multi-degree of Freedom test for Association what we actually have again because we don't have the individual level data we've actually observed is these marginal tiwas individual tus statistics um but if we know the correlation between these statistics then we can actually approximate the relationship or the effect under the three degree of Freedom or end degree of Freedom a test from these marginal effects the multi-scan paper also did some clever stuff where you have many tissues with highly correlated expression and you don't want to just throw them all into this model by using principal components analysis to First reduce the dimensionality of the expression down then just test the leading components of expression for Association in this P degree of Freedom test and this also in practice showed that it produces a much larger number of significant Gene trait associations again now we're sort of saying that if there's a little bit of signal here and a little bit of signal here and here then that can add up to a lot of signal across the the three degrees of freedom um the other I think interesting method or any other interesting method in this space is uh is now thinking about how to integrate together um many tiwas associations across a given Locus and so you've probably seen methods for GEOS fine mapping that try to identify the set of causal variants or variants that contain the causal variant with some predefined probability the same kind of methodology or the same sort of concept can be applied to tiwa statistics and so instead this work of Mancuso at all in 2019 reformulated this problem in terms of having multiple tiwas associations at a single Locus and then fine mapping these down to the likely to the set of likely causal genes and this is actually sort of starting to address some of the caveats that I outlined earlier when you have co-regulation of multiple genes or you have some tagging across multiple genes this is now an approach to put probabilities on which genes out of many are likely to be causal whereas which are likely to just be tags and to sort of estimate posterior probabilities of causality for a given Gene so just in the last couple of minutes I wanted to mention a bit about what else can be done with this framework and so everything so far that I've been talking about has involved gene expression or transcription but really the idea is that any molecular trait that is heritable and that can be predicted from data that we've measured is amenable to this sort of approach and this way of integrating with gwas and in particular we can go back to this model which maybe we've we've solved now in some sense and and sort of we can observe that this is this is also an over simplification in that most of the time for non-coding variants what we expect is that there's some regulatory element that sits in between the variant and the express Gene that maybe excuse me is the modifier or is the mediator of this gene expression so really there's probably an enhancer or a transcription factor or a combination of those features um through which this snip has an effect on the expression of the gene which then goes on to have an effect on the trait and with sufficient data we can actually start modeling these regulatory elements and the genetic predictors of these regulatory elements and we have some recent work to that end which we call a regulome or a system-wide Association study so we're sort of padding out the the the the letters of the alphabet here but the idea is that instead of learning predictors of a given Gene you can learn predictors of a some biochemical activity including transcription Factor binding chromatin state or chromatin accessibility and and additionally in this regime we can also leverage some allele specific information of variants that are inside these Peaks that we suspect to be modifying their activity and so we can boost power even further because because we can we can sort of Leverage signal within each individual in addition to Across the individuals and again we've shown that this approach is fairly robust that you can identify a very large number of predictive models and when we've applied this approach specifically to cancer gwas phenotypes so again this is cancer risk we're just talking about the predisposition to develop cancer we see going back to this plot that I started with we've now characterized each of these loci where we see that there were this Inner Circle here is the number of lots that it had a significant tiwas association with a gene but then actually when we incorporate these epigenetic features we see a much larger number of loci that additionally have associations through chromatin accessibility in this case many of which are do not actually exhibit a direct transcriptomic Association and so this is actually sort of interesting and somewhat mysterious in that we're able to identify loci where there seems to be a genetic regulatory effect that we don't see have a downstream Cascade on expression we do capture most of the low side that have the the the the tiwas association those were able to characterize but then we have this number of additional loci and we've sort of started to think about what those loci could be telling us one observation is that if you look at uh um if you look at the sort of distribution of evolutionary constraint across the genome you will see that as in regions with higher evolutionary constraint we see fewer tiwas models that can be built probably because selection is making it more difficult to detect or is decreasing the observed effect on expression making it harder to pick up the eqtls but we actually see more of these arwas or chromatin was models observed in those loci so this is maybe this Gap could potentially explain that sliver in the previous figure of these are low side that are very difficult to pick up in the expression framework but are not as difficult to pick up when we look directly at the intermediate chromatin phenotype and a sort of related observation that we've made is if you look at genes in terms of their tissue specificity so as we move from here to here from the left to the right these genes are more specifically expressed in prostate tissue we're looking at a prostate cancer GEOS again we see that the tus models there's fewer of them they're harder to fit for more tissue specific expression but for the chromatin based models and the transcription Factor based models there's more of them and they're easier to fit and so again this could be pointing towards a phenomenon where more tissue specific expression has lower power for the sort of eqtl and transcriptome-based models but higher power for these epigenetic based models and so with that I'll conclude um I hope I've been able to uh convince you at the very least that gene expression is a complex heritable and predictable trait and this predictability is something that we can leverage to integrate that trait into other data sets or we don't have it measured and specifically we derive this tiwas statistic which is a measure of the cisgenetic correlation between the gene expression and the disease as I noted at the end this is not just limited to transcription other molecular phenotypes can be used within the same framework and and again I want to sort of emphasize the caveats that go along with any kind of Association is that it's not a a causal inference and in fact causal inference in this space is I think a really interesting and sort of ongoing open problem how do we disentangle all of those different arrows that I was showing earlier and then also just to remind you that all the prediction here has been within the cislocus of the gene that's where we have power at the current sample size but there's a whole world of trans effects which we haven't uh really we sort of just barely scratched the surface in understanding and so that's something that I think as studies get larger and as we have more experimental data we'll also be able to fold into this framework and so with that I'll take your questions thanks thank you

## Methods for GWAS/TWAS {#sec-video2}

## TWAS Primer {#sec-video3}

# MWAS tutorial {#sec-video4}
