---
title: "Chapter 8.3: Gene Association Analysis (Video Transcript)"
---

# MAGMA {#sec-video1}

**Title**: How do we go from genetic discoveries from GWAS/WGS/WES to mechanistic disease insight?

**Presenter(s)**: Danielle Posthuma

Well, welcome back, this is part three of the session on how do we go from genetic discoveries to mechanistic disease insight, and in this last part I will focus a little bit on the software tool MAGMA for conducting gene based and pathway analysis. So in the practical you will learn how to work with MAGMA, and that's a tool that was created by Christiaan de Leeuw a couple of years ago and it can be downloaded from this website over here.

It is a tool for gene-set analysis and requires you to work with the Command Line interface, which should now be quite familiar to you and as an input you can either provide raw genotypic and phenotypic data or you can also provide summary statistics from already published results, but then you would also need some reference data. Because we need information on the LD structure of between the SNPs that are part of your analysis and then the other input is gene definitions. So that's with SNPs belong to which genes and I will come back to that in the next couple of slides and also definitions of gene sets. But if you download MAGMA, there's some files that have, some default files that you can use, that do this for you, but you're also free to use your own files if you want to. So just as an aside, if you want to have access to public summary statistics, we created a database and on this slide I've just added one example when you're interested in looking at one particular GWAS, then it gives you the Manhattan plot that gives you gene-based plots, it gives you the QQ plot. It also gives you gene sets outcome already. And it gives you some information about this GWAS with the link to the PubMed ID and where to download the data. So that's, you can use this database if you want to play around with any software tool that requires you to input summary statistics, then yeah you can just download summary statistics from this database but there are also some other databases that have the same purpose. In MAGMA gene set analysis, there are three main steps, so step one is the annotation where we match SNPs to genes, and so MAGMA needs to know which SNPs do I have to analyze as part of which gene. So that's step one.

Step 2 is the gene analysis, so that's where we compute the association of the gene with the phenotype. So here the unit of analysis is the gene and then step three Is the gene set analysis, where the association of gene sets is tested against your phenotype. And then, because it's a very general linear regression framework which can easily be extended, it's very easy to use continuous sets. So instead of having a dichotomous set where genes are either a member of the gene set or they're not a member of the gene set that you can also have quantitative, quantitatively defined set of genes where every gene has a value that indicates how likely it is to be part of a gene set, or that indicates the expression level of a gene in a cell type and then the cell type is the gene set. And it also allows you to do conditional and joint analysis and interaction analysis as was explained in part two of today's lectures.

*Annotation*

Now going back to the three main steps, annotation. If you download MAGMA it comes with a general annotation file and there SNPs are mapped to genes based on the physical location, and but you can also change this annotation file so you can, if you would like to have eQTLs included in it, you can map SNPs that are physically located outside of a gene but have a known eQTL link to the gene, or chromatin interaction, that's also possible to use. You can also add a window around the gene so you can say, well I would like to have maybe 1 megabases before and after the gene and those SNPs should also be analyzed as part of this gene. An one SNP can actually be linked, can be mapped to multiple genes.

*Gene Analysis*

Then if you run the analysis, there are four models that are available in MAGMA. If you have the raw genotypic data, then it will conduct a principal component linear regression analysis and that that can only be done when you have access to the raw data. So if you have, if you input summary statistics which most of you will probably do, then there are three different models that you can use to evaluate statistical significance of your genes and of your gene sets.

So the first model is the SNP-wise mean and it performs the test on the mean SNP association, so that evaluates the evidence for association of all of the SNPs in, that are located in gene and then uses the average association to evaluate whether the gene is actually associated. Or you can do to SNP-wise Top Model where where the focus of the analysis is on the strongest SNP association, and you can also combine these two models and get this, the SNPwise multi model where the evidence from both of these previous models is combined into one p-value for your gene. And yeah, deciding which model is best for you, that depends on what your actual hypothesis is. So what kind of sensitivity would you want? So there's no, we don't think that there's a best, best model. It really depends on the situation or your research question. That's why we provide multiple models in the MAGMA tool. So what's being done in the MAGMA tool, so when you do a gene set analysis, that's basically an analysis of genes. So instead of individuals being your unit of analysis or your data points, the genes are the data points in the analysis. So in this table we have listed 10 different gene IDs and each of these genes have been tested for association in the gene-based step in MAGMA. So they all have some kind of measure for the strength of the association with your phenotype based on your GWAS summary statistics. And then there's also an indication of whether or not they are part of the set of of your gene set that you would like to test.

So in this case the genes are the data points and the gene set is the grouping variable and the genetic association with the phenotype, that's the outcome that you would like to get, so this is basically a simple T-test testing whether the average association of the genes that are inside your gene set is different from the average association of the genes that are outside of your gene-set. Yeah, so that's basically just a one-sided test of genes because you have a very strong hypothesis of what association should be stronger. Now, there are two kinds of tests. So you could do a self-contained analysis where you ask whether the mean or the average genetic association of genes in a gene set is greater than zero. Yeah, so that's your null hypothesis and your alternative hypothesis, whereas in competitive analysis you ask whether the mean genetic association of genes in the gene set is greater that of the genes outside of the gene set.

Yeah, so that's your competitive analysis. And compare this with with a randomized controlled trial or any experimental setup, So in a self-contained analysis we would ask if the mean improvement of patients in the treatment group is greater than zero, whereas in a competitive analysis you would have a control group, so you would ask whether the mean improvement of patients in the treatment group is actually greater than that of patients in the control group. Now everybody would agree that we would want to do a competitive analysis. We would need a control group, otherwise we cannot really say that the treatment is causing the patients to improve. So that's also the reason why we think competitive analysis is the way to go in gene set analysis and that self-contained analyses are not informative for asking the question whether your set of genes that you tested is actually causally associated with your trait of interest. That's why we advise never to do a self-contained analysis, but always to to use a competitive gene set analysis.

OK. This just is stressing that same point and also in the, in part two of these lectures of today I've indicated this or if this is not clear than maybe you should go back to Part 2 of the lecture, so I hope that this message does come across and I'm looking forward to the MAGMA practical that is planned for later today. Thank you for listening and see you later!

------------------------------------------------------------------------

# FUMA {#sec-video2}

**Title**: FUMA: Functional mapping and annotation of genetic associations

**Presenter(s)**: Kyoko Watanabe

\
um hi everyone i'm a phd student at the fair university in amsterdam my my work is mainly focusing on how to understand the genetic association in biological context and today i'm going to introduce you a web application i have recently developed which is functional mapping and annotation of genetic associations

so i'm going to start with very quick look at what was zeros again so we basically start with genotyping a large number of individuals using snip slay which nowadays we can tag around million snips and by performing imputation with reference manual you end up with a maximum of 20 millions of snips so in a very simple case when you have case and control groups in your individual in your genotype individuals you you perform the statistical tests to see if the occulance of mineral in case and control groups are different from zero so at the end you basically get the p value for every single snip you have but as you can imagine the number of statistical tests be performed is same as the number of snips you have we of course have to correct for the multiple testing and the golden standard genomic significant p-value is five times ten to the minus eight so whenever you find the snips with p-value less than that they showed those genomic regions are called zeroes close eye

so the very very first jiva study was published in 2005 and since then the cost of genotyping dramatically decreased which allowed us to collect much larger number of individuals and nowadays the big consortium for meta-analysis usually use more than 100 000 of individuals and by increasing the sample size we also increase the statistical power to detect relatively weak effect size so for example the height study using around two hundred thousand of individuals they are not ended up with identifying more than hundred of these glow side so we've been performing jivas in last decade and the nowadays in the zeros catalog we have more than 3 000 of studies including more than 38 000 of unique sniper rate associations for over 600 of phenotypes so basically we have a lot of risk loci all over the genome

but however especially for complex trade which is highly progenic um we know that the association of single snips is very weak to detect those effects we need much larger number of sample size and luckily the uk biobank was just released this month and the qct data database has contains 500 000 of individuals for more than thousands of phenotypes so the uk by bank has the potential to identify novelist clauses for lots of human complex traits and we are expecting more and more viewers to be published in coming few months

so the question is what do we get benefit from jewish results so ideally we would like to identify the causal variants from genetic associations that can be used to improve diagnostics or prognostics or even you might be able to identify novel drug target or biomarkers however the association is not the cause association doesn't tell anything about cause id and also purely based on the p value from zeros you don't really know anything about the biology so identifying causal variance from zero zeros results is not straightforward so to overcome this problem we usually go through the several steps the first one is to correct for led which is non-non-democrats of snips so because of the led the most significant snips you find from a specific genome this glocus doesn't have to be the one actually causing the phenotype but there must be the snips that that is actually causing the phenotype and that snips must have the higher correlation with the most significant snips

so we don't want to miss those snips just based on the p value so the first step is to include all the snips that have higher correlation with the significant snips so once you get the list of snips you're interested in the second step is to check the functional consequence on the genes for example if you have a snips on the exonic regions or in the non-coding regions and there are several softwares that can perform this task however more than 90 percent of zero's findings are known to fall into non-calling regions so just knowing that you have jewish hit in the non-coding region doesn't really help you to understand what is actually going on in biological context um so you still need to annotate bioscale functions annotations there are several data resources that you can use for example the cut score is a directory as called as the details in score of snips and regular mdb is a categorical score tells you how likely the snips affect the regulatory elements and there are several eqtl databases for example the gtx having details in 44 different tissue types and especially for the non-coding regions you will also want to check the epigenetic status the data is available from roadmap and encode and i didn't bring any database name over here but the three digits in the field of cd genome more and more data is becoming available so including hi-c data might also be another option to map snips to the digital genes so using those functional information at the snips level you can end up with the list of genes you're interested in and then finally you you you need to take the expression proton in the different tissue types and also the cells share the biological functions such as pathways um so we've been performing these multiple steps manually as you can imagine that requires you to install different softwares and download different databases and sometimes you might have to reformat the data each time so this is very time consuming and elephant so we hope to make single platform that can perform all of them

so we developed a web application named FUMA that basically optimized four steps that i showed in the previous slides into one single platform so in the firmware there are two main processes the first one is snip to gene starting from zero summary statistics and we provide lists of candidate snips with the annotations and also the list of prioritized genes and those scenes can be passed to the second process which is jintu funk where we provide you the further virus confrontation at the gym level and another advantage of firma is we also provide intellectual visualize visualization in the web application so you don't have to use the external software to just for visualization so i'm going to go through what film actually does in each each process so in the snip to gene starting from the zero summary statistics we first characterize genomic disclose i by correcting for led and here we provide you the list of read snips and the genomic risk clause i and all the snips which are in led of lead snips are passed to the second step which is the notation of snips here we perform the anova and annotate several variant scores and eqdl and also the high c so using those informations we finally perform the gym mapping so currently we have three different criteria the first one is positional mapping using annotation from anova and eq thermoping and also the chromatin interaction mapping so before you perform this g mapping you can also filter snips based on the notation you obtain from step two and you can also combine different mappings together you can specify lots of different parameters when you when you submit the job and we provide the list of gene mapped by the based on the user defines parameters so this is just an example how the result page looks like so we provide manhattan pro on the top and the second one is we also perform gm-based tests using magma software so this is the manhattan board based on gmp value and the summary result per genomic risk loci and all the results are available as a table and you can also create a regional plot with all the annotations together and all the results and approaches are downloadable so this is just one example how you can utilize the eqter mapping so this is one of those cloud glocus on chromosome 14 from schizophrenia device and from from the top the most top product is the zoomed in manhattan broad and the genes and the cut score like rdb chromatin open chromatin states energy in the brain and the eq tails so as you can see the least clock the risk locus itself spans much proteins so if you don't know if you don't have any further information you end up with listing all the genes or you can manually take the function of genes and you can pick the one that has most interesting function in the phenotype however by performing eq demo ping we prioritize the single gene which have the eqtail in the brain so performing different types of mucus mapping you can also prioritize genes and another example is for the chromatin interaction mapping so the female currently lives in high c data set from stadium sheba at all which includes 14 different tissue types and cell lines but i already said the the field is growing very fast we also provide option to option to apply the custom chromogenetic matrix which is not limited to high c but you can also use capture high c and c5 um so the ground flight shows the risk loci on chromosome 16 from bmi zeros at the most layer is the manhattan product and the second the blue circle is the genome coordinates and these glossier highlighted in blue and inside of the circle the orange links are high ceiling and the green links are eqtls so as you can clearly see the high ck map snips to additional genes compared to the eqtl so that might helps you to identify a novel candidate genes which you have been boost so finally once you get the list of genes you can use into func process where we provide the heat gene expression heat map and tissue specificity by performing overrepresentation tests for differential express genes across different tissue types and enrichment testing gene sets and also the external link to omim and drunk bank to further investigate the individual genes

so in summary um we optimize the postgres annotation in a single platform as a web application so this might be the very first place to stop by to get the very um broad overview what's going on in uh in the zeros is close eye once you get the new zero zeros but also if you have a phenotype of interest there are lots of g1 summary statistics already available so you can start you can perform the firmware for the available zeros and start integrating with research and for the future updates we are thinking to extend fema to be able to accept the whole exome sequencing study and also the eros and so finally i would like to thank my supervisor core supervisor and whom is available online so please feel free to visit the website and i also have a poster this evening at b325 so if you have if you want to know for the details please feel free to visit me thank you

\[Applause\]

i've seen a couple of cases where even though the the british locus is associated with the same phenotype you see clear evidence of distinct haplotypes so it seems like fema would probably be able to show you cases like that where potentially you're getting the same phenotype from distinct variants that are affecting say the promoter of the gene or a nearby enhancer uh you mean like player therapy is um yeah well so it's more like the humor is just to annotate what's the functional information available so it just provides you the options which snips you're going to look at even further so it's not it's not removing the information so you might get multiple snips that have functions from one locus but yeah we cannot distinguish which is actually causal but i don't think you're going to miss that information you
