---
title: "Chapter 2.4: Linkage Disequilibrium (Video Transcript)"
---

# What is linkage disequilibrium? {#sec-video1}

**Title**: What is linkage disequilibrium?

**Description**:

**Presenter(s)**: Gábor Mészáros, PhD

hello everyone welcome back to the genomics boot camp to a series about lincoln this equilibrium in this video we just start to talk about this genomic phenomenon so therefore the question what it is exactly so what is linkage diskequilibrium and you will know it from this video before we start let's activate some of our prior knowledge that you might know from this channel or from other sources

and the two most important issues to highlight here is one that the single nucleotide polymorphisms or snips exist these are some markers that are widely used and currently in my opinion the most important marker types as for the genomics tens of thousands of such snips are being genotyped in a cost effective way on the entire genome so basically it is a very good way to get to know something about the genomes themselves and the other bit of information that is important here is the existence of the recombinations or recombination events that are of major biological importance this means that the nucleotides on our genome are not being inherited independently but in a form of shorter or longer genomic segments from the paternal and maternal side

*Mendels Law*

such statement about the non-independent inheritance is in conflict with the mendel's law of independent assortment which says genes do not influence each other with regard to the sorting of alleles into genes and every possible combination of alice for every gene is equally likely to occur mendel's law is of course valid when the genes or the parts of the genome are far from each other or for example on separate chromosomes but as you will see it's not valid when these genes or sneaps or parts of the genome are very close to each other

*Linkage Equilibrium*

the next few slides i took from the presentation of professor henner simiano from university of getting in to demonstrate what happens in such occasions and what in fact the linkages equilibrium is so for the sake of simplicity let's have an individual with two alleles and we ensure that this individual is entirely heterozygous then we mate it with another another individual which is entirely homozygous for these alleles a and b and when the law of independent assortment is valid then we get the four possible genotypes and if our sample is big enough then we have all these possible genotypes with a 25 of probability so in the our population we would see that all four genotypes appear in a proportion of 25 in some cases however we might notice that these four genotypes are in fact not appearing in an equal proportion but very very differently from each other in this case the alias denoted by the capital letters a and b seem to appear much more frequently together and also the alleles denoted by the lower case a and b are also appear to occur much more frequently with each other so in this case and 45 in comparison to a situation when the lower case a for example and a higher case b would be together in a single individual the reason for this is of course recombination so we have a non-recombinant gametes on the one side and the recombined gametes on the other side so in other words again in case a recombination happened between these two loci then we have a different occurrence for these for these gametes the degree of recombinations is measured by the recombination rate so

*Linkage Disequilibrium*

what we see is a departure from this equal distribution denoted as a linkage disequilibrium or ld which is the very frequently used abbreviation i wanted to underline that ld is in fact a parameter of the population so you need more individuals and in the best case scenario a large number of individuals to determine the ld between two loci in a population it is in fact a non-random association between the loci within this population which could be measured so we can tell that the two loci is strongly linked or weakly linked or not linked at all together in other words the ld tells us something about the strength of the information if we see an allele in a certain locus and what it can tell us about the occurrence of an other allele on another locus there are various methods to measure ld and we will talk about these measurement possibilities in the next video so here is an example of a part of

*Heatmap*

a genome with a heat map so here is basically the darker colors are high ld and lighter colors are low ld so it's measured with d prime in this example don't worry about this right now but basically what i want to show you that the snips that are close to each other tend to have a higher ld between each other and there are for example snips for example number nine and number 20 you see that there are relatively lower ld between them also there are segments on the genome that are so called ld blocks where all the pairwise combinations of the snips yield a high ld so basically this whole block is in inherited in one piece i really like this picture about the ld block structure in this particular case in a chicken genome that demonstrates a similar thing as in a previous picture or previous slide but on a much larger part of the genome also it compares to chicken lines where we see that the same part of the genome could be very different even within the species when we talk about different breeds or in this case the different lines and the reason for this could be that for example in this chicken breed in this particular part of the genome there are some genes that are very important for this breed therefore this part of the genome is quite well conserved with a very high pairwise lds in this part and the same genes are of course present also in the other breed but in this breed are not important or at least not selected for therefore we do not see such a strong ld block occurring in this breed this

*Why is LD important*

simple picture supposed to illustrate the use of ld and why is it important in the genomics as such so what we have here is let's say a chromosome and we have a snips on them and let's say we did some kind of analyzes where we found the significance of each snip so this would be the higher the snippets the higher the significance level now we see that there we have some kind of a signal here so the question is is this the gene of interest that influences our trait of course because we are speaking about snips that are themselves just markers and not the causal variants so this is in fact not the gene of interest but actually just shows what region of the genome is interesting for our case so we have a region of interest here and most likely the exact gene of interest resides somewhere here now the size of this region is determined by the linkage this equilibrium so ld because most likely this snip them the highest significant snips is connected to all of these nucleotides around also that the parts of the genome that were not observed and therefore helps us to find the gene that is actually interesting for us

*Summary*

so to sum up icl the important it actually defines how how far we are allowed to or supposed to look from the detected markers when we are looking for the causality on our genome and it shows the association strength between the observed snips and the unobserved genes or qtls also i want to underline that the ld itself is actually dependent on a population or species so most likely these association strengths or the in these regions or the size of the regions might differ between the different populations as for

*Applications*

the applications of ld in genomics it is really really wide ranging so i just mentioned a few examples here so in evolutionary biology it allows us to reflect on past events and gives insight into evolutionary history when it comes to genetic diversity it's a well similar as in a previous point but we often compute the effective population size for our populations and this one of the computation methods is via ld itself then we can tell something about the artificial or natural selection events on the genome when it again comes to the population with the so-called selection signatures and the genetic hitchhiking these are events when the actual selected gene drags along part of the surrounding genome and appearing in a population as a selection signature so this size is determined by the ld itself my small schematics showed an example of a genome-wide association study where we could search for causal genes and as it was mentioned there we could locate our search to these interesting regions where the most important or most significant snips reside and ld is also used in a genomic selection in an indirect way so here the question is that how many snips we need that are more or less equally distributed on the genome so the whole genome is covered so we could detect these very small qtls and very small genes that affect our trait of interest which is usually a quantitative trait so it is influenced by many genes of small effect we need to ensure that the whole genome is covered properly so that all of these small genes are connected to at least one snip in a high enough ld and therefore taken into account when the genomic reading value is calculated so again a slide to give an answer to the initial question what is linkage disequilibrium so linked to this equilibrium is a parameter that quantifies the non-random association between the low psi it shows if the frequency of a different alleles between two loss psi is higher or lower than it would be expected if the lows say loss i were independent from each other and associated randomly and

*Conclusion*

the very last summary slide for the entire presentation we talked about snips markers that allow to track the associations between parts of the genome and such associations and such connections could be non-random especially if the snips or markers or parts of the genome are very close to each other such non-random association is called the linkage this equilibrium or ld in a shortened abbreviated form and there are wide-ranging applications of lde within genomics in the following videos we will continue with the exact measurement techniques of the linkage this equilibrium also i will provide a very nice hands-on example but of course also show the way how the ld is computed for large data sets using blink for today i thank you for your time and see you at another video on the genomics bootcamp channel i wish you a very nice day

------------------------------------------------------------------------

# Measuring linkage disequilibrium {#sec-video2}

**Title**: How to measure linkage disequilibrium?

**Presenter(s)**: Gábor Mészáros, PhD

hello everyone welcome back to the genomics boot camp today we will continue to talk about linkages equilibrium before we start however i want to thank you for the 500 subscribers or the over 500 subscribers on the genomics boot camp channel it seems that there is a continued and ever increasing interest about the contents of the channel which is of course a very good news for me and a good motivation to move forward of course if you are not subscribed yet you can use this opportunity to do so so thank you thank you thank you and without further delay let's move on to further discussions about linkages equilibrium so let's start with

*Linkage disequilibrium*

a little bit of activation of our prior knowledge namely that single nucleotide polymorphisms or snips exists and these are actually the main marker types we speak about on this channel linkage disequilibrium as such also exists and we spoke about this in the previous video and in short it actually measures the non-random associations between snips and there is a wide ranging applications for ld in genomics but for today's question how to measure linkages equilibrium we saw this graph in the previous video and basically what we see is that some of the allele combinations are appearing much more frequently than others thus leaving two linkages equilibrium this non-random association between some alleles of a different loci can be visualized in such a two by two table where we have on one side locus a and with the alias capital a and small a and on the top side locus b with alias capital b and small b each of the combinations is then represented with a proportion so basically the p capital a capital b is the proportion of the combinations for these alleles if we are interested for example in the total proportion of the allele a then we have it in the right side the same we can tell about the proportion of all the other alleles so so the the lowercase a the capital b and this lowercase b there is one rule if we are talking about biolic systems that the sum of the proportion of the two alleles sums up to one in case of linkage equilibrium when the low psi a and b are not linked we can actually compute the genotype proportions based on the allele proportions this is shown for example in the first line and we can actually do the same for all the other genotypes one additional comment here so we use this feature of this previous two by two tables so i told you that basically the proportions for the two alleles are summing up to one so basically the p lowercase b we can also express as one minus p capital b this is of course a preferred setting because we reduce the number of unknowns from four to two so basically we just have to work with two variables p a and p b now this situation is valid for the linkage equilibrium as i mentioned so when the alleles are unconnected in case of linkage disequilibrium however these equations does not hold true in other words from the proportions of the alleles we cannot compute the proportion of the genotypes we cannot compute this because there is a difference between these two metrics that is the disequilibrium coefficient or capital d and it can be computed as the proportion of the a b together minus the proportion of a time proportion of b and when we include this this equilibrium coefficient to the equations as before we get the proportions of the genotypes so in other words this d or this equilibrium coefficient is a measure of the linkage this equilibrium or a coefficient that can be used to express it

*D prime*

but we have a bit of a problem with this this equilibrium coefficient because it's a bit hard to interpret and the sign is arbitrary so it's depending on which allele actually you consider first of course there is a common convention to set the capital n capital b to be the most common allele and the smaller case a and b to be the rare allele but of course the rarity of the alias or the minor allele frequencies can change from population to population even within the same species so that is not such a big win after all now there is a better version of this d or the disequilibrium coefficient called the d prime that is a scaled version that is computed as shown in this slide so basically is divided by the minimum of these two values in case the disequilibrium coefficient is lower than zero or the minimum or of these two values if the disequilibrium coefficient is higher than zero now we end up with a coefficient which is another a more popular version of the linkage this equilibrium and ranges between minus one and plus one extreme values in this case imply that at least one of the haplotype was not observed it has several advantages that is the d prime one or minus -1 means that the two snips are not separated by recombination so that they are in a complete ld it also means that if the earlier frequencies of the two loss i are relatively similar then the high d prime means that the markers are good surrogates for each other but in this case we have also some disadvantages namely that the d prime estimates are inflated in case the sample size is small and the d prime estimates might be also inflated when one of the alleles is rare then

*Correlation*

there is a follow-up question if there is a more intuitive way to measure linkage this equilibrium which to which the answer is yes and that it can be measured straight with the correlation coefficient or an adapted value anyway these correlation coefficients as we would expect expresses the mutual relationship between the alleles of the two loci now a little bit of refreshment on the correlation so this is the correlation as we know it from the statistics so it has a standardized abbreviation of lowercase r and ranges from minus one to plus 1. basically there could be a different values for the correlation coefficient we describe how the two variables behave in relation to each other so basically we have an x and a y and if with the increasing x the y decreases then we have a negative correlation coefficient and if the two values are increasing so with the increasing x also y increases then we have a positive correlation coefficients in between minus one and one so the extreme values so the correlation coefficient expresses the strength of such relationships and of course it could be situations when the two values are not correlated so then it's the correlation coefficient is zero now for linkages equilibrium we use the squared correlation coefficient so this is the r square so the squared correlation between the markers and therefore it ranges between zero and one r square of one implies that the markers provide exactly the same information and the r square of zero implied that the two markers are not connected at all so independent from each other so in other words the r square measures a loss of efficiency when a marker a is replaced with the marker b now the r square value could be computed from the allele frequencies themselves and what we need is actually just three values from here the joint appearance of a and b the proportion of capital a and the proportion of capital b based on these values we can put them into this very nice equation and we can compute the correlation coefficient r square now i realize that there is quite a jump from the correlation itself to this equation and i provide actually all the details in an additional video which i call the advanced or a quotation mark advanced video so you can look at it if you're interested it is already uploaded on the genomics bootcamp channel and once we have our measure of linkage d6 equilibrium of course we can compute the pairwise linkages equilibria between markers such it was done in canbary at all 2010 and for this example so in here what we have in on the y-axis is the link is this equilibrium measured in r square in the x-axis there is a distance between the markers in this case measured in morgans which can be of course translated to megabases so this would be one megabase two megabase difference three four and five megabase difference each dot on this graph is a linkage this equilibrium value for a marker pair so you see that if the two alleles are very close to each other then the linkage desecration could be very high up to one but then this value quickly goes down to a lower values as the differences between the markers or the genomic distances between the markers increase this sudden decrease of the average linkage disequilibrium between markers is shown with this dashed line and in fact its shape is very typical for all the organisms we refer to this sudden decrease as linkage disequilibrium decay the shape of this lddk is in general similar between the organisms but the starting values and the further developments can be different when it comes to different species or in this case as shown in perez o'brien 2014 it could be different even when we consider bostoros and boss indicus breeds this of course has some relevance when the ld is used for further genomic analysis as discussed in the last video so

*Summary*

to summarize this video the ld characterizes the degree of relationships between the need by loci there are various methods and possibilities to measure it one of them is the disequilibrium coefficient denoted by d or as a d prime that is a metric to use to quantify the ld but it has some disadvantages and then the other more commonly used metric is the r square or the squared correlation coefficient that is a robust measure of the ld so this is the end of this video if you're interested how we arrive to the equation to compute the r square then be sure to check out the next one a follow-up to this and if you say that is enough for today then of course i thank you for your time and wish you a very nice continuation of the day.

------------------------------------------------------------------------

# (Advanced) Measuring linkage disequilibrium {#sec-video3}

**Title**: How to measure linkage disequilibrium? (ADVANCED)

**Presenter(s)**: Gábor Mészáros, PhD

hello everyone welcome back to another video on the genomies boot camp this time is a follow-up or an add-on to the previous video how to measure linkages equilibrium this video is entitled advanced because we have a bit more equations here as on the average of the channel the reason i did not included these ones into the previous video is that perhaps not everybody is interested or they don't like to see you know too many equations in a educational video therefore i thought it makes sense to divide the two but you are here you are interested so let's get to it i repeat here a few slides that were in the original video in order to provide context but we will go through these ones in a quicker manner

*Rsquare*

we start with this video where we ended last time and this is our prior knowledge now and the most important part is the r square so there we have the commonly used robust measure of ld in the previous video i was just including the summary equation but in this video we will have more information how we arrive to this r square so again we have the correlation coefficient from a general statistics and abbreviated as a lowercase r ranges from -1 to 1 and there are various values of the correlation coefficients that are expressing the relationships between the two variables x and y of course the correlation coefficient is also computed somehow and it is actually computed as a covariance between x and y divided by the square root of variance x and variance y if we want to write out this one in a further detail so we have this would be the covariance between x and y and the variance of x and the variance of y in these brackets basically the goal of this video is to provide you with the link between this equation and the r square equation to compute the linkage disaquarium that was shown at the end of the last video and also will be shown more times also in this one so for refreshment we have the r square value that is the squared correlation between the markers therefore ranges between 0 and 1 and the value of 1 implies that the markers provide exactly the same information while the value of 0 for the r square implies that the markers are not connected at all to each other therefore it measured the loss of efficiency when a marker a is replaced by a marker b the r square itself could be computed based on the allele frequencies from a this two by two table so we have a locus a and the locus b and each of them has two values capital b and small b and a capital a in a small a so these values are valid for a bial analytic system so these kind of computations are valid only in a case that the markers have a two alleles but because the snips on the snip chips are biologic by design then we are good to go there are in fact only three values that we need that is the joint occurrence of a and b and the respective frequencies of capital a and capital b now we have the r square which is the measure of the linkage this equilibrium and that can be computed with this equation previously it was stated that the correlation itself so the r could be computed with this equation so basically the main question of this video is how to get from this to this now to answer this question we have to reformulate our example a little bit and we consider two loci a and b again on one gamete but each with a possible random realization we give the value of 1 to locus a with the probability of p capital a and the value of 0 with the probability of p 1 minus p capital a so this is basically the other allele so this is the p a and p capital a and then this is the b lower case a but because b lower case a and capital a equal to one so the sum of them is equal to one so then the b lower case a could be expressed as 1 minus capital a the very same thing happens with the locus b so we assign a value of 1 with the probability of b capital b and the value of 0 with the probability of one minus p capital b so this is a very particular example that we will follow up on in detail in the next video but still i thought it's useful to show it here so perhaps it's easy some understanding so basically what we have a two loci here locus one and locus two so we have uh allius capital a a lowercase a and for locus two it's a allele capital b and lowercase b and basically what we do is we replace with the capital a with that once and the lowercase a is with the zeros so then it looks like this so basically we just count how many capital a's we have so that would be a 5 and how many capital b we have that would be a 6 and the joint occurrence of capital a and capital b would be 4 in this particular example now of course we are interested in in a general example so we introduce a population size of n and we can compute the proportion of the allele a as the sum of occurrence of allele a divided by the population size n the same way we can compute the proportion of b with the sum of the allele capital b divided by the population size n and the joint occurrence of a and b we can compute when the jointly occurring a and b together divided by the population size n okay then the follow-up is what

*Follow up*

is the correlation between the realizations of zero and one of a random variable at the two loci and here comes again the equation for the correlation coefficient so that is the r this equation basically has three parts so that is one in this top and then the so basically the covariance and the variance of x and the variance of y and basically we have all these parts figured out already in the previous slide as shown here so it can be replaced by n times the joint occurrence of a and b and the sum x can be replaced by n times the proportion of a and the y could be replaced as the n times proportion of b and similarly all the other parts could be replaced and then in some cases also adjusted so we end up with such a beautiful and much more simple equations to provide you a link to the next slide i also color code them so basically we have this orange the green and the blue part which we then put back to the original correlation coefficient equation so this is the top part here is the correlation coefficient and then we put in the orange the green and the blue part and then you see here that the n is present all the way so it can be actually removed from the equation and then only this part remains so this part is basically the simple correlation between two loci one and two containing the alleles a and b and if we square that equation because we are interested in r square rather than a simple r so then we have this correlation or a squared correlation coefficient that is the measure of the linkage disequilibrium as we have shown in the previous video and also in this

*Summary*

one so for the summary the r square is a commonly used and robust measure of ld and the computation of the r-square is adapted based on the equation for the correlation coefficient and for the allele and genotype frequencies in reality of course we talk about computation of linkage disequilibria between a large number of markers or marker pairs so we use software for it for example blink or other software and also in the follow-up videos we will actually show how it is done using the plink itself for today i thank you for your time thank you for your interest also in this advanced content and i wish you a very nice continuation of the day

------------------------------------------------------------------------

# Computing linkage disequilibrium {#sec-video4}

**Title**: Compute linkage disequilibrium (Part 1)

**Presenter(s)**: Gábor Mészáros, PhD

hello everyone welcome back to another video on the genomics boot camp back again with another video on linkage disequilibrium this is the first part of the finishing two when we are actually computing the linkage disequilibrium this time by hand i believe this is an interesting piece of knowledge so it's worthwhile to look at so what we know so far so ld characterizes the degree of relationship between two loci as always and r square is the commonly used measure of the ld computed with the equation below and

here we are at the first of the two examples i will show you today the first one is the very basic one and actually you might recall this two by two table from the previous videos when we look at the r square equation it is clear what we need to compute and that is the proportion of a proportion of b and the proportion of the joint occurrence of a and b therefore we are interested what is in this segment or cell in in this other one for the b and the joint occurrence of a and b together because we are speaking about proportions we are need to divide them with the sum of all alleles so this is already how we compute the proportions so the proportion of a is the 5 divided by 10 so 0.5 in this case proportion of the b allele is 6 divided by 10 so 0.6 and the joint occurrence is 4 divided by 10 0.4 when we have our usual equation we basically have all the unknowns so the proportion of a b and a b and we put them in and after the computation we get the ld between these two low psi a and b as 0.167

so after this warm-up example i show you the real deal that is this ld calculation exercise that was shown by professor henry simiano at the

livestock conservation genomics data tools and trends workshop back in 2012 in pug island croatia i really like this example because it puts the ld calculation into context of haplotypes and all the allele combination using snips so first i give you the initial example how it was given back in 2012 then i give you the breakdown of the example and you will have the chance to stop the video and do the computations yourself and after that you can look up the solution in this video so the exercise looks like this so we have our genome here and we have a four snips first one is an eighty snip second one a cg snip third one is also a cg snip and the fourth is an at snip we have also our population and this are already the haplotypes where this haplotype for our genome or part of the genome appears 17 times this other one 14 times three times three times two times and this haplotypes appears just one time in the population the exercise is to calculate the ld between all pairs of loci now in this video i will show you how to compute ld between locus 1 and 2 in detail then i will show you the solutions for the two other pairs of loci and the last three you can follow up yourselves if you want

so to give you a bit of a heads up information i list here the low side that you need to compute the proportions for so for a locus one is the a the locus 2 is a c local 3 is also a c and locus 4 is an a you need to compute it for these alleles specifically because these are the major alleles for these loci so if you count them up so the a is the major allele for locus one uh dc is for locus two and the c is also for local three and a for locus four and obviously then you need to compute the joint occurrences between the two low side that you are following up and you want to compute ld for so if you actually have all these proportions you can compute the ld between all pairs or actually from four pairs one two one three and one four as we will list it in this video and putting these proportions into the well-known equation then you get the r square value for that pair of loci now if you want to go ahead and do the calculations yourself on with pen and paper then this is the moment to pause the video and well just go ahead with the computations and after you are done or at least you are done with the locus for example locus one two then to resume the video and see if you have done well or if there is something to be corrected

okay so we are continuing with the solution in three two one now actually what needs to be done is pretty similar to the previous smaller case but in this case you need to build up the two by two table yourself so here we have a locus one with an a t snip a locus two with the c g snip and because we told that the a is the actual major locus then you actually need to count them up so basically you need to fill up this spot in the 2x2 table for the locus 2 the c is the major logo so you need to fill up this spot on the 2x2 table and of course the joint occurrence as well so how to do that of course we will use the actual haplotype numbers as they are given in the exercise actually right now we are looking at locus 1 and locus 2 so basically these two columns are interesting for us so we can actually hide the other ones so that they don't bother us too much now perhaps the first thing you might consider is that well we need to calculate the proportions so we basically need to know how many haplotypes are there total so basically we add up 17 plus 14 plus 3 plus 3 plus 2 plus 1 and we when we do that we actually arrived number 40 so this is the total number of haplotypes now for locus 1 we need to count how many times the allele a appears so in this haplotype it there is an a so it's 17 times so it's 17 here in the locus one is a t so that is we don't count this also there is an a here an a here here there is a t here so we don't count this and the a here so basically we have a 17 plus 3 plus 3 plus 1. that is together 24. following the same logic for locus 2 in this case we count the c allele so here we have a c so if we count this here we have a g that so we don't count this count this this and this so everywhere where is the c so there is 17 plus 3 plus 3 plus 2 and that is together 25. the last thing we need to find out that how many times the a and c appear together so here the a c appear together so that we count this we don't count this but also ac here ac here here als although there is a c but there is not an a so we don't count

this part and also here is an a but there is a g here so it's not an ac combination so so we don't count this so basically we count this 17 these three and this three which is together 23 and because we are actually interested in proportions rather than the numbers themselves so we divide for the locus 1 the proportion of a is 24 divided by 40 that is 0.6 and therefore locus 2 the proportion of c is 25 divided by 40 that is 0.625 and the joint occurrence the same way so the 23 divided by 40 and that is this number 0.575 so with this we already have everything so we have a proportion of a proportion of c and the joint occurrence so we have our beautiful equation here so this is the joint occurrence and this would be the proportion of a and proportion of c so we put them in into the equation itself and then we end up with an ld between locus 1 and locus 2 as 0.711 so this is the ld between these two loci on this slide i give you all the results for the other low size so locus one two three and four and the joint occurrences of these three pairs and if we fill them in into the equation the well-known equation then we end up with well this is just a repetition from before and then there is the ld between locus 1 3 and the ld between locals 1 and 4. so this is basically it this is how you compute

ld by hand so for the summary an example of the ld computation by hand was shown which is not a very widespread piece of knowledge so i argue that is really good to have it in your inventory but on the other hand we also have to note that this manual mode of computation is not very effective for that we need to use software solutions that are able to handle computations of lde between thousands and hundreds of thousands of loci and such computations will be shown in the next video for today i thank you for your time and have a very nice rest of the day
