---
title: "Chapter 8.9: PheWAS (Video Transcript)"
---

------------------------------------------------------------------------

# PheWAS: Discovering gene-disease associations {#sec-video1}

**Title**: PheWAS: Discovering gene-disease associations

**Presenter(s)**: Dr. Joshua Denny, All of Us Research Program

**thank you very much it's a pleasure to**

be here as always and talking about just

the amazing stuff that's happening here

in the UK biobank

alright great so to start with you know

we've talked a lot about genome-wide

Association studies and sequencing and

and we've talked to them about phenome

wide Association studies as well that's

going to be the focus of my talk and

just to orient us you know essentially

what we're doing is thinking about an

independent variable than exploring what

phenotypes and the range of phenotypes

that are available and associated with

that really anchoring on the fact that

we have richly systematically phenotype

sets of individuals such as in the UK

biobank and other electronic health

record data sets which is where this

started usually that's based on things

like billing codes but I don't want to

limit us there you can think about

laboratory values you could think about

natural language processing and things

like that as well most of its been based

on billing codes so to start with I want

to orient us to a discovery study four

out of the electronic medical records

and genomics Network emerge in the US

and it was five sites that work together

for a carefully validated phenotype that

we used codes labs medications natural

language processing to find these cases

and manually validate who was a case for

autoimmune presumptive autoimmune

hypothyroidism and in control and we'd

identify the thyroid transcription

factor that was associated and

replicated this and then we did it took

took a variant the same variant that was

found here and did a fee wass on that

variant in a slightly larger population

and you know that was unselected our a

much larger population unselected for

any given phenotype and hypothyroidism

was the the you know the highest

associate a phenotype there but we also

had some other thyroid diseases that

came up and things like atrial flutter

as associated which we know that there's

you know hypothyroid or less likely to

manifest with atrial flutter and so so

this gives opportunity to look at the

performance of these two methods

and so you know on the left we have the

schematic of the algorithm we use and

then we use these mappings called fee

wass codes or fee codes and we usually

say there have to be two or more that

map into that fee code and you know you

can see the odds ratios are essentially

the same between the two approaches

within our population of individuals

with an emerge

you know we identified more cases and

with the fee wast codes than we did with

the algorithm and so you know there's

many approaches that fee Wasson just

talking about that you know said most

used billing codes in the u.s. that's

been historically icd-9 with the

clinical modifications and now icd-10

after 2015

there's about 65,000 icd-10-cm codes and

on the right you can see some of the

ways this works so the fee codes have

numbers that kind of look like icd-9

codes but they're actually not

and and then what we do is we group you

know like codes now across icd-9 and

icd-10 and icd-10-cm codes to you know a

given fee code and and so all the type 1

codes come together which is not obvious

from the icd-9 coding group system and

then each of those also define ranges of

control groups in addition to the feed

groupings or other groupings in the US

there's some the AHRQ has released some

software that groups things into about

300 diseases tree wasps is another thing

you can also use raw ICD codes for

instance that gives you a challenge

across of course mapping between icd-9

and icd-10 and you can do many other

things survey data has been run across

the UK biobank and other things that

I've talked about like the procedures so

here's another example fee wasps driven

by on EHR data looking at imputed HLA

types into the two and four digit types

of HLA and you can see you know quickly

it highlights the fact that there are

different associations between class one

and class two HLA alleles and what helps

you think about the range of

associations and and overall I think

there was a hundred or so significant

associations most of which were known

and a few of new ones but what's more

interesting as by doing in a single

population you can actually look across

those phenotypes

and and then look for pleiotropy and see

you know if you adjust in condition on

one and the other do you see you know

that they're that they're truly

independent associations and you can

also see we can a given HLA type that

you know one two HLA types that may put

you

similarly at risk for rheumatoid

arthritis it may had differential effect

for your risk on type one diabetes for

instance and so so that is a tool that

you know you can rapidly explore using

this kind of technique an important

aspect is validating its efficacy and so

one of the early things we did using our

icd-9 codes across the emerge was

replicating known associations in the G

wasps catalog we found eighty-six

phenotypes that were could be

represented in the electronic health

record and a number of snips about seven

hundred and fifty overall snip phenotype

pairs overall we replicated in two

hundred ten of them across the number of

disease classifications and sixty-six

percent of those for which we were

adequately powered in this population of

thirteen thousand people as well as

vying some novel associations on the top

of which we replicated it also allows us

to actually compare the effect size so

here you might you see something that

you would expect to see in that that the

effect sizes from the the\--why studies

are typically a little bit lower than

what's in the G wasps catalogue now some

of that's probably due to the winners

curse but some of it's also due to the

phenotype being not quite as accurate

and it helps you think about the ones

that where you have the most error so

the most common error and it was really

I universally type one diabetes is often

miss code in fact ninety six percent of

the time we found type one diabetics had

type two diagnosis codes and so so you

know it made it it made it and the

reverse is true fifty six percent of the

time so so it caused a lot of inaccuracy

in the type 1 diabetes phenotype and we

had trouble replicating some of those

snips and we've actually instituted

methods to fix that problem and we can

recover those associations so here's

here's a way you can use fee wasps in

concert with eg wasps so we did a GYN

emerge looking at longitudinal risk of

heart of a CO disease on a statin

and found variants that are tied to

expression of lipoprotein a were

associated with that outcome as a

longitudinal analysis and that risk is

increased for those that have you know

kind of ideal cholesterol levels at less

than 70 so we looked at a loss of this

lowest locus and you know as you'd

expect you see coronary atherosclerosis

near the top and and fortunately we see

most of the phenotypes are ones we would

expect to see which gets that the

question of if you were to target this

with a medication you know what

potential effects would you see and so

one of the things that's interesting and

wouldn't have been on a radar screen is

this point over here which is not quite

statistically significance was lung

cancer so you know this is a relatively

small population from 13,000 people as

it's explored more maybe that will turn

out to be true or not but it is a rapid

tool for highlighting especially when

you think about the scale you could buy

a bank I mentioned mapping these to

icd-10 and icd-10-cm codes just shows a

little bit of a process and the

vocabularies and systems that we used in

the process with some manual validation

it is in what we call it's still in beta

form but you can see it covers about 90%

of the billed ICD codes in the UK

biobank

and amongst the 10% that aren't there

most of those are not actual disease

codes only a small fraction those

represent true disease codes and we did

an evaluation at using our data with

icd-9 and icd-10 codes in terms of fee

wasps and you can see that the effect

sizes between this pandas population was

essentially the same for these two known

associations with that snip so I want to

give a few examples actually Kristen

showed this earlier doing a fee wass in

the UK biobank and just tons of

Association associated with atrial

fibrillation genetic risk score for afib

and and I when they condition for the

phenotypes the cardiovascular phenotypes

essentially those associations went away

and but it shows the power of a huge

population to show lots of things you

expect to see here's another one for

systolic blood pressure on a

a large G wasit was done across the

million veteran program as well as the

UK biobank and just a number number of

associations showing up with systolic

blood pressure they also did with

diastolic blood pressure and pulse

pressure to show that some of these

phenotypes overlap and you see

phenotypes that are not exactly

associated with a cardiovascular disease

in here coming out as well

endocrine being one of the more common

ones here's a resource Kristen also

talked about the sage approach - I'm use

saddle point approximation - I created

an efficient and accurate way of

calculating these kind of results at

scale to the UK biobank they have

produced a website where you can explore

those phenotypes calculated using the

same approaches for fee codes across the

UK biobank and this just shows a

particular a-fib snip in that website

and the URL there is there at the bottom

so you know we've talked about this and

and and in looking at individual

phenotypes I want to spend the last few

minutes talking about phenotypes in

clusters and how we think of them so if

you think about a Mendelian disease is

is a classic example that are often

syndromic presenting with many different

features and those features you know are

what we may bill in the electronic

medical record as physicians but it

doesn't necessarily represent you you

know the the disease is not always

recognized or you know may be recognized

later into the disease course as we you

know heard about earlier with

hemochromatosis and so through the

online Mendelian inheritance and man

resource and the linked human type human

phenotype ontology you know we can go

from a Mendelian disease to a list of

features of that disease which have a

vocabulary behind them and then so so

our lab mapped those hpo features to fee

codes so basically allowing you to

translate oh men features into EHR

phenotypes and then similar to a

polygenic risk or you know creating a

phenotype risk or that that looks

similar in process so aggregating

Phoenix

by their weights I'm to produce the

score for individual and essentially you

can crank this out across anything for

which you have a map and do it at scale

so let's look at the Cystic Fibrosis a

number of features from omen and each of

those is mapped to a human phenotype

ontology code and so I'm using our fee

code ontology of around 1,800 phenotypes

and you can map the ones that line up

fairly well to the CF they're not all

exact matches some are better matches

and others and then some that we don't

have in the HR which you know we're

familiar with and so let's play that out

on a couple individuals hypothetical

different conditions I mentioned they're

weighted so features like bronchiectasis

have a higher weight than features like

asthma and so when you go across this

you know individuals get a different

score and what you find is you can

separate case in the controls 460

fibrosis

you know just using the features of

disease so we're not using the disease

label but in this example we use

manually validated at cases versus

controls who don't have any evidence of

the disease in the in the text record

and we see a very significant result and

we've actually done this for 15 other

diseases now and in in every case except

for one we've seen very strong

separation between cases and controls

the one exception is phenylketonuria

which as you know in the u.s. is on

essentially every newborn screening test

and if you avoid and feel an exposure

you don't actually see the you know the

manifestations of the disease so it sort

of gives you a test of the effectiveness

and newborn screening in removing the

features of Z's in the population

because they generally do not have

elevated scores so we turn this on a

population of 21,000 people that had

eczema rate genotyping and looked at

6000 variants that were rare at at 1%

level or less and we found 18

significant Association most of which

were novel and we were able to change

the ACMG clinical interpretations for

eight of these variants towards likely

pathogenic or pathogenic and so so this

using our population as a paradigm that

I think can be explored with larger rich

phenotype populations such as what is in

the UK biobank so I want to end and with

represent a recognition of some of the

many people contributing to this work

and the middle row is probably the most

important row as as the folks actually

doing the work thank you very much

\[Applause\]

------------------------------------------------------------------------

# PheWAS and EHR {#sec-video2}

**Title**: Using EHR-based genomic approaches to understand the relationship between mental and physical health

**Presenter(s)**: Dr. Lea Davis, Vanderbilt University

Intro

**and the bio bank that's attached to them at Vanderbilt and so this opened up part**

of a newer area of investigation for me something that I've been interested in a long time but but hadn't had the

resources to investigate and so that's basically using EHR based genomic

approaches to try to better understand the relationship between mental health and physical health so that's kind of

the story that I'm going to be talking to you about today so if I can advance my slides oh there we go okay so so this

relationship between mental health and physical health is kind of well-known

that it's important and in one of the I think nice summary statements about the

importance of this comes from the World Health Organization on their website they state that there is no health without mental health and it's been

documented for some time that poor mental health is a risk factor for a number of physical conditions particularly chronic conditions and so

people with severe psychiatric illness are at much higher risk of experiencing

chronic physical conditions and vice versa people with chronic physical conditions

are also at risk of defining mental health and so I I've actually come to

People with severe psychiatric illness experience health disparities

think about this much like a disparity in fact that people with severe

psychiatric illness experience these kinds of healthcare disparities there we

go so this is really sort of punctuated by the observation that the lifespan for

people with severe psychiatric illness or schizophrenia bipolar disorder but also neuropsychiatric disorders like

autism spectrum disorders and cognitive impairment on average the life span is

shortened by 10 to 12 and a half years and it's thought that this is due to

many possible reasons one being a difference in the access to health care

so related to employment and insurance issues here in the states and that's

certainly been shown to play a role particularly for for psychiatric illnesses such as

schizophrenia where there's a much higher rate of homelessness but also

issues related to difficulty in expressing pain and so for example for

people who are non-verbal there's also some suggestion that there may be

altered interoception or among people with neurodevelopmental disorders and so

if for anybody who's not familiar with that term interoception is kind of your internal perception of pain discomfort

hunger thirst even sort of sensing your own heartbeat and and that appears to be

somewhat altered in people with developmental disorders but it's also

certainly possible well that there's some increased genetic or biological risk that's actually related to the

genetic risk for for the psychiatric disorder itself so imagine sort of

pleiotropic mechanisms and then of course increase exposure to environmental risk factors poor diet

lack of shelter lack of access to medication and those kinds of things and

I think it's also important to highlight that this is really a an under studied area for a couple of primary reasons

actually one is that until fairly recently a lot of people with severe psychiatric illness and

neurodevelopmental disorders received most of their health care in an institutionalized setting whether that

was a psychiatric institution or or prison or some other kind of group

institutionalized setting that was separated from community-based health

care and so there's this population has been under studied in the kind of

epidemiological and community-based studies and then there's also as I'm

sure everybody on this call is probably well aware has been kind of a historical

separation between psychiatry and the rest of Medicine and so it ends up being functional separation

so that refers to the fact that you know

often the mental health facilities are separate from the primary care

facilities and you might have like a community-based mental health clinic and a community-based primary care clinic

and even at a hospital there's often a separate psychiatric hospital you know

in a completely different building from the the general hospital so there's this

functional and cultural and financial separation of psychiatry and medicine

that I think has also contributed to this kind of healthcare disparity and in

the lack of research about it okay and

so like many of us in this field this is also a very an area of personal interest

for me so that this is a picture of me with my son Dylan Dylan is 24 years old

he has autism and severe cognitive impairment he's nonverbal he requires

24-hour support staff and he lives in a community integrated group home and so

even with a whole team of people who are therapists on Dylan's health we still

really struggle to get proper healthcare for him and so you know this is I think

an issue that it's important to me personally as it is to many families of

individuals with developmental disorders and our psychiatric disorders who are

getting older and aging and we don't really know what chronic health

conditions they may be at risk for so

while on the whole this has been a you know an under studied area there have

been a few kind of areas of focus particularly related to cardiovascular

and cerebrovascular disease and in recent years there have been some really

large meta analyses looking at the prevalence and cumulative incidence of

cardiovascular disease in people with severe psychiatric illness primarily

bipolar disorder schizophrenia and major depression and so this is actually a

recent paper with the meta-analysis of I think 92 studies the had really

impressive sample size so over three million patients with one of those three

disorders and and over a hundred million controls where they investigated the the

increased risk of both cardiovascular disease and cerebrovascular disease in these populations and so basically the

the take-home is that there is significantly increased risk for both disorder for both circular and

cardiovascular disease and that this risk persists even after adjusting for

some of the health behaviors but that may be related to to the incidence of

disease so things like smoking or poor diet or BMI and so based on these kinds

of epidemiological studies that are just now starting to be published we also are

interested in asking the same kinds of questions and better understanding the

relationship between men of mental and physical health using our EHR so in

particular we want to understand if there is some shared biology if it's the

case that for mental health causes for physical health or that for physical health causes for mental health and

really our model is that it's it's gonna be all of the above but understanding

you know what the primary risk factors are for for each type of chronic disease and

where their shared biology yeah I think this is going to be an important area of research another question that we have

is do these relationships between mental health and physical health transcends our diagnostic boundaries is it really

because that it's just people with severe psychiatric illness or diagnoseable more developmental

disorders that are who are at risk or is it the case that risk is continuous and

and that actually across the entire spectrum of genetic risk for these traits that there's also increased risk

for these chronic health conditions and

then are there particular health conditions for which people with developmental disabilities and severe

psychiatric illness are at high risk there's been a lot of work done on cardiovascular disease and cerebral

vascular disease but and and I think actually a fair bit of work on type 2

diabetes but outside of those kind of primary chronic health conditions um there's there's really not been very

much and so we really want to look genome-wide at these relationships and

then finally the wanna see if this can help us understand the best point of

intervention and and of course to identify any interventions that are typically used in healthy populations

that that may cause particular problems in patients with severe psychiatric

illness or developmental disorders so

A few of our research questions

maybe I'll pause there for just a minute if there are any questions

and don't forget you to unmute yourself if you have one and I mean one short

question - to move to the first slide so this is the shortening of the lifespan

for severe psychiatric diseases and so

the one thing that that was from my point was not missing on this list is

like suicide like big toe to to the psychiatric disease here is not the main

no it isn't actually um it's definitely a contributory cause but but actually

the primary causes chronic health condition so suicide is definitely a

increased risk in this population absolutely um but it doesn't account so

you mentioned already this like like being facilities being also prisons etc

so and is there other comparisons

between for example also like the US and Europe are the comparisons between

Caucasians and other industries have

honestly the literature is pretty scant so I'm I'm not sure if that's really

been investigated sort of systematically and with sufficient sample size um it

really hasn't been I think most of the papers looking at that these

associations have been published you know from the 90s and on so there's not

the kind of you know eighty year body of literature like there is for you know

cardiovascular disease and in healthy populations so I guess that yeah the

short answer is I'm I'm not sure that it's really been systematically investigated

two others have some questions if not

they said they'd like two or three on the car that are not muted so if again

so this is question if you own the car not muted please mute yourselves like right contractors um so we believe that

EHR based genomic approaches are actually a great way of investigating

several of these questions that we have so the EHR the electronic health record

allows us to investigate the relationships between phenotypes and the

biobank that we have also facilitates investigation of the genetic relationships between theatre phenotypes

and so we can then also compare the the genetic correlations to the phenotypic correlation better understand you know

what is there where there are the environmental risk factors that contribute to the matific relationships

and genetic risk factors that contribute to the genetic relationships between traits and it also allows us to utilize

the polygenic architecture of these complex traits and develop quantitative models so that we're not necessarily

relying relying completely on diagnostic categories but we can look at how

genetic risk as a quantitative trait is related to risk for various phenotypes

so maybe some of you have heard me talk about the Vanderbilt EHR and the bio

bank but just in case you haven't it can be kind of thought of as three entities

we have what we refer to as the synthetic derivative which is this

de-identified and continuously updated mirror image of the EHR or EMR that as

of now has a little over 2.8 million individuals if we look across just that

set of 2.8 million individuals the median length of the EHRs is only about

a year even though the EHR has been in existence now for 20 years and part of

the reason for this is that we're a tertiary care center so we get people coming in from all over the state of Tennessee and Kentucky and

and sort of all over the southeast particularly you know if they have a

they if they have the need to come see a special specialist in a specialty care

clinic um and so we end up drawing pretty sick people from all across the

state so in comparison to like the UK biobank that has maybe a healthier on

average population I think at Vanderbilt we have a sicker on average population

that said there is also a population of people who make Vanderbilt their medical

home so to speak and they get most of their primary care at the end result as

well and so our biobank which consists

of DNA samples that have been collected from just routine clinical blood draws

is enriched for that population of people but that actually make Vanderbilt

their medical home and so this is illustrated by the median length of the

EHR or bioview subjects which is about 10 years and so we have now somewhere

around 270,000 DNA samples that have been collected and a little over 50,000

of those subjects have been genotyped with some kind of G loss platform and on

average the age of those subjects is around 58 years old but we are trying to

genotype and accumulate pediatric care goals of all so any quick questions

about the structure of the bioview biobank or EHR that's central to the

rest of the so this is like all different kinds of diseases right all

cases of these are they are also healthy subjects here or is this all cases well

I mean yeah so there are people who don't have you know chronic diseases

certainly there are people who come in who's just for you know routine health care and they're in in the bio bank as well so

there's no ascertain meant at the bio bank level that said there it's a it's a

hospital ascertained population so so I think probably most people will be you

know a case for something okay and so how many of these \$50,000 psychiatric

patients how many of the 50,000 have

psychiatric codes I'm not exactly sure

yeah I actually don't have access yet to all 50,000 samples it's it's not an it's

not enriched for psychiatric cards that but but we've said you know the

genotyping data is actually it's a large project eventually we'll have a hundred thousand people genotype and so the the

data is coming through in waves and we actually were involved in pushing

several of the psychiatric diagnoses through but we haven't gotten those

genotype samples yet so um so they're right yeah we know where are they being

you know type here no on the mega

platform okay okay so as I mentioned

before we were very interested in taking advantage of the fact that most complex traits have a complex genetic

architecture with a measurable polygenic component and so we can look at how the

polygenic risk for all kinds of psychiatric disorders is also related to

disease status for other chronic health conditions

so I think probably everybody on the call is familiar with this approach but just in case one of the methods that

we're using is basically to calculate koala genetic risk scores for everybody in our bio Bank so using some kind of

large discovery gos and taking the effect sizes from from that gos to

create a linear weighted sum of the number of risk alleles and then looking

at how those polygenic risk scores segregate cases from controls across a

number of different phenotypes genome-wide so we started with actually

investigating this and both psychiatric disorders and in some of the previously

published chronic health conditions and so I'm starting with showing you the

coronary artery disease polygenic risk scores that are significantly associated

with the EHR definition for coronary artery disease so here we took the beta

weights from the cardiogram plus C for D consortium study that had about sixty

thousand cases and 123 thousand controls and applied it to a small subset of our

mega data target sample just to make sure that we that indeed coronary artery

disease in carefully ascertained research samples was related to the HRV

diagnosis for coronary artery disease and so this included covariates median

age across the EAP our sexes the top 10 pcs and actually it's not listed here but also genotyping batch and so you can

see that our apologetic risk score accounted for almost 3 percent of the variants and CID diagnosis within our

EHR so this was encouraging and we actually applied this model also to the

lipid traits that have been studied by the global lipids consortium so HDL LDL

triglycerides and try to also model the relationship the known relationships

between those risk factors and coronary artery disease oops

so I'm going to take just a short methodological detour here because one

of the other things that's pointer my attention in working with the Department

for biomedical informatics you know often when I present telegenic analyses

I get the question from people could be a lot of machine learning where is your

feature selection step right so in your genome-wide Association study training

the the weights but then how do you know which snips to include in your model and

the target data and typically really what we often do is just work across a

number of thresholds and you know and see how the r-squared might change if we

in just the genome-wide significant snips or you know everything at a p-value of less than 0.5 or less than

one and so kind of investigator cost different thresholds but we wondered really how well we would do if we

actually took a training set and use that to select a threshold for inclusion

for including snips in the apologetic risk score and then applied it to a validation set because this is actually

I think you know ultimately what we would do if we were they're really just

kind of out of curiosity we you know look at that now but we have our

Lipid traits and Coronary Artery Disease

discovery gos phenotype in the first column and the target phenotype and the

second column so the discovery gos either as the global the potential for the HDL LDL and triglycerides and you

can see my

this one person that is not muted or if it's Yulia sure no it's not me I heard

it too okay if we see you know me to see

your cars is okay all right great and so right so for for these lipid trades this was people and then our

target phenotypes were all measured in the bio Bank and so we we set up a

training sample of about 9,000 people and a validation sample oh yeah alright

so we had a training sample of about 9,000 people a validation sample of about 16,000 people and and so you can

see the key value threshold that was the best fit and the training sample the number of snips that was included in

that best friend we're truly r-squared value or the proportion of variance I have to say and and the p-value for that

because I have no experience so I'm not

sure I mean they they seem to be like they don't have the speaker on kro they don't even listen so it's a little bit

annoying so please everybody again here so if you hear us please mute yourself

or just if you don't listen if you're not listening just leave the call that's

fine as well okay so we used the same in

fresh hold that was identified in the training symbol to define the PRS in the

validation sample there's a you'll notice there's a different number of snips included and we think that this is

because the training sample was a me platform and the validation sample was

the mega platform even though they were both imputed to the same reference panel

and they're both European populations the mega sample seems to have better

overlap with the original discover AG loss and so the are squirts for you know

both the training and validation sample particularly for the lipid traits are really pretty impressive and and

actually start to approach the Senate based heritability or in particular for

HDL and so I put an asterisk here for

the p-value fold in the training sample was actually the same as the best fit

p-value threshold in the validation sample so I can pause there for any

question we were actually kind of surprised that there were any traits where the best of it men told was the

same across multiple samples and so I think this is actually really

interesting and potentially indicating that we're starting to approach kind of narcs and them information or an LDL in

our jaws

right in the interest of time I'll just move on so we took the same best-fit

thresholds and genetics course you're

looking at here is a Manhattan plot from us yaws where along the x-axis we have

kind of classified that the phenotypes and along the y-axis the minus log time

that the p-value for the PRS twin direction of the arrow indicates the

direction of risk so if there were higher polygenic risk scores among cases

this up arrow control to see a Down and so this view else was for LDL sorry for

HDL and so we see a protective effect of yell on type 2 diabetes and this has

been previously shown this was a good concept and we did a similar type of

analysis for our LDL polygenic risk score where again we see a strong

association with this lipid yneouws and

with coronary artery disease and so these are also known associations

interestingly so I don't have it here and in this presentation but but there's

a the phenotypic correlations between a chair between LDL and the diseases

tested in our genome wide analysis were much stronger so we saw lots of

different phenotypes associated with the measured LDL but these are the only phenotypes that are associated with

genetically predicted LDL so getting

into our primary interest which is the genetic risk for psychiatric disorders

associated with diseases across the phenome we asked

whether genetic risk for MDD was associated with heart disease codes and and actually all phenome wide codes and

so he took the most recent 2018 DD

meta-analysis results that are posted on the website and did the same kind of thing where we calculated a genetic risk

score and all of our 16,000 people in the mega example and we see really

strong associations with mood disorders and depression which we expect associations with bipolar disorder and

anxiety disorders but then we also see some of these party avascular rates

rising in significance as well and so we see an association with nonspecific

chest pain which is really kind of a catch-all you know as it states

nonspecific code but was definitely interesting to us and so we wanted to

kind of investigate this a little further and we asked whether we saw some

Do we see similar associations for related mental health traits?

similar associations for related mental health traits so related to major depression is also individual's

perception of loneliness and this is something that we've been working on with a consortium group now a Palmer

Dorit blooms fund and myself and others would call ourselves the lonely

consortium and we've amassed almost 500,000 samples and I think those of you on the MDD call have heard some reports

of this already and so we looked at the relationship between polygenic risk scores for loneliness and phenom wide

associations as well and so this is our Manhattan plot for our loneliness

seawalker sorry gee wasps and you can see also that we are observing some

enrichment of gene expression for our loneliness low sigh within

tissues that we expect to see some enrichment so this is in brain tissues in particular and we also have observed

several genetic correlations and I just have a few posted here but genetic correlations with phenotypes that are

associated with poor mental health including general tiredness lower

self-rated health and coronary artery disease and one of the reasons that we

were really interested in and actually looking at the genetic relationship between loneliness in these other traits

is that loneliness has in and of itself and identified as a risk factor for

increased morbidity and mortality and and so there have been several kind of

epidemiological studies looking at the temporal relationship between a person's self-reported loneliness and later

health consequences and so often the causal mechanism is inferred from that

temporal relationship but I think that's a little tricky because of course you know by the time I'm somebody actually

has a heart attack or has diagnosis of coronary artery disease that that

disease has been developing for for many years and so having that temporal

relationship is not always an indicator of a cause-and-effect relationship so a

graduate student in my lab julia c-loc has kind of led the effort on this work

well she looked at again the innate propensity to loneliness apologetic risk scores for loneliness to see whether

they were associated also with poor health outcomes and so in this case because we didn't have loneliness measured anywhere in our biobank we

weren't able to do this kind of foot training on a separate sample for a best fit so we just took everything at a

p-value of less than one and even with that we actually see a strong association with mood disorders and

depression with tobacco use disorders but then also you can see with a whole host of coronary artery disease type

phenotypes and so this is also getting at that question that we had about the relationship between sort

of dimensional traits and and diagnosis itself so again these are not

necessarily so the sample is not enriched for major depression and we're

not looking at the diagnosis of major depression our diagnosis of you know chronic loneliness or we're just looking

at the genetic risk factors so we

BMI and MDD diagnosis influence associations

actually one of the I think benefits of having this type of data as opposed to just looking at genetic correlations

with summary stats is that we can do a lot of conditional and sensitivity analyses to try to tease out some of the

relationships and so we looked at how

BMI and diagnosis of major depression might influence associations and so when

we adjust for BMI definitely we see an attenuation of the signal for coronary

artery disease although we do still see some of these phenotypes rising above

you know might significance and we also see again a strong association with the

you know disorders depression and tobacco use disorders when we adjust for the diagnosis of major depression we we

still observe a significant association with obesity and and even though our

coronary artery disease codes kind of fall below phenome wide significance they're still of course enriched among

our results and so this was actually a really important analysis because it's

also known well-known that after having a coronary event people become much more

susceptible to a major depression episode and so so he wanted to make sure

that our associations weren't completely driven by the major depression that may

be diagnosed after the fact and while we do see definitely an attenuation of the

signal I think we're think resample sighs these associations will probably still remain

phenome aren't significant we also were interested in seeing if

Differences between males and females?

there was a difference between males and females and so we stratified our few us

sample and and looked separately and while we do see some qualitative

differences so in females definitely the depression and mood disorder codes remain phenome like significant and in

males the mi and atherosclerosis codes

remain feel might significant these differences between them were actually not statistically significant so I think

this is really just reflecting the fact that more males have myocardial infarctions and more females

are diagnosed with depression Stephan

how am i doing on time good 10 to 12

minutes okay all right okay great so so

there's since we were really primarily interested in following up the associations between polygenic risk for

loneliness and the coronary artery disease codes we we focused in on the males and and looked there at

Association after adjusting for again MDD or BMI and and so again we see that

myocardial infarction remains significantly associated after adjusting

for either BMI or MDD and actually also remains associated after adjusting for

both MDD and BMI and so this now I think

is a really nice substrate for a

Mendelian randomization analysis as well

Part 2: Strategy to screen for biomarkers in lab data

so the the second part here is is actually just kind of an introduction to some of this the work that we have

planned and so I just wanted to sort of briefly go over it and and invite any

ideas or collaborations of people are particularly interested in certain

biomarkers for phenotypes that they that they're studying so within the EHR we

have access also to actually thousands of labs but many of those are kind of

unique special snowflakes so they may only have been ordered on you know a small handful of patients so when we

start looking at labs that have a larger sample size it turns out that we have

about 350 labs with over a thousand individuals and actually I should say

that all of these labs have at least a thousand observations so if we say that

you know we've got 350 labs with at least a thousand individuals at least a

thousand observations then everybody has been measured at least once but then

we've got a larger number of labs where we have a smaller number of individuals but a larger number of observations per

individual and so this data is also you know really rich for looking at

longitudinal associations between the relationship with you know psychiatric

illness and and changes over time in various biomarkers so in total we've got

about 500 labs with at least a hundred people measured at least a thousand

observations so this work has sorry this

Clinical lab data

data source really has not been utilized very much in the EHR space partially

because it's really messy data so it's taken us close to two years actually to to really carefully QC all of the the

labs that that had sufficient sample size it's

also challenging because again it's a hospital population and so you know a lot of times the the labs are being

drawn because somebody is actually sick and so this can be a challenge to interpretation but at the same time the

fact that we have the entire EHR allows us to investigate the relationship between diagnosis and and changes in lab

values some of the benefits of using this data are that we have a really

large sample size that is really rich for clinical data so it's longitudinal

we've got over 20 years worth of data and we can as I mentioned a couple of

times now test the effect of many possible mediating and moderating variables and then we can also go into

the charts themselves and validate by chart with you so developing tools to

both QC and visualize this data has been a really tremendous effort by Peter

Straub a programmer in my lab so he's developed a shiny app and a whole set of

tools that actually we're planning on making publicly available and and will

be allowed we'll be able to make the summary data for all of these different labs available so that we can look at

how they vary by age by sex by by race

and and that groups with bio banks and

and labs you know locally can also download these tools and and apply them to their data

and this is sort of the concept map for what we would eventually like to do so

this is you know that we would take the you know beta values from geo so many

psychiatric illnesses calculate polygenic risk scores across everybody

and bio view and then look at how those risk scores are related to median values

in hundreds of routinely collected labs and as I mentioned before we can

so use longitudinal models to see how they're related to change in lab values

over time so we've started doing this a

little bit for some of these traits that I've been talking about so far so this

is just focusing in again on HDL LDL and triglycerides and looking at how the

genetic risk scores for coronary artery disease loneliness or major depression

is associated with median lab values and

so in each of these plots we're looking at the r-squared values the proportion

of variance explained in HDL LDL and triglycerides on the y axis and then the

Discovery gos p-value threshold on the x axis so that we can see kind of across

the board how well do you know MDD risk scores CA D risk scores and loneliness

risk scores predict HDL LDL and triglycerides and so interestingly we

see that for HDL our loneliness risk scores actually tend to outperform our

CA D risk scores and for LDL you know it

looks like there's some difference but I don't think these are actually really meaningful because below 0.1% variance

it's not not a significant association with the between the risk score it's

often and the median value and then in our triglyceride analysis we do see the

the CAG risk score outperforming the loneliness risk score which is kind of what we would think about maybe

intuitively expect but the loneliness risk score does actually significantly

predict triglyceride levels as well

so like I've said a couple of times we

also are interested in testing for the mediating effect of these quantitative

traits and the moderating effects of sex and medications and and other diagnoses

on these relationships as well so this is kind of our an example of a general

model that we're interested in testing the relationship between polygenic viability and disease diagnosis that may

be again mediated through quantitative traits so kind of our future interests

are to mine this lab data for biomarkers for our psychiatric disorders using genetic risk scores from them are

publicly available jaws and I should say that you know we don't think it's likely

that we will identify a biomarker for you know major depression out of all of

these routinely collected labs but that actually we may identify by several quantitative traits that together may be

predictive of an individual's genetic twist or form a major depression or

schizophrenia for example we also are really excited

about using bi-directional Mendelian randomization analyses to better understand some of these possible causal

mechanisms and to compare the phenotypic correlations with the genetic

correlations to try to identify comorbidities that may actually be more of a consequence of environment then of

genetic causes which you know we're also equally interested in understanding so I

Acknowledgements

think that is it I'll wrap up with an acknowledgement of everybody in my lab a

really great group of people to work with and wonderful students and the work

of the lonely consortium again this has been like a just a phenomenal

collaboration that I think has yielded some really interesting results and so I

think with that all and and hopefully there's some time the questions thanks so much the other

fabulous presentations really it's a lot of data that's coming - therefore - it's

really exciting to see that you're diving into that with full speed here um

so is there I mean we have like around 30 people on the call we probably have

questions and they don't know how to unmute themselves or I still have

questions but I don't want to always step in did I hear somebody

okay so then here I still have likes like two or three night more short

questions I'm on the technical side so so two things here so first of all and these are you a because you're speaking

about to do know stuff but also is there

chance to actually give them the fact you might have seen some people with

especially high score screen your result is their chance to get in contact with

them again on the next visit or so or is this something that's so enormous there's no chance to to actually get in

contact with these individuals no they're yeah it's it's a completely de-identified data set yeah there is

also a what's called the research derivative so we work with in the synthetic derivative because we're also

working with the genetic data if we if we were to restrict ourselves just to

the phenotype associations we could apply for access to the research derivative which is an identified data

set but even then I think there are rules I don't know that that we could

actually recontact patients and and there's no although I don't understand

it why this is the case you can't work with genetic data in the identified

environment well I think most of the bio

banks within the US are de-identified that that issue about not working with

genetic data in the context of an identified environment that might be specific to Vanderbilt I'm not actually

sure what the what the you know rules around that where those

came from very interesting and a little

bit sad of course because that would be the and so the other issue a little bit

more technical side I mean like special project with scores are actively sensitive to so how is there

any chance for you or how do you want these dealing with this issue like yeah

you know that it's a good question we've been thinking about this a lot so so

there's a method I can't remember now the first author's name but I think it came from from Peter Fisher's group to

in the context that we're using where we have some individual level data and geo

summary statistics they've they've published an approach to actually first

just check to see if there is yeah what the probability of overlap actually is

everything up until now have to be honest is based on just the knowledge

that no investigators have contributed actively from Vanderbilt to these G wast

but I mean we haven't yet done the careful checking to make sure that

there's no sample overlap but that is on our list yeah it's it's really something

that I think we have to just actually baked into the pipeline and and we're

also part of a collaboration with other

bio bank sites through the emerge Network where we're doing a similar kind

of thing you know replicating these polygenic few associations in other bio

banks and so so we're really trying to develop a you know a robust pipeline for

for doing that across several EHR environments and and one of the things

we need to make sure to build into the pipeline is this kind of checking

especially when did you see it's good for for just testing reasons doesn't try to

see if we see some unexpected overlap there and we don't and this gives you a

little bit more security that yes like 200 cases coming up that we shared from

somebody else to PTC something that you need to be super careful at least did

you see everything in place we could test that we couldn't test done even without you know we have to check some

method where we can without sharing the genotype so it's something that could be

done on a relatively low level so so in

a genome-wide type of analysis how would you then basically like for each fee

wass category conduct a G was\--and check all of those some stat relationships

it's also possible that we have say somebody with schizophrenia who doesn't

actually have the diagnosis in our electronic health record right they have

a diagnosis of type 2 diabetes in our health record but maybe they've been inclusive in a geo somewhere else so so

I think you know I mean first we like

the check on the summary statistics level works for like a significant

overlap for for if you have the same field I think you're raising exactly

right questions that I think if you want to be really sure you need to really

test it on a genotype and the burning at

least you can do this like with a couple of these consortiums you at least

meaning if there's something unexpected or something expected and I think this gives already some support here and

maybe when you actually see the summary statistics level when you see actually

some significant overlap there Fineman somewhere then you you probably

want to go there ask these guys for can we actually test this on genotype 11 who are these individuals this is actually

correct is this I think we can test this on a couple of different levels yeah yeah that's that's a good idea

and that's basically what we wanted to start with was just to see first like do we see any evidence of overlap and then

if we do you know how should we go about dealing with it I mean to be honest the

overlap cannot be really really big because then it's so quickly we'll see

with the exploding a square values that's really impressive even if like if

you have a small data set like individuals all of these it's just 300

cases red ones are exploding so if there

is something little small thing can have

this small impact on these on these on these that seemed to fit so well right

yeah so do you have an intuition of like what you expect if there's an overlap of

say you know a dozen people you know like a really small overlap no this is

always something that I really wanted to test yeah somebody wants to go for it please no I don't have that I have these

experiences that it's like a couple hundred overlap it's really it's really

strong so much surprise surprisingly strong really that's why I'm always

careful about this it's not that you see dying are spread this is just jumping a little bit around it's really exploding

and so that's why I think my hunch is that even like it does know 20 could

have a small I mean some of the p-values I really like just passing the spin of white

trash out there right so that's why especially for these cases yeah

okay but that's the cheapest keep this ominous this discussion and I'm very

happy to it to be yeah and and so for

you know anybody on the call who's interested in looking at particular phenotypes and we've we've now see us

apologetic scores for all of the publicly available summary stats from PG

C and and so you know anybody who's like

interested in talking more about what schizophrenia looks like or what you know bipolar looks like or whatever

please feel free to get in touch with me because I would love to kind of have you know more folks to discuss these with

I'm sure the people will be more grave

okay so we are actually over the hour so I will have to close to the call now thanks so much for presenting yeah

thanks and this will continue here so

all the best
