---
title: "EWAS"
---

# Epigenome-Wide Association Studies

> In this tutorial, Drs. Wani and Katrinli demonstrate how to use the "ewastools" R package to perform QC of methylation data, and subsequent epigenome-wide association studies (EWAS). The first part of the video by Dr. Wani discusses QC of methylation data by evaluating idat file quality metrics, genotype checks, normalization, dye bias collection, sex checks, and ComBAT adjustment. In the second part of the video, Dr. Katrinli discusses the EWAS pipeline to estimate cell proportions, perform principal component analysis, smoking scores (for this specific analysis), and then finally to run the association test.

**Title**: How to Perform Epigenome Wide Association Studies

**Presenter(s)**:

-   Agaz Wani, PhD (Genomics Program, College of Public Health, University of South Florida)

-   Åžeyma Katrinli, PhD (Department of Gynecology & Obstetrics, Emory University School of Medicine)

**Level**: Intermediate

**Length**: 21:17

```{r, echo=FALSE}
library("vembedr")
embed_url("https://www.youtube.com/watch?v=uTEfOTHfHpk")
```

Link to video transcript [here](software_ewas_transcript.qmd).

Link to [ewastools GitHub](https://github.com/hhhh5/ewastools).
