---
title: "Software Tutorials: Datasets (Video Transcript)"
---

# FinnGen {#sec-video1}

**Title**: Intro to FinnGen

**Presenter(s)**: Aarno Palotie, MD, PhD (Institute for Molecular Medicine Finland (FIMM), University of Helsinki)

**Aarno Palotie:**

Good morning, good evening, good afternoon, wherever you are, you happen to be in your time zone, and thank you for attending this PGC meeting today. I'm going to tell you about three major collections in Finland that utilize some of the Finnish characteristics and special characters in the Finnish population.

So there they are: FinnGen, the Super Finland project, and the Northern Finnish Intellectual Disability Court. So, the success of genetic studies in disease genetics, especially specifically, are relying on basically four cornerstones: the population isolation, the National Health Registers, the long-standing epidemiological studies which have further developed biobanks, and then, of course, knowledge of the genome.

First, diving into what the National Registries in the Scandinavian countries are. They are ways to record the usage of the healthcare services, and these are very similar in all Scandinavian countries, not just in Finland. It means that whenever and wherever in the country you visit a hospital, your diagnosis is recorded in a central register. The same thing when you, when you purchase a prescription drug. These registers were originally developed for administrative purposes, but since they accumulate a lot of data of healthcare service usage, they become very interesting resources for research. And although they are silos as such, the typical individual identification number or social security number, whichever way you would like to look at it, then combines this data together once you get the appropriate permits.

The second aspect, the population, the history of Finland or the current population history of Finland, stems a few thousand years back where a small number of settlers were mainly living on the coastal regions of Finland, but then in the 6th Century, the Swedish King Gustav Vasa demanded Finns to move also to the East and the Northern parts of the country, which resulted in a second very strong internal migration and multiple bottlenecks. What happened then in all Nordic countries, after the 18th century, is a rapid population growth, which obviously then resulted in all the all the typical characteristics of a population isolate.

The Nationwide registers that I was describing. What is great about them is that every single individual -- every citizen, every resident -- is recorded, and they stem back decades back in a digital format. Some of them go back all the way to the 50s, but I think the most important time point is the late 1960s, when the hospital discharge data, the cause of death data, and the reimbursement data came into usage.

And this obviously provides you an opportunity to look at longitudinal data because this Health Data is available from birth to death, and this is one of the very key characteristics of this data. Even if it doesn't include some symptom-level data -- it's ICD codes, ATC codes -- but the great thing is, it's longitudinal.

And this is what the FinnGen research project is based on. It's a project that was initiated in 2017, and with the idea that it utilizes genetic strategies to understand disease mechanisms. It's aiming to collect 500,000 individuals, which is roughly 10% of the population, genotype them, and impute them against a national or population-specific deep whole genome sequence backbone and then integrating the National Health register data, which results in a dataset of 500,000 individuals with both genome and Health Data for our association analysis.

This project is a public-private collaboration -- a research project where all Finnish biobanks, which exist in all University hospitals around the country, all universities with a medical school are involved, as well as the Institute of Health and Welfare, and blood transfusion Center. Also, the Finnish Biobank Cooperative is a partner there, and currently, 13 Pharma partners are together with us working on our scientific goals.

This is a 10 year project and we have just completed year four. The sample collection will go on for still a little bit less than a couple of years, at which point we are full with our collection and then really have an opportunity to further focus on the analysis. Where are we in respect to the collection? We have currently collected 472,000 participants, which means that we are slightly ahead of schedule and seem well set to reach our goal. This includes not quite yet 200,000, but it will include some 200,000 Legacy Collections and 300,000 prospective collections. The prospective collection, what is crucial here most of them come from hospital biobanks, typically University Hospitals, which means that they are specifically enriched for diseases treated in these places. So, if you compare to the UK Biobank, this is a big difference in the content of the sample.

We produce data sets every six months. The current data freeze will consist of 350,000 individuals and more than 4,000 endpoints. All of these pheWAS analyses done from these data freezes are made public after 12 months once they become ready.

If you look at the type of phenotypes that we have, then the special thing is, again, that the age is relatively high here, which means that there are quite a lot of healthcare events already. The mean number of health events per individual is 340, including some 186 drug purchases. This means that there's quite a lot of data. And how we build these phenotype endpoints is that we combine data from different registers, whether drug usage or hospital visits, and create a strength and endpoint by this in this way. This needs quite a lot of understanding of how the data works, or the National register data, how it functions, what are the strengths, what are the drawbacks, and so forth.

And how do we then access the data? We have two levels; both of them work in the Google Cloud space. One is where we have the summary-level data, the results, in other words, from GWAS and pheWAS analyses. And then there is the secure environment, and this is something to stress in the way that those familiar with the European data regulations, which have become quite stringent. So this is coping with all those requirements and still being able -- we are able to - access the data from outside the European Union. The idea is that it's a secure environment where you cannot take data out; you can analyze it, you can look at it, but you cannot copy it out from there or download it to your own computer.

What has been characteristic in analyzing this data is that there are a large number of Finnish-specific variants that even in very extensively studied, genetically very extensively studied, diseases, like type 2 diabetes. All the ones with stars here marked are Finnish-enriched alleles and Finnish-specific hits. And this goes through, this is the characteristics through all traits that we actually have studied. There are typically Finnish-enriched variants.

The other way you can look at it, since we have the medication data from all individuals and the entire country, so you can ask questions the other way around: What about those who use medicines? Do we see variants enriched that actually give additional information for us, not just from the basic diagnosis? And this seems to be the case, at least in some of the cardiovascular study medications and those related medication GWASs.

So, when we look at the numbers of mental health diagnoses in FinnGen, out of the 321,000 individuals that were in the data freeze, any ICD codes for mental health were observed in 76,000 individuals, but even more individuals had depression medication, almost 8,000 more. Depression diagnosis was then seen in 33,000 individuals, schizophrenia in 11,000, sorry, schizophrenia, schizotypal and delusional disorders in 11,600, schizophrenia in 6,000, and bipolar disorder in almost 6,000 individuals as well.

A place where you can look into the distribution of diagnoses and how they have been constructed from the register data is called Risteys. If you go to the FinnGen website, you can be guided to this research website, which is open for everyone. You can see various age distributions of the individuals, years of first diagnosis, and comorbidities. It's quite a helpful site to get an understanding of how our endpoints are constructed.

Then we move to the second collection, which is the Super Finland collection. This has been critically supported by the Stanley Center, a collection that was done over the years 2016 to 2018. It consists of a little bit more than 10,000 patients with psychotic disorders. All samples have been GWASed and exome sequenced, a little bit less than 900 also have whole genome sequencing. Almost 90 percent of the individuals have consented for IPS research and can be re-contacted, and as of now, a little bit more than 5,200 PBMC lines have been collected for IPS cell generation. And this map is showing the distribution from where the patients are, which means that they pretty well represent the population density also in the country.

As you can imagine, there's a huge number of hard-working people behind it. They don't all fit here in the slide, but they worked very hard over the years to first collect them, and now later to analyze them.

The diagnostic groups in SUPER are such that a little bit more than half of them, a little bit less than 6,000 have a schizophrenia diagnosis. 2,600 have bipolar disorder, and the next largest group being schizoaffective disorders, psychotic depression, and other psychoses.

From the recruited individuals, we can clearly see that, as expected, the education completion is lower than in the general population. They are less married than non-psychotic individuals in the population and so forth. So, we have an understanding that quite a lot of these people are chronically ill and represent that type of psychosis spectrum.

We also have collected questionnaire and cognitive test data at the time of collection, so we don't have longitudinal cognitive data, just that the recruitment. We have the medical care register data and also the blood plasma stored.

Then, the third collection and the last one is the Northern Finnish Intellectual Disability Cohort. We currently have a little bit more than 3,000 participants. They are all collected from the northern regions, and if you remember when I was describing the late settlement movement, which happened after the 16th century, so these were areas that were very sparsely populated in small villages. We can see from the frequency, that the prevalence of both intellectual disability and schizophrenia are higher in these areas, and that was one of the reasons that stimulated us to collect individuals that don't have a clear ID diagnosis from this area currently. Currently, most of these have already been exome sequenced and GWAS analyzed. For instance, just for interest, 56 SCHEMA gene variants are observed in 80 carriers. It's kind of interesting that since we are dealing with a majority of mild ID cases, which is slightly unusual for this type of cohort. It's interesting to compare now the variant carriers specifically in this group and then in the SUPER Psychosis Study. Indeed, the same variants exist as well as the same genes having different variants in these two collections.

And I think everyone for listening and hope that these cohorts provide an opportunity for good harvest. Thank you.

------------------------------------------------------------------------

# PsychENCODE {#sec-video2}

**Title**: Introduction to PsychENCODE

**Presenter(s)**: Chunyu Liu, PhD (Department of Psychiatry, SUNY Upstate Medical University)

**Chunyu Liu:**

Hello. I'm Chunyu Liu at SUNY Upstate Medical University at Syracuse. I'm glad to be here to introduce the PsychENCODE project and explain how PsychENCODE data might be useful for your study of psychiatric genetics.

So, the PsychENCODE Consortium was established in 2014, funded by NIMH. Now, it has grown into a big Consortium with 47 grants supporting 12 institutes and more than 100 scientists. The Consortium focuses on human brain, and we do have individual projects covering non-human primates, and even mice, but the majority of the studies are on humans. We cover several major psychiatric disorders, including schizophrenia, bipolar disorder, autism, PTSD, and major depression. But a big chunk of the brain samples actually come from psychiatrically normal people. We also cover a wide range of life stages from prenatal fetal tissue, developmental early life, and adult brain.

The major features of PsychENCODE which make it different from other Consortia like ENCODE and GTEx are the following four points:

1\. Genetic variation: We use population samples and have hundreds, even thousands, of brain assays sequenced. With that, we can study genetic variation for its contribution or regulatory role in different omics.

2\. Brain focus: We focus on the brain over other organ tissues. The major brain region we study is frontal cortex. We do have a few studies that cover other brain regions, but frontal cortex is a major brain region we focus on.

3\. Major diseases: As I already said, schizophrenia, bipolar disorder, autism, major depression, and PTSD are the major diseases we study.

4\. Different omics: We have several different omics covered by different projects. For example, we have chromatin data, RNA-seq data, and Hi-C data.

In 2019, the Consortium had the first wave of publications. Eleven papers were published in the Science family journals. Since the papers have been published, I will not spend time on the major results of those studies.

I just want to describe how you can access the data. All the PsychENCODE produced data have been deposited into Synapse, which is managed by Mette Peters and Kelsey Montegomery. So, this is the website they created as a public portal where you can browse the data and request access to all the data produced so far. It's released to the public domain.

There are already more than 200 terabytes of data produced by the early phases of the PsychENCODE project. As you can see here, the biggest datasets are actually RNAseq data, followed by ChIP-seq data and ATAC-seq data. The other data types are relatively smaller, like Hi-C, NoMeSeq bisulfite sequencing, and so on.

A web interface or service that the Consortium is creating is this PsychSCREEN. This is a project led by Dr. Zhiping Wong group. They are creating a website for you to query PsychENCODE's analytical results. You can query the information by genes, regulatory elements, or genetic variants. The genetic environment is a simple path. We will talk a little more about how they relate to PGC.

The major thing I want to deliver today is how PsychENCODE actually can be connected to PGC. So, this blue box shows you that SNP associations with different disorders are actually the major products from PGC -- those GWAS results. The green oval actually covers the major product from psychENCODE. It contains all the brain omics for gene expression, for epigenetic measures, and protein abundance. With that, you can calculate QTLs, including expression QTLs (eQTLs), which basically associate SNPs with gene expression. You can also do QTLs for the other omics. Slowly, you can do differential analysis -- differential expression, for example. Other than that, you can also perform Mendelian randomization analysis, for example, to study the causal relationship between SNP and disease.

I want to spend a few minutes talking about the project in my own lab that is called BrainGVEX. It's one of the dozens of projects funded by PsychENCODE. My first collaborators on this project are Kevin White at Chicago and Junmin Peng from Children's Hospital. My most recent project is collaborating with Stella Dracheva from Mount Sinai and Eran Mukamel from UCSD. We also have peripheral collaborations from Wang lab at the University of Texas, and Central South University from China, working on data mining and analyzing the data we have generated so far. I want to especially thank the program officers Gita and Alexander, who have provided us with continued support.

From this BrainGVEX study, we have generated genotype data -- either by the array or low-pass whole genome sequencing -- on more than 400 brains. The majority or most of them have been sequenced for transcriptome using RNA-seq. We also have a good proportion of samples with ATAC-seq, riboseq, and proteomics by mass spectrometry.

So, through that, you can see that this really covers the major components of the central dogma, from genetic variants to gene expression, transcription, translation -- translating the mRNA binding to the ribosome, then translating into protein.

We use this multiomics data, along with ChIP-seq data, and we can perform QTL mapping. We can also build co-expression networks. With this information, we can further extend to connect to eQTLs from PGC. The goal is that we want to use all the information to understand and explain the GWAS signals, also trying to capture some causal relationships which are not obvious by just looking at the GWAS signals alone. On top of that, we can also perform something like TWAS analysis so we can identify normal risk changes in the pathway and further build a prediction model.

I want to share a few unpublished results. The first one is the QTL mapping of multiomics data. This is a study done by a former PhD student, Jiang Yi. So, he was analyzing our BrainGVEX data, which contains RNAseq, Riboseq, mass spec, and ATAC-seq from 200 to 400 brains.

The result shows different QTLs. Actually, can produce different, I mean, different omics can give you a different amount of QTLs. Like this panel A shows you eQTLs, splicing QTLs, has many more QTLs than riboQTLs, and protein QTLs. And if you look at the effect size of different QTLs, you can see that eQTLs have a stronger effect size than ribosome QTLs and protein QTLs. I think that's easy to appreciate because of gene expression, ribosome binding is closer to genetic environments than protein, as it involves more steps in the central dogma. They do explain a different level of heritability from the GWAS results as well.

When you compare the different types of QTLs obtained from different omics, you will see that for those SNPs associated with the same gene pair at different omics, they have some good consistency. This is reflected in the positive correlation diagonal. But at the same time, you do see some interesting pairs going in the opposite direction. For example, in this panel comparing eQTL with ribosome QTL, we do have a small number of SNP and gene pairs that show a negative correlation. The same scenario can be observed when comparing ribosome QTL to protein or gene expression to protein.

At the same time realize that many gaps remain to be filled by the PsychENCODE Consortium or the data analysis, particularly from the eQTL perspective. So, we still did not cover well for early development. We do not have good coverage for diverse populations, and we still need to cover in detail cell types.

We have several ongoing projects, and I'm going to describe one right now. Again, this is unpublished data, a collaboration between my lab and Michael Gandal at UCLA. This slide, prepared by Michael's student Cindy Wen, shows the scene we are looking at. We are generating eQTL isoforms, splicing QTLs from GTEx, on fetal brain tissue from nearly 700 samples covering three different trimesters of three population sources. This data actually shows you that the fetal brain does capture some unique early developmental QTLs. You can use this information to study and analyze the GWAS signal, and you see, they do enrich in the GWAS signal as well, as you observe in adult brain QTLs.

That's a study on fetal brain. We already have some cross-population studies, but in adult brain, the situation is also there. So, we don't have good coverage for diverse populations. The current data we generate from bulk tissue RNAseq predominantly originates from European population. We only have 18 African American brains, and that has been corrected to a degree in the single-cell RNAseq data. We have more representation from Hispanic and African American populations relative to the Caucasian sample. But so far, none of the non-European samples have been formally analyzed here.

We have a project ongoing to study East Asian, Han Chinese, brain eQTLs. This is a project with my collaborators and Chen Chao in China. These slides were prepared by our students, Chen Yu. Using 150 brain samples from China, we can identify a large amount of eQTLs and splicing QTLs, referring to thousands of genes. So, 80% of the eGenes and splicing QTLs are consistent across two populations, at the same time, it tells us that we still have a substantial amount of eQTLs that only appear in one population and not the other. And the important thing is, when we use the QTLs to explain GWAS signals, we see an interesting phenomenon. So, if you use East Asian QTLs to explain East Asian GWAS signals, you have a significantly higher proportion that can be explained, and then using the European eQTL data.

So, I said we have a gap to fill about the cell types eQTLs. That is a major task for this current phase of PsychENCODE. Actually, right now, hundreds of brains are under investigation, meaning they're being sequenced for single-cell RNAseq or ATAC-seq. We expect many major cell types will be covered.

So, stay tuned. In 2022, there should be publications coming out from our Consortium regarding cell types of eQTLs, early developmental stage QTLs, and some population-specific QTLs.

Because of the pandemic, this is our latest group picture. Several labs joined the Consortium later during the pandemic, so we don't have a group picture. As you can see, we have many senior investigators and young fellows in this picture. I believe everyone will have their own vision or interpretation of the PsychENCODE data. You definitely can approach them and listen to their explanation or presentation about PsychENCODE. Hopefully, that will be a very useful resource for you to study psychiatric genetics.

Thank you, bye.
