<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.353">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>PGC Video Textbook - Chapter 9.4: Interactions with Environmental Factors (Video Transcript)</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


<link rel="stylesheet" href="styles.css">
</head>

<body class="nav-sidebar docked nav-fixed">


<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="./index.html">
    <span class="navbar-title">PGC Video Textbook</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link active" href="./index.html" rel="" target="" aria-current="page">
 <span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./about.html" rel="" target="">
 <span class="menu-text">About</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./contact.html" rel="" target="">
 <span class="menu-text">Contact us</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="https://pgc.unc.edu/" rel="" target="">
 <span class="menu-text">PGC Website</span></a>
  </li>  
</ul>
            <div class="quarto-navbar-tools ms-auto">
</div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
      <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item">Chapter 9.4: Interactions with Environmental Factors (Video Transcript)</li></ol></nav>
      <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
      </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal sidebar-navigation docked overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
      <a href="./index.html" class="sidebar-logo-link">
      <img src="./pgc_logo_website_v3.jpeg" alt="" class="sidebar-logo py-0 d-lg-inline d-none">
      </a>
      <div class="sidebar-tools-main">
    <div class="dropdown">
      <a href="" title="" id="quarto-navigation-tool-dropdown-0" class="quarto-navigation-tool dropdown-toggle px-1" data-bs-toggle="dropdown" aria-expanded="false" aria-label=""><i class="bi bi-github"></i></a>
      <ul class="dropdown-menu" aria-labelledby="quarto-navigation-tool-dropdown-0">
          <li>
            <a class="dropdown-item sidebar-tools-main-item" href="">
            Source code
            </a>
          </li>
          <li>
            <a class="dropdown-item sidebar-tools-main-item" href="">
            Report a bug
            </a>
          </li>
      </ul>
    </div>
</div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./welcome.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Welcome to the PGC Video Textbook!</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./toc.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Table of Contents</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">
 <span class="menu-text">Chapters</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter1.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 1: Introduction</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter2.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 2: The Genome</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter3.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 3: Technologies</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter4.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 4: Study designs</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter5.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 5: GWAS analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter6.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 6: Polygenic Scores</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter7.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 7: Ancestry-Specific Analyses and Considerations</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter8.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 8: Post-GWAS bioinformatics</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter9.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 9: Advanced Topics</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter10.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 10: Other Considerations</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true">
 <span class="menu-text">Software Tutorials</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_cnvs.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">CNVs</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_conditional.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Conditional Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_crossdisorder.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Cross-disorder Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_datasets.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Datasets</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_ewas.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">EWAS</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_geneset.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Gene Set Identification</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_gwas.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">GWAS</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_genomicSEM.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Genomic SEM</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_MR.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Mendelian Randomization</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_mtag.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">MTAG</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_prs.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">PRS</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_correlation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">SNP Heritability and Genetic Correlation</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true">
 <span class="menu-text">Additional Resources</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./glossary.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Glossary</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./resources.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Tools/Software Resources</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./addreading.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Additional Reading</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="https://pgc.unc.edu/for-researchers/download-results/" class="sidebar-item-text sidebar-link">
 <span class="menu-text">PGC Summary Statistics</span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#statistics-of-the-interaction-term" id="toc-statistics-of-the-interaction-term" class="nav-link active" data-scroll-target="#statistics-of-the-interaction-term">Statistics of the Interaction Term</a>
  <ul class="collapse">
  <li><a href="#sec-video1" id="toc-sec-video1" class="nav-link" data-scroll-target="#sec-video1">Dummy variables: interaction terms explanation</a></li>
  <li><a href="#sec-video2" id="toc-sec-video2" class="nav-link" data-scroll-target="#sec-video2">Continuous variables: interaction term interpretation</a></li>
  </ul></li>
  <li><a href="#sec-video3" id="toc-sec-video3" class="nav-link" data-scroll-target="#sec-video3">Gene environment interaction in psychiatric genetics</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Chapter 9.4: Interactions with Environmental Factors (Video Transcript)</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<hr>
<section id="statistics-of-the-interaction-term" class="level1">
<h1>Statistics of the Interaction Term</h1>
<section id="sec-video1" class="level2">
<h2 class="anchored" data-anchor-id="sec-video1">Dummy variables: interaction terms explanation</h2>
<p><strong>Title</strong>: Dummy variables: interaction terms explanation</p>
<p><strong>Presenter(s)</strong>: Ben Lambert</p>
<p><strong>Ben Lambert</strong>:</p>
<p>so let’s think back to our example we had in the last video so let’s say we were interested in how wage rates varied between let’s say male and female people so the idea is that we regress wage on let’s say now that we sort of implicitly assuming that we have all these other variables so I’m not going to include them explicitly we’re just gonna have alpha plus let’s say beta 1 times the number of years of education plus beta 2 times our sort of sex variable where our sect variable takes on the value of 1 if the individual is a female and it takes on a value of naught if the individual is a male but then we included a further term which was let’s say B 2 3 where we multiplied sex times education so the idea is that we have collected all these variables across our individual or individuals or in the population or in our sample rather and we have included a multiplicative term in our regression specification so what does this multiplicative term mean what how do we interpret this beta 3 well let’s think about again what this sort of average wage rate would be for a female and compare it with the average wage rate for a male so the average wage rate for a female if they had a given number of years of education would be alpha plus beta 1 times the number of years of education which they had plus well this sector level now takes on the value of 1 so I’ve just got plus beta 2 and now our sex parable here is taking on a value of 1 as well so I’ve got plus beta 3 times the number of years of education ok and then we can sort of simplify this if we should’ve noticed that our alpha and our beta 2 of those constants here so you’re writing those both the stars of the model we get sort of alpha plus beta 2 let’s say and then we recognize that we have essentially got to education terms we’ve got this one and this one so I couldn’t simplify these as well by writing them or by combining them I just get beta 1 plus beta 3 times the number of years of education okay so that’s for the female case what do we have for the male so the idea for the sort of males in our sample is that the average wage rate is given by alpha plus beta 1 times the number of years of education because our sex pheromone takes on a value of zero so these sort of second two terms actually cancel or don’t don’t exist for the male so now we can think about what the effects of our sex variable has been on our sort of specification and our interpretation so what does b2t represent here well Pizza 2 represents the additional premium which females would have over males if they had zero years worth of education because if they had two years worth of education then both of these two terms would disappear and the only difference between males and females within factually and when it would in fact actually be our Peter 2 so just like your proves in the last video that is actually the wage premium which females do so it’s over males in the case in this case of having zero years worth of education okay so what does b23 represent well notice that the only difference between these two specifications in terms of the education variable is that essentially the partial effect of education for females has been boosted by an amount b23 relatives to the males so what does that mean well if beta 3 was greater than zero it means that the additional effect of one more year of education for females was in fact greater than that for males if it was less than zero then it would be the other way around so the additional effects of having one more year of education on average would tend to cause a smaller increase in female wage than it would do for males so we can sort of think about what these cross terms mean in our regression specification well essentially what they mean is if I’m interacting a dummy variable with a continuous variable is that it allows us to have different slopes of that particular continuous variable across the two different values which are dummy variable can take on so and that’s quite a sort of an inappropriate assumption to make in a whole host of different situations in this particular situation it kind of then you might suppose that there might be in a different effect of educate between males and females but there are a whole host of other ways in which this could be true across other sort of types of model</p>
</section>
<section id="sec-video2" class="level2">
<h2 class="anchored" data-anchor-id="sec-video2">Continuous variables: interaction term interpretation</h2>
<p><strong>Title</strong>: Continuous variables: interaction term interpretation</p>
<p><strong>Presenter(s)</strong>: Ben Lambert</p>
<p><strong>Ben Lambert</strong>:</p>
<p>hi there in this video I wanted toexplain what the sort of interpretation is when we have two continuous variables multiplied together in some sort of regression model okay so let’s think about a particular example the only deal here is that let’s say we were trying to explain a company’s level of sales but we are trying to do that in terms of let’s say the effective price and let’s say the effect of advertising yes a this might be the company’s level of advertising spend and this is just the company’s price is set for a particular product so traditional some Theory would sort of expect us to have a downward sloping demand curve so we would expect b21 to be less than zero because if you lower the price then in sales increase and we would sort of expect that if we spend more on advertising then sales tend to be higher as well so we’ve got Peter 1 being less than 0 and beta 2 being greater than zero but let’s say we included a third term here which is beta 3 and now included the product of price and the company’s spend on advertising what interpretation can we actually give to this piece of three well let’s think about this in two different situations so let’s say that the company was vet so spending a hundred thousand on advertising and let’s think about what the company sort of expected sales would be under that situation well the idea is that the company’s level of sales we would expect if advertising 100 would be equal to alpha plus beta 1 times the price plus now we’re going to get 100 beta 2 for this third term and then we’re going to get plus 100 and be two three times the price okay so what does this shown us well we can actually sort of think about the effective price because price is appearing twice in our model here we can sort of combine the price variables to create a sort of aggregate effective price so here we would have the aggregate effect the price would be beta 1 plus 100 Peter 3 yeah and then that would all be multiplied by the price so one interpretation is beta 3 and what sign would we expect Peter 3 to have in this case well we would actually respect that beta3 would be greater than zero why would we expect that well the idea here is that if you spend more money on advertising then that tends to decrease the sensitivity of your consumers to price changes in that product so notice that this appears because beta 1 is less than zero so if we’re adding 100-meter three where beats of three is greater than zero then the idea is actually we have decreased the sort of sensitivity of consumers to price changes or we should have made our customers less reactive to price changes which is something which you might expect accompanies shows to exhibit right you’ve might expect if you spend more money on advertising you increase this in a brand value or the sort of non tangible effect which consumers consider when they’re thinking about your brand so that might make them less price sensitive okay so that’s kind of what beetle 3 is representing in this case let’s think about another example whereby let’s say we had the price level be four to ten and let’s sort of say what we might predict the company sales to be and that’s that case so the idea is that the company sales on average when the price was ten would be equal to alpha plus ten beta 1 plus beta 2 times the level of advertising which we haven’t specified Plus now we can have 10 beat in three times the level of advertising so notice that again here we have two terms here which essentially have the same variable so we can combine these so we now have an aggregate effective advertising being put to beta 2 plus 10 beta 3 yeah so what does beta 3 represent in this case well remember that we found from the first example the beta 3 by theory should be greater than zero well what does it say in this case well it says if your price is higher so remember the prices are represented by this 10 here then the effect of advertising tends to be greater so that might be the case right if you have a higher to the premium price product you might have to demonstrate to consumers that it’s worthwhile buying show the effect of advertising is greater than if you have let’s say a low price product which consumers would flock towards anyway so Pizza three generally what does it mean what are we learned from these considering these two cases well it shows that the effects or price depends on the level of advertising spend and the effect of advertising tends to be determined by or tends to be affected by the level of price so beta3 is sort of a way of adjusting the effect of price and advertising to take into account their sort of multiplicative effects on one another</p>
<hr>
</section>
</section>
<section id="sec-video3" class="level1">
<h1>Gene environment interaction in psychiatric genetics</h1>
<p><strong>Title</strong>: Gene-environment interaction analysis</p>
<p><strong>Presenter(s)</strong>: Kenny Westerman, Broad Institute</p>
<p>Intro</p>
<p>uh yeah so i’m a little bit uh daunted by the the scope of presenting in general on gene environment interaction analysis because there’s so much that that can be encompassed here but i’ll do my best to provide a broad overview and happy to to go into any of the topics in more depth</p>
<p><em>Outline</em></p>
<p>all right so to start off here i’ll generally present this in three chunks here we can start with gene environment interactions in general what are they why are we interested in them and how might we go about assessing or measuring them the second a little bit of investigation into why statistical power is a particular issue for gene environment interactions or g by e and talk about some of the opportunities for addressing those and improving power and then number three will be a bit of a grab bag of some additional topics that are relevant and potentially interesting for a future investigation</p>
<p><em>Phenylketonuria: a high-impact gene- environment interaction</em></p>
<p>so i want to start with an example here and that example is of phenylketonuria which is known as pku and this is an example of a high environment a high impact gene environment interaction and basically the the idea here is that phenylketonuria is a disease in which the phenylalanine hydroxylase gene which is responsible for converting phenylalanine to tyrosine uh loses most or al most or all of its function um this leads to phenylalanine buildup and that can uh downstream lead to intellectual disability and developmental problems quite early in life in babies if not addressed um it’s so impactful that it has been long part of routine neonatal testing and that’s partly because it can be treated in a sense by specifically using a low phenylalanine diet in babies and throughout life and that is able to largely remove these symptoms the interesting thing here and the reason i call it a gene environment interaction is that if if we look at the top row of this table here where we say we sort of have a normal phenylalanine hydroxyl alanine hydroxylase gene it doesn’t necessarily make a difference a low phenylalanine diet or a standard diet that’s not going to change the disease phenotype and likewise in the case of a low phenylalanine diet the you will see a normal phenotype whether the gene is normal or the gene is dysfunctional but it’s only in this sort of two-hit synergistic manner that having both the dysfunctional phenylalanine hydroxylase gene and a standard diet will result in these disease symptoms and this takes us towards the general</p>
<p><em>What is a gene-environment interaction (GxE)?</em></p>
<p>idea of a gene environment interaction which is can be viewed from one of two perspectives one i call it genotype centric where we’re interested in genetic effects but that genetic effect depends on some exposure in terms of how either it’s direction of effect or its magnitude of effect likewise we can think of it although it’s equivalent in an exposure-centric perspective where we care about the effect or the association of some exposure with an outcome but that depends on a genotype and so what this looks like in practice here’s my little cartoon here if we look at the left panel there’s both an effect of the genotype the phenotype increases as we go from genotype a to b and the phenotype also increases as we go from environment one to two but there’s no interaction here because the these effects are independent each other another way to say this is that these lines are parallel so in the right panel instead here we have an example of the g by e interaction because in genotype a we see that the difference between environment one and two is different than in genotype b and vice versa</p>
<p><em>Other common concepts in genetics can reflect GxE</em></p>
<p>so just to expand this intuition just a little bit other common concepts in genetics can reflect gene environment interactions in a way that we might not typically think of and so other concepts that might might encompass this include variable penetrance so we talk about our well this genetic mutation is impactful in some people but it’s not impactful in others that might be because of interactions with environment when we stratify genetic associations let’s say by sex and we say there’s a genetic association in males if that’s different than females this is an interaction between sex and the genetic variant and also pharmacogenomics because the outcome of interest in pharmacogenomics is response to some exposure which is a pharmaceutical drug in this case this can also be thought of of a thought of as a g by e</p>
<p><em>Why do we care about GxE?</em></p>
<p>so why do we care in general there are a whole host of reasons why gene environment interactions could be interesting first is simply a better understanding of biology so if we see that a genetic variant we think we understand how it’s acting and it turns out that that depends on the environment that could give really interesting clues as to sort of refining our understanding of the biology there if we think in this genotype centric manner that i talked about earlier we might be interested in understanding genetic effects but in the context of environment so maybe this helps us explain misinheritability you know we can’t find this we can’t find the right genetic effects to explain heritability because they are dependent on some way and environment or it might help us detect genetic associations that we otherwise wouldn’t because again they’re hidden in some way if we move to the exposure-centric perspective where we care about the effects of exposure being modulated by genotype this is where we get into the realm of precision medicine right and the idea that for me versus for you based on our genotype we should be taking different drugs so we should be eating different diets or something like that so the statistical model to set up</p>
<p><em>Statistical model</em></p>
<p>we can start with a standard genome-wide association study which you know we’re going to put one that’s very straightforward here we have a genotype term and we have covariates and the g by e extension is not too complicated essentially we’re going to add one term an exposure term which is just like a covariate and it’s going to be treated as such and then we have a g by e term and this is this is the nuts and bolts of it right we have a product term just a point wise product between the genotype and the environment and it’s that beta g by e that’s often of interest when we evaluate such a model but once we’ve trained or fit such a model we can then get quite a lot of things from it in some sense so one we have an interaction test and that would be evaluating this beta g by e and our we would maybe for that be trying to understand biology or understand again specifically this synergy between the gene and the environment in impacting the outcome we can also evaluate jointly and there are statistical ways to do this we can jointly evaluate both the gene genotype and the gene environment interaction term and this gives us some idea that maybe we can better detect genetic associations by incorporating both of these terms jointly then only the genetic term by itself and likewise though this is not a statistical test that’s often performed we if we’re thinking about precision medicine or an exposure-centric manner we’re in some sense trying to holistically evaluate both the interaction term and the effect of the exposure so that we understand let’s say some notion of a context specific impact overall of the exposure</p>
<p>Note: we can use “environment” liberally i’ll make one note here we can use environment we’re using the term gene environment interaction for me that’s largely historical in the way i use that term but we can expand far beyond what we think of as an environment let’s say that might be pollution or social factors but we can expand to lifestyle smoking or diet for example we can expand to demographic factors like sex we can expand to even physiological factors that are sometimes used as outcome like bmi those could be used as exposures in a gene environment interaction test here specifically i’m going to focus just because of my area of expertise and in order to constrain the space of where i’m getting samples from i’ll be focusing on specifically dietary exposures and using bmi as an outcome but that’s by no means the the limits of the scope of gene environment interactions</p>
<p><em>GxE discovery as intended: saturated fat, APOA2, and BMI</em></p>
<p>so the pku the phenylketonuria example is a particularly stark one because we have a phenotype that’s very risk that’s very notable and it’s very responsive to this interaction but in practice we expect to find more modest effects and so i’ll give one little vignette here of a series of studies looking into one interaction that might give a little intuition for how we maybe hope or expect that this could work in practice so in a study in 2009 in the framingham heart study a group was able to look at stratifying individuals by both their saturated fat intake which you see on the in the groups here on the x-axis as well as genotype at a variant in apoe2 and you see those each of the different colored bars or the different shaded bars here are those genotypes and they’re interested in seeing how those different strata and that interaction of saturated fat and the genotype at apoa2 influenced bmi and what we can see here in this diagram is that in individuals with low saturated fat intake this genotype while there might be a little decrease we don’t see a statistically significant difference in bmi however if we look at the high saturated fat consumers more than 22 grams per day we see there is a significant difference in bmi between the two genotypes cc carriers versus everyone else and so they originally discovered this in framingham heart study and here you don’t need to look at everything for these figures let’s just focus on the relative heights of the bars we can see that they were able to see a relatively similar type of pattern in the golden study which is also in the us as well as the boston puerto rican study which is different ancestry individuals but also from the us we move on additional studies by this group and others were able to see that they could replicate again the same sort of shape of these relative bars you know an increase based on genotype in high but not low saturated fat consumers and additional populations we have a mediterranean population chinese and asian indians and iranian individuals with type 2 diabetes so in all sorts of ancestry ethnicities and even physiological states like type 2 diabetes this interaction seems to replicate and finally a mechanistic follow-up type study was able to be performed and they understood that genotype and saturated fat come to mutually influence a regulatory region in apoa2 that then downstream is able to affect dna methylation then transcription then affect certain metabolites especially tryptophan and branched-chain amino acid metabolites ultimately downstream affecting bmi and are looking into a clinical trial to test this interaction so this is just an example of how we hope that interaction analysis will work and i can step in uh in a moment into maybe how things often turn out in practice but i’ll stop there for questions just at the moment thank you kenny i don’t see any questions just yet so i will let you know at the next break point sounds great all right um so getting into sort of if reality hits a little bit we have often some statistical power difficulties in gene environment interactions juwas discovery this plot on the left</p>
<p><em>GxE discovery is difficult</em></p>
<p>some of you may have seen this this is even from four years ago or so we’ve noted that juwas discovery in step with both time passing and sample sizes increasing we see a nice increase in locus discovery but g by e analysis has just has lagged a little bit behind this and so one major reason for this is that</p>
<p><em>Statistical power is a major obstacle</em></p>
<p>statistical power is a major obstacle to gene environment interaction so at least in the case of a binary exposure uh one heuristic we often use is that interaction detection statistically will often require almost four times the sample size compared to a main effect that has a similar magnitude of effect on the outcome and so to get a little handle of this we can look at just some quick ad hoc power calculations that i performed for this if we are able to simulate an effect size of 0.05 standard deviations in some let’s say some continuous outcome if we want intuition for this we could think about bmi and think about fto variant that is you know one of the best known geos variants and one of the strongest that associates with about .4 bmi units per allele so we’d be talking about an interaction effect that would correspond to about half of that let’s say 0.2 bmi units that this interaction is responsible for in some sense and so we can see across different minor allele frequencies if we’re looking towards rare variants of let’s say 0.005 all the way up 2.5 these are naturally given the same effect size they’re explaining different amounts of variants and will require different sample sizes so if we’re conducting a gwas to try to detect this type of sample size add a genome-wide significance threshold um for the 0.5 minor low frequency let’s say maybe 34 000 ish individuals here is what’s necessary and then we’re moving all the way up for this rarer variant if we’re not aggregating it might take quite a lot of individuals even for a gwas almost reaching 2 million to detect this effect now if we extend to g by e suddenly you know we’re expanding this is not exactly a factor of four but we need a substantially greater number of individuals based on a set of reasonable assumptions to get to the same type of power for the interaction and we’re getting towards here we see moving from 34 000 which is you know very common these days for a gos we’re getting towards the need for biobank scale population sizes in order to even identify an interaction effect for a common a common variant and we’re getting towards maybe even unfeasible sample sizes for something like a very rare variant that might be you know the .005 minor low frequency with current sample sizes that are available</p>
<p><em>Exposure measurement can be an additional limiting factor</em></p>
<p>so in additional beyond simply that general difficulty in statistical power we have the additional obstacle of exposure measurement a couple obstacles this poses one is that we can have imprecise or biased measurement you know a lot of exposures are self-reported and this can result in both lower power and less strong interaction effect estimates we have cultural and environmental heterogeneity resulting in more difficult replication uh and we have the potential for reverse causation uh in in addition and this could lead not necessarily that you know maybe an interaction effect is equally strong but it could lead us to conclude incorrectly if we are thinking in one direction like diet soda affects obesity and maybe it turns out to be the opposite</p>
<p><em>Approaches to increase statistical power</em></p>
<p>so taking those two these two obstacles into account we have a couple of types of approaches that can be used to increase statistical power in this context we’re going to focus we’re going to say okay we have some genotype or genotypes some exposures and they’re interacting to affect body mass index and we can focus on the genotypes or the exposures in trying to increase statistical power so first focusing on the genotypes here a very common type of approach is to use</p>
<p><em>Variant prioritization for two-stage testing</em></p>
<p>a two-stage testing in which we prioritize variants in stage one that for some reason we think they’re more likely to participate in interactions and then in stage two we can test for g by e and greatly reduce the multiple testing burden necessary so most straightforward approach we can choose variance or variance from genes based on some prior biological knowledge we have genes related to bmi let’s use those that’s you know very simple straightforward and is often done you take one step forward and say based on a general idea we have we have some reason to believe that interactions will often show up maybe in known loci that are already found from a standard geos this is one analysis looking at waste tip ratio as a function of genes sex and interaction between genes and sex and here from one of our papers recently and what we’re showing here is a manhattan plot for interaction p values going up on the top and marginal genetic effects going down and what what you generally see here is that almost every peak that we see for interaction we see showing up as a marginal effect locus as well indicating that maybe we can simply prioritize gios loci and then look for interaction effects at those loci</p>
<p><em>Variant prioritization: variance-QTLs</em></p>
<p>yet another idea would be this idea of variance qtls and so here the point is that interactions can induce genetic associations with the variance of some phenotype and so what we’re looking at here and we might actually start in the panel on the right uh in b here and point out that if there’s a genetic interaction let’s say the genotype increases the phenotype in one environment let’s say the blue and in the red it’s actually decreasing the phenotype this may not happen a lot in practice but let’s suppose it happens we can then look at the actual distribution of the values of these phenotypes and now look to the left and see that if we were to test for difference in variance across genotypes a a a g and gg we might actually detect that this locus is interesting even if we don’t know or we haven’t measured the red green blue environment in panel b there and so multiple studies have looked at this but one for example took a look at finding these variants qtls and then ultimately looking at those vqtls if we find interactions and try to replicate them internally how well do they replicate indicating sort of how enriched maybe these variants are for actual interaction if they take a bunch of random snips they see essentially negligible replication of the interactions that they find as might be expected if they take main effect only so that was what i was talking about in the last slide maybe a juwas locus they find that okay we see better replication you know if they find 25 percent of the if they find a certain set of interactions then maybe about 25 of them are replicating in this held out internal sample but finally if they choose vqtls we can see that we’re getting a lot higher here and that we’re more than doubling the amount of variance or the amount of interactions that seem to be real or that seem to replicate and so this is quite a promising strategy and i’ll actually be talking in the main in the main mpg talk today about one investigation levering leveraging this idea another genotype focused approach is to use polygenic scores and this is</p>
<p><em>Polygenic scores for interaction</em></p>
<p>probably the most straightforward of all here so we have an example here of sugar sweetened beverages genotypes and bmi this study was a quite a well-known study from 2012 where they simply constructed a bmi polygenic score not in not based on interaction just based on the typical polygenic score methods and they were able to see that on the y-axis here we have the effect size of sugar sweetened beverages on bmi and we can see that across each of these three cohorts and when we pull them together there’s this step increase such that individuals as their genetic predisposition to higher bmi increases the effect of sugar sweetened beverages goes from negative or essentially zero up to a quite substantial effect and so this is another quite promising path that a lot of that’s quite often used to increase power for interaction detection in addition we have exposure focus strategies right so i i’ll put this in</p>
<p><em>Exposure-focused strategies</em></p>
<p>two categories here first we can think about the data sets that we actually collect one approach is better measurement so maybe wearable technologies for physical activity for sleep etc these might give us better estimates of some of these behavioral components than we might get from self-report we can incorporate more diverse populations where uh ethnic and cultural heterogeneity leading to differences in the distribution of environments and exposures this can actually be quite helpful in some instances in aid discovery especially in meta-analysis that sort of thing and finally we can collect longitudinal data which will also help us better pin down better estimates of the outcome and the exposure in different individuals likewise on the analysis side there are multiple very interesting methods cropping up related to use of multiple environments at once there’s the idea of a lifestyle risk score that’s come out recently from the charged gene lifestyle interaction group where they’re trying to aggregate and say well are multiple healthy behaviors in some sort of score does that interact with genotype likewise another approach taken in a 2019 paper here was this idea of high dimensional environments where we construct an environment relationship matrix that’s similar to what we might think of as a genomic relationship matrix and use that to encompass a similarity of environments across individuals across a wide scope of these exposures and finally longitudinal methods here in the analysis will help us better take advantage of the longitudinal data that i mentioned above to again better identify these interactions so i will take a break here for a moment and i saw a couple questions here thank you very much yes i will voice the one that’s here and ask anyone else again who has questions to please post them in the q a so the question um here is isn’t the assumption that interactions are more likely to be found in main effect do you ask local flawed because the larger variance introduced by interaction effects would lower the likelihood that a given variant reaches significance uh thus main effect geos could be biased toward non-interacting loci so i think this is a really good point and something that i glossed over at the beginning when i introduced interactions and i think is a key piece here is this distinction between what we’ll often call quantitative interaction where the genotype let’s say the effect of the environment is just being tuned up or down uh you know it’s it’s more strong of an effect in one environmental strata than compared to the other then we have this idea of qualitative interactions and i can actually uh back up real quick here so this slide if we’re looking at panel b on the left here that’s an example what what’s being shown here is what we might call a qualitative interaction so this is a particular instance where uh the uh the person asking the question is totally right that in this particular instance we’d expect a potentially strong interaction and no marginal genetic effect potentially whatsoever uh the what i would point out is that we don’t expect to often see this in practice it’s it at least intuitively it seems a bit more likely that we’d find a let’s say a genetic variant that’s sort of tuning the effect let’s if we go to our saturated saturated fat bmi example we might not expect that in in you know high or low saturated fat intake the effect of a genotype is going to totally reverse direction we might rather expect that it’s going to be modest in one group of the population and more substantial in another group so i think that is a great point and that’s why it’s useful to to conduct genome-wide studies of interaction and not just neglect that but for certain especially depending on your hypothesis of what the interaction might look like it will often it can still be worth the trade-off to use main effect let’s say thank you very much and there’s there’s one other question um before you go on so um the the attendee thanks you for your talk um and says i’d like to know how to conduct the joint test g by e plus g and the other test g by e plus e uh yeah so the joint test specifically there are a couple of different uh software tools um we our lab is has particularly released one software recently it’s called the this a gem tool and this was the you can read about that in our paper right here the bioinformatics paper that is</p>
<p><em>Variant prioritization marginal genetic effects</em></p>
<p>in press at the moment so that is one approach is to use one of a couple different software so some software that’s used for calculating gene environment interactions especially genome-wide will be able to output directly the joint test another approach though in general if you have let’s say one interaction is that ultimately the joint test is simply it’s basically a chi-squared test with two degrees of freedom that is taking into account both the genetic effect its variance the g by effect its variance and the covariance between those two terms and so you can if it’s maybe for one locus or something you can sort of construct this yourself if need be and a final thing i’ll note there’s also a patch that my advisor alisa manning and her collaborator han chen input to metal where metal you can use metal in a way that performs a meta-analysis and can take both the interaction terms and the main effects and will output this joint test as well and i’m happy to feel free to get in touch with me after and i’m happy to talk specifics thank you so much those are the questions for the moment so all right so here as i mentioned before we can step into a little bit of a grab bag here and just some some interesting or or thought-provoking questions in interaction testing that i think are worth thinking about in this context</p>
<p><em>Interactions masquerading as main effects</em></p>
<p>so what i’m calling interactions masquerading as main effects uh and the idea here uh we’re going to set up a situation where we simulate a pure interaction and that’s what was done this was not my work but um was in the hugo astrodad’s paper from 2016 here so if we simulate a pure interaction we don’t expect that the genotype by itself or the environment by them itself is explaining any effects per se only insofar as they participate in the interaction but our question that we’re going to ask here and and we’re going to look at in this plot is how much additional variance is explained by the interaction beyond what we could find by environment and genotype alone what we can see if we make some basic assumptions here for example we’re going to assume that the mean of the environment is non-zero what you can see in this plot is that as we move across the risk allele frequency spectrum there is you know there’s a large change in some of these parameters how much variance is explained and which components are explaining it the interaction the genotype by itself or the environment but what you’ll note is most prominently here is that the vast majority of the variants explained across the soleal frequency spectrum are explained by things that are not the interaction and so the idea is i mean for intuition we can think about the fact that if let’s say as we assumed here the mean of the environment is two or let’s just say it’s non-zero the inter the interaction term becomes correlated with the gene term and or the environment term and because of that we expect that performing a simple g-wasp might very well uncover loci that in reality are pure interactions but we’re going to uncover them even in a test for a marginal genetic effect and this also gets back to the question that was asked earlier which uh you know about whether we expect interaction effects to show up in gwas loci or not this is one of the reasons why we expect that and so just to sort of summarize our takeaway at this point we don’t necessarily expect that g by e will explain a huge amount of additional variance because of this effect that we’re seeing here to the left where gwas will find interactions even though it’s not explicitly looking for them and so we might not expect that incorporation of g by e is going to hugely improve prediction but what it can do is help us identify causal players and that’s why that’s one of the specific reasons why this interaction analysis is particularly interesting because even if it’s not going to improve prediction so much for some phenotypes at least we can expect that it might reveal interesting causal associations</p>
<p><em>Additive vs.&nbsp;multiplicative interactions</em></p>
<p>okay so second idea here is additive versus multiplicative interactions and so here we’re going to set up another hypothetical example we have an exposure that has two categories and we have a genotype that has two categories just to simplify here let’s say each of them doubles risk for some binary outcome uh so we move on the genotype equals zero row we move from exposure one to exposure exposure zero to one we double our risk move from genotype 0 to genotype 1 in the exposure zero stratum double our risk if the if this continues to double risk in that final bottom right column then we expect that sort of the relative risk here is four now if this is the case then modifying the exposure in some sense matters more in risk genotype carriers than it does in non-risk genotype carriers if that’s the case then we might care more about it for let’s say public health reasons right because if we think that this is some disease for example and then maybe if we can find risk carriers they’re going to be generally have a greater burden of the disease and changing the exposure even if there’s no multiplicative interaction or there’s no interaction where we find something new biologically it still might be more compelling to adjust the exposure if we have i don’t know limited resources something like that in people who have the risk genotype and so this is what we call an additive interaction where if you have effects of both the genotype and the exposure it might be of interest to understand how non-additive they are and in this case they’re not multiplicative there’s no multiplicative interaction it’s simply multiplying a relative risk of two with a relative risk of two but that doesn’t mean that uh the two aren’t important for each other in determining relative amounts of absolute risk that’s being changed this is particularly relevant for binary outcomes there there is a additive and multiplicative interactions can exist for continuous outcomes but it tends to be less relevant just a quick note on some other ongoing areas investigation areas of investigation one is that we have whole genome approaches to heritability and variants explained due to interactions as i mentioned before there are some particular considerations in thinking about whether we expect to uncover lots of new heritability but it still might be quite interesting to get some large-scale view whether we know the environments some of for some of these methods we need to know the environment for some of them we don’t even need to know the environment at all but there are a whole host of methods that can look in a genome-wide way at the heritability contributed to by interactions in some sense for rare variants we have the same trade-off that exists in gus right that we expect that we very well might find some higher impact rare variants but this is going to be traded off with the fact that as we already described power is quite an obstacle for gene environment interaction analysis and even more so when we start to reduce the minor low frequency as we saw in that simulation so rare variants are certainly an ongoing area of interest but the the power trade-off has to be taken into account and finally something i just think is a lot of fun i i don’t know if this is an official method or not but a a group from the uk has looked uh for a couple different studies have looked into this idea of mendelian randomization g by e where the idea is that mendelian that g by e interaction analysis can help prove the robustness of an mr analysis so our example here if we have an mr analysis that’s proxying smoking and trying to understand a causal effect of smoking if we can look at it from a gene environment interaction analysis perspective where there’s some modifying factor which is does the population smoke at all we’d expect that if an even genetic variance proxying smoking will not have any effect and should not affect the outcome if it’s a population where there is no smoking for example so it’s just another interesting uh area for investigation as well and so just to sum up here gene</p>
<p><em>Summary</em></p>
<p>environment interaction analysis can be a critical building block for biological understanding as well as informing personalized medicine i talked about that in general the biological understanding may be from just looking specifically at the interaction term and understanding synergies between environments writ large and genotypes as well as for informing personalized medicine and that again comes from this idea of being able to look at the effect of the exposure holistically on conditional on some genotype those statistical power as i talked about in the second section is the major obstacle and we saw that that’s due to both general difficulties in power for gene environment interactions as well as the fact that often the exposure will not be measured as well as the genotype or the outcome and if that’s the case then that will further dilute statistical power but we have a series of approaches that can help deal with that we can prioritize variants any number of ways geowas variants variants qtls etc we can use multi-genotype approaches like a polygenic risk score to collapse genotypes into one and we can use multi-exposure approaches this could be like a lifestyle risk score or this could be like some of the more advanced maybe high dimensional environment random effect type of approaches to understanding the impact of multiple environments interacting at once and to make progress on this a lot of these takeaways are similar to what we might say for gos right we can use larger samples we can try to actively seek more diverse populations and i think unique from geos is the fact that particularly in one thing we can mean by diversity here even more than uh for gus is that where you can specifically see populations where there’s diversity of these exposures whether sort of environmental differences in these groups and finally better exposure measurement will always be helpful in pinpointing the measurement and identification and so thank you for your attention happy to take some additional questions here thank you so much that was really wonderful um there are a couple questions um and before i voice them um a reminder to everyone who’s attending that if you have any questions please continue posting them in the q a um so the first question is is posted there and um it’s a question about terminology and the attendee says are the additive rather than multiplicative interactions sometimes called gene environment correlations rather than interactions yeah yeah so i’m aware of although i don’t i’m not hugely in i don’t i’m not too familiar with the literature on this there’s i think a lot especially in the social sciences there’s a lot of uh discussion of and references to this gene environment correlation where we’re thinking about if our exposure let’s say is some behavior uh let’s say it’s a behavior um where that that can have impacts on both our uh our statistical inference and how we think about the mechanism of an interaction so that is definitely worth thinking about and it’s something that can be a really key component to consider when you’re understanding when you’re testing for interactions and trying to understand what’s going on with that said i don’t know and i’m happy to to hear if there’s an element that i’m not thinking about but the additive versus multiplicative i don’t know that those two pieces are necessarily correlated because at least in the way i presented it here if we’re thinking about this hypothetical setup we’re not necessarily claiming that there’s any impact of the genotype on the exposure but simply that changing one of them will have more impact on the absolute number of people let’s say getting some disease in one strata versus the other of exposure or genotype etc got it great thank you and and to the the attendee if you wanted to follow up please of course post things in the in the q a i will follow up with one question also that’s something you’ve already touched on a bit but it’s something that is really frequently occurs to me which is that often times even just with you know a simple g wasp or you know a just um genotype phenotype needs um there’s often varying levels of confidence in the um in the phenotypes and so sometimes you’re absolutely certain as well as one can be of a phenotype and for other individuals you may have a decent guess but you have some probability associated with that um and then of course when you add the environmental component that adds a whole new opportunity for variation in the certainty among individuals</p>
<p><em>Other ongoing areas of investigation</em></p>
<p>and i know you’ve addressed aspects of this before i’m wondering what your sense is if if you have um complete as much as that can be the case phenotype and environment data for a subset of individuals and then only phenotype data for others is there a way to include all of the individuals bearing in mind that um the amount or the types of information is variable or the certainty is variable among them or is it better just to exclude things that fall individuals that fall far below the information standards for the others yeah i think this is this is a really tricky one and i think um i my mind’s going to a couple different ways one is that you know the idea of unsupervised learning you know if we’re thinking in a machine learning context right that you know we have some labels we have some not labels and can we use all the data in a useful way i think one way naively that we could address a situation like this let’s just say we decide half of our individuals are just they’re they’re straight out we can’t use them whatsoever for these individuals we might still be able to use that sample in some way when we if we’re thinking about some of the variant prioritization approaches that i talked about if we think about okay let’s stratify our sample let’s use the first half who don’t have as good exposure data so</p>
<p>let’s think about for example variance qtls right that’s what i was talking about it might you might very well be able to take individuals from the same population but for whom you don’t know the environment you might be able to prioritize ten percent of your variance five percent of your variance you know you could use a very liberal cut off and just say all right we’re trying to narrow down by some substantial amount the set of variants that we’re looking at in uh and then you could look at that in individuals with the exposure another way in general and especially for more noisy noisily measured exposures like let’s say self-reported physical activity or diet something like that an expert in an approach i’ve seen explored is to take a look at only extremes and where we say you know binarization or sort of discrete discretization of individuals where the environment isn’t measured as well maybe could still allow us to to use that information um and maybe be diluted a little bit less so these are all kinds of approaches towards that great thank you very much that’s really thought-provoking and i think we have time for one more question um and this goes back to the previous question um and uh so to go back to the earlier question about traditional geos being biased toward main effects for any interacting variant not just the example you mentioned the standard deviation will be larger and thus the wider confidence intervals may impair achieving genome-wide significance for the interacting variant compared to a non-interacting variant with the same main effect correct uh yes i would yeah is that the question or that’s the first part that is that is the question as written and again please please um follow up if you would like to ask more questions but yes yeah so i think the the idea there would be yes we might expect that the confidence interval could be it could be a little more difficult to detect that variant compared to a main effect variant of the same effect but i think that the important maybe comparison here is that you know g wasp power more power is always helpful more sample size is always helpful but we have pretty solid sample sizes for a lot of phenotypes for gwas so much so that we might expect that even those that are a little more difficult to identify that at this point we might be getting towards identifying those and that the comparison of that with a main geos is less important than the comparison between that and an interaction test where we have maybe mismeasurement of exposure etc and so i think that all contributes</p>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
          // target, if specified
          link.setAttribute("target", "_blank");
          // default icon
          link.classList.add("external");
      }
    }
});
</script>
</div> <!-- /content -->



</body></html>