<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.353">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>PGC Video Textbook - Chapter 7.1: Trans-ancestry analysis (Video Transcript)</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


<link rel="stylesheet" href="styles.css">
</head>

<body class="nav-sidebar docked nav-fixed">


<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="./index.html">
    <span class="navbar-title">PGC Video Textbook</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link active" href="./index.html" rel="" target="" aria-current="page">
 <span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./about.html" rel="" target="">
 <span class="menu-text">About</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./contact.html" rel="" target="">
 <span class="menu-text">Contact us</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="https://pgc.unc.edu/" rel="" target="">
 <span class="menu-text">PGC Website</span></a>
  </li>  
</ul>
            <div class="quarto-navbar-tools ms-auto">
</div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
      <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item">Chapter 7.1: Trans-ancestry analysis (Video Transcript)</li></ol></nav>
      <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
      </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal sidebar-navigation docked overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
      <a href="./index.html" class="sidebar-logo-link">
      <img src="./pgc_logo_website_v3.jpeg" alt="" class="sidebar-logo py-0 d-lg-inline d-none">
      </a>
      <div class="sidebar-tools-main">
    <div class="dropdown">
      <a href="" title="" id="quarto-navigation-tool-dropdown-0" class="quarto-navigation-tool dropdown-toggle px-1" data-bs-toggle="dropdown" aria-expanded="false" aria-label=""><i class="bi bi-github"></i></a>
      <ul class="dropdown-menu" aria-labelledby="quarto-navigation-tool-dropdown-0">
          <li>
            <a class="dropdown-item sidebar-tools-main-item" href="">
            Source code
            </a>
          </li>
          <li>
            <a class="dropdown-item sidebar-tools-main-item" href="">
            Report a bug
            </a>
          </li>
      </ul>
    </div>
</div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./welcome.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Welcome to the PGC Video Textbook!</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./toc.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Table of Contents</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">
 <span class="menu-text">Chapters</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter1.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 1: Introduction</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter2.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 2: The Genome</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter3.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 3: Technologies</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter4.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 4: Study designs</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter5.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 5: GWAS analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter6.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 6: Polygenic Scores</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter7.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 7: Ancestry-Specific Analyses and Considerations</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter8.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 8: Post-GWAS bioinformatics</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter9.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 9: Advanced Topics</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter10.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 10: Other Considerations</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true">
 <span class="menu-text">Software Tutorials</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_cnvs.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">CNVs</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_conditional.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Conditional Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_crossdisorder.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Cross-disorder Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_ewas.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">EWAS</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_geneset.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Gene Set Identification</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_gwas.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">GWAS</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_genomicSEM.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Genomic SEM</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_imaging.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Imaging</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_MR.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Mendelian Randomization</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_prs.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">PRS</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_correlation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">SNP Heritability and Genetic Correlation</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true">
 <span class="menu-text">Additional Resources</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./glossary.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Glossary</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./resources.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Software Resources</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./addreading.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Additional Reading</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="https://pgc.unc.edu/for-researchers/download-results/" class="sidebar-item-text sidebar-link">
 <span class="menu-text">PGC Summary Statistics</span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#sec-video1" id="toc-sec-video1" class="nav-link active" data-scroll-target="#sec-video1">Cross-ancestry analysis</a></li>
  <li><a href="#sec-video3" id="toc-sec-video3" class="nav-link" data-scroll-target="#sec-video3">Local Ancestry and Admixed Populations</a></li>
  <li><a href="#sec-video2" id="toc-sec-video2" class="nav-link" data-scroll-target="#sec-video2">Ancestry-specific PRS</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Chapter 7.1: Trans-ancestry analysis (Video Transcript)</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<section id="sec-video1" class="level1">
<h1>Cross-ancestry analysis</h1>
<p><strong>Title</strong>: Cross-ancestry PTSD and Polygenic Findings, and the Cross-Population SIG of the PGC</p>
<p><strong>Presenter(s)</strong>: Laramie Duncan</p>
<p><strong>Caroline Nievergelt:</strong></p>
<p>to the PTC worldwide late meeting I see that people are still calling in so let’s give them a minute okay looks like it’s stable now so today’s topic is analysis across ends history we have three speakers today we have Laramie Duncan Elizabeth Atkinson and Alicia Martin all three talks are centered around Shiva’s analysis across industries which is a really timely topic today we’re gonna do something a little bit different than usual we are actually gonna wait after to take questions until all three talks are over and then we will allow you to unmute yourself so if you have a question for any of the presenters then you can just mute yourself and ask a question directly and if that’s not gonna work then we can still take questions we had a chat function okay so let’s get started with Laramie Duncan who is currently assistant professor of psychiatry at Stanford University Larry Moe Laramie received a joint PhD in neuroscience and clinical psychology from the University of Colorado and then did a postdoc in statistical genetics in the labs of Jordan smaller and Mark Daly at MGH in Harvard med school today she will be talking about call centers to analysis in PTSD intercourse population especially interest cope of the PGC so take it away Laramie</p>
<p><strong>Laramie Duncan</strong>:</p>
<p>okay great um thanks so much Caroline I really appreciate this opportunity to present so thank you to you and pat like you mentioned I’m going to talk about three topics today related across ancestry and analyses first the pgc PTSD analyses second telegenic or scoring analyses and third the cross population special interest group within the PGC so starting with pgc PTSD the original wave of the pgc PTSD analysis was led by dr. Kirsten Conan and my postdoctoral mentor as Caroline mentioned with dr. mark Bailey they provided incredible leadership and mentorship for this I wanted to mention them upfront um and the reason that we’re talking about pgc PTSD is that actually compared to most large-scale Jewess our samples were relatively diverse so as you can see here we actually had a minority of European ancestry participants and we had a rather large sample of african-americans as well as some Latino Hispanic individuals this is again for the first wave of PTSD analyses within the PTC at the time when I made this figure I looked at the other large psychiatric to us and like the rest of medical genetics we can see that most samples are of European ancestry the samples the art of European ancestry across medical genetics tend to be from East Asian populations so despite the fact that pgc PTSD does have greater representation of more populations it’s important to note that it’s still not reflective of world ancestry but nevertheless in conducting these analyses the fact that we had these more diverse samples necessitated modifications to our analysis and so when we approach this problem we did what what anyone does we thought about what might be appropriate talk to experts and we tried a number of different analysis approaches and there were a lot of modifications that were necessary just showing one here we decided ultimately to conduct a trans ancestry meta analysis and so in order to do that we needed to assign ancestry for each of the participants in our study such that within each cohort we could individually cut conduct ancestry specific Jewesses prior to meta-analysis within each ancestry and then transits ancestry meta-analysis so actually though this work has continued in terms of making further analytical improvements to the pipeline and I just want to point everyone here - Caroline’s work on this and Adam a hopper and her group they’ve continued to improve the pipeline with some really nice additions and changes to quality control imputation steps and the analysis and this second round of the pgc PTSD analysis is now available on bio archive the link is here so for questions about the most up-to-date improvements I would suggest that people check out this paper or talk to - Caroline going back to PTSD the first wave that the PTSD analyses though I’d like to point out something that was that was interesting and informative from these analyses so purely by chance alone it happened to be the case that we had about 10,000 individuals of African American ancestry and European ancestry with about 25% cases each despite this comparable these comparable sample sizes it turned out that we had no significant results and the African American samples whereas as expected we had significant snip heritability estimates in the European ancestry individuals as well as apologetic predictions for example using schizophrenia external schizophrenia GYC results to predict PTSD in the European ancestry samples and based on power calculations we thought that we would have enough power in these European ancestry samples but due to expectations of lower transferability and African American samples we didn’t know if we would have enough power and and in this in these samples it turned out that we did not the reasons for this are are at least somewhat well understood number one we have there was worse coverage of African ancestry variants in our samples this is both because African ancestry individuals have greater genetic diversity than other populations and so for any given number of variants you can’t cover as much as of the genome but also as as many people know there’s a bias towards European ancestry variants on typing trip the chips so both of these problems probably contributed also the lack of external data resources from non-european populations is really a problem particularly for these African ancestry samples but this is true for other populations as well and then finally there are also methodological and that inadequacies currently for example differences in linkage disequilibrium and allele frequencies are still not handled as well as they could be in many different analyses so the good news is that in the second wave of the PTSD analyses this paper that I mentioned before from Caroline we had larger sample sizes for all populations and with these larger sample sizes we could actually estimate heritability and have polygenic predictions in the african-american samples so it does look like this was an issue of power and so certainly as expected it helps to have larger sample sizes I’m going to turn it out to a couple of quick points about polygenic risk scores so I don’t know if it’s partly just being in Silicon Valley and being near 23andme but everyone is talking about polygenic risk score scare not you know not within my psychiatry building but just people are very interested in these and a couple of years ago when for example 23andme is releasing results without any apologetic results without any mention that these scores might or would certainly would have differential performance the cross ancestry group so we thought that this was a really important topic to spend some time exploring so to give a bit of background as early as 2009 the ISC paper Purcell at all demonstrated that polygenic risk scores that are derived from European ancestry populations have poor performance and African American samples and this flora performance and non-european ancestry samples has been shown many times since and lots of different polygenic risk scoring publications so what you know kind of the new question that we wanted to look at what we wanted to quantify the decrement of performance between European ancestry performance and other ancestor ease and we looked at a couple of ways of doing</p>
<p>this I’ll just show one bit of information here we looked at 10 years worth of polygenic scoring studies first to just see who was included in these studies as expected these studies also have primarily European ancestry but then getting to the performance apologetics scores across different ancestry groups we made comparisons and for example one of the results is that we found that both apologetic scores performed about three times better in European ancestry individuals and in african-american samples I don’t know if I think Alicia may or may not mention an additional data that she has here that has I think even better analyses addressing this question so in summary regarding polygenic scores the performance it’s worse from non-european ancestry populations this is not surprising this was you know known for quite some time but efforts are underway to quantify how much words the performances in different populations I think it’s important to know what to expect both for scientific research or we want to have accurate power calculations and then to the extent that anyone is using these scores in clinical practice ultimately or just you know sort of in their own lives getting 23andme reports it’s important to know how well they might work efforts are also underway to improve prediction across populations but getting to a bigger picture question I think that they questioned a lot of us have on our minds is what do we do about the substantial under-representation of most population of most populations and genetic studies so this gets me to about my last topic and at first or as a point of background information there are many potential solutions to this to this problem and I know that many researchers within the pgc have been working on this in this area for considerable time and so we’re just mentioning one additional approach we you know if I had more time I would mention other efforts that have been ongoing so something that we do it we did is that with high laying mom who is now junior faculty and Mark Bailey’s group I’m at the Broad we started a cross population special interest group within the PDC with the goals of improving applicability of genetic results conducting analyses and supporting the involvement of researchers from different parts of the world in one of our very first meetings</p>
<p>Roseanne Peterson had a really great idea to write a best practices paper describing the ways in which genetic analyses ought to be modified for different samples of diverse ancestry and in particular admin samples and just mixtures of different different samples and so some of the topics that are addressed in this paper include specific recommendations for how to modest and to modify quality control steps which parameters should be modified and how and why as well there are recommendations for imputation and for how to analyze samples that issue us or mixed model context and this is absolutely a huge team effort um I want to just say thank you to this group I think that the reason that such a best practices paper doesn’t already exist is that they’re actually you know really isn’t any one person in the world that could have written a best practices paper and it’s really thanks to the analysts in this group especially that have expertise in particular areas that we were able to cover these different components of analysis and really write about the different approaches and what the pros and cons are for taking one approach versus another so that’s what’s in this best practices paper I’m highly eggs spoke to an editor at sell and they were very interested in this paper and so they suggested that we write it up as a primer so that’s the current format of the paper we’ll be submitting in on Monday and so that’s our first work product and the special interest group is open to anyone it’s technically within the stats group but if you’re interested we have meetings on the first Wednesday of the month at 1:00 p.m. currently and you can get in touch with us and just thanks to the group all these groups for these incredible collaborative efforts so</p>
</section>
<section id="sec-video3" class="level1">
<h1>Local Ancestry and Admixed Populations</h1>
<p><strong>Caroline Nievergelt</strong>:</p>
<p>thank you thank you very much Naomi and thanks for being on time this was a really nice presentation and I’m sure that there are many comments and questions but let’s keep them until the end we are now moving on to our next speaker and it’s Elizabeth Atkinson who is currently a postdoc at MGH in that broad with Mark Daly and peniel Elizabeth did her PhD at Washington University in st. Louis and is currently developing resources that allow for improved genetic analysis in admixed populations Clemmy she’s a member of the PTC PTSD group and she has developed local ancestry analysis for us today Elizabeth will be talking about her framework enabling well-calibrated genomic analysis of psychiatric traits across admixed populations then the slides are up let’s see okay I’ll see you then so take it away Lisa B</p>
<p><strong>Elizabeth Atkinson</strong>:</p>
<p>all right thank you so much again for inviting me to talk about this project I’m really excited to hear the feedback so yeah this Caroline said today I’m going to tell you about a project I’ve been working on hoping to improve the ability to do sophistical genomic</p>
<p>analyses on admixed populations so to start off there we go a whole reiterate</p>
<p>a point that you know we’re hoping is now becoming common knowledge in the GOS community and that laramie just spoke deeply about the vast majority of our</p>
<p>association studies are conducted on European cohorts and if we look more closely at this sort of wedge of the pie</p>
<p>that is non-european we’ll notice that only a few handful of percent are populations that are admixed and just to</p>
<p>make sure that we’re all on the same page starting off when I say admix I’m talking about an individual whose ancestry is not homogeneous but rather</p>
<p>is comprised of several ancestral populations so importantly there’s actually many more samples out there who</p>
<p>have been genotyped or sequenced alongside phenotypes but they’re not actually making it into this figure they’re being excluded from analysis due</p>
<p>to being too admixed and there’s also several significant large-scale efforts</p>
<p>to collect psychiatric phenotypes alongside genomic data in diverse populations some of them led right here</p>
<p>at the Stanley Center so there really is a pressing need for the development of novel methods that will allow the easy</p>
<p>inclusion of admix people into association studies we really can’t afford to kind of leave large swaths of</p>
<p>our data on the table anymore and you know let alone the other problematic aspects of leaving entire ethnic groups</p>
<p>understudied so at mixed individuals are generally</p>
<p>removed into the challenges of accounting for their complex ancestry so that there are kind of concerns about</p>
<p>population structure infiltrating analysis and biasing results which can lead to false positive associations</p>
<p>studies that do include admixed individuals tend to generally only correct for PCs so pcs actually I think</p>
<p>I can make a laser pointer here let me do that all right so pcs generally only collect correct</p>
<p>for kind of average ancestry so if somebody is say you know 75% African 25%</p>
<p>European or vice versa and they actually you know can’t really consider a lot of the more fine scale population structure</p>
<p>that can still be present on the data and this is important which you know this this sort of fine scale pattern</p>
<p>might be different for example in one case in control cohorts and still leave the door open for false positives so here I’m going to present a novel</p>
<p>analytical framework to hopefully rectify this issue and allow for the easy incorporation of admix people into</p>
<p>association studies we specifically do this by accounting for local ancestry</p>
<p>which takes into account this finer scale of population structure so to get</p>
<p>a little bit more into the actual kind of method we’re calling this method</p>
<p>tractor for the moment the core feature of the proposed framework relies on a</p>
<p>certain population structure as I said informed by local ancestry so the first step is this automated pipeline to call</p>
<p>local ancestry tracts in your sample so just in case you know you haven’t seen</p>
<p>any painted carry grim figure before here’s an example of a Latino individual the autosomes are along the x axis</p>
<p>position along the chromosomes on the Y and then the two strands of each chromosome are painted according to the</p>
<p>ancestral origin of that tract so we use this information after we collect it to</p>
<p>kind of improve long-range phasing and haplotype tract recovery which I’ll go</p>
<p>into in a lot more detail in coming slides and then the end goal is basically to be able to extract out the</p>
<p>component ancestries of interest so for example if you have a large European cohort and you have this admixed</p>
<p>population you could use tractor to scoop out the European bits of your admixed cohort to include alongside your</p>
<p>European individuals and you know same goes for the African and Native American components in this example so in this</p>
<p>way you can kind of leverage the information and add mix people you don’t have to exclude them from analysis anymore so since the first step in this</p>
<p>pipeline involves calling local ancestry I wanted to validate that it was performing well and kind of the target populations that we had in mind so the</p>
<p>use case I’ll be talking about for the rest of this talk is the african-american demographic context and</p>
<p>we modeled this after PGCE PTSD african-american cohorts so to test</p>
<p>local ancestry inference using one of the very worthy existing methods to call local ancestry our F mix I simulated a</p>
<p>truth dataset that resembles these realistic pgz PTSD populations using</p>
<p>real haplotype data so this will retain the LD patterns and you know other genomic features present and real data</p>
<p>that are sometimes hard to simulate and using this truth dataset with known</p>
<p>phase and local ancestry I then ran chapter step one calling you know local</p>
<p>ancestry and quantified how often we got it correct and hardening ly about 98% of</p>
<p>the time no matter how you slice it we were getting correct slope lines which recall so this seems to perform very</p>
<p>well next I wanted to see how the pipeline would work in kind of a more</p>
<p>real looking data set so usually we don’t have perfect phase we have statistically phase data so I took our</p>
<p>truth data set and ran it through a standard phasing pipeline shape at two</p>
<p>using a thousand genomes as a reference panel I mean noticed something that we kind of hadn’t initially expected to see which</p>
<p>is that a lot of the originally long trucks were really broken up by these</p>
<p>switch errors and phasing so even though local industry inference is performing really well you know if it’s calling</p>
<p>well if we have zero one or two African tracks at a given position these long range</p>
<p>haplotypes are being disrupted due to switch errors so we ended up building in an additional step in tractor to find</p>
<p>and unkink is the way we’re using as if you know I’m kicking this garden hose to recover these long range chaplet types</p>
<p>so here’s our karyogram again now after this I’m kinking step has been</p>
<p>implemented we noticed that there were still sort of stretches where due to the local ancestry you know we painted the</p>
<p>chromosomes on they statistically phase data so there’s some overhang areas that were still not able to be recovered so</p>
<p>we decided to implement one more iteration of our F mix on this sort of</p>
<p>ironed out data set to see if this could help recover these full Apple types and</p>
<p>indeed this did dramatically improve the situation so just to convince you that not only</p>
<p>did this you know make our carry grams look prettier but this actually made the data look more realistic I modeled the</p>
<p>track distributions using a plus sign waiting time centered at nine which was the number of generations ago that the</p>
<p>pulse of admixture occurred so given this this is what we would expect the European tracts in our data set in this</p>
<p>you know the European tracks of these African American individuals to look like you know given given all things</p>
<p>being correct and we can use this to put a p-value on how likely it is to get the</p>
<p>distributions in the various treatments of the data so in our truth data set</p>
<p>again focus on these red tracks which is the European segments we do see pretty</p>
<p>close to expectations distribution here almost around nine after statistical</p>
<p>phasing however the tracks get extremely short they’re actually P times 10 to the</p>
<p>negative 28 likely and to observe a distribution of this many tiny tracks so very very unlikely after on kinking it’s</p>
<p>about twice as improved and with one iteration of the pipeline we have gotten</p>
<p>things almost back up to where they should be and just sort of to zoom out and put this all in into perspective</p>
<p>another way you can see you know how we really dramatically improved things from this original</p>
<p>purple phased situation back down to this yellow line which is as close to</p>
<p>the truth dataset in black here as we’ve gotten it so far so we are indeed making</p>
<p>the situation look statistically significantly more realistic by recovering these long-range haplotypes</p>
<p>which which can be important too if you’re concerned with things like LD for example in your data set we’ve also</p>
<p>tested this and a bunch of other demographic scenarios and it performs you know similarly well so with</p>
<p>different admixture fractions putting a different you know demographic models in</p>
<p>so pulse times at different points in in the past and also between different</p>
<p>ancestries with varying kind of divergence times so in the last couple</p>
<p>minutes here I’d like to kind of take a step back and look at one of the main applications of this framework which is</p>
<p>really to show that it improves performance in Ag wast context so not only should we better correct for</p>
<p>population structure by using this local ancestry information but we can actually identify a new low sy through an</p>
<p>increase in power from local ancestry as well so the statistical model that’s</p>
<p>built into tractor basically tests each snip for an association with the phenotype using the logistic regression</p>
<p>model that I’ve show here where X is the number of copies of the risk allele from the you know the first of your ants this</p>
<p>trees x2 is the number of copies from your second ancestry and then you can put in whatever other code various</p>
<p>importantly including global pcs you know as for the rest of your parameters</p>
<p>you also notice that this is currently written as a two-way admix context but it could be scaled up to an arbitrary</p>
<p>number of ancestries and you can further test if the risk allele is ancestry</p>
<p>specific by evaluating the difference between beta 1 and beta 2 using AZ test</p>
<p>so we wanted to see if there were any kind of power gains using this model and</p>
<p>I want to acknowledge Adam here who heroically ran a whole lot of these simulations with me</p>
<p>so to do this we we simulated again a realistic african-american demographic</p>
<p>scenario we drew a by allelic disease variant with the probability of each</p>
<p>genotype copy drawn from the minor allele frequency we then simulated a disease phenotype</p>
<p>assuming a 10% disease prevalence and an individual’s risk of developing the phenotype was modified basically by</p>
<p>their percent admixture and the presence of the minor allele on the African genetic background so this is basically</p>
<p>assuming no effect in a European genetic context but an effect on an African genetic context which would be you know</p>
<p>equivalent to modifying effect sizes due to a tagged snip being present in African background and not in Europeans</p>
<p>for a shared causative mutation so we ran a whole lot of simulations varying</p>
<p>all of these kind of parameters to sort of characterize the landscape of how tractor would improve your power this is</p>
<p>the simplest context just showing the allele frequency held constant at a twenty percent minor low frequency and</p>
<p>in all of these plots the dotted line or the dash line is tractor and the solid line is sort of the traditional G wast</p>
<p>method which does incorporate PCs and you’ll notice that both the same sample</p>
<p>size is shown in blue and black we do see a significant power gain using local</p>
<p>ancestry and this can get very large and you know this game can get really big in</p>
<p>different genomic contexts so for example when you introduce a minor allele frequency difference between the</p>
<p>ancestry is now shown in this red line this sort of gap between the methods</p>
<p>grows very dramatic and I kind of want to dwell in this for a moment just to show how much you know how significant</p>
<p>this power improvement is so if you pick your favorite power let’s say 80% you’d be able to detect variants that are of</p>
<p>an odds ratio of about point one lower using our local ancestry incorporating method compared to your traditional G</p>
<p>wasps or if you pick your favorite odds ratio let’s say 1.2 you would basically have you know very little power to find</p>
<p>this variant in a normal context that we’d be able detected at very high power from using</p>
<p>our our method so we’ve also done a lot</p>
<p>of sort of tests characterizing other aspects of the genomic landscape since I’m running a little low on time I’ll</p>
<p>just sort of briefly mention the most significant one which is the effect size difference across ancestries so now</p>
<p>we’ve introduced an effect in both backgrounds not only in the African genetic context and the sort of major</p>
<p>take-home point is that you know if you have the effects so on the right hand</p>
<p>side here we have the effect now swapped only in the European background rather than the African background and you’ll</p>
<p>notice that we would have effectively zero power to detect it using a traditional model but can detect it at</p>
<p>reasonable power with Traktor this is you know because European ancestry only</p>
<p>makes up about 20% of the sample so the signal would really have gotten swamped</p>
<p>out if we hadn’t deke involved the local ancestry tracts and importantly in a</p>
<p>context where we would actually expect no improvement from local ancestry the effect is the same everything is the</p>
<p>same in both ancestries we don’t see much of a power loss a very minimal</p>
<p>power loss from using tractor so there doesn’t seem to be you know a major hit from including this in your analyses so</p>
<p>you know if you’re uncertain of the effect of ancestry it seems like it’s not a massive problem to incorporate</p>
<p>this and importantly we’re talking about you know the perceived effect of this tag snip not the real causative mutation</p>
<p>and so you know even if there’s the shared causative mutation is the same across ancestries you know what you</p>
<p>actually would detect in your Jie wasps could differ depending on the amount little frequency or LD patterns or environmental interactions or many other</p>
<p>factors that could be different across these ancestries alright to sum up today</p>
<p>I’ve talked about this readily implemented pipeline that should allow you to be able to include admixed</p>
<p>individuals in your jaw studies in a well calibrated manner I am currently optimizing it with</p>
<p>members of the Hale team from the Broad Institute so it should be a very quick</p>
<p>and work in all systems once it’s finalized there but it’s already written up in</p>
<p>Python and I can you know distribute it to whoever’s interested we have also</p>
<p>built in these extra features to improve long range phasing and I’ve shown you</p>
<p>today that we not only you know boost your sample size by allowing inclusion of admixed individuals but we can</p>
<p>actually further boost your power by leveraging this information from admix populations to find novel variants we’re</p>
<p>also hoping to take this in a few kind of directions I’m starting testing on an</p>
<p>empirical data set that has a very well-established you know ancestry specific hit so blood lipids and</p>
<p>african-american individuals to make sure this really does work not only in simulation but in real miracle data as</p>
<p>well we’re hoping that since recently admixed individuals have disrupted LD</p>
<p>blocks we should be able to leverage this to more kind of fine scale pinpoint</p>
<p>causal variants so we’re sort of diving into this now too and developing a</p>
<p>heterogeneity test to try to automatically suggest and eliminate candidate sites based on your admixed</p>
<p>populations and i’m also working with alicia martin who’s speaking immediately next and some other collaborators here</p>
<p>at the road and mph to try to build a polygenic risk or framework that would actually produce reliable estimates for</p>
<p>admixed populations which remains another kind of notable gap in current methods and I’ll just close with saying</p>
<p>not only is this procedure useful in the</p>
<p>G wast context but could really be applied to any situation where you need to control for population structure at a</p>
<p>very fine scale manner so for example even things such as you know evolutionary study is doing genome-wide</p>
<p>scans of selection so with that I’ll acknowledge my collaborators and</p>
<p>advisers and I guess I’ll take questions at the very end but thank you for your attention</p>
</section>
<section id="sec-video2" class="level1">
<h1>Ancestry-specific PRS</h1>
<p><strong>Caroline Nievergelt</strong>:</p>
<p>thank you so much Elizabeth that was really interesting Thanks our third speaker is Alesha Martin a militia is main structure at mth and affiliate researcher at the board in Harvard she received her PhD in genetics from Stanford University and her current research focus is on developing novel statistical methods to improve the generalizability of genetic risk prediction from your centric genetic studies so her talk today is entitled clinical use of current Polytechnic whiskers may May X activate health disparities go ahead Alicia</p>
<p><strong>Alicia Martin</strong>:</p>
<p>thanks Caroline um sorry try to move this other way well hopefully this goes away so I’m excited to talk to you guys today and thanks for having me here can you guys all see this part of the</p>
<p>zoom screen on the screen you can see your slides you can see my slides okay</p>
<p>perfect so I think an important point to start</p>
<p>at is in thinking about human population history so all of our genetic</p>
<p>differences across populations are sort of shaped by how we originated and how</p>
<p>we migrated and mixed as we moved out of Africa so of course humans originated in Africa this has been shown through</p>
<p>genetic evidence as well as archaeological and linguistic evidence so many sources I’m showing this and</p>
<p>then humans migrated Out of Africa and as they did so they took a subset of genetic diversity with them as they</p>
<p>populated Europe Asia Australia and into</p>
<p>the Americas another point that I think is worth talking about sorry here is the</p>
<p>fact that gos are becoming super increasingly powerful and so that’s very very exciting because we’re making so</p>
<p>many more bio medical discoveries these days some drugs are even being developed out of this and this has been growing it</p>
<p>such an exponential rate that it’s been really you know impressive and somewhat challenging even to keep track of all of this progress so that’s</p>
<p>been really really awesome to watch unfortunately however as laramie</p>
<p>mentioned earlier and Elizabeth did as well genetics has this diversity problem</p>
<p>so now shading the same growth and progress in genetics by the populations</p>
<p>that are represented in these genetic studies at the individual level you can see of course that the vast majority of</p>
<p>participants in genetic studies about 80% these days are of European descent and this is far out of step with the</p>
<p>global population where about 16% of the world is of European ancestry and</p>
<p>perhaps even more troubling Lee if we look at the fraction of individuals that have been in these jiwa studies as an</p>
<p>overall proportion the progress in diversifying genomics has somewhat stalled or perhaps slid a little bit</p>
<p>since about 2014 and so this is a really big issue if we’re trying to generalize studies to everybody and so one of the</p>
<p>nuances I want to mention here is that it’s not that the studies have been getting smaller studies have generally</p>
<p>been either staying the same or have been growing in different populations but one thing that we’ve seen is that</p>
<p>studies have been growing much much much more rapidly in European descent populations than elsewhere and so one</p>
<p>question that I really focus on is in understanding how to do ancestry study</p>
<p>biases in genetics impact the generalizability of the knowledge that we can learn from these studies and so I</p>
<p>break that down in a few ways in different aspects of our work but in general I want to highlight a few key</p>
<p>points that I think are worth keeping in mind throughout all of these questions so one of these is that the fundamental</p>
<p>biology is really shared across different populations so one person from</p>
<p>one population will probably have if they’re gonna have a heart attack it’s probably for the same underlying reasons</p>
<p>for example as an individual with another ancestry and that’s true in genetics face as well so when we’ve</p>
<p>looked across many different biomedical domains and tried to understand what causal variant effects are in</p>
<p>different populations these really tend to be mostly shared so it’s not that there’s anything special about genetics</p>
<p>that’s different from other biology in general the causal genetic variants seem to be the same and shared across</p>
<p>populations but there are some complications and interpreting genetics across populations for several different</p>
<p>reasons so of course there’s it’s also worth keeping in mind that there’s more genetic variation within than between</p>
<p>populations so populations are not substantially genetically differentiated</p>
<p>to the point where we’re finding completely distinct genetic populations in fact most of the genetic variation is</p>
<p>shared that’s common across populations and there’s more genetic variation within than between another point that I</p>
<p>think is really worth keeping in mind is the LD structure this correlation structure of the genome is one driving</p>
<p>factor that’s created a lot of challenges because this really differs across population as a function of human history going back to the first map of</p>
<p>slaw that I showed you so to try to address this question of how</p>
<p>generalizable generalizable our genetic studies are I started with computing</p>
<p>polygenic risk scores so laramie talked about this earlier this great enthusiasm in polygenic risk score space and how</p>
<p>it’s been really impressive to see the growth in this area in the past few</p>
<p>years just so we’re all on the same page I think most of you are familiar with a polygenic score but in general this is</p>
<p>just predicting an individual’s phenotype from genotype so we’re basically taking genotypes from some</p>
<p>target individual some effect size estimates from aji wasps that exists multiplying these together summing them</p>
<p>up across the genome and that’s basically our phenotypic prediction that’s a very simplistic method there</p>
<p>are other methods but some considerations that sort of cut across all of these different methods for computing polygenic scores are which</p>
<p>snips we should include what weights we should use and one thing that we always</p>
<p>need to address in terms of the utility of our polygenic scores is how accurate</p>
<p>is the score and this is really going to vary a lot with sample size the heritability the genetic architecture of</p>
<p>the trait and a number of different factors so definitely worth keeping in mind all of</p>
<p>these complexities and interpreting polygenic scores across the literature so a study that we had done previously</p>
<p>showed that population history really impacts genetic risk prediction across</p>
<p>diverse populations and so a few points here are the genetic prediction accuracy</p>
<p>decays with increasing genetic divergence between the discovery and target populations so and this figure on</p>
<p>the right I’m showing you a polygenic score distribution computed in the thousand genomes project so you can see</p>
<p>that we predicted European populations for example to be taller than American and South Asian populations and we</p>
<p>predicted East Asian and African populations to be the shortest globally but these differences really don’t line</p>
<p>up with anthropometric studies and they’re rather misleading these distributional shifts are really really</p>
<p>massive and so this is clearly not necessarily reflecting reality we also</p>
<p>set up some coalescence simulations alongside some statistical genetics</p>
<p>simulations and showed that these polygenic scores can differ across populations arbitrarily and these are not necessarily meaningful and we also</p>
<p>showed that neutral human evolution alone can be sufficient to explain these differences we don’t necessarily rule</p>
<p>out selection but we do say that neutral evolution and drift in particular may be driving some of these differences</p>
<p>there’s a really interesting couple of papers that I want to point out by Michelle Sohail Jeremy Berg and colleagues that were</p>
<p>published in Elife yesterday on this topic so we wanted to look at this in a</p>
<p>large-scale setting and so to do that we looked at the UK biobank data the UK</p>
<p>biobank is of course mostly consisting of European descent individuals and so</p>
<p>we used those individuals to conduct G wasps for several different traits 17</p>
<p>traits that were all quantitative so things like height and BMI and then a number of blood panel traits and then we</p>
<p>used the diversity in the UK biobank in this non European subset to try to</p>
<p>understand how generalizable prediction accuracy is and so in general what we saw is that if we normalize prediction</p>
<p>accuracy across all of these 17 traits to how accurately we predicted in europeans we</p>
<p>saw a pretty substantial drop-off and prediction accuracy across these different populations so for example on</p>
<p>the rightmost part of this graph we saw a four and a half fold improvement in</p>
<p>prediction accuracy in Europeans with respect to how well we’re doing in African descent populations we’re also</p>
<p>doing about twice as well predicting in European populations as in East Asians and you can see how well we’re doing in</p>
<p>these other populations so these are really really quite large disparities here so why is this happening a lot of</p>
<p>people have written about this before and it’s been well covered in the literature but in general there’s a</p>
<p>pretty predictable basis of polygenic risk for disparities and it obviously relates to who we’re studying so why</p>
<p>though well as you know G wasps are best powered to discover variants that are common in the population so if we’re</p>
<p>studying European populations over and over again in general we’re detecting variants that are most common in</p>
<p>European populations which are then able to explain more of the phenotypic variation in European populations than</p>
<p>those variants that are less common in other populations additionally the LD differences across populations mean that we are probably</p>
<p>getting better tagged snips in the G wasps in European populations than in</p>
<p>other populations and then there are other really much more complex topics</p>
<p>that are influencing this generalizability so difference is an environment selection and other more complicated differences but I want to</p>
<p>emphasize that there’s a lot of consistent promise from diversifying efforts so far so for example the pgc</p>
<p>schizophrenia working group led by hi Liang Wong and with a first author max</p>
<p>Lam have been working to really increase the sample size of East Asian individuals and schizophrenia studies so</p>
<p>right now the European case control studies in schizophrenia are still about</p>
<p>three fold larger than the East Asian Studies but the progress in the East Asians have been really really rapid and</p>
<p>massive lately and so when we looked at Paulo were able to predict east-asian</p>
<p>schizophrenia risk in these case control cohorts in general what we saw was that</p>
<p>we were much better able to predict east-asian schizophrenia risk using the</p>
<p>ancestry matched east-asian data despite the fact that the European training data was about three fold</p>
<p>larger so that indicates that there’s really a lot of promise and value in doing ancestry specific studies hearing</p>
<p>so lastly I want to turn to this effort that we did to try to compare some bio Bank scale analyses so generally what</p>
<p>we’re really interested in was doing equal size G wasps in the UK biobank and in the bio Bank Japan where we have a</p>
<p>lot of overlapping traits that have been deeply studied then we’re interested in predicting in the European populations</p>
<p>and in the Japanese population to try to understand what prediction accuracy looks like to see if this was symmetric</p>
<p>and comparable across populations we also tested prediction accuracy in the UK biobank African descent populations</p>
<p>so in general what we saw was the ancestry match disease prediction was most accurate so on the Left I’m showing</p>
<p>you along the x-axis the disease and on the y-axis I’m showing you the prediction accuracy and the ancestry</p>
<p>matched results are indicated by matching colors so on the y axis is the target of prediction and the summary</p>
<p>statistics that we’ve generated that we’ve used to generate the predictors are shown in blue for the biobank Japan</p>
<p>individuals and in red for the UK biobank individuals and so in general we</p>
<p>did better in both scenarios with the ancestry matched Disney use prediction one thing that’s sort of interesting</p>
<p>that we learned was the overall disease prediction was more accurate in Japan and that was sort of a consequence of</p>
<p>how these cohorts were constructed so the biobank Japan cohort is a more Hospital based disease ascertained</p>
<p>cohort whereas the UK biobank is a healthier and wealthier than average population-based cohort a similar</p>
<p>finding emerges from the quantitative traits so for these general health</p>
<p>measures anthropometric and blood panel traits in general the ancestry match quantitative trait prediction is also</p>
<p>most accurate taking a look at these y axes you can again see notable differences and help predict how</p>
<p>accurate prediction is in each of these populations and the quantitative traits are predicted most accurately overall in</p>
<p>general in the UK European samples and that’s again because of this ascertain</p>
<p>montano the-- populations of individuals</p>
<p>okay so I want to stop here and wrap up and think about next steps so in general</p>
<p>I think polygenic scores are really exciting and interesting and may have some power to improve clinical models</p>
<p>but right now I’m a little concerned or kind of a lot concerned that they’re likely to increase health disparities</p>
<p>due to these vast Eurocentric gee wasp IOC’s so I see this as a call for a few</p>
<p>pushes that we really need to make concerted efforts on one is we need much more diverse jiwa studies and another is</p>
<p>that we need new statistical methods to address these major issues and on this topic since it can be a bit sensitive i</p>
<p>just want to urge everyone that’s working on these cross-cultural you know widespread topics to communicate their</p>
<p>research responsibly and widely and anticipate the implications of your research thinking about the potential</p>
<p>negative consequences that will happen I also wanted to point out that there are large efforts for example going on in</p>
<p>Africa to scale up jiya studies there especially in psychiatric space for for</p>
<p>example developmental disorders and for schizophrenia and I think this is really exciting and it’s taught us a lot about</p>
<p>how to do cross-cultural research and an ethically responsible way that a company</p>
<p>is some capacity-building efforts so I want to acknowledge that this is a huge</p>
<p>effort involving a ton of really awesome people including my advisor and mentor</p>
<p>Marc Daly and with that I’m happy to stop and have everyone take questions I</p>
<p>suppose excellent presentation Alicia very</p>
<p>interesting thank you so much all right so Tammy can you make sure</p>
<p>that people can unmute themselves yes it</p>
<p>looks like they can and they can also ask questions in the chat box or the Q&amp;A box at the bottom of their screens okay</p>
<p>let’s see are there any questions</p>
<p>so I see people raising their hand let’s see um según would you like to go ahead</p>
<p>okay so thank you very much my name is chef at sumo so today I’m I’m just</p>
<p>calling from the South Floyd on that Monica I have many questions but I would</p>
<p>just ask them the most important one to me today so I have one question each for each presenter but my but what my fall</p>
<p>special goes to Elizabeth so I want to answer is that the method design which</p>
<p>is called structure and she has implemented this method by evaluating</p>
<p>its battle with the common way of checking for for population structure</p>
<p>that’s the principle principle component I wanted to parks Elizabeth are you also</p>
<p>a compare method with other other model that is used to control for population</p>
<p>structures such as the linear linear mixed model which was implemented in</p>
<p>many software so I would like to know how if your model is much more better</p>
<p>than endodermis model which is commonly used now in Africa to do Cheevers sure yeah</p>
<p>thanks so much for your question so yeah we started off kind of doing the simulations with a case control</p>
<p>framework and I guess it’s also important to note that in all the comparisons that I showed you today for</p>
<p>tractor versus the traditional GUI smart we do include principal components in</p>
<p>both so that is you know improvements even beyond the sort of incorporation of</p>
<p>global ancestry this really is like a local ancestry specific game but yeah</p>
<p>we’re also hoping to kind of think about other you know phenotype contexts beyond</p>
<p>the case control framework so yeah some some things were to be done but I think</p>
<p>we should expect similar power gains across contexts that answers your</p>
<p>question all right</p>
<p>next sem Leah Davies hey can you hear me</p>
<p>yes okay great thanks so much these were great presentations and I guess so one</p>
<p>question you know we’ve been struggling with this in the context of the biobank</p>
<p>in terms of performance of polygenic risk scores and trying to think about</p>
<p>ways to potentially squeeze out more information from the european central GA</p>
<p>us that would be translatable to non-european populations and one thing</p>
<p>that we’ve thought about but haven’t yet tried and i’m curious if any of the panelists have tried this is instead of</p>
<p>taking a clumping approach that’s based on p-values which will be in part driven</p>
<p>by the allele frequency in the discovery sample using a comb perch that’s</p>
<p>basically based on the effect size so that you’re retaining the snip was the</p>
<p>largest effect but also the highest allele frequency in the target</p>
<p>population I’m wondering if that could possibly coming of course it’s not going</p>
<p>to solve the problem certainly but could it possibly improve the predictions a little bit anybody looked at that we’ve</p>
<p>looked at it a little bit yeah um so I guess in terms of two different thing</p>
<p>both in terms of prediction accuracy and then also in terms of Alicia showed the</p>
<p>slide where the different thousand genomes populations have apparently different distributions apologetic</p>
<p>wrists courts mm-hmm so when we clump on all the ancestry is instead of clumping</p>
<p>on European ancestry and when we use a smaller set of snips that accounts for a</p>
<p>greater LD across all the populations we actually see pretty much normalized distributions across all ancestors so</p>
<p>that was kind of interesting we we tried to look at how well the predictions</p>
<p>worked across ancestry groups doing a you know clumping ideas like you mentioned and I think unfortunately in</p>
<p>what we looked at we didn’t have enough power to really determine if it was making a difference so I can’t say for</p>
<p>sure but it seems like a great idea to look into an adequately powered datasets yeah adding to what Laramie said there</p>
<p>are a bunch of different approaches with how you would do clumping and how you would potentially evaluate even in you</p>
<p>know sort of other methods and they can you know various methods can improve on</p>
<p>prediction accuracy across populations so for example LD pred does a little bit better than printing a threshold</p>
<p>approach but not as much as we would like and so one issue I think is that</p>
<p>there’s no method currently that I’m aware of that looks at multiple LD</p>
<p>panels across populations to try to translate sort of the effect size the</p>
<p>marginal effect sizes from one population to a marginal effect size to another population and so that’s an area</p>
<p>that we’re actively working on developing new methods in and so stay tuned and hopefully we’ll have some new</p>
<p>new method that you can try out there yeah awesome well if you if you are looking for a biobank size test case</p>
<p>that was a lot of diversity let us know that’s great thanks created all right we</p>
<p>can take a few more questions I see you David correct yes</p>
<p>yeah hi so I guess people know me so two things really one is where I’m coming from is that I think</p>
<p>quite likely that polygenic risk scores will never ever be committee useful so with that as the context my question is</p>
<p>really it is don’t don’t we think there’s a risk that in African populations because of the structure</p>
<p>that diverse the diversity and lack of linkage disequilibrium that the kind of phenomena that we see with gee wisest in</p>
<p>population have been through the Out of Africa bottleneck in East Asia in Europe the that that stuff just won’t work in</p>
<p>African populations the polygenic risk scores not work in African populations all would require at least sample sizes</p>
<p>of ten times a hundred times what we’re using to generate progenitor scores now</p>
<p>and I think the question is if polishing it with scores are gonna be clinically useful are we gonna massively invest in</p>
<p>getting these huge African samples and seeing whether they work or not or are we at this point can I look at it carefully and think maybe there won’t be</p>
<p>crimpy useful and therefore will deal with other things in Africa but not that</p>
<p>yeah David that’s an interesting question and we have also been kind of thinking along the same lines like how</p>
<p>would you improve things for everybody the fastest and so clearly the way that</p>
<p>the you know way that you would improve things for everybody is by starting to look at the most genetically diverged</p>
<p>group from what’s already been studied the most so looking at the most genetically diverged group from</p>
<p>Europeans means looking in Africa and so the way that we would generally help African populations have higher</p>
<p>prediction accuracy East Asian South Asians you know admixed populations everybody would be to start in Africa to build up</p>
<p>these very very large cohorts and so I think that’s part of the reason we’ve had such a concerted effort with trying</p>
<p>to push much larger studies in Africa because it’s been so vastly underrepresented in genetic studies to</p>
<p>kind of your first point will they ever be useful there has so I mentioned super</p>
<p>briefly that there is some promise from diversifying efforts so the ancestry matched results do a whole lot better in</p>
<p>non-european populations than just using a larger European population even when there’s several fold the sample size and</p>
<p>sort of a meta-analysis approach might improve things even more so I think in general having a</p>
<p>very very large African cohort is probably like step one and I’m diversifying and meda analyzing a lot of</p>
<p>different groups together is probably step two and yeah we’re you’re right we’re a long ways off I think that’s</p>
<p>kind of the direction we need to move in thank you Alicia that kind of and also</p>
<p>fit into the question that Geeta sent ill and posted what kind of sample size</p>
<p>is do we need across various populations to compute and just a specific PRS so do</p>
<p>you have any idea of it’s probably very difficult to say right yeah that’ll</p>
<p>depend a lot on genetic architecture of the traits you’re looking at how shared the genetic like what the genetic</p>
<p>correlation across ancestries is yeah it’s a hard question to answer without</p>
<p>without some more specifics unfortunately for sure some more funding would help right alright so I see we</p>
<p>have time for one more question Monica do you wanna sure excuse me can</p>
<p>everyone hear me yeah great well first of all I’d like to compliment our three presenters today for doing a fabulous</p>
<p>job and in particular bringing to a larger audience our attention on this</p>
<p>very important problem that you know it sounds like you individually and collectively been working on for some</p>
<p>time but sometimes I don’t think the larger scientific community is as plugged into this as they need to be so</p>
<p>you’re doing a great service today by presenting what you’ve been working on so so well my question is I think really</p>
<p>for Elizabeth and in particular in relation to something that Laramie</p>
<p>mentions so Larry mentioned that our abilities to sort of detect these</p>
<p>ancestry differences was informed a little bit by the PGC ptsds more diverse</p>
<p>cohorts that included a larger proportion of individuals from African ancestry in particular and so these</p>
<p>methods have been developed using those data if I understand correctly that Elizabeth presented well those methods</p>
<p>be available soon for other pgc cohorts to use or even for other cohort</p>
<p>that focus specifically on psychiatric outcomes sure so thanks so much for the</p>
<p>comment to us all and for the question yeah certainly so we’re using a you know</p>
<p>PTSD in an african-american demographic context as kind of our our initial use</p>
<p>case just to test the performance of this method but it certainly could be applied to any complex you know type you</p>
<p>know it’s built around psychiatric phenotype so certainly psychiatric you ties phenotypes but even kind of more</p>
<p>broadly and medical genetics complex traits in general should be all served well by by attractor it’s also it should</p>
<p>also perform well provided local ancestry performs well its perform well in other populations so I’m starting to</p>
<p>do some work looking at more complex demographic scenarios so for example and you know three-way add mixed Hispanic</p>
<p>Latino individuals are a little bit of a trickier situation when a two-way admixed population so I’m scaling it up</p>
<p>now and making sure that it’ll still perform well in those contexts as well but yeah basically any any time that you</p>
<p>can accurately call local ancestry which depends mostly on divergence between the component populations doctors should be</p>
<p>should be able to to handle it and so do you have a timeline for making the code</p>
<p>available I know you said we could email you but to deploy it on the broad website or something like that</p>
<p>yeah I’m working with some developers for this you know Hale software which is a hosted by the</p>
<p>Broad Institute and is uh you know going to make it a lot fancier and run faster</p>
<p>and work in every computational environment so I think the timeline for that is certainly within six months it’s</p>
<p>uh it’s currently a set of Python scripts that runs but probably will run nicer once I get the Hale developers to</p>
<p>pretty it up so yeah within six months there should be a bi archive paper with</p>
<p>with the software attached to it I would say thanks so much everybody I think we</p>
<p>are past the hour now and I would like to thank all the speakers and all the</p>
<p>many participants this was a really world late meeting thanks so much thanks</p>
<p>Caroline guy thanks everyone yeah thanks so much</p>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
          // target, if specified
          link.setAttribute("target", "_blank");
          // default icon
          link.classList.add("external");
      }
    }
});
</script>
</div> <!-- /content -->



</body></html>