<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.353">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>PGC Video Textbook - Chapter 8.1: SNP Heritability (Video Transcript)</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


<link rel="stylesheet" href="styles.css">
</head>

<body class="nav-sidebar docked nav-fixed">


<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="./index.html">
    <span class="navbar-title">PGC Video Textbook</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link active" href="./index.html" rel="" target="" aria-current="page">
 <span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./about.html" rel="" target="">
 <span class="menu-text">About</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./contact.html" rel="" target="">
 <span class="menu-text">Contact us</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="https://pgc.unc.edu/" rel="" target="">
 <span class="menu-text">PGC Website</span></a>
  </li>  
</ul>
            <div class="quarto-navbar-tools ms-auto">
</div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
      <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item">Chapter 8.1: SNP Heritability (Video Transcript)</li></ol></nav>
      <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
      </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal sidebar-navigation docked overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
      <a href="./index.html" class="sidebar-logo-link">
      <img src="./pgc_logo_website_v3.jpeg" alt="" class="sidebar-logo py-0 d-lg-inline d-none">
      </a>
      <div class="sidebar-tools-main">
    <div class="dropdown">
      <a href="" title="" id="quarto-navigation-tool-dropdown-0" class="quarto-navigation-tool dropdown-toggle px-1" data-bs-toggle="dropdown" aria-expanded="false" aria-label=""><i class="bi bi-github"></i></a>
      <ul class="dropdown-menu" aria-labelledby="quarto-navigation-tool-dropdown-0">
          <li>
            <a class="dropdown-item sidebar-tools-main-item" href="">
            Source code
            </a>
          </li>
          <li>
            <a class="dropdown-item sidebar-tools-main-item" href="">
            Report a bug
            </a>
          </li>
      </ul>
    </div>
</div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./welcome.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Welcome to the PGC Video Textbook!</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./toc.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Table of Contents</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">
 <span class="menu-text">Chapters</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter1.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 1: Introduction</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter2.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 2: The Genome</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter3.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 3: Technologies</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter4.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 4: Study designs</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter5.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 5: GWAS analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter6.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 6: Polygenic Scores</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter7.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 7: Ancestry-Specific Analyses and Considerations</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter8.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 8: Post-GWAS bioinformatics</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter9.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 9: Advanced Topics</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter10.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 10: Other Considerations</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true">
 <span class="menu-text">Software Tutorials</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_cnvs.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">CNVs</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_conditional.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Conditional Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_crossdisorder.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Cross-disorder Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_ewas.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">EWAS</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_geneset.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Gene Set Identification</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_gwas.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">GWAS</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_genomicSEM.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Genomic SEM</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_imaging.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Imaging</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_MR.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Mendelian Randomization</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_prs.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">PRS</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_correlation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">SNP Heritability and Genetic Correlation</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true">
 <span class="menu-text">Additional Resources</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./glossary.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Glossary</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./resources.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Software Resources</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./addreading.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Additional Reading</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="https://pgc.unc.edu/for-researchers/download-results/" class="sidebar-item-text sidebar-link">
 <span class="menu-text">PGC Summary Statistics</span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Chapter 8.1: SNP Heritability (Video Transcript)</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<p><strong>Title</strong>: Heritability and SNP Heritability</p>
<p><strong>Presenter(s)</strong>: Alkes Price, Broad Institute</p>
<p>again good morning everyone and welcome to our first virtually presented mpg session we’re fortunate today to have Dr.&nbsp;Alkes Price from Harvard School of Public Health speaking he’s a professor of statistical genetics in the Department of Epidemiology and today we’ll be talking about heritability and snip heritability and the relationship to the genetic architecture of disease a few notes about questions so I guess has kindly offered to answer questions both at the end and also importantly during the talk itself to sort of make sure that those are are voiced in this this virtual format and please plan on typing them into the Q&amp;A box which at least on my screen is at the very bottom there you can type your question and also upvote questions posted by others that are of particular interest to you and I will do my best to keep an eye on that and to interject as as needed during the talk also if you have questions that you prefer to save until the end of the talk I will keep an eye on that and we can aim to address those then so again</p>
<p><strong>Alkes Price</strong>:</p>
<p>thanks very much for joining and I think we can get started with the talk all right well good morning everybody I’m alkis price from the Harvard School of Public Health and welcome to this first virtual mpg primer session I’m going to talk about heritability and snip heritability which is basically a sort of introduction to the genetic architectures of disease and as Diane mentioned I do encourage everybody to jump in with questions throughout the talk in the format that Diane communicated so let’s get started I think that everybody in this audience will be aware that genome-wide Association studies have already been very successful at producing important biological insights this is just one example from the schizophrenia landmark schizophrenia G was published in the year 2014 in nature with a lot of contributions from people at broad and this landmark study had a huge number of interesting and important findings and yet at the same time I think everybody in the audience will be aware that even though genome-wide Association studies have found a lot of things they certainly have not found everything and in the case of schizophrenia we know that the findings of the 2014 paper explained about 3% of heritability whereas the heritability of schizophrenia has been estimated at about 64 percent from twin studies so there’s this big gap between what we found and what we believe is out there and this gap is classically known as the missing heritability and this story about missing heritability goes back to around 2008 or so this is a commentary of mehar 2008 nature and in the year 2008 people didn’t really know what the cause of the missing heritability was now I think it’s much better understood and this talk introducing the concept of snip heritability will also provide a review of what we know about the answer to this mystery of missing heritability so with that in mind here’s an outline of my talk I’ll start with an introduction and a definition of heritability then I’ll talk about genome-wide Association studies and missing heritability and then we’ll delve into this idea of heritability explained by snips also known as snip heritability and if we happen to have extra time there are some extra bonus topics pertaining to heritability that we might have time to scratch the surface on so let’s start with an introduction to heritability</p>
<p>so heritability is generally defined as the proportion of phenotypic variance that is due to genetic effects and most of the time when people are talking about heritability they’re talking about narrow-sense heritability which is the proportion of phenotypic variance due to additive genetic effects you could also talk about broadsides heritability which could include epistatic or recessive dominant effects but those are generally harder to estimate and so generally when people are talking about heritability they’re usually talking about narrow-sense heritability denoted as lowercase h squared and so people have been trying to estimate narrow-sense heritability for a long time at least back to the year 1886 and one way that you could do this is you could take some relatives and see if they are somewhat phenotypic ly similar because if a trait is genetically heritable then you would expect that relatives are going to be somewhat phenotypic ly similar and there’s a method called Haseman Elston regression but I’m not going to describe in detail but here you’re basically regressing the phenotypic similarity on the genetic similarity or the expected genetic relationship amongst a particular pair of relatives so you know a parent-child pair or a sabzi pair have an expectation about 50% share genetics and you could ask how a phenotypic Lee similar are those pairs what’s their phenotypic correlation and so on this graph this is sort of a an amalgamation of different results that were compiled in the paper of issue at all 2010 in which they each points or represents one study and you can kind of see this is a study this is a sort of amalgamation of studies of height that generally people who are very closely related to each other tend to have very correlated values of height and this is generally sex adjusted standardized height that people are studying whereas on the other hand people who are a little bit genetically similar to each other tend to have height that’s a little bit you know typically correlated and that sort of stands to reason of what you’d expect for a trait that’s largely but not completely heritable and the slope of this line in this paper is estimated at point seven four seven which might mean that height is something like 75 percent heritable in terms of the narrow sense heritability but there’s a little bit of a surprising finding here which is that there’s an intercept that you know as you’d think that it as your as your familial genetic relationship goes to zero the phenotypic correlation ought to go to zero right if you’re sharing close to zero genetics you should have close to zero phenotypic correlation so you’d really expect this red line to go through the point zero zero and yet surprisingly this red line seems to not go through the point zero zero there’s this large intercept and there’s a lot of speculation you know different hypotheses why what could this be due to probably the simplest explanation is shared environments that you know even people who are just cousins they come from a you know a type of socio-economic background or something like that that makes them more disposed to be taller or more disposed to be shorter as a consequence of environmental effects and that is one possible explanation for why this line does not go through zero zero and there’s more complicated expect explanations as well involving genetic ancestry and Anna sort of mating and so on and so on but we just have to keep in mind as we think about narrow-sense heritability that is complicated and there’s a lot of room for different sorts of confounding and complex effects as we try to estimate this quantity that I have defined and probably the most popular way right now to try to estimate narrow-sense heritability is using the classic monozygotic die gods dizygotic twin study and so the idea here is the hope not the guarantee but the hope is that monozygotic twins versus dizygotic twins have basically the same amount of shared environment and really the only thing that differs between monozygotic twins and dizygotic twins is that monozygotic twins share 100% of their genetics whereas dizygotic twins share only 50% of their genetics and so the difference in how much phenotypic correlation you see between monozygotic twins on one hand versus dizygotic twins on the other hand should give you a sense of how you know how heritable the trait is and this twin based approach is really the it’s widely considered as the gold standard in ways to estimate narrow sense heritability and it should work unless there’s a difference in the amount of shared environment between monozygotic twins on one hand and dizygotic twins on the other hand for example due to effects in the womb or due to sort of societally you know effect for the family influences differences in the ways that man has got a twins versus dizygotic twins are treated and we have to take seriously this possibility that there could be still some confound into the differences in the amount of shared environment between monozygotic twins and dizygotic twins where by monozygotic twins have more shared environments and that might actually inflate the twin based estimates of heritability and I will provide some evidence later in the talk that decision fact likely to be the case that that the twin based estimates may in fact be inflated all right and so keeping in mind that these estimates might be inflated there have been a lot of a lot of work published you know that probably the broad and recent reference worth looking at is polar men at all 2015 Nature Genetics but for example for height the most quoted value from twin studies is 0.8 and then you know recent studies of cancer have have produced estimates of around 0.3 to 0.5 and generally speaking most of the disease diseases and complex traits that people tend to be interested in height is a bit of an outlier that’s extremely heritable but most of the diseases and complex traits that people tend to be interested in seem to have narrow-sense heritability estimates from twin studies that might be on the order of around 0.5 or a little bit less than 0.5 typically somewhere in that range all right so that does a brief introduction to narrow-sense heritability and now i’m gonna delve just a little bit more into the missing heritability problem from</p>
<p>genome-wide Association studies that I’ve already defined early at this talk so again the missing heritability is originally was defined as this gap between what we discover from the genome-wide significant low-side 4g wasps versus the estimated narrow-sense heritability from twin studies and again using schizophrenia as an example 3% is the hair ability explained by the 108 gos Llosa from PG c 2014 nature versus on the other hand 64% the narrow-sense heritability that’s been estimated from twin studies that’s a very large gap and in the early days of gos people were really very interested to try to understand what the cause of this gap is like why doesn’t G us find everything why is it so incredibly far away from finding everything and there’s a lot of explanations out there but these are the four that I think are worth most worth discussing in this MGP primer format so</p>
<p>I’m gonna go over these four explanations one at a time so the first explanation is common causal variants of exceedingly low effect size and so we could imagine some different possible values of the genetic architecture of a disease or trait one possible genetic architecture is that you have ten common risk variants which each explained about 1/10 of the heritability and this is what people thought you know a long time ago you know way back only maybe early 2000s this is what people thought disease architectures might look like there’d be 10 low sigh you’d run into us you’d find the 10 low sigh and if story that’s what people were expecting but it might be more complicated than that you know there could be 10 there could be 100 or a thousand or even 10,000 common risk variants that each explained a tiny tiny tiny tiny proportion of heritability and in this extreme case at the bottom of this slide were we have 10,000 common risk variants that each explained on the order of one ten-thousandth of narrow-sense heritability maybe more in some cases and less in some other cases you can imagine that even at a very large sample size Egeus is going to be very underpowered to find them and you might only find a very very small fraction of the true common causal risk variants and that’s why you’re only going to explain a very very small proportion of the narrow-sense heritability with the ones that you’re actually lucky enough or well powered enough to actually find and this is what a lot of people believe that you know schizophrenia happens to be an example of a particularly polygenic trait with a lot of causal variants and so this may be what we’re looking at we may be looking at something like 10,000 common risk variants they need to explain a tiny tiny tiny proportion of heritability and even at very large sample sizes you’re going to be under powered to detect most of them you’re only going to detect the ones with largest effects or you’re going to attack a few because you get lucky or something like that but most of those common risk variants you’re just not going to detect them as being genome-wide significant in the G loss and we can even go to a greater extreme the so called the infinitesimal model nobody believes that this is realistic but it can be useful to think about as a theoretical construct and so the infinitesimal model is a model image all of the common snips in the genome are causal risk variants causal risk variants with causal effect sizes something like you know normally distributed with mean 0 and variance H squared divided by M where m is the number of snips and so there you’re sort of spreading the the narrow-sense heritability across literally all the common snips in the genome so this is one very plausible explanation as to what is going on with missing heritability we’ve got a lot of causal a lot of common causal snips of tiny effects gos even at large sample size are not finding most of them and you know we do know that there’s a lot of traits that are extremely polygenic I’ve already said that schizophrenia is a particularly polygenic trait this is a Manhattan plot from the blood pressure G wasps of Evangeline at all 2008 teenager genetics and this is another another illustration of an extremely polygenic trait which clearly has an extraordinarily large number of of causal Llosa so a second explanation that people have been interested in for a while is the explanation of rare and low frequency causal variance and we know that G wasps are not well powered to identify rare and low frequency causal variants because the power sort of scales with the allele frequency if you have a really rare variant it doesn’t occur very often in the sample and so you’re not going to be well powered to detect it and mathematically we know that etta had a specific fixed per allele effect size our our power scale something like with the sample size times P times 1 minus P where P is the minor allele frequency and so if you have a really rare variant then you’re definitely not going to be well powered to detect it as being genome-wide significant energy loss where you’re conducting single marker tests I mean I’m not I’m not going to get into the topic of sort of multi 8 marker gene level test which is a topic for a different MPG primer session and so G wasps are not well powered to identify these rare and low frequency causal variants as being genome I significant and maybe that’s where a lot of the missing heritability is and that’s an explanation people have been interested in but recent work from Zhang a tall 2018 nature genetics as well as other work from our group shook at all it is all at all are suggesting that there’s not really a lot of heritability coming from rare and low-frequency snips so according to shook at all it’s less than 10% of snip heritability I know I haven’t defined snip heritability yet we’ll get to that later but less than 10% of snip heritability is coming from snips with minor allele frequency less than 1% now there’s a little bit of a complicated story here involving negative selection it under under simplified assumptions where you don’t have any selection and you might have to also assume equal a constant effective population sizes across time but under some some assumptions I’m not going to get into you might expect that snips with math less than 1 per sentence only</p>
<p>ought to explain about 1% of snip heritability so there is some excess here where they actually explain I couldn’t do this paper about 9% of snip heritability which is a lot more than 1% and that excess where by 9% is more than 1% is a consequence of the action of negative selection which sort of causes really important snips that have important effects too generally those important effects are usually bad for the organism and because they’re bad for the organism those snips cannot rise to high frequency and stay as rare snips and so you might expect that rare snips will tend to have larger effects and in fact that’s exactly what you see and that’s why snips at math minor leave frequency less than 1% explains somewhere around 9 percent of snip heritability which is a lot more than 1% of snip heritability but even so at the end of the day according to these papers these rare and low frequency snips are not really explaining a lot of heritability and they are probably not the primary explanation for missing heritability now to be clear I don’t I specifically do not want to say that rare and low frequency causal variants are not important and we shouldn’t study them rather what’s going on here is that if you’re interested in heritability or in explaining a lot of heritability then rare and low frequency causal variants don’t contribute a lot and by extension if you’re interested in polygenic prediction which kind of piggy box piggy backs off of heritability then rare and low frequency causal variants aren’t very important but on the other hand if what you’re interested in is discovering interesting disease biology that could lead to a drug target rare and low-frequency calls of variants may be really important you may identify a rare coding variant that explains a miniscule amount of heritability but it has a really biologically interesting sort of mechanism behind it that might lead to a drug target totally fine if it explains a miniscule amount of heritability if it’s going to lead to an actionable drug target so let’s just keep in mind that even though they’re not so relevant for heritability they could be very important for disease biology and developing drug targets ah the next explanation that people I see that there’s a QA Diane do you want to read out the QA thank you yes um I will do that now so the question from now son is how they heard how doesn’t heritability of rare how is the room sorry how is the heritability of rare variants estimated for a jiwa study that is underpowered</p>
<p>okay so the question is how is the heritability of rare variants estimated for a jiwa study that is underpowered and I the topic of snip heritability that I will delve into in the third part of this talk delves into estimating the aggregate heritability contributed by all the variants in the genome may be I mean mostly what I’m going to talk about is estimating the heritability explained by all common collectively explained by all common variants in the genome but something related to that that you could do is you could estimate the heritability explained by all rare variants in the genome and so these studies that I’ve quoted on this slide shook it all saying it’ll cos all and there’s others they’re doing something sort of like that they’re doing something they’re sort of extending the methods to estimate snip heritability or heritability explained by all snips in the genome which I will be talking about in the third part of this talk they will be extending that to estimate the heritability explained by all snips in a particular minor allele frequency class such as the heritability explained by all rare variants and I will aim to return to this question in the third part of the talk when I talk about snip heritability hopefully what I just said will become more clear after I delve into what’s going on with sniff heritability all right so moving on to explain explenation number three on my list which is copy number variation and it is in principle possible that copy copy number variants are biologically important and contribute a lot to heritability but are not well tagged by common snips and there’s another question</p>
<p>see I don’t see the question in the Q&amp;A box okay in that case I will keep going</p>
<p>thank you so it’s possible that the copy number variance could be important for disease but their effects may not be so well tagged by snips so that if you’re only looking at snips you won’t see them and we have to take that possibility seriously way back in the year 2010 this paper that I’ve cited suggested that common copy number variants that could be tight on existing platforms and that’s an important qualification did not contribute much but that may be more at saying more about the technology than about the biology and more recent work of segments at all 2015 is suggesting that structural variants are enriched on haplotypes identified by G Y so some of the common snips that we identify as genome-wide significant in G wasps may be tagging causal copy number variants so I don’t think we have a conclusive answer to this question right now as to whether you know untight and an untagged or only partially tagged copy number variants are responsible for a lot of the missing heritability III would like to go out on a limb and hypothesize if this story might maybe be sort of similar to this story with rare variants and some of these copy number variants of course are likely to be rare because you can’t you know you can’t knock out a big chunk of genome and and and have it not have a huge effect which would then due to negative selection keep the variant rare I hypothesized that it might be true that just like rare variants copy number variants might not explain a lot of heritability and might not be important for quantifying heritability or by extension and for apologetic risk prediction but on the other hand they might be really important for understanding disease biology and they might lead us to examples where we understand the biological mechanism and can develop drug targets so I hypothesized that they might be not so important for heritability but at the same time very important for disease biology and drug targets and then finally the fourth explanation that I wanted to talk about is the possibility that narrow-sense heritability was overestimated in the first place and I already alluded earlier in this talk to the possibility that narrow sense estimates of narrow-sense heritability might be inflated due to shared environment and there’s some other complicated explanations whereby if you have G by G interaction which is not supposed to be included in estimates of narrow-sense heritability which is defined as including only additive effects and not G by G interaction effects then according to look at all 2012 that could inflate your twin based estimates so there’s people out there who made who believed that the twin based estimates may be inflated despite the fact that they’re sort of the best well until recently at least they’re the best thing we have I’d like to highlight this paper of young at all 2018 a nature genetics which introduced a new method called relatedness disequilibrium regression which is predicated on having at your disposable at your disposal a really large data set with a lot of related individuals which those authors happen to have because they analyzed the decode genetics data set from Iceland and they claimed that they have an estimation procedure that is robust to these types of effects of shared environment and they claim that the narrow-sense heritability of height is only about 0.55 and we have to take those claims seriously they are John fairly consistent with a couple earlier studies of Zaillian at all from our group and I think people at this point in time tend to believe that point eight four for height for the twin based estimates of a neuro sensor to be live height that point eight probably was an over estimate and the truth might actually be some number closer to about point six or so I think that’s what people most people believe and so summing it all together I think in terms of heritability the two explanations for missing heritability that are most prevalent are number one you’ve got common variants of exceedingly low effect size that cannot be detected by G wasps as being genome-wide significant and number two the narrow that the twin base narrow-sense heritability estimates are somewhat too high obviously this is not going to get you you know a slightly overestimated narrow sense notability is not going to explain a difference between point zero three and 0.64 for schizophrenia which is a humongous difference and that’s probably more dominated by number one so where does this leave us well there there’s a sort of fundamental question that was asked originally in a landmark paper of yang at all 2010 Nature Genetics from Peter Fisher’s group maybe we can try to estimate the heritability explained by all the snips in the genome or maybe all the common snips in the genome not just the snips that are genome-wide significant but in fact all the snips collectively even we don’t know which ones are the causal ones we can still try to estimate the heritability jointly explained by all of those snips together and so that’s this concept of snip heritability is really the main focus of this primer and I will start to now delve into that and so the distinction between on the one hand narrow-sense heritability and on the other hand snip heritability the hair ability explained by snips specifically uh it really sort of rests on the distinction between two important ideas</p>
<p>IBD or identity by descent and IBS or identity by state so identity by descent or IBD means two people are related they have similar genetics IBS or identity by state means two people who are not related may still by chance have genetics they’re a little bit similar and you might be able to do something with that to learn about complex trade architectures so let me delve into a little bit of detail let’s start with IBD let’s suppose you take two individuals who are related so the two individuals I depicted on this slide are brothers and we could ask ourselves what is the proportion of the genome that these two brothers share identical by descent which means inherited from a recent common ancestor and the answer well the answer is not exactly point five the answer is approximately point five because it can vary a tiny bit from one step to the next but an expectation is 0.5 and you can imagine if you have sort of a large set of related individuals who you’re analyzing genetic data from you can build an IBD matrix quantifying the IBD for individuals and for a pair of individuals who are brothers the entry would be 0.5 because point five is the IBD of those two brothers and this cave matrix or this IBD matrix you could use it to estimate narrow-sense heritability and no worries if you if you don’t feel like sorting through all these equations but this is what it would look like in terms of math that you have a vector Y of phenotypes and you’re you’re sort of decomposing the the phenotypic variance into the part coming from genetic effects called U and the part coming from environmental effects called epsilon and the variance of U the genetic effects isn’t it sort of proportional to the IBD matrix and where’s the environmental effects on the other hand if we make a strong assumption of no shared environment then the environmental variance is proportional to the identity matrix that is saying we have no cross terms at all between distinct individuals and then you you sort of parameterize the overall phenotypic variance of E in that way estimate these parameters and estimate the narrow-sense heritability and there’s methods you know max likelihood methods or restricted max and likelihood methods for estimating that parameter or actually you’re estimating the two parameters Sigma square G and Sigma square e from the previous slide I’m going to just kind of skip most of mathematical details you can read about it in these various papers and you know no worries if if you’re not following all the math so on the other hand we could have two unrelated individuals such as the two individuals depicted on this slide and even though these two individuals are completely unrelated it is possible that they might be just a little bit more genetically similar than average or just a little bit less genetically similar than average just by statistical chance and we can sort of quantify that in using the equation on the right half of this slide where we can compute something this is sometimes called the genetic relationship matrix or it’s sometimes called could be called an IBS matrix but it’s basically you just sort of like compute the correlation across snips between these individuals genotypes suitably normalized and typically if you have two unrelated individuals you can practically guarantee that that number is going to be really really close to zero but in these standardized units in which the mean is going to be zero on average it might be a tiny bit bigger than zero like zero point zero zero four or it might be a tiny bit less than zero like- it’s zero point zero zero four and you might expect that if you have a heritable trait then two individuals who just by chance are a tiny bit more genetically similar than average hot-hot to have slightly concordant phenotypes whereas on the other hand individuals who are slightly less genetically similar than average ought to have slightly less concordant phenotypes and this is something that</p>
<p>you can use to estimate the heritability explained by snips or heritability explained by genotype snips in the way that it was originally employed and the real question here is what is the set of snips and well the answer is related to the snips that you used to compute this genetic relationship matrix if you only use snips on chromosome 1 to compute the genetic relationship matrix then you’ll get an answer that has something to do with snips on chromosome 1 if you use all common snips to compute the genetic relationship matrix then you’ll get an answer that has something to do with all common steps and so on and so on and so forth and so once again we can model the phenotypic covariance of V as sort of a linear combination of this genetic relationship matrix or IBS matrix that’s sort of the the genetic part of the genetic part coming from snips and then everything else which is the environmental part or strictly speaking it’s everything except the genetic part coming from snips and and then we can estimate this quantity called snip heritability and again reverting to a little bit of math here I feel that when we if we’re going to talk about estimating a quantity we really wanted to find that quantity in the entire population and this is a formal definition of that entire quantity of that quantity in the entire population and we can skip the mathematical symbols here and in words this is just the maximum amount of phenotypic variance that you could explain using any linear combination of snips and that’s the</p>
<p>definition of snip heritability that’s a definition in the entire population that definition does not depend on a particular sample although it does depend on which set of snips you’re looking at if you’re looking at you know just snips on chromosome 1 the answer will be smaller than if you’re looking at all snips in the genome or you might be looking at just the genes of just a small just a few hundred thousand genotype snips or a large number of imputed snips or common snips or common and rare snips or whatever and the answer will be different in each case but this is a quantity that you can define in the entire population and then after you’ve defined it in the entire population then you could obtain an estimate of that quantity an estimate with noise of that quantity in a finite sample and this is what was done in this landmark paper viewing at all 2010 Nature Genetics and this is the same math that I showed on an earlier slide for IBD this is just the same math with the IBD matrix K replaced with the IBS or gr M matrix called capital A and we’re now intuitively thinking about this in terms of unrelated individuals and snip heritability but mathematically all the math is the same and all the maximum likelihood or restricted maximum likelihood computations are the same so I’m going to choose to not focus on the math in this talk and I’m gonna gonna focus more on intuition and so now we have these two quantities H squared that’s total narrow-sense heritability and that’s sort of that’s corresponds to</p>
<p>the question how phenotypic ly similar are two relatives there’s a question there is yes thank you I think the question is asking or the question savanna lewis is asking the following when you say the entire population do you mean the entire human population okay so this is this is a good point</p>
<p>this is something I’ve really glossed over and you know if I wanted to delve deep into the population genetics that could be a separate MPT primer but generally when people talk about a population they’re talking about a populace if ik population of a particular continental ancestry one example the population would be European Americans another example the population would be the British ancestry individuals from the UK biobank and so on and so forth now if I want to be a strict population geneticist I might define a population as a set of Penn matically randomly mating individuals now know believes that European Americans are a set of ethnically randomly mating individuals nobody believes that British ancestry individuals from the UK or you know East Asians from Japan or you know Nigerians or whatever population you’re talking about nobody believes that that’s strictly speaking of magnetically mixing set of individuals we might just choose to pretend that’s the case as a sort of approximation and I mean I mentioned earlier at the beginning of this talk that there’s opportunities for all sorts of confounding one sort of confounding has to do with confounding due to population stratification do the differences in genome-wide ancestry amongst different individuals that again could be a topic for another MPG primer we should be aware of the possibility that if you are studying a population such as European Americans or any population in which there are differences in genome-wide ancestry amongst different individuals in that population we have to be a little bit careful because there is the possibility for confounding to the population stratification but the short answer is we’re thinking about a specific population of a specific continental ancestry all right and so getting back</p>
<p>to this slide narrow-sense heritability it corresponds to the question you know how phenotypic ly similar are to relatives and it’s kind of implicit that two relatives could be phenotypically similar due to like whatever genetics they happen to be caring maybe they’re carrying the same rare variant maybe they’re selling the same copy number variant maybe they’re carrying the same common variant it’s going to include all of that on the other hand we have a smaller quantity snip heritability originally this was called the heritability explained by genotype snips because people liked it to estimate this just using genotype snips before imputation became kind of universally popular so that’s why you’ll see this terminology on some of these slides but you can apply this to any set of snips and this this corresponds to the idea if</p>
<p>I have two unrelated individuals and I use a specific set of snips and the set of snips is important to quantify how genetically similar they are just by chance then how phenotypic ly similar will they be and again this is a function of a very specific set of snips and because it’s a it’s only capturing the heritability causally explained by a very specific set of snips in general it’s expected to be lower than the total narrow-sense heritability which is the capturing additive effects from all possible genetic variance all right and then finally there’s a third quantity that has already appeared in this talk and I’m going to call that H for G us and H squared G us is the heritable explained by genome-wide significant snips and I mean if you’re not worried about LD then it’s basically just the sum of the variants contributed by each of your genome-wide significant snips in turn but it’s a little bit more complicated if you have LD and strictly I can define it as the maximum proportion of phenotypic variants that you can explain with any linear combination of genome-wide significance snips so just focusing on those snips that are genome-wide significant and that’s that’s the number that’s the number that was in a particular study at point zero three for schizophrenia and I have to be a little bit careful when I say that I’ve defined this it’s not really a pump it’s not really a true population level parameter because it’s a function of which snips up which snips pop up is being genome-wide significant in a particular gos at a particular sample size so it’s really a function of a specific G watch that identifies a specific set of genome-wide significance snips it’s really a function of what the set of genome-wide significance snips is in a particular study and so now we have we have sort of inequality with three different quantities we have h4g watch that’s the smallest number which is just the genome-wide significant snips we have H squared G that’s the snip heritability that’s all the snips in the genome maybe all the genotype snips maybe all the common snips whatever flavor of snips you want to be choose to study at a particular point in time but it’s in some level the heritability explained by all the snips in the genome including all the snips that are not genome-wide significant but might contain some signal and then funding to largest quantity total narrow-sense heritability that’s the additive heritability explained by all genetic variants which includes not only every possible type of snip but other types of variants as well copy number variance and so on and so so this has been a little bit of a dry technical discussion so now we might want to look at some real data to sort of get a sense of</p>
<p>having some intuition of how this works in practice and so I’m going to start with height which is probably the the trait most well studied by geneticists and in terms of narrow-sense heritability well from the twin studies we have 0.8 as I mentioned earlier that’s probably an overestimate we’re probably closer to 0.6 but for now I’m just going to say 0.8 from the twin base studies and then we have snip heritability this is the quantity that yang at all 2010 Nature Genetics in their landmark paper estimated at 0.45 and then the hair adjust the heritability explained by Gy snips well way back in the year 2010 that was at about zero point one zero although it’s gone up a little bit with some studies of wood at all 2014 Nature Genetics and enu at all I forget which year 2018 or 2019 human likely genetics it’s a little bit higher than 10% now but you can see that this concept of snip heritability can explain most not all but it can actually explain most of the missing heritability and this goes back to explanation number one of my four explanations that I quoted earlier where you know there’s just a lot of snips with really small effects that you know that gee wasps do not detect as being genome-wide significant and if you define and estimate this quantity H to RG which quantifies the heritability explained by all snips or all snips of a particular category such</p>
<p>as genotype snips not just the ones that are genome-wide significant then you get a much much much larger number like such</p>
<p>as 0.45 then you get from just the genome I significant ones such as point</p>
<p>1 0 and I like to use the terminology</p>
<p>hidden heritability as do others hidden heritability to explain this gap between h4g wasps in each for a G so that</p>
<p>heritability we know it’s there we know it’s in those snips we just don’t know which snips it is because we don’t have</p>
<p>enough sample size to do an infinitely power G loss to figure out which which you know which snips are which are all</p>
<p>the causal snips in the genome but we know that heritability is there it’s just kind of hiding and then on the other hand there’s heritability that’s</p>
<p>still missing which is this gap between 0.45 and 0.8 although as I said a moment ago maybe it’s really a gap now between</p>
<p>0.45 and 0.6 if we believe point 6 for height now and that’s heritability that’s still missing and then we don’t</p>
<p>have an X we don’t currently have an explanation for so this is just a</p>
<p>generalization published by yang at all 2011</p>
<p>to some other quantitative traits and I’m not going to go over this in detail other than to say that I mean of course</p>
<p>these traits are less Harold other than less heritable height but qualitatively the story is pretty similar to what’s</p>
<p>going on with height we still have one question in the Q&amp;A that I think it’s maybe timely at this</p>
<p>moment which is from Matthew Worman who says hey because if one were to use a p-value of 10 to the minus 5 and instead</p>
<p>of 10 to minus 8 for D us what fraction of snips exceeding this less stringent threshold are false positives okay so I</p>
<p>do think I do think it is of interest to say hey what happens if I choose some other threshold let’s say 10 to the</p>
<p>minus 5 and I could define a set of snips which are snips that come in at P</p>
<p>less than 10 to the minus 5 in my G wasps and I guess I would like to say</p>
<p>three different things about that well the first thing that I want to say is that just as is the case with the set of</p>
<p>genome wisely he would snips at five times ten to the minus eight the nature and characteristics of the snips that</p>
<p>would come in at ten to the minus five is very much a function of your sample size I mean it’s a function of a lot of things like the genetic architecture of</p>
<p>the trade and so and so forth but it’s really very much a function of your sample size so that’s sort of like very dependent on a particular study</p>
<p>you know the snips that come in at P less than ten to the minus five killer set of characteristics that’s the first</p>
<p>thing that I wanted to say the second thing that I wanted to say is that although it might depend on that the</p>
<p>particular study in a particular genetic architecture and and so on and so forth that the sample size I think intuitively</p>
<p>it’s appropriate for us to would think intuitively that most of the snips that come in you know at P less than ten to</p>
<p>the minus five most of those are going to be false positives in the sense that they’re not truly causal variants or they’re not truly tagging causal</p>
<p>variants that is that in general that is what I suggest is likely to be the case and the third thing that I want to say</p>
<p>is that even if most of them are false positives a very important fraction of</p>
<p>them will be true positives and those true positives will be contributing more heritability and so if I if I define</p>
<p>something called H square G was 10 H squared G was less than 10 to the minus 5 which is the heritability explained by</p>
<p>a particular set of snips that comes in at P less than ten to the minus five in a particular G wasps then that will lie</p>
<p>somewhere in between h4g wasps which is just the john of genome I seen big ones and h4g and it would probably be</p>
<p>substantially greater than each square G loss because even though I would I’m going to say that in general make you’d</p>
<p>expect that most of those coming in at ten less than ten of - I would be false positives a very very important fraction</p>
<p>of them would be true positives and that would get you a lot closer to H square G the true snip irritability although I’m</p>
<p>still going to say that in general you’d expect that there’s still a lot more signal that is not even captured by P</p>
<p>less than ten to the minus five and you’d still have a gap between h 4g whilst 10 to minus 5 and h 4g so h 4g</p>
<p>west 10 to the minus 5 might lie somewhere in the middle between h 4g west and h 4g with the details that</p>
<p>depending on how polygenic the architecture is and what the sample size is and so on and so forth all right very</p>
<p>good and so so generally speaking H for G is less than H squared so it’s an</p>
<p>imperative builds us the narrow-sense heritability that could be due to rare and low frequency variants as well as other types of variants like copy number</p>
<p>variants that are not captured by the snip heritability of a very specific set of snips usually either genotype common</p>
<p>snips or genotype and imputed snips that are mostly common or something like that that you’re estimating this an impaired</p>
<p>ability of and then on the other hand we have a completely different phenomenon each board gos</p>
<p>less than H squared G so that’s each word us less than snip heritability where we need larger G while sample</p>
<p>sizes or as Matt alluded to maybe we need just a less less stringent genome-wide its significance threshold</p>
<p>or whatever I mean H for G is just sort of in the limit of you know any snip with P less than or equal to 10 to the 0</p>
<p>and your G loss right and and so in the limit of large sample sizes we might</p>
<p>hope that as our jaws gets hugely large then we’ll identify all the you know</p>
<p>associated snips and each for gos will approach H squared G but there is a bit of a caveat here that even if you’ve got</p>
<p>a humongous sample size like 758 thousand samples for blood pressure if it’s a really polygenic trait like blood</p>
<p>pressure 758 thousand samples is not enough and it’s not even remotely close to being enough and each board gos</p>
<p>point zero six may still be a lot lower than this nip heritability of point two one all right so I’ve been I think I’ve</p>
<p>been saying repeatedly that snip heritability is a function of the set of snips and that’s definitely true but we</p>
<p>should also keep in mind that it’s a function of the particular population that we’re looking at and again as I</p>
<p>mentioned earlier one population that some people sometimes study is British</p>
<p>ancestry individuals from UK biobank which is mentioned on this slide and another population that people sometimes</p>
<p>study is European Americans and that happens to be the population that was analyzed in reference to on this slide</p>
<p>and it turns out that if you estimate snip heritability in British ancestry UK biobank samples you consistently get</p>
<p>higher numbers than you do if you estimate heritability in other cohorts such as European American cohorts and</p>
<p>there’s different possible explanations for this but probably the most likely explanation is that snip heritability is</p>
<p>truly higher in a British ancestor UK biobank population than it is in a</p>
<p>European American population and that might be due to just less environmental variance in a set of British ancestry</p>
<p>individuals residing in the UK or more precisely less environmental variance in</p>
<p>a set of British ancestry individuals residing in the UK that the UK biobank captures which is actually it’s not a</p>
<p>perfectly random subset of of British individuals it’s British individuals who choose to respond to surveys and might</p>
<p>have higher than average SES and so on and so on and so forth and there seems to be less environmental noise and thus</p>
<p>larger snip heritability in that population than in for example a general</p>
<p>European American population so let’s just keep in mind that’s an impaired ability depends on the population studied as well as on the set of snips</p>
<p>and of course I should say that total narrow-sense heritability also depends on the particular population that you’re</p>
<p>studying so one question that comes up a lot is</p>
<p>which assumptions do we need for estimates of snip heritability to be valid and one thing that the world now</p>
<p>understands very well is that le dependent architectures can lead to bias and estimates and because I see that I’m</p>
<p>running out of time I’m going to choose to just gloss over the details of that I want to say very carefully what I mean</p>
<p>by LD dependent architectures le dependent architectures we’re not talking about tagging here of course we</p>
<p>know that you know a system that’s not a causal snip can tag a different snip that is a causal snip but LD dependent</p>
<p>architectures means it means that causal effects vary with the amount of LD s</p>
<p>snip has and this could be just due to minor allele frequency that’s kind of trivial that rarer snips you know</p>
<p>explain less heritability percent because they’re rare and they also have less LD that’s kind of trivial but even</p>
<p>if you condition on math even in a specific fixed math we know that low LD</p>
<p>snips actually have larger causal effect sizes than hi LD snips and the reasons</p>
<p>for this are quite complicated you know we believe they have something to do with negative selection but it turns out</p>
<p>that that that violates some of the assumption of some of these estimation methods and it can lead to biased in</p>
<p>estimates and it’s something that we have to watch out for and it’s something that the world now understands well and</p>
<p>knows that we have to watch out for and I’m just going to leave it at that and gloss over some of these other slides</p>
<p>and then a second question that comes up which is the second bullet fifth point now on this slide over here is the</p>
<p>question is it okay if effect sizes have a non infinitesimal distribution and so I haven’t really emphasized this point</p>
<p>but the sort of the the estimation procedures that I’ve described that were you know computer genetic relationship</p>
<p>matrix from unrelated individuals and then use restricted maximum likelihood the fit variance components to estimates</p>
<p>and impaired ability there’s kind of some assumptions underlying their underlying that about an infinitesimal</p>
<p>architecture where infinitesimal as I defined earlier refers to a genetic architecture in which all snips are</p>
<p>causal with a normal or Gaussian distributed distribution of causal</p>
<p>effects and the methods assume that but we now know that even though that assumption is clearly an incorrect assumption that</p>
<p>does not lead to any bias it’s merely the case that you might be leaving some precision on the table we’re by methods</p>
<p>that are a little bit more sophisticated in accounting for the fact that that through the distribution of causal</p>
<p>effects is a little bit more sparse than that can actually produce more precise estimates but that’s an issue of precision of getting a lower standard</p>
<p>error more precise estimate it’s not an issue of bias and this does not lead to bias and that is well understood all</p>
<p>right um I do want to mention at least briefly the important point that whereas</p>
<p>I’ve been talking about the math in terms of quantitative traits like height people are very interested in case</p>
<p>control traits because most of the disease traits that were really ultimately interested in terms of</p>
<p>medical action ability and drug targets and treatment or whatever it is we’re aiming for you know even polygenic</p>
<p>prediction we’re generally aiming for it in the context of medically important disease traits like schizophrenia or type 2</p>
<p>diabetes or whatever disease you like to study and we need a little bit more math to get this to work right and the type</p>
<p>of math that is most commonly used is called the liability threshold model and again I’m going to try to avoid going</p>
<p>over the mathematical symbols here but I want to provide the intuition because this is so important that the liability</p>
<p>threshold model models that there’s some unobserved underlying continuous number</p>
<p>called the liability and if the liability is above some number you have the disease and fortunately we have an</p>
<p>example which is type 2 diabetes which provides some intuition here where maybe you’re doing a G loss of type 2 diabetes</p>
<p>and all you know is that somebody has told you whether or not they have type 2 diabetes or whether or not they’ve been</p>
<p>told by a doctor that they have type 2 diabetes but there’s an underlying continuous quantity which we call the</p>
<p>liability which is your fasting blood glucose level which is one of the the diagnostic criteria for type 2 diabetes</p>
<p>and we could think of it as you know we don’t get to see the liability but if you’re if you’re fasting blood glucose</p>
<p>is above some level then you are you are typed you have type 2 diabetes and so what we want to do is we want to do some</p>
<p>math that operates on this unobserved continuous valued liability and that’s</p>
<p>what liability threshold modeling is about and I’m not going to go into the details but all the work that’s been</p>
<p>done in this space on snip heritability has relied on the liability threshold model and contingent on using the</p>
<p>liability threshold model it has generally produced a qualitatively similar story to the story that I</p>
<p>communicated earlier for height and I’m not going to go into those results in detail I do want to mention briefly that</p>
<p>I’ve been talking mostly about methods that use individual level genotypes there’s been a lot of interest dating</p>
<p>back to the year 2012 and methods that instead only requires summary Association statistics as input and</p>
<p>there are methods now that they do make that do input summary Association statistics to learn about various</p>
<p>flavors of heritability and snip heritability and I just want to provide one point of intuition that this is</p>
<p>related to the fact an important observation made by yang at all 2011 that the average chi-square statistic in</p>
<p>a Jie wasps is very closely related to snip heritability you should not be surprised if your average chi-square</p>
<p>statistic and in G wasps is above 1 that does not imply that you have confounding but rather that is directly related to</p>
<p>the amount of polygenic signal in that G wasps and an increasingly sophisticated set of methods can use that observation</p>
<p>to estimate heritability from summary statistics and I’m going to gloss over all of the details of that so</p>
<p>conclusions we can use family data such as twin studies to estimate the</p>
<p>narrow-sense heritability of a trait for even now in the year 2020 most G wasps</p>
<p>have the property that the genome-wide significant low-side that they discover do not explain all the heritability we</p>
<p>have missing heritability and there’s this concept of snip heritability that can really close that gap so these are</p>
<p>some of the bonus topics really within the heritability space that I think are super interesting and that I have some</p>
<p>bonus slides on and that as expected I’m not going to have the time to delve in to in detail but you can check out those</p>
<p>bonus slides if you want to as you know the slides were sent out to the whole group and I’d like to acknowledge all the members of my group who have figured</p>
<p>all this stuff out and explained it to me and I’d like to put in a brief advertisement for my course advance</p>
<p>population medical genetics which would be offered in spring 2001 and with that I’ll take any additional questions that</p>
<p>people may have thank you very much ok as there are no questions in the Q&amp;A box at the moment</p>
<p>please do post to the whole group please do posting a few if you have further questions it was a question that was</p>
<p>just asked oh great um it’s from Matthew Orman who’s commenting thanks Elka I really appreciate the clarity your</p>
<p>presentation so I know the second to that and thank you and I do have one question on which is I’m curious about</p>
<p>the with the identity by state definition the concept of unrelated individuals does this require that the</p>
<p>variant shared by the two individuals be of independent origin or just okay so</p>
<p>thank you for raising that point and and unrelated individuals that this is another term just like the term</p>
<p>population where it’s a little bit tricky to define it and if you define it strictly then it’s probably nothing like</p>
<p>what you have in real life so if I want it to be real strict about it I could say unrelated individuals is like let’s</p>
<p>think I have a simulation and in my simulation I’ve got some minor allele frequencies and I’m you know or</p>
<p>haplotype frequencies and I’m you know generating the the genome of individual number one and then completely</p>
<p>independently and generating the genome of individual number two and the whole process is completely independent but in</p>
<p>real life of course it’s not like that you know you take any two people who think they’re unrelated to each other no</p>
<p>you could probably go back you know five or ten generations and find some common ancestor that’s usually the way that it</p>
<p>works and so the the detailed answer to that question is if you’ve got cryptic relatedness you’ve got this set of</p>
<p>individuals from a G wash they’re supposed to be unrelated to each other you don’t think there’s any siblings in there but there’s probably some cryptic</p>
<p>relatedness that I mean I believe that yang at all 2010 Nature Genetics I came</p>
<p>up with some threshold that I think might have been 0.05 or maybe they later they changed it to 0.025 or something</p>
<p>like that and as long as he as long as you apply this threshold where you stringently check that there are no two</p>
<p>individuals in the data set who are related at a level level greater than 0.05 or greater than 0.025 then it is</p>
<p>hypothesized that the met that the method is sort of robust to that and then even though the people are not strictly unrelated you should be okay</p>
<p>now on the other hand what if you have a data set that consists of like related and unrelated individuals and you don’t</p>
<p>you want to throw away half of them because then you’d be zapping your power then what do you do well it turns out you can fit two variants components to</p>
<p>jointly estimate narrow-sense heritability and impaired ability I’m not going to go over the details there’s a reference date at all 2013 plus</p>
<p>genetics from our group so I think that I mean that in summary the shortest answer to your question is as long as</p>
<p>you make sure that there you know that any particularly related people are only a little little bit related then you’re</p>
<p>kind of okay and everything that I said still kind of holds thank you very much there were three more questions posted</p>
<p>but I’m afraid we’re out of time because MGP talk will be starting shortly so I think we’ll keep those questions in</p>
<p>reserve and share them with you all because if you if you wish all are welcome to email me offline a price at HS pH don’t harbor any deer great thank</p>
<p><strong>you so much for your talk and thanks to the audience for joining in today so thanks okay</strong></p>



</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
          // target, if specified
          link.setAttribute("target", "_blank");
          // default icon
          link.classList.add("external");
      }
    }
});
</script>
</div> <!-- /content -->



</body></html>