<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.353">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>PGC Video Textbook - Chapter 9.6: Therapeutic Implications</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


<link rel="stylesheet" href="styles.css">
</head>

<body class="nav-sidebar docked nav-fixed">


<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="./index.html">
    <span class="navbar-title">PGC Video Textbook</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link active" href="./index.html" rel="" target="" aria-current="page">
 <span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./about.html" rel="" target="">
 <span class="menu-text">About</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./contact.html" rel="" target="">
 <span class="menu-text">Contact us</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="https://pgc.unc.edu/" rel="" target="">
 <span class="menu-text">PGC Website</span></a>
  </li>  
</ul>
            <div class="quarto-navbar-tools ms-auto">
</div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
      <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item">Chapter 9.6: Therapeutic Implications</li></ol></nav>
      <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
      </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal sidebar-navigation docked overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
      <a href="./index.html" class="sidebar-logo-link">
      <img src="./pgc_logo_website_v3.jpeg" alt="" class="sidebar-logo py-0 d-lg-inline d-none">
      </a>
      <div class="sidebar-tools-main">
    <div class="dropdown">
      <a href="" title="" id="quarto-navigation-tool-dropdown-0" class="quarto-navigation-tool dropdown-toggle px-1" data-bs-toggle="dropdown" aria-expanded="false" aria-label=""><i class="bi bi-github"></i></a>
      <ul class="dropdown-menu" aria-labelledby="quarto-navigation-tool-dropdown-0">
          <li>
            <a class="dropdown-item sidebar-tools-main-item" href="">
            Source code
            </a>
          </li>
          <li>
            <a class="dropdown-item sidebar-tools-main-item" href="">
            Report a bug
            </a>
          </li>
      </ul>
    </div>
</div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./welcome.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Welcome to the PGC Video Textbook!</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./toc.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Table of Contents</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">
 <span class="menu-text">Chapters</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter1.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 1: Introduction</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter2.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 2: The Genome</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter3.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 3: Technologies</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter4.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 4: Study designs</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter5.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 5: GWAS analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter6.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 6: Polygenic Scores</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter7.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 7: Ancestry-Specific Analyses and Considerations</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter8.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 8: Post-GWAS bioinformatics</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter9.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 9: Advanced Topics</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter10.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 10: Other Considerations</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true">
 <span class="menu-text">Software Tutorials</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_cnvs.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">CNVs</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_conditional.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Conditional Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_datasets.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Datasets</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_ewas.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">EWAS</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_geneset.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Gene Set Identification</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_gwas.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">GWAS</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_genomicSEM.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Genomic SEM</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_MR.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Mendelian Randomization</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_mtag.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">MTAG</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_prs.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">PRS</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_correlation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">SNP Heritability and Genetic Correlation</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true">
 <span class="menu-text">Additional Resources</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./glossary.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Glossary</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./resources.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Tools/Software Resources</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="https://pgc.unc.edu/for-researchers/download-results/" class="sidebar-item-text sidebar-link">
 <span class="menu-text">PGC Summary Statistics</span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Chapter 9.6: Therapeutic Implications</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<p><strong>Title</strong>: Pharmacogenomics knowledge for personalized medicine</p>
<p><strong>Presenter(s)</strong>:</p>
<ul>
<li><p>Michelle Whirl-Carrillo, PhD (Stanford University School of Medicine)</p></li>
<li><p>Cristina Rodriguez-Antona, PhD (CNIO, Spanish National Cancer Research Centre)</p></li>
</ul>
<p><strong>Cristina Rodriguez-Antona [Host]:</strong></p>
<p>Director of the Pharmacogenomics Knowledge Base from PharmGKB at Stanford University. She leads the PharmGKB team and is responsible for the development of new content, projects, and futures. For the last 20 years, she has led pharmacogenomics research and its application to personalized medicine and personal genomics. She has leadership roles in multiple NIH-funded pharmacogenomic projects and, in addition, formed PharmGKB, including the Clinical Pharmacogenetics Implementation Consortium and the Pharmacodynamics Clinical Annotation Tool. She has served on multiple national and international pharmacogenomic working groups and steering committees. So, her research interests include the translation of human genome sequencing data to clinical implementation using curated pharmacogenomics knowledge. Michelle, thank you.</p>
<p><strong>Michelle Whirl-Carrillo:</strong></p>
<p>Thank you, thank you so much for the introduction, and thank you for the opportunity to speak with you all today. I really appreciate it, and I’m well aware that I’m standing between you and dinner for most people, so I’ll try to move it along.</p>
<p>Oh, just a disclosure really fast, I don’t really have anything to disclose other than my public funding.</p>
<p>So we had an excellent overview of pharmacogenomics from the previous speaker, Munir did a great job explaining it, so I’m just going to start with talking about how the amount of pharmacogenomic information that we’ve had over the past 20 years has really gone up. This is almost a very similar diagram to what Munir showed, where I just started in the year 2000, there were only about 387 publications for pharmacogenomics that were found in PubMed for that year, versus in 2021, in that year alone, we’ve got over 2,600 publications. So we are accumulating knowledge all the time. There are over 35,000 publications in total in PubMed right now about pharmacogenomics and genetics.</p>
<p>So it’s great to have all this information. More and more information means that we are more informed, but just the information alone can be tricky. What do we do with that? So, this raises some issues about how do we organize all this information and how can we standardize it across all these different publications? People use different terminologies or are measuring different things, and it’s not standardized currently. It also leads to questions about how can we use this information for clinical actionability. So, we really need centralized resources to help us be able to organize this information, standardize it, so we can easily search across it. So, I’m going to talk to you today about a few resources that are available that help centralize this information.</p>
<p>But first, I’m going to talk about a few sources of pharmacogenomic knowledge that we can accumulate together in order to get a better picture of the field. One is the peer-reviewed published studies. I already showed you what’s in PubMed right now – thousands and thousands of studies. We also know that there is some information in regulatory agency-approved drug labels. You heard a little bit about that from Munir as well, that some labels from the EMA, for example, from FDA, do have information on them about testing for particular genetic variants or having a metabolizer status and how you can choose a drug or change the dosage accordingly. But that’s very few drug labels that have that information on them currently, although it is getting more and more all the time.</p>
<p>We also have published guidelines, so there are different groups that write actual clinical guidelines for clinicians on how to go from the genotype to dosing the patient. Most of those guidelines are derived from the peer-reviewed published literature, though, so they’re kind of a derivative of that data source. We also know there’s a lot of unpublished data out there, right? There are clinical trials and other experiments that may be proprietary from pharmaceutical companies. That information is more difficult to use and to standardize and also to vet as a public resource. It’s hard to deal with that information, but we know that a lot of that information is used in those regulatory agency-approved drug labels as well. A lot of that is submitted to these agencies when the drugs are going for approval.</p>
<p>So, one of the centralized sources for dealing with organizing this type of information that I just discussed is PharmGKB, which is the Pharmacogenomics Knowledge Base. This is a website with a database backend. It’s publicly available. It’s probably the biggest resource that’s publicly available in the world right now. And we started this around the year 2000. It’s based at Stanford University, but our mission is basically to collect, encode, and disseminate pharmacogenomic knowledge for uses ranging from research and discovery all the way through clinical actionability and implementation.</p>
<p>So, what do we do at PharmGKB? We take the information that is out there that I kind of already went over: guidelines, drug labels, what’s published in the literature, and we think of this as our knowledge information stream. This is all publicly available right now for everyone. But at PharmGKB, we have a team of scientists that extract information from these sources, and using expert manual curation, we take the relevant parts of what drug, what genotype, what gene, etc., what variants, we standardize the terminologies, and we aggregate this information together. We’re always working towards clinical implementation and adoption, and in the end, we present some resources that I’m going to show you in a minute to actually go back out into this knowledge stream, and we hope enrich it and make it a better source of information for people who are trying to implement pharmacogenomics in their clinics.</p>
<p>So, one source of information at PharmGKB are these guidelines that I mentioned earlier. The two groups that probably publish the most information about clinical implementation of pharmacogenomics would be CPIC and the Royal Dutch Pharmacists Association’s Dutch Pharmacogenetics Working Group (KNMP) out of the Netherlands.</p>
<p>So, PharmGKB, this is just an example webpage. I don’t expect you to read it. That’s just an example. But, for example, the Dutch working group has some guidelines about the use of amitriptyline and how CYP2D6 variants can affect the implementation or affect dosage of this particular drug. The Dutch working group puts out a PDF that is text-based, that people can read and get this information themselves. But what we tried to do at PharmGKB is make this a little bit simpler presentation through tables and by using all the information available from the Dutch group itself.</p>
<p>We have on the web page a way for people to input what the particular variants are of a specific patient or an example. And pull up what the recommendation is for that specific genotype. So just a little bit of an aid for people to be able to take the information from the Dutch group and get right to the source without having to understand the mapping from the actual genotype in the gene to the metabolizer phenotype, such as poor metabolizer, etc., to the actual recommendation.</p>
<p>So, we have on our website just the broken-down information from multiple groups, including CPIC, the Dutch group, and there are a few other groups that have some one-off recommendations as well. You can compare across these, and due to resource issues and so forth, not all the same drugs have been have guidelines written about them from both organizations. And so, sometimes one organization may have something about a drug that the other one doesn’t. Most of the time, the recommendations agree or at least are very similar, but sometimes there’s a little bit of differences, and you can compare those at the PharmGKB to see what the differences or similarities are. We also highlight where guidelines tell you whether to give you testing guidance – what to test and when to test.</p>
<p>So, another source of information that we organize would be drug labels. We annotate drug labels from FDA but also from EMA, Canada, and we’ve had a couple of collaborations with groups in both Japan and Switzerland. So, we have some labels that have been annotated from those groups as well, and we’re always keen to collaborate with others. So, if anybody is interested in working with us on that, please let me know. But yes, our biggest group of drug labels would be from FDA, and then followed by EMA.</p>
<p>So, we saw an example of a drug label earlier from Munir, and so sometimes on drug labels, they do highlight that there is information, but much of the time, the language on drug labels can be not precise or somewhat vague. And so, we had feedback from many of our users that what they really wanted to know from these drug labels is, “Just tell me which labels say I need to test, which one is recommended testing, and which ones have any kind of information that I can use.” So, we came up with this labeling system where we can tag these labels that we curate with those categories. And also we get very specific if there is prescribing information based on pharmacogenetics on a drug label, such as changing the dose or changing the drug, we highlight that as well.</p>
<p>This is just an example screenshot (I don’t expect you to read it) of annotation of the EMA label for abacavir and HLA-B. You heard a little bit about this earlier. So yes, this is a very kind of famous example of pharmacogenomics, very well understood. And so yes, testing is required according to the label. The label gives you specifics about what variant it is, specifically in HLA-B, that you should be interested in, and gives you advice about what to do. This is an example of a nicely laid out drug label, and we were able to capture this information very easily. But not all drug labels are quite so simple.</p>
<p>So, again we have a page at PharmGKB. Please go ahead and check it out for yourself at some point if you are able and inclined. We have drug labels from FDA, EMA, and a few other groups as well, and you can use this table to compare across, and you can drill down to see that exact annotation like I just showed you for the EMA one in abacavir. But what’s interesting is that some regulatory agencies have different information on their labels, depending on either they may be lacking information altogether about pharmacogenomics on that label, whereas other countries or groups have a very high level of actionability for pharmacogenomics. So, it’s kind of interesting to look at this webpage to see overall, again, where the similarities and differences are, and it just has to do with what is submitted to these regulatory agencies and what is deemed actionable at the time.</p>
<p>We also heard from Munir a little bit about randomized controlled trials. Again, yes, this is considered the gold standard, and a lot of clinicians prefer to have information that comes from RCTs, and this is where they would like to see any kind of proof of clinical actionability. But as was pointed out, it’s really not feasible. A lot of times, we have small populations in these studies. Again, when you mentioned that we don’t have huge cohorts like in diabetes, etc. So, the small populations do often make statistical significance difficult. But we do know that statistical significance is not always just the same as clinical significance anyway. And there is a lack of standardization again across all of these studies and publications that come out. But we can use replication of data from multiple different sources to help us address some of these issues, right?</p>
<p>So, we can take an article like this (this is from the New England Journal of Medicine) and is talking about a particular association between a variant and SLCO1B1 and a reaction to Simvastatin. And if we take this and manually curate it at PharmGKB, again we’re standardizing across all the publications, the gene names, the gene variants. There are many different names for genetic variants, as I’m sure you’re all aware, right? People can use RSIDs, they can use cDNA changes, protein, amino acid changes, etc. So, we can standardize so that we can see across all publications which ones are talking about the same variant. We can standardize the drug terms, etc. And we take a bunch of information out about you know statistics and what type of study and the study size. We can collect the information from each article.</p>
<p>And over time, there are more and more articles, right, about a particular association that can be published. Many times, these articles replicate the original findings, but sometimes they do not. So, sometimes an association is published, and the next paper that comes out might refute actually that association. But if you look at a large enough cohort of papers and like I said, you’re able to compare across because of the standardization, we can start seeing the bigger picture and write summaries about the association based on the corpus of evidence. And then we can also take the information from those drug labels and the guidelines that I referenced earlier that are published and annotated at PharmGKB as well, and we can get an even bigger and better picture of what’s going on with a particular association. Once we have all this evidence together, we can not only write a summary about what’s going on, but we can assign a level of evidence about how confident we are and the strength of this particular association.</p>
<p>So, this is just a screenshot again for a particular association with the same variant with simvastatin, and down here, we can see that there are two dosing guideline annotations available in PharmGKB – one from CPIC and one was from the Dutch group. And you know, in this case, we only have nine publications, but there can actually be many, many more publications depending on the phenotype that’s studied.</p>
<p>So, based on that grouping of information, we can come up with different levels of evidence. You can read more about the system that we have come up with for standardizing when we’re trying to collect evidence together or visit the webpage. I can’t go into a lot of detail here because it would take a while to explain. But basically, we score the information that we take from every single publication. We take that information and we add it together across all the different publications, taking into account if any regulatory labels are available and the information on those labels, and also guidelines if their information is available. We come up with a total score for that particular summary annotation. Then, based on that score for the summary annotation, we can figure out which cutoff for a level that it meets and assign accordingly. So, for all of these summaries, you can go, if you’re on a PharmGKB page, you can get a feeling for how much support there is for a particular association based on the little labels on the left-hand side that are color-coded and also have the level written in them. In most cases, as a level one or instances where we know of a guideline or a label that really confirms basically actionability or clinical implementation can be instituted for a particular association. Level two would be those that are very close but maybe don’t have a guideline or a label yet, but there’s a lot of information published and the association looks very strong. Some people, on the other hand, are very interested in low-level associations if they’re not looking to implement clinically but they’re looking at it from a research perspective.</p>
<p>Okay, so I’m going to switch gears for just a couple of minutes to tell you a little bit more about CPIC. Again, this is one of the groups that writes guidelines for how to implement pharmacogenomics in the clinic. The Dutch group also writes guidelines as well. I happen to be involved in CPIC. I’m going to give you a little bit of the background of that particular group. The goal of CPIC is not to tell people whether or not to test or even what to test, per se, but if you are a clinician and you have genetic results in hand, we want you to be able to understand and quickly be able to know what you can do with that information. Preemptive genotyping is becoming more widespread. There are direct-to-consumer genotyping companies out there, and some patients are going to their doctors, at least in the states, with information about their genetic variants, and they want their doctors to be able to understand that information and act on it. But most clinicians, as we heard, they don’t have the time for patients to sit and really research for themselves what a particular genetic variation might mean or might imply for a particular drug prescription. So, they need a facile way to access information that can tell them upfront what it is that they should do.</p>
<p>So, these guidelines are written by expert groups put together. These groups include clinicians, gene experts, pharmacists, and, in some cases, just research scientists as well. And again, we start with a PubMed literature review. So, these guidelines are based on PubMed literature reviews. But the information is collated together and graded for every single outcome by the clinical authors, the authors of the guideline. And so, a consensus is reached from that particular group, and then a statement about what you could do with a particular genetic test result.</p>
<p>So, as part of this process, we have to understand for every genetic allele what the functional implication is. Somebody had mentioned that before. So yes, we do have to care about what the function is for all of these variants. And that’s part of the process for creating CPIC guidelines – to basically do a deep dive literature review of every allele for the particular gene in that guideline and see if we can come to an agreement about what the function of that allele is.</p>
<p>Then, once we have a function defined for an allele, we can put the two alleles together and map to a phenotype, such as a poor metabolizer or an ultra-rapid metabolizer that you heard about in the previous talk. Once we have the metabolizer status for a given genotype for a given patient, then we can come up with what the therapeutic recommendation would be for that particular phenotype.</p>
<p>CPIC also makes available clinical decision support flow charts and wording for CDS alerts to help aid with clinical implementation for groups that don’t have the resources to do that as well. So, these are just example charts and CDS language that people could use if they want to implement in their institution.</p>
<p>There’s a bunch of CPIC guidelines right now. The last count was 26, but that might have changed really recently because we’re always publishing. But please go and check out the website for yourself. We also have a database and API for people that are interested to access the information computationally and import it into their own databases.</p>
<p>So, another challenge with implementing pharmacogenomics would be our nomenclature. You heard a little bit about the star nomenclature already. I get people ask me all the time who aren’t in the pharmacogenomics field but more in clinical genomics, “What’s the deal with the star alleles? Why do you guys have them, and what do they mean?” So, just really briefly, right? We have the gene symbol and that star with the number after it is just an allele designation. And so, for example, this CYP2D6 star 8. This is HGVS representation for it. So, there are, you know, three different variants across the gene that define this allele. In pharmacogenomics, with most of the cytochrome P450s, we’re interested in what the combination of variants are across the entire allele. It’s not necessarily one particular variant or another, but what’s the combination of variants across the entire gene? It’s the haplotype that matters. And so, to define the haplotypes, the star allele shorthand was born. And these variants, these are SNPs, so that’s quite simple. But in many cases, the variation across the gene can include repeats or indels, there can be structural variants and copy number variations as well. The variation occurs not just in the exons but also introns or upstream or downstream of the genes, sometimes. And they all have functional effects and are very important to capture and understand what the variation is across. And so, this shorthand helps the community understand what the variation is.</p>
<p>I’ve been asked many times, “Well, can the star alleles just go away?” from people who are clinicians but deal with Mendelian diseases many times or other kinds of genetics. Unfortunately, I don’t think that’s going to happen. It’s used throughout the literature, lab tests, and assays, test reports all refer to the star allele nomenclature right now. And it’s used by many of the prescribing guidelines. And this started in the 1990s, and it was organized to a great extent in the early 2000s by the Human Cytochrome P450 Allele Nomenclature Database in Sweden. And they, I think, were the first group to really start tracking these alleles and defining them as what variation is in each allele, and they’re the ones that started naming them with the star alleles. That transitioned in 2017 to a group called PharmVar.</p>
<p>The PharmVar group has taken over the reins there. So, PharmVar, please check out this group as well. So when we’re talking about standardization, PharmVar is critical to standardization for these pharmacogenetic alleles, and it is the central repository. People who find new allelic variation can submit that variation to the PharmVar group, and there’s a very rigorous process and rules that must be followed for a new star allele to be assigned. But if that’s the case, if the submission reaches the level of a new star allele, it will be assigned.</p>
<p>Just to show you a little bit about this – this is not the most complicated example by any means, but just to show you a little bit what I’m talking about – so this is again a CYP2D6 illustration. We’re looking at the star alleles star 10, 36, 37, etc., and then this is a list of variants down the side. And so, what we see here is the 100C&gt;T variant is a part of all of these different alleles. So is this variation down here. So, if you, for example, were just going to test this 100C&gt;T, you might not really know which star allele you’re talking about. For simplicity’s sake, many times people just call it a star 10 because okay, that’s the defining variant in that particular star allele. But you can see that there are many other star alleles that also contain that variant and then other variants as well. So, it can get much more complicated than this as well, and some genes, such as CYP2D6, have structural variants, and it gets crazy. But these star alleles really are important for understanding what the variation is, and that all has functional implications. So, the functionality of these different star alleles can and do differ.</p>
<p>That leads to one of the other challenges of pharmacogenomics and implementation – the type of the test, what is tested, can affect your results, right? There have been studies done where they’ve sent the same samples out to different labs and gotten different results back, and a lot of times, that has to do with whether you’re talking about a panel or an exome or whole genome sequencing, as someone had brought up earlier. And it’s true that if you have a panel and not everything is on that panel that you can see in whole genome sequencing, you could potentially miss genetic variation that exists in that particular patient. And that’s why we have to be really careful with this. Often in the field, if you don’t see any variation, it is defaulted to what’s called a star 1 allele, which implies that you don’t have any genetic variation or wild type. But of course, depending on what’s tested, that may or may not be true. So, just something to keep in mind when people are trying to implement pharmacogenomics – that transparency about the tests and the test results can be very important. There can be situations where a patient is tested for pharmacogenomic alleles and none are found, and it goes into their EMR or their record that they have no genetic variation in a particular gene. But we know that that may not always be 100% true, right? For example, there are some known existing genetic variations where we know of the genetic change, but we don’t know what the function is. We don’t know what to do with that change. But over time, you know, research is going to eventually elucidate that for us. Also, there’s new variation out there that may not be covered, so, the current star allele nomenclature can’t cover everything that’s not known yet. So, as new discoveries happen, they’re submitted and we have new star alleles that may not have been tested for previously. So, at the very least,</p>
<p>[audio cuts out]</p>
<p>knowing what the results are can be very key to both clinicians and patients, especially going forward in the future. Another project that I’m involved in is called the Pharmacogenomics Clinical Annotation Tool. This is a software where the goal is to take the output from a genetic test result report – sorry, the output from a genetic test, such as a panel test or whole genome or exome sequencing – and come up with the star allele designations for those genes, which can be very complicated, as was mentioned earlier. So, this software aims to be able to take whole genome sequencing and determine that without using those tag alleles that are used on the SNP panel, and then connect the resulting genotypes with clinical guidance. And we’ve started with the clinical guidance from CPIC, but we’ll be expanding that to other groups as well, such as the Dutch guidelines and also what’s available on regulatory agency drug labels. That’s the goal.</p>
<p>So basically, the way this tool works is it takes a VCF file, which is just a way that genetic variation can be output from testing. And then we take those allele definitions from PharmVar, which has the definitions of what variants are in each star allele, and the information from the guidelines that can be accessed via the API in the database. And then we also have some messaging from the PharmCAT tool as well, with caveats and disclaimers, etc. We take all of that together. So, what the user supplies is the actual genetic variation that is detected, and we come up with a genotype summary, which tells you what the star alleles, what the diplotypes are or genotypes for that particular patient based on that VCF file, and give you a summary of the output. Then, the recommendations by drugs – so you may be a CYP2D6 star 4 star 8, for example. What does that mean? We have recommendations for each drug that we know can be affected or have guidelines showing that the drug response can be affected by CYP2D6. And then we would tell you exactly what the recommendations are for that particular genotype.</p>
<p>And then there’s a section of the report that is very key but not necessarily, you know, first and foremost in any clinician’s mind or patient’s, for that matter. But very importantly, we highlight what variation was given to us in the VCF file versus what is known to date. So, we can easily see in this section of the report if not every known genetic variant that could be part of a star allele, etc., was tested. In this way, we can see with an assigned genotype at the end if information was essentially missed – if there were known star alleles that were not tested for. In this case, you would know then if, say, a patient is a star 1 star 1 as the result, whether or not there are potentially variations that that patient is carrying in that gene that was not covered by that test.</p>
<p>It’s great to have this as a report. You get a report out of PharmCAT, and that’s a report that a patient could have and take to their doctor, or a clinician could look at. But in many cases, as we know, clinicians don’t have the time to sit and read reports either. So having this information go automatically into an EMR or EHR system is extremely important, and that is a big challenge since they are not standardized across all hospitals and hospital systems. And so, that’s another area that we’re working on for the future.</p>
<p>One of the final challenges I’m going to talk about today would be allele frequency, and this again was mentioned a little bit earlier. But the issue is that in many cases, in Mendelian disease, a low frequency is a sign of maybe pathogenicity, right? So, frequency can be used as a way to figure out if something might be very important in a particular gene. But unfortunately, in pharmacogenomics, some of these variations can be quite common, up to, you know, 30 percent or even more in certain populations. So, you cannot use frequency the same way in pharmacogenomics as a sign for what could be effective function. And yet, you know, knowing the frequency of the alleles is very helpful. To know, for example, if we can only test a handful of alleles, which ones should you be testing? Unfortunately, we don’t know the frequency for many of the defined alleles. They may be undiscovered in a particular patient, submitted to PharmVar, for example, gotten a star allele, now we know we can test for them. But if people haven’t tested for them previously, we don’t really know what the frequency in any given population is. So many times we think in pharmacogenetics, in the field, we think certain star alleles are rare because it’s not published much in the literature, we can’t find statistics about it. But that’s not necessarily the case. So, we do have to be careful there. Also many populations around the globe are understudied. So again, we want to make pharmacogenomics available for all diverse populations, but we haven’t done a lot of studies yet in many of the populations around the globe - lot of them have been focused on European populations or perhaps Asian populations as well. So, research is needed there to understand what the frequency is for these variants. But we always must remember that when we’re trying to implement this, the population frequency is never a proxy for what a patient actually has. So, testing the individual patient is really important.</p>
<p>So, the last challenge I’m going to talk about is just the separation of pharmacogenomics from the rest of clinical genomics. So, I gave you a little bit of an overview of all these four resources: PharmGKB, PharmCat, PharmVar, and CPIC. These resources talk to each other all the time, they trade information back and forth. In the US, we have very big projects for clinical genomics. ClinVar and ClinGen are huge resources that are funded through the NIH, but they’re kind of isolated from pharmacogenomics. I should say pharmacogenomics is isolated from them because they are very large and capture most of the genetic clinical genomics that are documented today. But pharmacogenomics kind of exists as a silo. We are trying to address that by depositing information from both CPIC and PharmGKB into both ClinVar and ClinGen. There are some issues with trying to take pharmacogenomic information and wedge it into a clinical genome database, and so it’s a little bit tricky at times, and the deposition of this information is slow. But in addition to depositing it, we are looking forward to being able to exchange information with ClinVar and ClinGen as well. So what we really want to do is not have pharmacogenomics be isolated and siloed from clinical genomics overall and genomic medicine.</p>
<p>And our vision, right, would be full integration of pharmacogenomics with genomic medicine. And this is just a diagram to try to illustrate that all those pharmacogenomic resources, and more, could, you know, eventually be combined together in some way to have a much easier interface, maybe one place that people can go, and that group could interact very closely with both ClinVar and ClinGen. And efforts are underway. We have a panel that we’re starting within the ClinGen group to address pharmacogenomics and how to better cross-talk between pharmacogenomics and genomic medicine in general.</p>
<p>So, in many cases, we don’t know the implications of pharmacogenomic discoveries for clinical implementation, but we definitely know that the clinical utility is there for a couple of dozen examples, at least. And then we know that integrating this together with the risk of genomic medicine is going to push implementation forward. Having to try to address this separately is slow going, as Munir described. It’s not being implemented across as many places as we’d like pharmacogenomics to be yet. It does differ from genomics disease models. The haplotypes and those star alleles, and the fact that you have to be aware of the diplotypes and map that to metabolizer phenotypes in many cases is very different, and we are aware of that. So, we do still need some specialized resources, but integrating would be the way to go in the future. And as of right now, we have a few centralized knowledge sources that people can use in the meantime.</p>
<p>And with that, I’d like to just thank all my colleagues. There are many different projects that I was talking about today, and that takes many people in many different institutions to help make all these projects a reality. So, I’d like to thank them and thank you for your attention, your time.</p>
<p>[Applause]</p>
<p><strong>Cristina</strong>: We have questions. I will start from the chat. One question regards SLCO1B1 and statins, and basically what they ask is, are there studies regarding drug adherence and pharmacogenetic testing or genotyping? I mean, does it influence adherence of the patients?</p>
<p><strong>Michelle</strong>: Adherence, yes. I’m not aware of specific studies about adherence to drugs, but we know colloquially, just anecdotally, that this is true, and that patients who have adverse effects that are not as severe as what we saw in the previous talk, but just myopathy, maybe, you know, pain or some other kind of nausea, etc., can lead patients to stop taking their medication. And so yes, it has been shown that because of the myopathy that’s associated with a particular genetic variation, there have been cases, like I said anecdotally, where patients just stop taking their medicine because of it. Yeah.</p>
<p><strong>Audience member</strong>: Hi Michelle, over here. Thank you for a great presentation. Fabulous resources, I think incredibly useful for the community. Can I just ask, are there plans or thoughts about translating those resources into different languages to make them more accessible?</p>
<p><strong>Michelle</strong>: That is an excellent question. We would love to do that, unfortunately, resource-wise, being government-funded, we don’t have the resources to pay for that, but if there are people that would like to collaborate with us, we could see what we could do. Yes, that’s a question we get a lot. You know, can this be in multiple languages? And we’re happy to do it, it’s just a resource issue right now.</p>
<p><strong>Cristina</strong>: Okay, another question from the chat is, despite the fact that pharmacogenomics data is growing rapidly and the importance of testing is clear, the integration in the clinics is very slow. So maybe you could comment on the barriers or the reasons for that.</p>
<p><strong>Michelle</strong>: Right, yes, I think a lot of the reasons for the slow uptake were just mentioned previously, just having the education, I think, of clinicians, for them to be familiar with this type of data, familiar with how to use it. Also, there are issues putting this kind of information in medical records, so getting this into the EHR system, I think, would be key. That’s one barrier we know to uptake. And then just also, you know, testing and reimbursement. What are you supposed to test for? When should you test? Who’s going to pay for it? These kinds of issues definitely affect implementation as well.</p>
<p><strong>Cristina</strong>: Last question.</p>
<p><strong>Audience member</strong>: Yeah, thank you as well for a great presentation. We, in our company, started testing the PharmCat, and I’d like to ask how many drugs are, in this moment, supported? Because we have only a limited number, and also if it can be adjusted by the users, the parameters, and what are you going to expand it?</p>
<p><strong>Michelle</strong>: Yeah, absolutely. So yes, if you check out pharmcat.org, we’ve updated a lot of the documentation, maybe since you last checked it out. All of the drugs that are covered by CPIC guidelines are currently covered by PharmCat and they’re in the report. All the genes, and yes, you can alter—well, first of all, it’s freely accessible, so if you want to download the tool and branch off, you can change the code however you like. But it is also customizable in terms of if you have genes that you want to include and you have the definition files for those, you can add those as you want, and same with the recommendations as well. And we are going to be expanding those within the next year, is the goal, to include drugs and recommendations from the Dutch Pharmacogenetic Working Group and also what information we can find on the FDA labels. We’ll probably start with FDA, but then ultimately can expand from there as well.</p>
<p><strong>Audience member:</strong> Okay, those are bioinformatics that I work in our company on, so I’m not completely familiar, but is it possible to speak to your IT staff?</p>
<p><strong>Michelle</strong>: Yeah, sure, sure. You know, send—I didn’t have the email up here, but if you send an email to just feedback@pharmgkb.org, we will make sure it gets to PharmCat. I think there’s also pharmcat@pharmgkb.org, but I can’t swear to that. So, if you send it to feedback@farmgkb.org and you say, “This is a question about PharmCat,” we can deal with that. We’re a very small team, and we are the same people working on many projects at once, but yeah, but yes, definitely.</p>
<p><strong>Audience member:</strong> Thank you. Thanks.</p>



</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
          // target, if specified
          link.setAttribute("target", "_blank");
          // default icon
          link.classList.add("external");
      }
    }
});
</script>
</div> <!-- /content -->



</body></html>