<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.353">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>PGC Video Textbook - Chapter 9.6: Therapeutic Implications</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


<link rel="stylesheet" href="styles.css">
</head>

<body class="nav-sidebar docked nav-fixed">


<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="./index.html">
    <span class="navbar-title">PGC Video Textbook</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link active" href="./index.html" rel="" target="" aria-current="page">
 <span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./about.html" rel="" target="">
 <span class="menu-text">About</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./contact.html" rel="" target="">
 <span class="menu-text">Contact us</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="https://pgc.unc.edu/" rel="" target="">
 <span class="menu-text">PGC Website</span></a>
  </li>  
</ul>
            <div class="quarto-navbar-tools ms-auto">
</div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
      <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item">Chapter 9.6: Therapeutic Implications</li></ol></nav>
      <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
      </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal sidebar-navigation docked overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
      <a href="./index.html" class="sidebar-logo-link">
      <img src="./pgc_logo_website_v3.jpeg" alt="" class="sidebar-logo py-0 d-lg-inline d-none">
      </a>
      <div class="sidebar-tools-main">
    <div class="dropdown">
      <a href="" title="" id="quarto-navigation-tool-dropdown-0" class="quarto-navigation-tool dropdown-toggle px-1" data-bs-toggle="dropdown" aria-expanded="false" aria-label=""><i class="bi bi-github"></i></a>
      <ul class="dropdown-menu" aria-labelledby="quarto-navigation-tool-dropdown-0">
          <li>
            <a class="dropdown-item sidebar-tools-main-item" href="">
            Source code
            </a>
          </li>
          <li>
            <a class="dropdown-item sidebar-tools-main-item" href="">
            Report a bug
            </a>
          </li>
      </ul>
    </div>
</div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./welcome.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Welcome to the PGC Video Textbook!</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./toc.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Table of Contents</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">
 <span class="menu-text">Chapters</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter1.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 1: Introduction</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter2.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 2: The Genome</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter3.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 3: Technologies</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter4.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 4: Study designs</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter5.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 5: GWAS analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter6.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 6: Polygenic Scores</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter7.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 7: Ancestry-Specific Analyses and Considerations</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter8.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 8: Post-GWAS bioinformatics</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter9.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 9: Advanced Topics</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter10.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 10: Other Considerations</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true">
 <span class="menu-text">Software Tutorials</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_cnvs.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">CNVs</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_conditional.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Conditional Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_crossdisorder.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Cross-disorder Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_ewas.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">EWAS</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_geneset.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Gene Set Identification</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_gwas.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">GWAS</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_genomicSEM.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Genomic SEM</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_imaging.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Imaging</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_MR.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Mendelian Randomization</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_prs.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">PRS</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_correlation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">SNP Heritability and Genetic Correlation</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true">
 <span class="menu-text">Additional Resources</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./glossary.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Glossary</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./resources.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Software Resources</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./addreading.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Additional Reading</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="https://pgc.unc.edu/for-researchers/download-results/" class="sidebar-item-text sidebar-link">
 <span class="menu-text">PGC Summary Statistics</span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Chapter 9.6: Therapeutic Implications</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<p><strong>Title</strong>: Pharmacogenomics knowledge for personalized medicine</p>
<p><strong>Presenter(s)</strong>: Michelle Whirl-Carrillo</p>
<p><strong>Michelle Whirl-Carrillo:</strong></p>
<p>director of the pharmacogenomics knowledge base from jkb at the Stanford University she leads the farm jkb team and is responsible for develop the development of new content projects and futures for the last 20 years she has led pharmacogenomics research on its application to personalize medicine and personal genomics she has leadership roles in multiple nih-funded pharmacogenomic projects and in addition from Farm jkb including the clinical pharmacogenetics implementation Consortium and the pharmacodynamics clinical annotation tool and has served on multiple national International pharmacogenomic working groups and steering committees so her research interests include the translation of human genome sequencing data to clinical implementation using curated pharmacogenomics knowledge so</p>
<p><strong>Cristina Rodriguez-Antona:</strong></p>
<p>Michelle thank you thank you thank you so much for the introduction and thank you for the opportunity to speak with you all</p>
<p>today I really appreciate it and I’m well aware that I’m standing between you</p>
<p>and dinner for most people so I’ll try to move it along</p>
<p>oh just a disclosure really fast I I don’t I don’t really have anything to disclose other than my um Public Funding</p>
<p>so we had an excellent overview of pharmacogenomics um from the previous speaker munir did a</p>
<p>great job explaining it so um just going to start with uh talking about how</p>
<p>the amount of pharmacogenomic information that we’ve had over the past 20 years has really gone up and this is</p>
<p>almost a a very similar diagram to what munir showed where I just started in the</p>
<p>year 2000 there were only about 387 Publications for pharmacogenomics that</p>
<p>were found in PubMed for that year versus in 2021 in that year alone we’ve</p>
<p>got over 2 600 Publications so we are accumulating knowledge all the time</p>
<p>there’s over 35 000 Publications in total in PubMed right now about pharmacogenomics and genetics so it’s</p>
<p>great to have all this information more and more information means that we are more informed but</p>
<p>just the information alone can be tricky what do we do with that so this raises some issues about how do we organize all</p>
<p>this information and how can we standardize it across all these different Publications people use</p>
<p>different terminologies or are measuring different things and it’s not standardized currently</p>
<p>it also leads to questions about how can we use this information for clinical</p>
<p>actionability so we really need centralized resources</p>
<p>to help us be able to organize this information standardize it so we can</p>
<p>easily search across it so I’m going to talk to you today about a few resources</p>
<p>that are available that help centralize this information but first i’m going to talk about a few</p>
<p>sources of pharmacogenomic knowledge that we can accumulate together in order to get a better picture of the field so</p>
<p>one is the peer review published studies I already showed you what’s in PubMed right now thousands and thousands of</p>
<p>studies we also know that there are some information and Regulatory agency</p>
<p>approved drug labels so you heard a little bit about that from munir as well that some labels from the EMA for</p>
<p>example from FDA do have information on them about testing for particular</p>
<p>genetic variants or having a metabolizer status and how you can choose a drug or</p>
<p>change the dosage accordingly but that’s a very few drug labels that have that</p>
<p>information on them currently although it is getting more and more all the time</p>
<p>we also have published guidelines so there are different groups that write</p>
<p>actually clinical guidelines for clinicians so how to go from the</p>
<p>genotype to dosing the patient most of those guidelines are derived from the</p>
<p>peer-reviewed published literature though so they’re kind of a derivative of that data source we also know there’s a lot of</p>
<p>unpublished data out there right so there’s clinical trials and other experiments that may be proprietary from</p>
<p>pharmaceutical companies that information is more difficult to to use</p>
<p>and to standardize and also to vet as a as a public resource it’s hard to deal</p>
<p>with that information but we know that a lot of that information is used in those regulatory agency approved drug labels</p>
<p>as well a lot of that submitted to these agencies when they’re the drugs are going for approval</p>
<p>so one of the sources centralized sources for dealing with organizing this</p>
<p>type of information that I just discussed is Farm gkp which is the pharmacogenomics knowledge base</p>
<p>this is a website with a database backend it’s publicly available it’s probably the biggest resource that’s</p>
<p>publicly available in the world right now and we started this around the year</p>
<p>2000 it’s based at Stanford University but our mission is basically to collect</p>
<p>encode and disseminate from code genomic knowledge for uses ranging from research</p>
<p>and Discovery all the way through clinical actionability and implementation</p>
<p>so what do we do at Farm gkb we take the information that is out there that I</p>
<p>kind of already went over guidelines drug labels what’s published in the literature and we think of this as our</p>
<p>knowledge information stream this is all publicly available right now for everyone but at from gkb we have a team</p>
<p>of scientists that extract information from these sources and using expert</p>
<p>manual curation we take the relevant parts of of uh like what</p>
<p>what drug what genotype What gene</p>
<p>Etc what variance we standardize the terminologies and we aggregate this information together and we’re always</p>
<p>working towards clinical implementation and adoption and in the end we present some resources that I’m going to show</p>
<p>you in a minute to actually go back out into this knowledge stream and we hope enrich it and make it a a better source</p>
<p>of of information for people who are trying to implement pharmacogenomics in their clinics</p>
<p>so one source of information at Pharmacy KB are these guidelines that I mentioned earlier and the two groups that are</p>
<p>probably published the most information about clinical implementation of pharmacogenomics would be cpic and the</p>
<p>Royal Dutch pharmacists association’s Dutch pharmacogenetics working group out of Netherlands</p>
<p>so I’m from gkb we this is just an example web page I don’t expect you to</p>
<p>read it that’s just an example but so for example the Dutch working group has some guidelines about the use of</p>
<p>amitriptyline and how sub 2d6 variants can affect implementation of effect dosage of this</p>
<p>particular drug and uh the Dutch working group puts out a PDF that is Tech space that people can</p>
<p>read and get this information themselves but what we tried to do it from dkb is make this a little bit simpler</p>
<p>presentation through tables and by using all the information available from the</p>
<p>Dutch group itself we have um on the web page a way for people to</p>
<p>input what the particular variants are of a specific patient or an example</p>
<p>and pull up what the recommendation is for that specific genotype so uh just a</p>
<p>little bit of an aid for people to be able to take the information from the Dutch group and um and uh get right to</p>
<p>the source without having to understand the mapping from the actual genotype in the gene to the metabolizer phenotype</p>
<p>such as poor metabolizer Etc to the actual recommendation so we have on our website uh just the</p>
<p>broken down information from multiple groups including cpic the Dutch group and there are a few other groups that</p>
<p>have some one-off recommendations as well and you can compare across these</p>
<p>um due to Resource issues and so forth not all the same drugs have been</p>
<p>um have guidelines written about them from from both organizations and so sometimes one organization may have</p>
<p>something about a drug that the other one doesn’t most of the time the recommendations agree or at least are</p>
<p>very similar but sometimes there’s a little bit of differences and you can compare those at the at Farm gkb to see</p>
<p>what the differences or similarities are and we also highlight where guidelines tell you whether give you testing</p>
<p>guidance what to test and when to test so another source of information that we</p>
<p>organize would be drug labels we annotate drug labels from FDA but also</p>
<p>from EMA Canada and we’ve had a couple of collaborations with groups in both</p>
<p>Japan and Switzerland so we have some labels that have been annotated from those groups as well and we’re always</p>
<p>Keen to collaborate with others so if anybody is interested in working with us</p>
<p>on that please let me know but yes but our biggest group of drug</p>
<p>labels would be from FDA and Then followed by with email so</p>
<p>we saw an example of a drug label earlier from munir and so sometimes on</p>
<p>drug labels they do highlight that there is information but much of the time the</p>
<p>language on drug labels can be not precise or somewhat vague and so we</p>
<p>had feedback from many of our users that what we they really wanted to know what these drug labels is just tell me which</p>
<p>labels say I need to test which one is recommend testing and which ones have any kind of information that I can use</p>
<p>so we came up with this a labeling system where we can tag these labels</p>
<p>that we curate with um with those categories and also we get very specific</p>
<p>if there is prescribing information based on pharmacogenetics on a drug label such as changing the dose or</p>
<p>changing the drug we highlight that as well so this is just an example screenshot I</p>
<p>don’t expect you to read it of annotation of the EMA label for abacavir and hlab you heard a little bit about</p>
<p>this earlier so yes this is a very kind of famous example of pharmacogenomics</p>
<p>very well understood and so yes testing is required according to the label the</p>
<p>label gives you specifics about what variant it is specifically in HLA B that</p>
<p>you should be interested in and gives you advice about what to do so this is an example of a nicely laid out drug</p>
<p>label and we were able to capture this information very easily but not all drug labels are quite so simple</p>
<p>so um again we have a page at farm gkb please go ahead and check it out for</p>
<p>yourself at some point if you are able and inclined we have drug labels from FDA EMA and a few other groups as well</p>
<p>and you can use this table to compare across and you can drill down to see</p>
<p>that exact annotation like I just showed you for the EMA one in a Bakka beer but what’s interesting is that some</p>
<p>Regulatory Agencies have different information on their labels depending on either they may have be lacking</p>
<p>information altogether about pharmacogenomics on that label whereas other countries or groups have a very</p>
<p>high level of actionability for pharmacogenomics so it’s kind of interesting to look at this webpage to</p>
<p>see overall again where the similarities and differences are and it just has to do with what is submitted to these</p>
<p>Regulatory Agencies and what is deemed actionable at the time</p>
<p>so we also heard from you near a little bit about randomized control trials</p>
<p>again yes this is considered the gold standard and a lot of um clinicians prefer to have information</p>
<p>that comes from rcts and this is where they would like to see any kind of proof</p>
<p>of clinical actionability but as was pointed out it’s really not feasible a</p>
<p>lot of times we have small populations in these studies again when you mentioned that you know we don’t have</p>
<p>huge uh cohorts like in diabetes Etc so the small populations do often make</p>
<p>statistical significance difficult and but we do know that statistical significance significance is not always</p>
<p>just the same as clinical significance anyway um and there is a lack of standardization again across all of</p>
<p>these studies and Publications that are uh that come out but we can use a replication of data</p>
<p>from multiple different sources to help us address some of these issues right so we can take an article like this this</p>
<p>is from New England Journal of Medicine and is talking about a particular</p>
<p>association between a variant and slcl1v1 and a reaction to Simvastatin</p>
<p>and if we take this and manually curate it at Farm gkb again we’re standardizing</p>
<p>across all the Publications the gene names the gene variants there are many different names for genetic variants as</p>
<p>I’m sure you’re all aware right people can use rsids they can use cdna changes</p>
<p>um protein amino acid change Etc so we can standardize so that we can see</p>
<p>across all Publications which ones are talking about the same variant we can standardize in drug terms Etc and we</p>
<p>take a bunch of information out about you know statistics and what type of</p>
<p>study and the study size we can collect the information from each article and</p>
<p>over time there are more and more articles right about a particular Association that can be published</p>
<p>many times these articles replicate the original findings but sometimes they do not so sometimes an association is</p>
<p>published and the next paper that comes out might refute actually that Association but if you look at a large</p>
<p>enough cohort of papers and like I said you’re able to compare across because the standardization we can start seeing</p>
<p>the bigger picture and write summaries about the association based on the Corpus of evidence and then we can also</p>
<p>take the information from those drug labels and the guidelines that I referenced earlier that are published in</p>
<p>an annotated at Pharmacy KB as well and we can get an even bigger and better picture of what’s going on with a</p>
<p>particular Association once we have all this evidence together we can not only write a summary about</p>
<p>what’s going on but we can assign a level of evidence about how confident we</p>
<p>are and the strength of this particular Association so this is just a screenshot again for a</p>
<p>particular association with the same variant with simvastatin and down here</p>
<p>we can see that there are two dosing guideline annotations available in farm gkb one from cpic and one was from the</p>
<p>Dutch group and you know in this case we only have nine Publications but there can actually be many many more</p>
<p>Publications depending on the phenotype that’s studied so based on that grouping of information</p>
<p>we can come up with a different levels of of evidence you can read more</p>
<p>about the system that we have come up with for standardizing when we’re trying to collect evidence together or visit</p>
<p>the web page I can’t go into a lot of detail here because it would take a while to explain but basically we score</p>
<p>the information that we take from every single publication we take that information and we add it together</p>
<p>across all the different Publications and taking into account if any</p>
<p>regulatory labels are available and the information on those labels and also guidelines of their information if</p>
<p>they’re available and we come up with a total score for that particular summary annotation then based on that score for</p>
<p>the summary annotation we can figure out which cutoff for a level that it meets</p>
<p>and assign accordingly so for all of these summaries you can go if you’re on a farm gkb page you can see uh get a</p>
<p>feeling for how much support there is for a particular Association based on the little labels on the left</p>
<p>hand side that are color coded and also have the level um written in them</p>
<p>for in most cases as a level one or instances where we know of a guideline</p>
<p>or a label that really confirms basically actionability or clinical</p>
<p>implementation can be instituted for a particular Association and level two</p>
<p>would be those that are very close but maybe don’t have a guideline or a label</p>
<p>yet but information on a label but there’s a lot of information published and the association looks very strong</p>
<p>some people on the other hand are very interested in low-level associations if they’re not looking to implement clinically but they’re looking at from</p>
<p>research perspective okay so I’m going to switch I’m just</p>
<p>years for just a couple minutes to tell you a little bit more about cpic again this is one of the groups that writes</p>
<p>guidelines for how to implement uh pharmacogenomics in the clinic the Dutch</p>
<p>group also writes guidelines as well I happen to be involved in cfix I’m going</p>
<p>to give you a little bit of the background of that particular group so the goal of cpic is not to tell people</p>
<p>whether or not to test or even what to test per se but if you are a clinician</p>
<p>and you have genetic results in hand we want you to be able to understand and quickly be able to know what you can do</p>
<p>with that information preemptive genotyping is becoming</p>
<p>more widespread there are direct to Consumer genotyping companies out there</p>
<p>and some patients are going to their doctors at least in the states with information about their their genetic</p>
<p>variants and they want their doctors to be able to understand that information</p>
<p>and act on it but most clinicians as we heard they don’t have the time for</p>
<p>patient to sit and and really research for themselves what a</p>
<p>particular genetic variation might mean or might imply for a particular drug prescription</p>
<p>so they need a facile way to access information that can tell them up front what it is that they should do</p>
<p>so these guidelines are written by expert groups put together these groups involve a clinician include</p>
<p>clinicians Gene experts Pharmacists</p>
<p>and sometimes in some cases just research scientists as well</p>
<p>and again we start with a PubMed literature review so these guidelines are based on PubMed literature reviews</p>
<p>but the information is collated together</p>
<p>and graded for every single outcome by the the</p>
<p>um the clinical authors the authors of the guideline and so a consensus is reached from that particular group and</p>
<p>then a statement about what you could do with a particular genetic um</p>
<p>um test result so as part of this process we have to</p>
<p>understand for every genetic allele what the functional implication is somebody</p>
<p>had mentioned that before so yes we do have to care about what the function is for all of these variants and so that’s</p>
<p>part of the um the process for creating stupid guidelines is to look at the</p>
<p>basically do a deep dive literature review of every allele for the particular Gene and that guideline and</p>
<p>see if we can come to agreement about what the function of that allele is then once we have a function defined for an</p>
<p>allele we can put the two alleles together and map to a phenotype such as</p>
<p>a poor metabolizer or an ultra rapid metabolizer that you heard about in the previous talk</p>
<p>once we have the metabolizer status for a given genotype for a given patient</p>
<p>then we can come up with what the therapeutic recommendation would be for that particular phenotype</p>
<p>so cpic also makes available clinical decision support flow charts and</p>
<p>um and wording for CDs alerts to help aid with clinical implementation for</p>
<p>groups that don’t have the resources to do that as well so this is just example charts and CDs language that people</p>
<p>could use if they want to implement in their Institution so there’s a bunch of Civic guidelines</p>
<p>right now the last count was 26 but that might have changed really recently</p>
<p>because we’re always publishing but please go and check out the website for</p>
<p>yourself we also have a database and API for people that are interested to access the information computationally and</p>
<p>import it into their own databases so another challenge with</p>
<p>implementing pharmacogenomics would be our nomenclature you heard a little bit</p>
<p>about the star nomenclature already I get people ask me all the time who</p>
<p>aren’t in the farm codes next field but more clinical genomics what what’s</p>
<p>what’s the deal with the star wheels why do you guys have them and what do they mean so just really briefly right we</p>
<p>have the the symbol the gene symbol and that star with the number after it is just an allele designation and so for</p>
<p>example this uh D6 star 8. this is hgbs representation for it so there are you</p>
<p>know three different um uh three different variants across the gene that Define this allele so in</p>
<p>pharmacogenomics with the most of the cytochrome p450s we’re interested in</p>
<p>what the combination of variants are across the entire allele it’s not necessarily one particular variant or</p>
<p>another but what’s the combination of of uh variance across the entire Gene it’s</p>
<p>a haplotype that matters and so to define the haplotypes the the star</p>
<p>allele shorthand was um was born and these variants these are Snips so that’s</p>
<p>quite simple but in many cases the the variation across the gene can include repeats or indels there can be</p>
<p>structural variants and copper number variation as well and the variation occurs not just in the</p>
<p>exons but also in trons or upstream or Downstream of the genes sometimes and they have all have functional effect and</p>
<p>are very important to to capture and understand what the variation is across</p>
<p>and so this shorthand helps the community understand what the variation</p>
<p>is I’ve been asked many times well the star alleles just can they just go away from</p>
<p>from people who are you know clinicians but deal with a mendelian disease many</p>
<p>times or other kinds of genetics and unfortunately I don’t think that’s going to happen so it’s used throughout the</p>
<p>literature lab tests and assays test reports all refer to the start allele nomenclature right now and it’s used by</p>
<p>many of the prescribing guidelines and this started in the 1990s and it was</p>
<p>organized to a great extent in the early 2000s by the human cytochrome p450 allele</p>
<p>nomenclature database in Sweden and they I think they were the first group to really start tracking these alleles and</p>
<p>defining them as what’s in what variation is in each allele and they’re the ones that started naming them with</p>
<p>the star alleles that has transitioned in 2017 to a group called Farm bar</p>
<p>the farm far group has taken over the reins there and so Farm bar please check</p>
<p>out this group as well so when we’re talking about standardization Farm fire is critical to standardization for these</p>
<p>pharmacogenetic alleles and it is the central repository people who find new allelic variation can submit that</p>
<p>variation to the farm bar group and uh and there’s a a very rigorous process</p>
<p>rules that must be followed for a new star allele to be assigned but if uh if</p>
<p>that’s the case if the submission reaches the level of a new star allele it will be assigned</p>
<p>so just to show you a little bit about the this is not the most complicated example by any means but just to show</p>
<p>you a little bit what I’m talking about so this is again a sip 2d6 illustration</p>
<p>um we’re looking at the star alleles star 10 3637 Etc and then this is a list</p>
<p>of variants down the side and so what we see here is the the 100 C to T variant</p>
<p>is um is a part of all of these different alleles so is this variation</p>
<p>down here so if you for example we’re just going to test this 100 C to t you</p>
<p>might not really know which star allele you’re talking about for simplicity’s</p>
<p>sake many times people just call it a star 10 because okay that’s the defining variant in that particular star allele</p>
<p>but you can see that there’s many these other star alleles also contain that variant and then other variants as well</p>
<p>so it can get much more complicated than this as well and some genes such as 2d6</p>
<p>of structural variance and it gets it gets crazy but these star alleles really are important for understanding what the</p>
<p>variation is and that all has functional implications so the functionality of</p>
<p>these different star alleles can and do differ so that leads to one of the other</p>
<p>challenges of uh of pharmacogenomics and implementation is that the type of the</p>
<p>test what is tested can affect your results right so there have been studies done where they’ve sent the same samples</p>
<p>out to different labs and gotten different results back and and that a lot of times has to do with whether</p>
<p>you’re talking about a panel or an exome or a whole genome sequencing is someone had brought up earlier and it’s true</p>
<p>that if you have a panel and not everything is on that panel that you can see in whole genome sequencing you could</p>
<p>potentially miss genetic variation that exists in that particular patient and</p>
<p>that’s why we have to be really careful with this often in the field if you</p>
<p>don’t see any variation it is defaulted to what’s called a star one allele which</p>
<p>implies that you don’t have any genetic variation or wild type but of course depending on what’s tested</p>
<p>that may or may not be true so just something to keep in mind when people are trying to implement pharmacogenomics</p>
<p>that transparency about the tests and the test result can be very important there can be situations where a patient</p>
<p>is tested for pharmacogenomic alleles and none are found and it goes into</p>
<p>their the EMR or their record that they have</p>
<p>no genetic variation in a particular Gene but we know that that may not always be 100 True right for example</p>
<p>there are some non-existing genetic variations where we</p>
<p>know of the genetic change but we don’t know what the function is we don’t know what to do with that change but over</p>
<p>time you know research is going to eventually elucidate that for us also there’s new variation out there</p>
<p>that may not be covered so the current star allele nomenclature can’t cover everything that’s not known yet so as</p>
<p>new discoveries happen they’re submitted and we have new star alleles that may not have been tested for previously so</p>
<p>at the very least</p>
<p>and what the results are is can be very key to both clinicians and patients</p>
<p>especially going forward in the future so a um another project that I’m</p>
<p>involved in is called the pharmacogenomics clinical annotation tool this is a software where the goal</p>
<p>is to take the output from a genetic test result report sorry the output from</p>
<p>a genetic test such as a panel test or whole genome or exome sequencing</p>
<p>come up with the star allele designations For Those Jeans which can be very complicated as was mentioned</p>
<p>earlier so this software aims to be able to take whole genome sequencing and</p>
<p>determine that without using those tag alleles that are used on the snip panel and then connect the resulting genotypes</p>
<p>with clinical guidance and we’ve started with the clinical guidance from cpic but we’ll be expanding that to other groups</p>
<p>as well such as the Dutch guidelines and also what’s available on a regulatory</p>
<p>agency drug labels that’s the goal so basically the way this tool works is</p>
<p>it takes a VCF file which is just um a way that genetic variation can be output</p>
<p>from testing and then we take those allele definitions like from Farm bar</p>
<p>what that has the definitions of what variance are in each star allele and the information from the guidelines</p>
<p>that can be accessed via the API in the database and then we also have some messaging</p>
<p>from the farm cat tool as well with caveats and disclaimers Etc and we take</p>
<p>all of that together so the what’s the user supplies is the actual genetic</p>
<p>variation that are detected and we come up with a genotype summary which tells</p>
<p>you what the star alleles what the diplotypes are or genotypes for that particular patient based on that VCR</p>
<p>VCF file and give you a summary of the output then the recommendations by drugs so you</p>
<p>may be a sip2d6 star 4 star 8 for example what does that mean we have</p>
<p>recommendations for each drug that we know that can be affected or have guidelines showing that the drug</p>
<p>response can be affected by sip2d6 and then we would tell you exactly what the recommendations there are for that</p>
<p>particular genotype and then there’s a section of the report</p>
<p>that is uh very key but not necessarily you know first and foremost in any</p>
<p>clinician’s mind or patient for that matter but very importantly we highlight what variation was given to us in the</p>
<p>VCF file versus what is known to date so we can easily see in this section of the</p>
<p>report if not every known genetic variant that could be part of a star allele Etc was tested so in this way we</p>
<p>can see with an assigned a genotype at the end if information was essentially missed if</p>
<p>there was a known star alleles that were not tested for in this case you would know then if say a patient is a star one</p>
<p>star one as the result whether or not there are potentially</p>
<p>variation that that patient is carrying in that Gene that was not covered by</p>
<p>that test so um it’s great to have this as a report you</p>
<p>get a purport out of farm cat and that’s a report that a patient could have and take to their doctor or a clinician</p>
<p>could look at but in many times as we know clinicians don’t have the time to sit and read reports either so having</p>
<p>this information go automatically into an EMR or EHR system is extremely important and that is a a big challenge</p>
<p>since they are not standardized across all hospitals and or Hospital systems</p>
<p>and so that’s another area that we’re working on for the future</p>
<p>um one of the final challenges I’m going to talk about today would be allele frequency and this again was mentioned a</p>
<p>little bit earlier but the issue is that in many cases a mendelian disease a</p>
<p>frequency a low frequency is a is a sign of maybe pathogenicity right so</p>
<p>frequency can be used as a way to figure out if something might be very important in a particular Gene but unfortunately</p>
<p>in pharmacogenomics some of these variations can be quite common up to you</p>
<p>know 30 percent or even more in certain populations so</p>
<p>so you cannot use frequency the same way and pharmacogenomics is a sign for what could be effective function</p>
<p>and yet um you know knowing the frequency of the alleles is very helpful to know for</p>
<p>example what are the if we can only test a handful of alleles which ones should you be testing unfortunately we don’t</p>
<p>know the frequency for many of the defined alleles they may be undiscovered in a particular patient submitted to</p>
<p>farmvar for example gotten a star allele now we know we can test for them but if people haven’t tested for them</p>
<p>previously we don’t really know what the frequency in any given population is so</p>
<p>many times we think of in pharmacogenetics in the field we think certain star alleles are rare because</p>
<p>it’s not published much in the literature we can’t find statistics</p>
<p>about it but that’s not necessarily the case so we do have to be careful there also</p>
<p>um you know many populations around the globe are understudied so again we want to make pharmacogenomics available for</p>
<p>all a diverse populations but we haven’t done a lot of studies yet in many of the</p>
<p>populations around the global a lot of them have been you know focused on European populations or perhaps Asian</p>
<p>populations as well so research is needed there to understand what the frequency is for</p>
<p>these for these variants but we always must remember that when we’re trying to implement this the population frequency</p>
<p>is never a proxy for what a patient actually has so testing the individual patient is really important so the last</p>
<p>challenge I’m going to talk about is just the separation of pharmacogenomics from the rest of clinical genomics so I</p>
<p>gave you a little bit of an overview of all these four resources farm gkb farm cat Farm bar and C pick these resources</p>
<p>talk to each other all the time they trade information back and forth</p>
<p>in the US we have very big projects for clinical genomics clinvar and Clinton are a huge resources</p>
<p>that are funded through the NIH but they’re kind of uh isolated these are</p>
<p>isolated from pharmacology I should say pharmacogenomics is isolated from them because they are very large and and</p>
<p>capture most of the genetic clinical genomics that are documented today but</p>
<p>pharmacogenomics kind of exists as a silo we are trying to address that by</p>
<p>depositing information from both cpic and farm gkb into both clinvar and Clinton</p>
<p>there are some issues with trying to take a pharmacogenomic information and</p>
<p>wedge it into a clinical genome database and so it’s a little bit tricky at times</p>
<p>and the deposition of this information is slow but in addition to depositing it we are</p>
<p>looking forward to being able to exchange information with clinvar and Clinton as well so what we really want</p>
<p>to do is not have pharmacogenomics be isolated and siled from Clinical genomics overall and genomic medicine</p>
<p>and our vision right would be full integration of pharmacogenomics with</p>
<p>genetic genomic medicine and this is just a diagram to try to to illustrate</p>
<p>that all those pharmacogenomic resources and more could you know eventually be</p>
<p>combined together in some way to have a much easier interface maybe one place that people can go and that that group</p>
<p>could interact very closely with both clinvar and Clinton and efforts are</p>
<p>underway we we have a panel that we’re starting within the Clinton group to</p>
<p>address pharmacogenomics and how to better crosstalk between pharmacogenomics and genomic medicine in</p>
<p>general so um in many cases we don’t know the um</p>
<p>the implications of pharmacogenomic discoveries for clinical implementation</p>
<p>but we definitely know that the clinical utility is there for for a couple dozen</p>
<p>examples at least and then we know that integrating this together with the risk of genomic mess</p>
<p>and is going to push implementation forward having to try to address this separately is slow going as meniere</p>
<p>described it’s not being implemented across as many places as we’d like pharmacogenomics to be yet</p>
<p>um it does differ from genomics disease models the haplotypes and those star</p>
<p>alleles and the fact that you have to be aware of the diplotypes and map that to metabolizer phenotypes in many cases is</p>
<p>very different and we are aware of that so we do still need some specialized resources but integrating would be the</p>
<p>way to go in the future and as of right now we have a few knowledge centralized</p>
<p>knowledge sources that people can use in the meantime and with that I’d like to just thank all my colleagues there’s</p>
<p>many different projects that I was talking about today and that takes many</p>
<p>people in many different institutions to help make all this all these projects a</p>
<p>reality and so I’d like to thank them and thank you for your attention your time [Applause]</p>
<p>we have questions I will start from the chat one question regards</p>
<p>sclo1b1 and statins and basically what they they ask is are there studies uh</p>
<p>regarding drug adherence and pharmacogenetic testing or genotyping I</p>
<p>mean does it influence adherence of the patients adherence yes</p>
<p>um I’m not aware of specific studies about adherence to drugs but we know</p>
<p>colloquially just anecdotally that this is true and that patients who have</p>
<p>adverse effects that are not as severe as like what we saw in the previous talk but just myopathy maybe uh you know pain</p>
<p>or some other kind of nausea Etc can lead patients to stop taking their</p>
<p>medication and so yes it has been shown that because of the myopathy that’s</p>
<p>associated with a particular genetic variation um there there have been cases like I</p>
<p>said anecdotally where patients just stop taking their medicine because of it yeah</p>
<p>hi Michelle over here thank you a great presentation fabulous resources I think</p>
<p>incredibly useful for the community can I just ask are there plans or thoughts</p>
<p>about translating those resources into different languages to make them more</p>
<p>accessible to do that unfortunately resource wise</p>
<p>being government funded we don’t have uh the resources to pay for that but if</p>
<p>there are people that would like to collaborate with us we could see what we could do yes that we get that question a</p>
<p>lot you know can this be in multiple languages and we’re happy to do it uh just a resource issue right now</p>
<p>okay another question from the chat is despite the fact that pharmacogenomics data is growing rapidly and importance</p>
<p>of testing is clear the integration in the clinics is very slowly so maybe you</p>
<p>could comment on on the barriers or the reasons for that right um yes I think um a lot of the reasons</p>
<p>for the slow uptake we just were mentioned previously just having</p>
<p>um the education I think of clinicians to for them to be familiar with this</p>
<p>type of data familiar with how to to use it also there are issues putting this kind</p>
<p>of information in medical records so getting this into the ahr system I think would be key that’s one barrier we know</p>
<p>to uptake and then just also you know testing and reimbursement what are you</p>
<p>supposed to test for when should you test who’s going to pay for it these kinds of issues definitely in effect</p>
<p>implementation as well last question yeah thank you as well for a great</p>
<p>presentation we in our company started testing the</p>
<p>farm cat and I’d like to ask how many drugs are in this moment</p>
<p>supported because we have a only a limited number and also if it can be adjusted by the</p>
<p>users the parameters and what are you going to expend it yeah absolutely so</p>
<p>yes so if you um check out farmcat.org so we’ve updated a lot of the</p>
<p>documentation maybe since you have checked it out so all of the drugs that</p>
<p>are covered by cpic guidelines are currently covered by Farm cat and their end report all the genes and yes you can</p>
<p>you can you can alter well first of all it’s freely accessible so if you want to</p>
<p>download the tool and Branch off you can change the code however you like but it</p>
<p>is also a customizable in terms of if you have jeans that you want to include</p>
<p>and you have the definition files for those you can add those as you want and same with uh with the recommendations as</p>
<p>well and we are going to be expanding those within the next year is the goal to to include</p>
<p>drugs and recommendations from the Dutch pharmacogenetic working group and also</p>
<p>what information we can find on the first the FDA labels we’ll probably start with FDA but then ultimately can</p>
<p>expand from there as well okay uh those are bioinformatics that I</p>
<p>work in in our company on it so I’m not completely familiar but is it possible</p>
<p>to speak to your its yeah sure sure um you know send um I didn’t have the</p>
<p>email up here but if you send email to just feedback at farmgkb.org we will make sure it gets to</p>
<p>farm cat I think there’s also Farm cat at Farm gkb.org but I can’t swear to</p>
<p>that so if you send it to feedback at farmgkb.org and you say this is a question about Farm cat we can deal with</p>
<p>that we’re a very small team and we are the same people working on many projects</p>
<p>at once but um yeah but yes definitely thank you thanks</p>
<p>okay thank you so I I want just to remind you that if you continue</p>



</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
          // target, if specified
          link.setAttribute("target", "_blank");
          // default icon
          link.classList.add("external");
      }
    }
});
</script>
</div> <!-- /content -->



</body></html>