<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.353">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>PGC Video Textbook - Chapter 8.9: PheWAS (Video Transcript)</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


<link rel="stylesheet" href="styles.css">
</head>

<body class="nav-sidebar docked nav-fixed">


<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="./index.html">
    <span class="navbar-title">PGC Video Textbook</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link active" href="./index.html" rel="" target="" aria-current="page">
 <span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./about.html" rel="" target="">
 <span class="menu-text">About</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./contact.html" rel="" target="">
 <span class="menu-text">Contact us</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="https://pgc.unc.edu/" rel="" target="">
 <span class="menu-text">PGC Website</span></a>
  </li>  
</ul>
            <div class="quarto-navbar-tools ms-auto">
</div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
      <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item">Chapter 8.9: PheWAS (Video Transcript)</li></ol></nav>
      <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
      </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal sidebar-navigation docked overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
      <a href="./index.html" class="sidebar-logo-link">
      <img src="./pgc_logo_website_v3.jpeg" alt="" class="sidebar-logo py-0 d-lg-inline d-none">
      </a>
      <div class="sidebar-tools-main">
    <div class="dropdown">
      <a href="" title="" id="quarto-navigation-tool-dropdown-0" class="quarto-navigation-tool dropdown-toggle px-1" data-bs-toggle="dropdown" aria-expanded="false" aria-label=""><i class="bi bi-github"></i></a>
      <ul class="dropdown-menu" aria-labelledby="quarto-navigation-tool-dropdown-0">
          <li>
            <a class="dropdown-item sidebar-tools-main-item" href="">
            Source code
            </a>
          </li>
          <li>
            <a class="dropdown-item sidebar-tools-main-item" href="">
            Report a bug
            </a>
          </li>
      </ul>
    </div>
</div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./welcome.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Welcome to the PGC Video Textbook!</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./toc.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Table of Contents</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">
 <span class="menu-text">Chapters</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter1.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 1: Introduction</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter2.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 2: The Genome</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter3.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 3: Technologies</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter4.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 4: Study designs</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter5.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 5: GWAS analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter6.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 6: Polygenic Scores</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter7.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 7: Ancestry-Specific Analyses and Considerations</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter8.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 8: Post-GWAS bioinformatics</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter9.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 9: Advanced Topics</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter10.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 10: Other Considerations</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true">
 <span class="menu-text">Software Tutorials</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_cnvs.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">CNVs</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_conditional.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Conditional Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_crossdisorder.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Cross-disorder Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_ewas.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">EWAS</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_geneset.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Gene Set Identification</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_gwas.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">GWAS</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_genomicSEM.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Genomic SEM</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_imaging.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Imaging</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_MR.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Mendelian Randomization</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_mtag.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">MTAG</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_prs.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">PRS</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_correlation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">SNP Heritability and Genetic Correlation</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true">
 <span class="menu-text">Additional Resources</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./glossary.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Glossary</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./resources.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Software Resources</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./addreading.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Additional Reading</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="https://pgc.unc.edu/for-researchers/download-results/" class="sidebar-item-text sidebar-link">
 <span class="menu-text">PGC Summary Statistics</span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#sec-video1" id="toc-sec-video1" class="nav-link active" data-scroll-target="#sec-video1">PheWAS: Discovering gene-disease associations</a></li>
  <li><a href="#sec-video2" id="toc-sec-video2" class="nav-link" data-scroll-target="#sec-video2">PheWAS and EHR</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Chapter 8.9: PheWAS (Video Transcript)</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<hr>
<section id="sec-video1" class="level1">
<h1>PheWAS: Discovering gene-disease associations</h1>
<p><strong>Title</strong>: PheWAS: Discovering gene-disease associations</p>
<p><strong>Presenter(s)</strong>: Dr.&nbsp;Joshua Denny, All of Us Research Program</p>
<p><strong>thank you very much it’s a pleasure to</strong></p>
<p>be here as always and talking about just</p>
<p>the amazing stuff that’s happening here</p>
<p>in the UK biobank</p>
<p>alright great so to start with you know</p>
<p>we’ve talked a lot about genome-wide</p>
<p>Association studies and sequencing and</p>
<p>and we’ve talked to them about phenome</p>
<p>wide Association studies as well that’s</p>
<p>going to be the focus of my talk and</p>
<p>just to orient us you know essentially</p>
<p>what we’re doing is thinking about an</p>
<p>independent variable than exploring what</p>
<p>phenotypes and the range of phenotypes</p>
<p>that are available and associated with</p>
<p>that really anchoring on the fact that</p>
<p>we have richly systematically phenotype</p>
<p>sets of individuals such as in the UK</p>
<p>biobank and other electronic health</p>
<p>record data sets which is where this</p>
<p>started usually that’s based on things</p>
<p>like billing codes but I don’t want to</p>
<p>limit us there you can think about</p>
<p>laboratory values you could think about</p>
<p>natural language processing and things</p>
<p>like that as well most of its been based</p>
<p>on billing codes so to start with I want</p>
<p>to orient us to a discovery study four</p>
<p>out of the electronic medical records</p>
<p>and genomics Network emerge in the US</p>
<p>and it was five sites that work together</p>
<p>for a carefully validated phenotype that</p>
<p>we used codes labs medications natural</p>
<p>language processing to find these cases</p>
<p>and manually validate who was a case for</p>
<p>autoimmune presumptive autoimmune</p>
<p>hypothyroidism and in control and we’d</p>
<p>identify the thyroid transcription</p>
<p>factor that was associated and</p>
<p>replicated this and then we did it took</p>
<p>took a variant the same variant that was</p>
<p>found here and did a fee wass on that</p>
<p>variant in a slightly larger population</p>
<p>and you know that was unselected our a</p>
<p>much larger population unselected for</p>
<p>any given phenotype and hypothyroidism</p>
<p>was the the you know the highest</p>
<p>associate a phenotype there but we also</p>
<p>had some other thyroid diseases that</p>
<p>came up and things like atrial flutter</p>
<p>as associated which we know that there’s</p>
<p>you know hypothyroid or less likely to</p>
<p>manifest with atrial flutter and so so</p>
<p>this gives opportunity to look at the</p>
<p>performance of these two methods</p>
<p>and so you know on the left we have the</p>
<p>schematic of the algorithm we use and</p>
<p>then we use these mappings called fee</p>
<p>wass codes or fee codes and we usually</p>
<p>say there have to be two or more that</p>
<p>map into that fee code and you know you</p>
<p>can see the odds ratios are essentially</p>
<p>the same between the two approaches</p>
<p>within our population of individuals</p>
<p>with an emerge</p>
<p>you know we identified more cases and</p>
<p>with the fee wast codes than we did with</p>
<p>the algorithm and so you know there’s</p>
<p>many approaches that fee Wasson just</p>
<p>talking about that you know said most</p>
<p>used billing codes in the u.s. that’s</p>
<p>been historically icd-9 with the</p>
<p>clinical modifications and now icd-10</p>
<p>after 2015</p>
<p>there’s about 65,000 icd-10-cm codes and</p>
<p>on the right you can see some of the</p>
<p>ways this works so the fee codes have</p>
<p>numbers that kind of look like icd-9</p>
<p>codes but they’re actually not</p>
<p>and and then what we do is we group you</p>
<p>know like codes now across icd-9 and</p>
<p>icd-10 and icd-10-cm codes to you know a</p>
<p>given fee code and and so all the type 1</p>
<p>codes come together which is not obvious</p>
<p>from the icd-9 coding group system and</p>
<p>then each of those also define ranges of</p>
<p>control groups in addition to the feed</p>
<p>groupings or other groupings in the US</p>
<p>there’s some the AHRQ has released some</p>
<p>software that groups things into about</p>
<p>300 diseases tree wasps is another thing</p>
<p>you can also use raw ICD codes for</p>
<p>instance that gives you a challenge</p>
<p>across of course mapping between icd-9</p>
<p>and icd-10 and you can do many other</p>
<p>things survey data has been run across</p>
<p>the UK biobank and other things that</p>
<p>I’ve talked about like the procedures so</p>
<p>here’s another example fee wasps driven</p>
<p>by on EHR data looking at imputed HLA</p>
<p>types into the two and four digit types</p>
<p>of HLA and you can see you know quickly</p>
<p>it highlights the fact that there are</p>
<p>different associations between class one</p>
<p>and class two HLA alleles and what helps</p>
<p>you think about the range of</p>
<p>associations and and overall I think</p>
<p>there was a hundred or so significant</p>
<p>associations most of which were known</p>
<p>and a few of new ones but what’s more</p>
<p>interesting as by doing in a single</p>
<p>population you can actually look across</p>
<p>those phenotypes</p>
<p>and and then look for pleiotropy and see</p>
<p>you know if you adjust in condition on</p>
<p>one and the other do you see you know</p>
<p>that they’re that they’re truly</p>
<p>independent associations and you can</p>
<p>also see we can a given HLA type that</p>
<p>you know one two HLA types that may put</p>
<p>you</p>
<p>similarly at risk for rheumatoid</p>
<p>arthritis it may had differential effect</p>
<p>for your risk on type one diabetes for</p>
<p>instance and so so that is a tool that</p>
<p>you know you can rapidly explore using</p>
<p>this kind of technique an important</p>
<p>aspect is validating its efficacy and so</p>
<p>one of the early things we did using our</p>
<p>icd-9 codes across the emerge was</p>
<p>replicating known associations in the G</p>
<p>wasps catalog we found eighty-six</p>
<p>phenotypes that were could be</p>
<p>represented in the electronic health</p>
<p>record and a number of snips about seven</p>
<p>hundred and fifty overall snip phenotype</p>
<p>pairs overall we replicated in two</p>
<p>hundred ten of them across the number of</p>
<p>disease classifications and sixty-six</p>
<p>percent of those for which we were</p>
<p>adequately powered in this population of</p>
<p>thirteen thousand people as well as</p>
<p>vying some novel associations on the top</p>
<p>of which we replicated it also allows us</p>
<p>to actually compare the effect size so</p>
<p>here you might you see something that</p>
<p>you would expect to see in that that the</p>
<p>effect sizes from the the--why studies</p>
<p>are typically a little bit lower than</p>
<p>what’s in the G wasps catalogue now some</p>
<p>of that’s probably due to the winners</p>
<p>curse but some of it’s also due to the</p>
<p>phenotype being not quite as accurate</p>
<p>and it helps you think about the ones</p>
<p>that where you have the most error so</p>
<p>the most common error and it was really</p>
<p>I universally type one diabetes is often</p>
<p>miss code in fact ninety six percent of</p>
<p>the time we found type one diabetics had</p>
<p>type two diagnosis codes and so so you</p>
<p>know it made it it made it and the</p>
<p>reverse is true fifty six percent of the</p>
<p>time so so it caused a lot of inaccuracy</p>
<p>in the type 1 diabetes phenotype and we</p>
<p>had trouble replicating some of those</p>
<p>snips and we’ve actually instituted</p>
<p>methods to fix that problem and we can</p>
<p>recover those associations so here’s</p>
<p>here’s a way you can use fee wasps in</p>
<p>concert with eg wasps so we did a GYN</p>
<p>emerge looking at longitudinal risk of</p>
<p>heart of a CO disease on a statin</p>
<p>and found variants that are tied to</p>
<p>expression of lipoprotein a were</p>
<p>associated with that outcome as a</p>
<p>longitudinal analysis and that risk is</p>
<p>increased for those that have you know</p>
<p>kind of ideal cholesterol levels at less</p>
<p>than 70 so we looked at a loss of this</p>
<p>lowest locus and you know as you’d</p>
<p>expect you see coronary atherosclerosis</p>
<p>near the top and and fortunately we see</p>
<p>most of the phenotypes are ones we would</p>
<p>expect to see which gets that the</p>
<p>question of if you were to target this</p>
<p>with a medication you know what</p>
<p>potential effects would you see and so</p>
<p>one of the things that’s interesting and</p>
<p>wouldn’t have been on a radar screen is</p>
<p>this point over here which is not quite</p>
<p>statistically significance was lung</p>
<p>cancer so you know this is a relatively</p>
<p>small population from 13,000 people as</p>
<p>it’s explored more maybe that will turn</p>
<p>out to be true or not but it is a rapid</p>
<p>tool for highlighting especially when</p>
<p>you think about the scale you could buy</p>
<p>a bank I mentioned mapping these to</p>
<p>icd-10 and icd-10-cm codes just shows a</p>
<p>little bit of a process and the</p>
<p>vocabularies and systems that we used in</p>
<p>the process with some manual validation</p>
<p>it is in what we call it’s still in beta</p>
<p>form but you can see it covers about 90%</p>
<p>of the billed ICD codes in the UK</p>
<p>biobank</p>
<p>and amongst the 10% that aren’t there</p>
<p>most of those are not actual disease</p>
<p>codes only a small fraction those</p>
<p>represent true disease codes and we did</p>
<p>an evaluation at using our data with</p>
<p>icd-9 and icd-10 codes in terms of fee</p>
<p>wasps and you can see that the effect</p>
<p>sizes between this pandas population was</p>
<p>essentially the same for these two known</p>
<p>associations with that snip so I want to</p>
<p>give a few examples actually Kristen</p>
<p>showed this earlier doing a fee wass in</p>
<p>the UK biobank and just tons of</p>
<p>Association associated with atrial</p>
<p>fibrillation genetic risk score for afib</p>
<p>and and I when they condition for the</p>
<p>phenotypes the cardiovascular phenotypes</p>
<p>essentially those associations went away</p>
<p>and but it shows the power of a huge</p>
<p>population to show lots of things you</p>
<p>expect to see here’s another one for</p>
<p>systolic blood pressure on a</p>
<p>a large G wasit was done across the</p>
<p>million veteran program as well as the</p>
<p>UK biobank and just a number number of</p>
<p>associations showing up with systolic</p>
<p>blood pressure they also did with</p>
<p>diastolic blood pressure and pulse</p>
<p>pressure to show that some of these</p>
<p>phenotypes overlap and you see</p>
<p>phenotypes that are not exactly</p>
<p>associated with a cardiovascular disease</p>
<p>in here coming out as well</p>
<p>endocrine being one of the more common</p>
<p>ones here’s a resource Kristen also</p>
<p>talked about the sage approach - I’m use</p>
<p>saddle point approximation - I created</p>
<p>an efficient and accurate way of</p>
<p>calculating these kind of results at</p>
<p>scale to the UK biobank they have</p>
<p>produced a website where you can explore</p>
<p>those phenotypes calculated using the</p>
<p>same approaches for fee codes across the</p>
<p>UK biobank and this just shows a</p>
<p>particular a-fib snip in that website</p>
<p>and the URL there is there at the bottom</p>
<p>so you know we’ve talked about this and</p>
<p>and and in looking at individual</p>
<p>phenotypes I want to spend the last few</p>
<p>minutes talking about phenotypes in</p>
<p>clusters and how we think of them so if</p>
<p>you think about a Mendelian disease is</p>
<p>is a classic example that are often</p>
<p>syndromic presenting with many different</p>
<p>features and those features you know are</p>
<p>what we may bill in the electronic</p>
<p>medical record as physicians but it</p>
<p>doesn’t necessarily represent you you</p>
<p>know the the disease is not always</p>
<p>recognized or you know may be recognized</p>
<p>later into the disease course as we you</p>
<p>know heard about earlier with</p>
<p>hemochromatosis and so through the</p>
<p>online Mendelian inheritance and man</p>
<p>resource and the linked human type human</p>
<p>phenotype ontology you know we can go</p>
<p>from a Mendelian disease to a list of</p>
<p>features of that disease which have a</p>
<p>vocabulary behind them and then so so</p>
<p>our lab mapped those hpo features to fee</p>
<p>codes so basically allowing you to</p>
<p>translate oh men features into EHR</p>
<p>phenotypes and then similar to a</p>
<p>polygenic risk or you know creating a</p>
<p>phenotype risk or that that looks</p>
<p>similar in process so aggregating</p>
<p>Phoenix</p>
<p>by their weights I’m to produce the</p>
<p>score for individual and essentially you</p>
<p>can crank this out across anything for</p>
<p>which you have a map and do it at scale</p>
<p>so let’s look at the Cystic Fibrosis a</p>
<p>number of features from omen and each of</p>
<p>those is mapped to a human phenotype</p>
<p>ontology code and so I’m using our fee</p>
<p>code ontology of around 1,800 phenotypes</p>
<p>and you can map the ones that line up</p>
<p>fairly well to the CF they’re not all</p>
<p>exact matches some are better matches</p>
<p>and others and then some that we don’t</p>
<p>have in the HR which you know we’re</p>
<p>familiar with and so let’s play that out</p>
<p>on a couple individuals hypothetical</p>
<p>different conditions I mentioned they’re</p>
<p>weighted so features like bronchiectasis</p>
<p>have a higher weight than features like</p>
<p>asthma and so when you go across this</p>
<p>you know individuals get a different</p>
<p>score and what you find is you can</p>
<p>separate case in the controls 460</p>
<p>fibrosis</p>
<p>you know just using the features of</p>
<p>disease so we’re not using the disease</p>
<p>label but in this example we use</p>
<p>manually validated at cases versus</p>
<p>controls who don’t have any evidence of</p>
<p>the disease in the in the text record</p>
<p>and we see a very significant result and</p>
<p>we’ve actually done this for 15 other</p>
<p>diseases now and in in every case except</p>
<p>for one we’ve seen very strong</p>
<p>separation between cases and controls</p>
<p>the one exception is phenylketonuria</p>
<p>which as you know in the u.s. is on</p>
<p>essentially every newborn screening test</p>
<p>and if you avoid and feel an exposure</p>
<p>you don’t actually see the you know the</p>
<p>manifestations of the disease so it sort</p>
<p>of gives you a test of the effectiveness</p>
<p>and newborn screening in removing the</p>
<p>features of Z’s in the population</p>
<p>because they generally do not have</p>
<p>elevated scores so we turn this on a</p>
<p>population of 21,000 people that had</p>
<p>eczema rate genotyping and looked at</p>
<p>6000 variants that were rare at at 1%</p>
<p>level or less and we found 18</p>
<p>significant Association most of which</p>
<p>were novel and we were able to change</p>
<p>the ACMG clinical interpretations for</p>
<p>eight of these variants towards likely</p>
<p>pathogenic or pathogenic and so so this</p>
<p>using our population as a paradigm that</p>
<p>I think can be explored with larger rich</p>
<p>phenotype populations such as what is in</p>
<p>the UK biobank so I want to end and with</p>
<p>represent a recognition of some of the</p>
<p>many people contributing to this work</p>
<p>and the middle row is probably the most</p>
<p>important row as as the folks actually</p>
<p>doing the work thank you very much</p>
<p>[Applause]</p>
<hr>
</section>
<section id="sec-video2" class="level1">
<h1>PheWAS and EHR</h1>
<p><strong>Title</strong>: Using EHR-based genomic approaches to understand the relationship between mental and physical health</p>
<p><strong>Presenter(s)</strong>: Dr.&nbsp;Lea Davis, Vanderbilt University</p>
<p>Intro</p>
<p><strong>and the bio bank that’s attached to them at Vanderbilt and so this opened up part</strong></p>
<p>of a newer area of investigation for me something that I’ve been interested in a long time but but hadn’t had the</p>
<p>resources to investigate and so that’s basically using EHR based genomic</p>
<p>approaches to try to better understand the relationship between mental health and physical health so that’s kind of</p>
<p>the story that I’m going to be talking to you about today so if I can advance my slides oh there we go okay so so this</p>
<p>relationship between mental health and physical health is kind of well-known</p>
<p>that it’s important and in one of the I think nice summary statements about the</p>
<p>importance of this comes from the World Health Organization on their website they state that there is no health without mental health and it’s been</p>
<p>documented for some time that poor mental health is a risk factor for a number of physical conditions particularly chronic conditions and so</p>
<p>people with severe psychiatric illness are at much higher risk of experiencing</p>
<p>chronic physical conditions and vice versa people with chronic physical conditions</p>
<p>are also at risk of defining mental health and so I I’ve actually come to</p>
<p>People with severe psychiatric illness experience health disparities</p>
<p>think about this much like a disparity in fact that people with severe</p>
<p>psychiatric illness experience these kinds of healthcare disparities there we</p>
<p>go so this is really sort of punctuated by the observation that the lifespan for</p>
<p>people with severe psychiatric illness or schizophrenia bipolar disorder but also neuropsychiatric disorders like</p>
<p>autism spectrum disorders and cognitive impairment on average the life span is</p>
<p>shortened by 10 to 12 and a half years and it’s thought that this is due to</p>
<p>many possible reasons one being a difference in the access to health care</p>
<p>so related to employment and insurance issues here in the states and that’s</p>
<p>certainly been shown to play a role particularly for for psychiatric illnesses such as</p>
<p>schizophrenia where there’s a much higher rate of homelessness but also</p>
<p>issues related to difficulty in expressing pain and so for example for</p>
<p>people who are non-verbal there’s also some suggestion that there may be</p>
<p>altered interoception or among people with neurodevelopmental disorders and so</p>
<p>if for anybody who’s not familiar with that term interoception is kind of your internal perception of pain discomfort</p>
<p>hunger thirst even sort of sensing your own heartbeat and and that appears to be</p>
<p>somewhat altered in people with developmental disorders but it’s also</p>
<p>certainly possible well that there’s some increased genetic or biological risk that’s actually related to the</p>
<p>genetic risk for for the psychiatric disorder itself so imagine sort of</p>
<p>pleiotropic mechanisms and then of course increase exposure to environmental risk factors poor diet</p>
<p>lack of shelter lack of access to medication and those kinds of things and</p>
<p>I think it’s also important to highlight that this is really a an under studied area for a couple of primary reasons</p>
<p>actually one is that until fairly recently a lot of people with severe psychiatric illness and</p>
<p>neurodevelopmental disorders received most of their health care in an institutionalized setting whether that</p>
<p>was a psychiatric institution or or prison or some other kind of group</p>
<p>institutionalized setting that was separated from community-based health</p>
<p>care and so there’s this population has been under studied in the kind of</p>
<p>epidemiological and community-based studies and then there’s also as I’m</p>
<p>sure everybody on this call is probably well aware has been kind of a historical</p>
<p>separation between psychiatry and the rest of Medicine and so it ends up being functional separation</p>
<p>so that refers to the fact that you know</p>
<p>often the mental health facilities are separate from the primary care</p>
<p>facilities and you might have like a community-based mental health clinic and a community-based primary care clinic</p>
<p>and even at a hospital there’s often a separate psychiatric hospital you know</p>
<p>in a completely different building from the the general hospital so there’s this</p>
<p>functional and cultural and financial separation of psychiatry and medicine</p>
<p>that I think has also contributed to this kind of healthcare disparity and in</p>
<p>the lack of research about it okay and</p>
<p>so like many of us in this field this is also a very an area of personal interest</p>
<p>for me so that this is a picture of me with my son Dylan Dylan is 24 years old</p>
<p>he has autism and severe cognitive impairment he’s nonverbal he requires</p>
<p>24-hour support staff and he lives in a community integrated group home and so</p>
<p>even with a whole team of people who are therapists on Dylan’s health we still</p>
<p>really struggle to get proper healthcare for him and so you know this is I think</p>
<p>an issue that it’s important to me personally as it is to many families of</p>
<p>individuals with developmental disorders and our psychiatric disorders who are</p>
<p>getting older and aging and we don’t really know what chronic health</p>
<p>conditions they may be at risk for so</p>
<p>while on the whole this has been a you know an under studied area there have</p>
<p>been a few kind of areas of focus particularly related to cardiovascular</p>
<p>and cerebrovascular disease and in recent years there have been some really</p>
<p>large meta analyses looking at the prevalence and cumulative incidence of</p>
<p>cardiovascular disease in people with severe psychiatric illness primarily</p>
<p>bipolar disorder schizophrenia and major depression and so this is actually a</p>
<p>recent paper with the meta-analysis of I think 92 studies the had really</p>
<p>impressive sample size so over three million patients with one of those three</p>
<p>disorders and and over a hundred million controls where they investigated the the</p>
<p>increased risk of both cardiovascular disease and cerebrovascular disease in these populations and so basically the</p>
<p>the take-home is that there is significantly increased risk for both disorder for both circular and</p>
<p>cardiovascular disease and that this risk persists even after adjusting for</p>
<p>some of the health behaviors but that may be related to to the incidence of</p>
<p>disease so things like smoking or poor diet or BMI and so based on these kinds</p>
<p>of epidemiological studies that are just now starting to be published we also are</p>
<p>interested in asking the same kinds of questions and better understanding the</p>
<p>relationship between men of mental and physical health using our EHR so in</p>
<p>particular we want to understand if there is some shared biology if it’s the</p>
<p>case that for mental health causes for physical health or that for physical health causes for mental health and</p>
<p>really our model is that it’s it’s gonna be all of the above but understanding</p>
<p>you know what the primary risk factors are for for each type of chronic disease and</p>
<p>where their shared biology yeah I think this is going to be an important area of research another question that we have</p>
<p>is do these relationships between mental health and physical health transcends our diagnostic boundaries is it really</p>
<p>because that it’s just people with severe psychiatric illness or diagnoseable more developmental</p>
<p>disorders that are who are at risk or is it the case that risk is continuous and</p>
<p>and that actually across the entire spectrum of genetic risk for these traits that there’s also increased risk</p>
<p>for these chronic health conditions and</p>
<p>then are there particular health conditions for which people with developmental disabilities and severe</p>
<p>psychiatric illness are at high risk there’s been a lot of work done on cardiovascular disease and cerebral</p>
<p>vascular disease but and and I think actually a fair bit of work on type 2</p>
<p>diabetes but outside of those kind of primary chronic health conditions um there’s there’s really not been very</p>
<p>much and so we really want to look genome-wide at these relationships and</p>
<p>then finally the wanna see if this can help us understand the best point of</p>
<p>intervention and and of course to identify any interventions that are typically used in healthy populations</p>
<p>that that may cause particular problems in patients with severe psychiatric</p>
<p>illness or developmental disorders so</p>
<p>A few of our research questions</p>
<p>maybe I’ll pause there for just a minute if there are any questions</p>
<p>and don’t forget you to unmute yourself if you have one and I mean one short</p>
<p>question - to move to the first slide so this is the shortening of the lifespan</p>
<p>for severe psychiatric diseases and so</p>
<p>the one thing that that was from my point was not missing on this list is</p>
<p>like suicide like big toe to to the psychiatric disease here is not the main</p>
<p>no it isn’t actually um it’s definitely a contributory cause but but actually</p>
<p>the primary causes chronic health condition so suicide is definitely a</p>
<p>increased risk in this population absolutely um but it doesn’t account so</p>
<p>you mentioned already this like like being facilities being also prisons etc</p>
<p>so and is there other comparisons</p>
<p>between for example also like the US and Europe are the comparisons between</p>
<p>Caucasians and other industries have</p>
<p>honestly the literature is pretty scant so I’m I’m not sure if that’s really</p>
<p>been investigated sort of systematically and with sufficient sample size um it</p>
<p>really hasn’t been I think most of the papers looking at that these</p>
<p>associations have been published you know from the 90s and on so there’s not</p>
<p>the kind of you know eighty year body of literature like there is for you know</p>
<p>cardiovascular disease and in healthy populations so I guess that yeah the</p>
<p>short answer is I’m I’m not sure that it’s really been systematically investigated</p>
<p>two others have some questions if not</p>
<p>they said they’d like two or three on the car that are not muted so if again</p>
<p>so this is question if you own the car not muted please mute yourselves like right contractors um so we believe that</p>
<p>EHR based genomic approaches are actually a great way of investigating</p>
<p>several of these questions that we have so the EHR the electronic health record</p>
<p>allows us to investigate the relationships between phenotypes and the</p>
<p>biobank that we have also facilitates investigation of the genetic relationships between theatre phenotypes</p>
<p>and so we can then also compare the the genetic correlations to the phenotypic correlation better understand you know</p>
<p>what is there where there are the environmental risk factors that contribute to the matific relationships</p>
<p>and genetic risk factors that contribute to the genetic relationships between traits and it also allows us to utilize</p>
<p>the polygenic architecture of these complex traits and develop quantitative models so that we’re not necessarily</p>
<p>relying relying completely on diagnostic categories but we can look at how</p>
<p>genetic risk as a quantitative trait is related to risk for various phenotypes</p>
<p>so maybe some of you have heard me talk about the Vanderbilt EHR and the bio</p>
<p>bank but just in case you haven’t it can be kind of thought of as three entities</p>
<p>we have what we refer to as the synthetic derivative which is this</p>
<p>de-identified and continuously updated mirror image of the EHR or EMR that as</p>
<p>of now has a little over 2.8 million individuals if we look across just that</p>
<p>set of 2.8 million individuals the median length of the EHRs is only about</p>
<p>a year even though the EHR has been in existence now for 20 years and part of</p>
<p>the reason for this is that we’re a tertiary care center so we get people coming in from all over the state of Tennessee and Kentucky and</p>
<p>and sort of all over the southeast particularly you know if they have a</p>
<p>they if they have the need to come see a special specialist in a specialty care</p>
<p>clinic um and so we end up drawing pretty sick people from all across the</p>
<p>state so in comparison to like the UK biobank that has maybe a healthier on</p>
<p>average population I think at Vanderbilt we have a sicker on average population</p>
<p>that said there is also a population of people who make Vanderbilt their medical</p>
<p>home so to speak and they get most of their primary care at the end result as</p>
<p>well and so our biobank which consists</p>
<p>of DNA samples that have been collected from just routine clinical blood draws</p>
<p>is enriched for that population of people but that actually make Vanderbilt</p>
<p>their medical home and so this is illustrated by the median length of the</p>
<p>EHR or bioview subjects which is about 10 years and so we have now somewhere</p>
<p>around 270,000 DNA samples that have been collected and a little over 50,000</p>
<p>of those subjects have been genotyped with some kind of G loss platform and on</p>
<p>average the age of those subjects is around 58 years old but we are trying to</p>
<p>genotype and accumulate pediatric care goals of all so any quick questions</p>
<p>about the structure of the bioview biobank or EHR that’s central to the</p>
<p>rest of the so this is like all different kinds of diseases right all</p>
<p>cases of these are they are also healthy subjects here or is this all cases well</p>
<p>I mean yeah so there are people who don’t have you know chronic diseases</p>
<p>certainly there are people who come in who’s just for you know routine health care and they’re in in the bio bank as well so</p>
<p>there’s no ascertain meant at the bio bank level that said there it’s a it’s a</p>
<p>hospital ascertained population so so I think probably most people will be you</p>
<p>know a case for something okay and so how many of these $50,000 psychiatric</p>
<p>patients how many of the 50,000 have</p>
<p>psychiatric codes I’m not exactly sure</p>
<p>yeah I actually don’t have access yet to all 50,000 samples it’s it’s not an it’s</p>
<p>not enriched for psychiatric cards that but but we’ve said you know the</p>
<p>genotyping data is actually it’s a large project eventually we’ll have a hundred thousand people genotype and so the the</p>
<p>data is coming through in waves and we actually were involved in pushing</p>
<p>several of the psychiatric diagnoses through but we haven’t gotten those</p>
<p>genotype samples yet so um so they’re right yeah we know where are they being</p>
<p>you know type here no on the mega</p>
<p>platform okay okay so as I mentioned</p>
<p>before we were very interested in taking advantage of the fact that most complex traits have a complex genetic</p>
<p>architecture with a measurable polygenic component and so we can look at how the</p>
<p>polygenic risk for all kinds of psychiatric disorders is also related to</p>
<p>disease status for other chronic health conditions</p>
<p>so I think probably everybody on the call is familiar with this approach but just in case one of the methods that</p>
<p>we’re using is basically to calculate koala genetic risk scores for everybody in our bio Bank so using some kind of</p>
<p>large discovery gos and taking the effect sizes from from that gos to</p>
<p>create a linear weighted sum of the number of risk alleles and then looking</p>
<p>at how those polygenic risk scores segregate cases from controls across a</p>
<p>number of different phenotypes genome-wide so we started with actually</p>
<p>investigating this and both psychiatric disorders and in some of the previously</p>
<p>published chronic health conditions and so I’m starting with showing you the</p>
<p>coronary artery disease polygenic risk scores that are significantly associated</p>
<p>with the EHR definition for coronary artery disease so here we took the beta</p>
<p>weights from the cardiogram plus C for D consortium study that had about sixty</p>
<p>thousand cases and 123 thousand controls and applied it to a small subset of our</p>
<p>mega data target sample just to make sure that we that indeed coronary artery</p>
<p>disease in carefully ascertained research samples was related to the HRV</p>
<p>diagnosis for coronary artery disease and so this included covariates median</p>
<p>age across the EAP our sexes the top 10 pcs and actually it’s not listed here but also genotyping batch and so you can</p>
<p>see that our apologetic risk score accounted for almost 3 percent of the variants and CID diagnosis within our</p>
<p>EHR so this was encouraging and we actually applied this model also to the</p>
<p>lipid traits that have been studied by the global lipids consortium so HDL LDL</p>
<p>triglycerides and try to also model the relationship the known relationships</p>
<p>between those risk factors and coronary artery disease oops</p>
<p>so I’m going to take just a short methodological detour here because one</p>
<p>of the other things that’s pointer my attention in working with the Department</p>
<p>for biomedical informatics you know often when I present telegenic analyses</p>
<p>I get the question from people could be a lot of machine learning where is your</p>
<p>feature selection step right so in your genome-wide Association study training</p>
<p>the the weights but then how do you know which snips to include in your model and</p>
<p>the target data and typically really what we often do is just work across a</p>
<p>number of thresholds and you know and see how the r-squared might change if we</p>
<p>in just the genome-wide significant snips or you know everything at a p-value of less than 0.5 or less than</p>
<p>one and so kind of investigator cost different thresholds but we wondered really how well we would do if we</p>
<p>actually took a training set and use that to select a threshold for inclusion</p>
<p>for including snips in the apologetic risk score and then applied it to a validation set because this is actually</p>
<p>I think you know ultimately what we would do if we were they’re really just</p>
<p>kind of out of curiosity we you know look at that now but we have our</p>
<p>Lipid traits and Coronary Artery Disease</p>
<p>discovery gos phenotype in the first column and the target phenotype and the</p>
<p>second column so the discovery gos either as the global the potential for the HDL LDL and triglycerides and you</p>
<p>can see my</p>
<p>this one person that is not muted or if it’s Yulia sure no it’s not me I heard</p>
<p>it too okay if we see you know me to see</p>
<p>your cars is okay all right great and so right so for for these lipid trades this was people and then our</p>
<p>target phenotypes were all measured in the bio Bank and so we we set up a</p>
<p>training sample of about 9,000 people and a validation sample oh yeah alright</p>
<p>so we had a training sample of about 9,000 people a validation sample of about 16,000 people and and so you can</p>
<p>see the key value threshold that was the best fit and the training sample the number of snips that was included in</p>
<p>that best friend we’re truly r-squared value or the proportion of variance I have to say and and the p-value for that</p>
<p>because I have no experience so I’m not</p>
<p>sure I mean they they seem to be like they don’t have the speaker on kro they don’t even listen so it’s a little bit</p>
<p>annoying so please everybody again here so if you hear us please mute yourself</p>
<p>or just if you don’t listen if you’re not listening just leave the call that’s</p>
<p>fine as well okay so we used the same in</p>
<p>fresh hold that was identified in the training symbol to define the PRS in the</p>
<p>validation sample there’s a you’ll notice there’s a different number of snips included and we think that this is</p>
<p>because the training sample was a me platform and the validation sample was</p>
<p>the mega platform even though they were both imputed to the same reference panel</p>
<p>and they’re both European populations the mega sample seems to have better</p>
<p>overlap with the original discover AG loss and so the are squirts for you know</p>
<p>both the training and validation sample particularly for the lipid traits are really pretty impressive and and</p>
<p>actually start to approach the Senate based heritability or in particular for</p>
<p>HDL and so I put an asterisk here for</p>
<p>the p-value fold in the training sample was actually the same as the best fit</p>
<p>p-value threshold in the validation sample so I can pause there for any</p>
<p>question we were actually kind of surprised that there were any traits where the best of it men told was the</p>
<p>same across multiple samples and so I think this is actually really</p>
<p>interesting and potentially indicating that we’re starting to approach kind of narcs and them information or an LDL in</p>
<p>our jaws</p>
<p>right in the interest of time I’ll just move on so we took the same best-fit</p>
<p>thresholds and genetics course you’re</p>
<p>looking at here is a Manhattan plot from us yaws where along the x-axis we have</p>
<p>kind of classified that the phenotypes and along the y-axis the minus log time</p>
<p>that the p-value for the PRS twin direction of the arrow indicates the</p>
<p>direction of risk so if there were higher polygenic risk scores among cases</p>
<p>this up arrow control to see a Down and so this view else was for LDL sorry for</p>
<p>HDL and so we see a protective effect of yell on type 2 diabetes and this has</p>
<p>been previously shown this was a good concept and we did a similar type of</p>
<p>analysis for our LDL polygenic risk score where again we see a strong</p>
<p>association with this lipid yneouws and</p>
<p>with coronary artery disease and so these are also known associations</p>
<p>interestingly so I don’t have it here and in this presentation but but there’s</p>
<p>a the phenotypic correlations between a chair between LDL and the diseases</p>
<p>tested in our genome wide analysis were much stronger so we saw lots of</p>
<p>different phenotypes associated with the measured LDL but these are the only phenotypes that are associated with</p>
<p>genetically predicted LDL so getting</p>
<p>into our primary interest which is the genetic risk for psychiatric disorders</p>
<p>associated with diseases across the phenome we asked</p>
<p>whether genetic risk for MDD was associated with heart disease codes and and actually all phenome wide codes and</p>
<p>so he took the most recent 2018 DD</p>
<p>meta-analysis results that are posted on the website and did the same kind of thing where we calculated a genetic risk</p>
<p>score and all of our 16,000 people in the mega example and we see really</p>
<p>strong associations with mood disorders and depression which we expect associations with bipolar disorder and</p>
<p>anxiety disorders but then we also see some of these party avascular rates</p>
<p>rising in significance as well and so we see an association with nonspecific</p>
<p>chest pain which is really kind of a catch-all you know as it states</p>
<p>nonspecific code but was definitely interesting to us and so we wanted to</p>
<p>kind of investigate this a little further and we asked whether we saw some</p>
<p>Do we see similar associations for related mental health traits?</p>
<p>similar associations for related mental health traits so related to major depression is also individual’s</p>
<p>perception of loneliness and this is something that we’ve been working on with a consortium group now a Palmer</p>
<p>Dorit blooms fund and myself and others would call ourselves the lonely</p>
<p>consortium and we’ve amassed almost 500,000 samples and I think those of you on the MDD call have heard some reports</p>
<p>of this already and so we looked at the relationship between polygenic risk scores for loneliness and phenom wide</p>
<p>associations as well and so this is our Manhattan plot for our loneliness</p>
<p>seawalker sorry gee wasps and you can see also that we are observing some</p>
<p>enrichment of gene expression for our loneliness low sigh within</p>
<p>tissues that we expect to see some enrichment so this is in brain tissues in particular and we also have observed</p>
<p>several genetic correlations and I just have a few posted here but genetic correlations with phenotypes that are</p>
<p>associated with poor mental health including general tiredness lower</p>
<p>self-rated health and coronary artery disease and one of the reasons that we</p>
<p>were really interested in and actually looking at the genetic relationship between loneliness in these other traits</p>
<p>is that loneliness has in and of itself and identified as a risk factor for</p>
<p>increased morbidity and mortality and and so there have been several kind of</p>
<p>epidemiological studies looking at the temporal relationship between a person’s self-reported loneliness and later</p>
<p>health consequences and so often the causal mechanism is inferred from that</p>
<p>temporal relationship but I think that’s a little tricky because of course you know by the time I’m somebody actually</p>
<p>has a heart attack or has diagnosis of coronary artery disease that that</p>
<p>disease has been developing for for many years and so having that temporal</p>
<p>relationship is not always an indicator of a cause-and-effect relationship so a</p>
<p>graduate student in my lab julia c-loc has kind of led the effort on this work</p>
<p>well she looked at again the innate propensity to loneliness apologetic risk scores for loneliness to see whether</p>
<p>they were associated also with poor health outcomes and so in this case because we didn’t have loneliness measured anywhere in our biobank we</p>
<p>weren’t able to do this kind of foot training on a separate sample for a best fit so we just took everything at a</p>
<p>p-value of less than one and even with that we actually see a strong association with mood disorders and</p>
<p>depression with tobacco use disorders but then also you can see with a whole host of coronary artery disease type</p>
<p>phenotypes and so this is also getting at that question that we had about the relationship between sort</p>
<p>of dimensional traits and and diagnosis itself so again these are not</p>
<p>necessarily so the sample is not enriched for major depression and we’re</p>
<p>not looking at the diagnosis of major depression our diagnosis of you know chronic loneliness or we’re just looking</p>
<p>at the genetic risk factors so we</p>
<p>BMI and MDD diagnosis influence associations</p>
<p>actually one of the I think benefits of having this type of data as opposed to just looking at genetic correlations</p>
<p>with summary stats is that we can do a lot of conditional and sensitivity analyses to try to tease out some of the</p>
<p>relationships and so we looked at how</p>
<p>BMI and diagnosis of major depression might influence associations and so when</p>
<p>we adjust for BMI definitely we see an attenuation of the signal for coronary</p>
<p>artery disease although we do still see some of these phenotypes rising above</p>
<p>you know might significance and we also see again a strong association with the</p>
<p>you know disorders depression and tobacco use disorders when we adjust for the diagnosis of major depression we we</p>
<p>still observe a significant association with obesity and and even though our</p>
<p>coronary artery disease codes kind of fall below phenome wide significance they’re still of course enriched among</p>
<p>our results and so this was actually a really important analysis because it’s</p>
<p>also known well-known that after having a coronary event people become much more</p>
<p>susceptible to a major depression episode and so so he wanted to make sure</p>
<p>that our associations weren’t completely driven by the major depression that may</p>
<p>be diagnosed after the fact and while we do see definitely an attenuation of the</p>
<p>signal I think we’re think resample sighs these associations will probably still remain</p>
<p>phenome aren’t significant we also were interested in seeing if</p>
<p>Differences between males and females?</p>
<p>there was a difference between males and females and so we stratified our few us</p>
<p>sample and and looked separately and while we do see some qualitative</p>
<p>differences so in females definitely the depression and mood disorder codes remain phenome like significant and in</p>
<p>males the mi and atherosclerosis codes</p>
<p>remain feel might significant these differences between them were actually not statistically significant so I think</p>
<p>this is really just reflecting the fact that more males have myocardial infarctions and more females</p>
<p>are diagnosed with depression Stephan</p>
<p>how am i doing on time good 10 to 12</p>
<p>minutes okay all right okay great so so</p>
<p>there’s since we were really primarily interested in following up the associations between polygenic risk for</p>
<p>loneliness and the coronary artery disease codes we we focused in on the males and and looked there at</p>
<p>Association after adjusting for again MDD or BMI and and so again we see that</p>
<p>myocardial infarction remains significantly associated after adjusting</p>
<p>for either BMI or MDD and actually also remains associated after adjusting for</p>
<p>both MDD and BMI and so this now I think</p>
<p>is a really nice substrate for a</p>
<p>Mendelian randomization analysis as well</p>
<p>Part 2: Strategy to screen for biomarkers in lab data</p>
<p>so the the second part here is is actually just kind of an introduction to some of this the work that we have</p>
<p>planned and so I just wanted to sort of briefly go over it and and invite any</p>
<p>ideas or collaborations of people are particularly interested in certain</p>
<p>biomarkers for phenotypes that they that they’re studying so within the EHR we</p>
<p>have access also to actually thousands of labs but many of those are kind of</p>
<p>unique special snowflakes so they may only have been ordered on you know a small handful of patients so when we</p>
<p>start looking at labs that have a larger sample size it turns out that we have</p>
<p>about 350 labs with over a thousand individuals and actually I should say</p>
<p>that all of these labs have at least a thousand observations so if we say that</p>
<p>you know we’ve got 350 labs with at least a thousand individuals at least a</p>
<p>thousand observations then everybody has been measured at least once but then</p>
<p>we’ve got a larger number of labs where we have a smaller number of individuals but a larger number of observations per</p>
<p>individual and so this data is also you know really rich for looking at</p>
<p>longitudinal associations between the relationship with you know psychiatric</p>
<p>illness and and changes over time in various biomarkers so in total we’ve got</p>
<p>about 500 labs with at least a hundred people measured at least a thousand</p>
<p>observations so this work has sorry this</p>
<p>Clinical lab data</p>
<p>data source really has not been utilized very much in the EHR space partially</p>
<p>because it’s really messy data so it’s taken us close to two years actually to to really carefully QC all of the the</p>
<p>labs that that had sufficient sample size it’s</p>
<p>also challenging because again it’s a hospital population and so you know a lot of times the the labs are being</p>
<p>drawn because somebody is actually sick and so this can be a challenge to interpretation but at the same time the</p>
<p>fact that we have the entire EHR allows us to investigate the relationship between diagnosis and and changes in lab</p>
<p>values some of the benefits of using this data are that we have a really</p>
<p>large sample size that is really rich for clinical data so it’s longitudinal</p>
<p>we’ve got over 20 years worth of data and we can as I mentioned a couple of</p>
<p>times now test the effect of many possible mediating and moderating variables and then we can also go into</p>
<p>the charts themselves and validate by chart with you so developing tools to</p>
<p>both QC and visualize this data has been a really tremendous effort by Peter</p>
<p>Straub a programmer in my lab so he’s developed a shiny app and a whole set of</p>
<p>tools that actually we’re planning on making publicly available and and will</p>
<p>be allowed we’ll be able to make the summary data for all of these different labs available so that we can look at</p>
<p>how they vary by age by sex by by race</p>
<p>and and that groups with bio banks and</p>
<p>and labs you know locally can also download these tools and and apply them to their data</p>
<p>and this is sort of the concept map for what we would eventually like to do so</p>
<p>this is you know that we would take the you know beta values from geo so many</p>
<p>psychiatric illnesses calculate polygenic risk scores across everybody</p>
<p>and bio view and then look at how those risk scores are related to median values</p>
<p>in hundreds of routinely collected labs and as I mentioned before we can</p>
<p>so use longitudinal models to see how they’re related to change in lab values</p>
<p>over time so we’ve started doing this a</p>
<p>little bit for some of these traits that I’ve been talking about so far so this</p>
<p>is just focusing in again on HDL LDL and triglycerides and looking at how the</p>
<p>genetic risk scores for coronary artery disease loneliness or major depression</p>
<p>is associated with median lab values and</p>
<p>so in each of these plots we’re looking at the r-squared values the proportion</p>
<p>of variance explained in HDL LDL and triglycerides on the y axis and then the</p>
<p>Discovery gos p-value threshold on the x axis so that we can see kind of across</p>
<p>the board how well do you know MDD risk scores CA D risk scores and loneliness</p>
<p>risk scores predict HDL LDL and triglycerides and so interestingly we</p>
<p>see that for HDL our loneliness risk scores actually tend to outperform our</p>
<p>CA D risk scores and for LDL you know it</p>
<p>looks like there’s some difference but I don’t think these are actually really meaningful because below 0.1% variance</p>
<p>it’s not not a significant association with the between the risk score it’s</p>
<p>often and the median value and then in our triglyceride analysis we do see the</p>
<p>the CAG risk score outperforming the loneliness risk score which is kind of what we would think about maybe</p>
<p>intuitively expect but the loneliness risk score does actually significantly</p>
<p>predict triglyceride levels as well</p>
<p>so like I’ve said a couple of times we</p>
<p>also are interested in testing for the mediating effect of these quantitative</p>
<p>traits and the moderating effects of sex and medications and and other diagnoses</p>
<p>on these relationships as well so this is kind of our an example of a general</p>
<p>model that we’re interested in testing the relationship between polygenic viability and disease diagnosis that may</p>
<p>be again mediated through quantitative traits so kind of our future interests</p>
<p>are to mine this lab data for biomarkers for our psychiatric disorders using genetic risk scores from them are</p>
<p>publicly available jaws and I should say that you know we don’t think it’s likely</p>
<p>that we will identify a biomarker for you know major depression out of all of</p>
<p>these routinely collected labs but that actually we may identify by several quantitative traits that together may be</p>
<p>predictive of an individual’s genetic twist or form a major depression or</p>
<p>schizophrenia for example we also are really excited</p>
<p>about using bi-directional Mendelian randomization analyses to better understand some of these possible causal</p>
<p>mechanisms and to compare the phenotypic correlations with the genetic</p>
<p>correlations to try to identify comorbidities that may actually be more of a consequence of environment then of</p>
<p>genetic causes which you know we’re also equally interested in understanding so I</p>
<p>Acknowledgements</p>
<p>think that is it I’ll wrap up with an acknowledgement of everybody in my lab a</p>
<p>really great group of people to work with and wonderful students and the work</p>
<p>of the lonely consortium again this has been like a just a phenomenal</p>
<p>collaboration that I think has yielded some really interesting results and so I</p>
<p>think with that all and and hopefully there’s some time the questions thanks so much the other</p>
<p>fabulous presentations really it’s a lot of data that’s coming - therefore - it’s</p>
<p>really exciting to see that you’re diving into that with full speed here um</p>
<p>so is there I mean we have like around 30 people on the call we probably have</p>
<p>questions and they don’t know how to unmute themselves or I still have</p>
<p>questions but I don’t want to always step in did I hear somebody</p>
<p>okay so then here I still have likes like two or three night more short</p>
<p>questions I’m on the technical side so so two things here so first of all and these are you a because you’re speaking</p>
<p>about to do know stuff but also is there</p>
<p>chance to actually give them the fact you might have seen some people with</p>
<p>especially high score screen your result is their chance to get in contact with</p>
<p>them again on the next visit or so or is this something that’s so enormous there’s no chance to to actually get in</p>
<p>contact with these individuals no they’re yeah it’s it’s a completely de-identified data set yeah there is</p>
<p>also a what’s called the research derivative so we work with in the synthetic derivative because we’re also</p>
<p>working with the genetic data if we if we were to restrict ourselves just to</p>
<p>the phenotype associations we could apply for access to the research derivative which is an identified data</p>
<p>set but even then I think there are rules I don’t know that that we could</p>
<p>actually recontact patients and and there’s no although I don’t understand</p>
<p>it why this is the case you can’t work with genetic data in the identified</p>
<p>environment well I think most of the bio</p>
<p>banks within the US are de-identified that that issue about not working with</p>
<p>genetic data in the context of an identified environment that might be specific to Vanderbilt I’m not actually</p>
<p>sure what the what the you know rules around that where those</p>
<p>came from very interesting and a little</p>
<p>bit sad of course because that would be the and so the other issue a little bit</p>
<p>more technical side I mean like special project with scores are actively sensitive to so how is there</p>
<p>any chance for you or how do you want these dealing with this issue like yeah</p>
<p>you know that it’s a good question we’ve been thinking about this a lot so so</p>
<p>there’s a method I can’t remember now the first author’s name but I think it came from from Peter Fisher’s group to</p>
<p>in the context that we’re using where we have some individual level data and geo</p>
<p>summary statistics they’ve they’ve published an approach to actually first</p>
<p>just check to see if there is yeah what the probability of overlap actually is</p>
<p>everything up until now have to be honest is based on just the knowledge</p>
<p>that no investigators have contributed actively from Vanderbilt to these G wast</p>
<p>but I mean we haven’t yet done the careful checking to make sure that</p>
<p>there’s no sample overlap but that is on our list yeah it’s it’s really something</p>
<p>that I think we have to just actually baked into the pipeline and and we’re</p>
<p>also part of a collaboration with other</p>
<p>bio bank sites through the emerge Network where we’re doing a similar kind</p>
<p>of thing you know replicating these polygenic few associations in other bio</p>
<p>banks and so so we’re really trying to develop a you know a robust pipeline for</p>
<p>for doing that across several EHR environments and and one of the things</p>
<p>we need to make sure to build into the pipeline is this kind of checking</p>
<p>especially when did you see it’s good for for just testing reasons doesn’t try to</p>
<p>see if we see some unexpected overlap there and we don’t and this gives you a</p>
<p>little bit more security that yes like 200 cases coming up that we shared from</p>
<p>somebody else to PTC something that you need to be super careful at least did</p>
<p>you see everything in place we could test that we couldn’t test done even without you know we have to check some</p>
<p>method where we can without sharing the genotype so it’s something that could be</p>
<p>done on a relatively low level so so in</p>
<p>a genome-wide type of analysis how would you then basically like for each fee</p>
<p>wass category conduct a G was--and check all of those some stat relationships</p>
<p>it’s also possible that we have say somebody with schizophrenia who doesn’t</p>
<p>actually have the diagnosis in our electronic health record right they have</p>
<p>a diagnosis of type 2 diabetes in our health record but maybe they’ve been inclusive in a geo somewhere else so so</p>
<p>I think you know I mean first we like</p>
<p>the check on the summary statistics level works for like a significant</p>
<p>overlap for for if you have the same field I think you’re raising exactly</p>
<p>right questions that I think if you want to be really sure you need to really</p>
<p>test it on a genotype and the burning at</p>
<p>least you can do this like with a couple of these consortiums you at least</p>
<p>meaning if there’s something unexpected or something expected and I think this gives already some support here and</p>
<p>maybe when you actually see the summary statistics level when you see actually</p>
<p>some significant overlap there Fineman somewhere then you you probably</p>
<p>want to go there ask these guys for can we actually test this on genotype 11 who are these individuals this is actually</p>
<p>correct is this I think we can test this on a couple of different levels yeah yeah that’s that’s a good idea</p>
<p>and that’s basically what we wanted to start with was just to see first like do we see any evidence of overlap and then</p>
<p>if we do you know how should we go about dealing with it I mean to be honest the</p>
<p>overlap cannot be really really big because then it’s so quickly we’ll see</p>
<p>with the exploding a square values that’s really impressive even if like if</p>
<p>you have a small data set like individuals all of these it’s just 300</p>
<p>cases red ones are exploding so if there</p>
<p>is something little small thing can have</p>
<p>this small impact on these on these on these that seemed to fit so well right</p>
<p>yeah so do you have an intuition of like what you expect if there’s an overlap of</p>
<p>say you know a dozen people you know like a really small overlap no this is</p>
<p>always something that I really wanted to test yeah somebody wants to go for it please no I don’t have that I have these</p>
<p>experiences that it’s like a couple hundred overlap it’s really it’s really</p>
<p>strong so much surprise surprisingly strong really that’s why I’m always</p>
<p>careful about this it’s not that you see dying are spread this is just jumping a little bit around it’s really exploding</p>
<p>and so that’s why I think my hunch is that even like it does know 20 could</p>
<p>have a small I mean some of the p-values I really like just passing the spin of white</p>
<p>trash out there right so that’s why especially for these cases yeah</p>
<p>okay but that’s the cheapest keep this ominous this discussion and I’m very</p>
<p>happy to it to be yeah and and so for</p>
<p>you know anybody on the call who’s interested in looking at particular phenotypes and we’ve we’ve now see us</p>
<p>apologetic scores for all of the publicly available summary stats from PG</p>
<p>C and and so you know anybody who’s like</p>
<p>interested in talking more about what schizophrenia looks like or what you know bipolar looks like or whatever</p>
<p>please feel free to get in touch with me because I would love to kind of have you know more folks to discuss these with</p>
<p>I’m sure the people will be more grave</p>
<p>okay so we are actually over the hour so I will have to close to the call now thanks so much for presenting yeah</p>
<p>thanks and this will continue here so</p>
<p>all the best</p>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
          // target, if specified
          link.setAttribute("target", "_blank");
          // default icon
          link.classList.add("external");
      }
    }
});
</script>
</div> <!-- /content -->



</body></html>