<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.353">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>PGC Video Textbook - Chapter 2.3: Evolutionary signatures (Video Transcript)</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


<link rel="stylesheet" href="styles.css">
</head>

<body class="nav-sidebar docked nav-fixed">


<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="./index.html">
    <span class="navbar-title">PGC Video Textbook</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link active" href="./index.html" rel="" target="" aria-current="page">
 <span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./about.html" rel="" target="">
 <span class="menu-text">About</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./contact.html" rel="" target="">
 <span class="menu-text">Contact us</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="https://pgc.unc.edu/" rel="" target="">
 <span class="menu-text">PGC Website</span></a>
  </li>  
</ul>
            <div class="quarto-navbar-tools ms-auto">
</div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
      <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item">Chapter 2.3: Evolutionary signatures (Video Transcript)</li></ol></nav>
      <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
      </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal sidebar-navigation docked overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
      <a href="./index.html" class="sidebar-logo-link">
      <img src="./pgc_logo_website_v3.jpeg" alt="" class="sidebar-logo py-0 d-lg-inline d-none">
      </a>
      <div class="sidebar-tools-main">
    <div class="dropdown">
      <a href="" title="" id="quarto-navigation-tool-dropdown-0" class="quarto-navigation-tool dropdown-toggle px-1" data-bs-toggle="dropdown" aria-expanded="false" aria-label=""><i class="bi bi-github"></i></a>
      <ul class="dropdown-menu" aria-labelledby="quarto-navigation-tool-dropdown-0">
          <li>
            <a class="dropdown-item sidebar-tools-main-item" href="">
            Source code
            </a>
          </li>
          <li>
            <a class="dropdown-item sidebar-tools-main-item" href="">
            Report a bug
            </a>
          </li>
      </ul>
    </div>
</div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./welcome.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Welcome to the PGC Video Textbook!</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./toc.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Table of Contents</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">
 <span class="menu-text">Chapters</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter1.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 1: Introduction</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter2.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 2: The Genome</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter3.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 3: Technologies</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter4.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 4: Study designs</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter5.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 5: GWAS analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter6.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 6: Polygenic Scores</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter7.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 7: Ancestry-Specific Analyses and Considerations</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter8.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 8: Post-GWAS bioinformatics</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter9.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 9: Advanced Topics</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter10.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 10: Other Considerations</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true">
 <span class="menu-text">Software Tutorials</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_cnvs.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">CNVs</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_conditional.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Conditional Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_crossdisorder.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Cross-disorder Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_ewas.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">EWAS</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_geneset.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Gene Set Identification</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_gwas.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">GWAS</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_genomicSEM.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Genomic SEM</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_imaging.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Imaging</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_MR.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Mendelian Randomization</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_prs.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">PRS</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_correlation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">SNP Heritability and Genetic Correlation</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true">
 <span class="menu-text">Additional Resources</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./glossary.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Glossary</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./resources.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Software Resources</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./addreading.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Additional Reading</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="https://pgc.unc.edu/for-researchers/download-results/" class="sidebar-item-text sidebar-link">
 <span class="menu-text">PGC Summary Statistics</span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#sec-origins" id="toc-sec-origins" class="nav-link active" data-scroll-target="#sec-origins">Origins of Genetic Variation</a></li>
  <li><a href="#sec-selection" id="toc-sec-selection" class="nav-link" data-scroll-target="#sec-selection">MPG Primer: Natural selection &amp; human genetic variation</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Chapter 2.3: Evolutionary signatures (Video Transcript)</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<hr>
<section id="sec-origins" class="level1">
<h1>Origins of Genetic Variation</h1>
<p><strong>Title</strong>: Origins of Genetic Variation</p>
<p><strong>Description</strong>:</p>
<p><strong>Presenter(s)</strong>: Jessica Pamment, Professor, DePaul University</p>
<p>[Music]</p>
<p>Next time you’re in the classroom look around you and you’ll see that although all your peers to humans the same species as you know to individuals in the class will look exactly the same unless you have identical twins in the room this variation in traits is true not only for human population but for any species some of the differences observed within a population are caused by the environment and experiences of each individual for example hormonal changes brought on by cooler temperatures result in the fur of an arctic fox turning from brown to white although the environment definitely plays a role in introducing variation in a population most of the variations seen in populations are caused by differences in genes for example one gene is responsible for determining where the rats will have brown or black fur with exception of clones such as identical twins each individual within a population carries a unique set of genes half of which were received from one parent and half from the other the total set of genes of all individuals in a given population is called the gene pool a gene is a discrete unit of hereditary information consisting of a specific nucleotide sequence in DNA so nucleotides are the building blocks of DNA and therefore of genes differences between individuals can be measured all the way down to the level of individual nucleotides however measuring differences within a gene pool at this level is not particularly useful because much of the variation lives within non-coding regions of the DNA meaning that these variations don’t result in an observable difference it’s often better to measure variation at the gene level because there is at this level that both quantitative and discrete traits are coded so how does genetic variation arise in a population well one of the ways is as a result of mutation which result in a change in the original DNA sequence mutations can occur as mistakes during DNA replication however if the mutation does not happen in a Cell that is passed down to offspring such as an agro sperm cell the change cannot lead to nearly Oh which is an alternative version of a gene variation Canosa rice at the chromosome level during the process of meiosis this is a modified type of cell division found only in sexually reproducing organisms which results in the production of gametes the two ways in which variation is introduced in meiosis crossing over an independent assortment crusted OVA happens early on in meiosis in prophase one and results in the exchange of DNA between homologous chromosomes so between the paternal and maternal chromosome of each chromosome pair this results in recombinant chromosomes the second way in which variation is introduced is as a result of the random arrangement of chromosome pairs on the cell plate during metaphase one in humans the random assortment of chromosomes gives rise to over 8.4 million possible combinations of chromosomes and this is without taking crossing event account which introduces even more variation another mechanism that contributes to genetic variation in sexually reproducing organisms is random fertilization as I just mentioned in humans each male and female gamete represents one of about 8.4 million possible chromosome combinations due to independent assortment the fusion of a male gamete with a female gamete during fertilization is completely random and will produce a cell with any of about 70 trillion chromosome combinations if we factor in variation brought in by crossing over the number of combinations is even higher hopefully you can see how unique you really are now that we’ve learned how genetic variation is introduced into a population of sexually reproducing organisms it isn’t to remember the evolutionary significance of this natural selection is a driving force behind evolution and natural selection results and the accumulation of genetic variations favored by the environment another way of thinking about this is that genetic variation is the raw material needed for evolution to occur</p>
<p>[Music]</p>
<hr>
</section>
<section id="sec-selection" class="level1">
<h1>MPG Primer: Natural selection &amp; human genetic variation</h1>
<p><strong>Title</strong>: MPG Primer: Natural selection &amp; human genetic variation</p>
<p><strong>Description</strong>:</p>
<p><strong>Presenter(s)</strong>: Stephen Schaffner, Computational Biologist, Broad Institute</p>
<p>good morning hello everyone thank you for coming I have the privilege of introducing myself I’m Steve Schaffner I’m a staff scientist here computational biologist and I’ve been here for a long time I used to be part of mpg back in the thousand genomes and half map Iran beating before that these days I work more in pathogens malaria and viruses but I’m still interested in human genetics and natural selection humans which is why I’m talking to you today about natural selection and in humans and particularly its effect on human genetic variation and what we can learn from from this and how we can detect it so I think you probably all know what natural selection is I’ll just stated for the record that it’s the principle that alleles that make an organism more</p>
<p><strong>Natural selection</strong></p>
<p>successful in terms of survival and reproduction are likely to be transmitted more to the next generation and therefore likely to increase in frequency while they’re being selected and we can for convenience we can divide the kind of selection into several different kinds one first is balancing selection which is selection that maintains multiple alleles in the population at some intermediate frequency and there’s purifying selection which is the elimination of new mutations that are deleterious and finally there’s positive selection there’s a selection for some beneficial trade and actually the last two are really sort of flip sides of the same things you’re selecting for one trait you’re selecting against some other trait you only have you have to be choosing one but it’s convenient to distinguish them based on what’s the but starting out is rare you can think of purifying selection as eliminating new rare things and I’ll start by talking about balancing selection this is probably the rarest kind of selection it’s kind of cool and you can find it but it’s a little difficult to detect and probably doesn’t happen very often there is one very well-known case in humans sick of the sickle cell trait sickle cell allele which was one of the first cases probably I</p>
<p><strong>Sickle cell</strong></p>
<p>think is the first case of identified natural selection in humans it it was holding in the 50s noticed that certain diseases involving hemoglobin were much more common in places where there was a lot of malaria and yep hypothesis that there was natural selection was playing some sort of role and he was correct and these two maps the top map shows where malaria occurs worldwide and the bottom map shows where there’s a high prevalence of the sickle cell variant sickle cell trait and the reason is is quite clearly quite clear even it’s easy</p>
<p><strong>Heterozygous individuals survive best</strong></p>
<p>to find out to check you can actually measure the difference in Fitness if you have one copy of if you’re heterozygote if you have one copy of this allele then you have concern against malaria so if you have no copies you’re exposed to malaria or likely to die younger because malaria is a big killer if you have one copy you have a real benefit if you have two copies then you get sickle-cell disease and you’re also likely to die young because without modern medical care it’s very serious disease and so you could look at the the survival this was without that kind of medical care you look at the survival it’s clearly clear that that hair is I gets in this case have an advantage and that means that there’s selection pressure to maintain this allele at some intermediate level so that the maximum number of heterozygotes in the population it’s not very pleasant solution but it is an evolutionarily stable solution in toilets with severe selective pressure you can get balancing</p>
<p><strong>Balancing selection selection for diversity</strong></p>
<p>selection in other ways by selection for diversity there are a lot of cases in which it’s good to be different from other members of your species like escaping from predators if the predator is used to finding a purple people and eating them if you’re green that’s it can be kind of good to be green then and it’s quite common in resistance disease if a new virus enters the village and it can infect most of the people in the village and you’re different and you know have a different genotype than most of the people then you have an advantage so the whole thing that that’s spreading through the rest of halation you’re immune from and you can see the effects of this of the selection for diversity in terms of disease resistance in the HLA region which is critical for immune response to</p>
<p><strong>Selection for diversity HLA</strong></p>
<p>pathogens and if you just look on chromosome six that the density of snips in that region it’s much higher there than elsewhere in the genome because there’s a lot of selection obviously we’re constantly exposed to different kinds of infectious disease so there’s a lot of selection for having diversity there and there’s the HLA region if it couldn’t guess where it was and this is actually an example of frequency dependent selection that is whether decree allele is ’event Aegeus depends on how frequent it is when it’s rare it’s good to be different so that’s advantageous if it increases in frequency then it becomes it can become less advantageous and also it obviously can vary quite a lot with your local environment what pathogens happen to be nearby or what predators and in humans it’s more mostly pathogens we worry about not so much predators these days so it can fluctuate quite a bit on small geographic scales and on temporals short temporal time scale as a side note pathogens also evolve and they do similar some similar things this is one of the chromosomes of Plasmodium falciparum which causes the most severe kind of malaria and where that sharp spike in diversity is showing diversity across the chromosome that short sharp spike in diversity is a gene that codes for protein that’s exposed to the immune system a lot of malaria proteins are not exposed because they hide in red blood cells but this this protein is exposed to the human immune system and again there’s a lot of pressure to be diverse so that if you are the first parasite entering a village you want to be different than all the other parasites that the populations have been exposed to so that you can happily infect people without being you know having that unpleasant immune system triggered immediately and they also they’ve all faster than we do by the way so they’re constantly evolving as well so that’s balancing selection purifying selection is the most common kind of selection it’s sort of a little dull because all it is is its removal of new Dilla Tereus mutations most organisms are pretty well adapted to their environment and a functional change you know something that changes their phenotype in an important way is probably bad for them it’s gonna be eliminated by natural selection it’s not gonna be passed on for very long and that’s it’s very clear to see this in humanity that humanity netic data this is a plot of the top figure here is from thousand genomes data it’s the diversity across the whole genome looking at different parts of genes and I’ve worked where some exons are this percents on the middle exon and the last exon within genes throughout the genome and you can see the diversity is much lower within these coding exons because changes to the to a protein are probably bad and they tend to be eliminated and you can see it see that the effect is strong and just read off from there how many mutations have occurred in these exons and that have been eliminated by selection over time and although they’re eliminated they may not be eliminated immediately obviously if something makes you non-viable then you’re not going to see in the population but lots of mutations are mildly d latias they might make you more likely be sick or stupid or less attractive whatever which might be bad in certain circumstances and so you can see these these alleles are hanging around the population but they they won’t rise to very high frequency</p>
<p><strong>Purifying selection eliminates deleterious mutations</strong></p>
<p>because they are bad and they’re less successful so if you just plot the allele frequency of nonsynonymous mutations and compare it to synonymous mutations that’s when I show here on the left as a function of frequency in the 1% being that first bin you see there’s an excess of nonsynonymous mutations compared to the synonymous mutations because they look more of these or functional and so that access represents mildly deleterious allele that are going to be eliminated eventually by selection but haven’t been eliminated yet even within the nonsynonymous you can break it down further on the right has plotted different categorization of nonsense mutations as to whether or not whether or not they’re likely to damage the protein has changed the effect of the protein so in pink and red those are changes that are have a pretty good chance of changing the proteins function and those are the ones that are see more of them at low frequency so those are the the ones are going to be eliminated so in terms of evolutionary biology these are not interesting this is going to sludge that’s against being eliminated by a natural selection all the time in terms of medical genetics or human genetics of disease these are probably a lot of the ones that were interested in because these are one of the ways of being dilatoriness it does it makes you more likely to get sick and so these are some of the things that are causing genetic diseases or increasing your risk for early onset diseases of various kinds so there are great medical interest in some cases but not of tremendous interest to evolutionary biologists the effect of purifying selection can be seen not only in the allele itself that’s that’s being selected against it can also be seen on some of the surrounding variation the effect of purifying selection is that it reduces diversity in that region and you can think of like there’s a say there’s a gene where deleterious mutations keep happening when Julie - this mutation happens that from the chromosome it happens on is going to be removed from the population eventually and so effectively it’s not part of the population now in terms of the long term success so you have a right around genes or other functional elements you effectively have a smaller population size and that means you can sustain less diversity because some of the diversity gets taken you know any diversity that’s on that chromosome that gets the bad mutation is going to be removed from the population and it’s very easy to see this - this again a thousand genomes data here they plotted the distance from the beginning from the start or the stop of a gene across the whole genome the three colors are three different populations and you can see in what’s plotted this diversity and there’s a dip significant dip in diversity around a gene it’s on the order of fifty or hundred KB so it’s a substantial stretch where there’s a notable reduction in diversity and this is you know affecting the diversity distribution of diversity throughout the genome this is an ongoing effect why why is the absolute level well there’s just less diversity the red and the blue are the two non-african populations and there’s just less diversity outside of Africa because they pass through a bottleneck leading Africa and so there’s isn’t that isn’t that much there there may also be a small difference in how much diversity is reduced there there’s no question of whether you’re pure of like selection has been less effective outside of Africa because the effective population size was smaller but that’s a pretty minor effect and finally there’s positive selection which is what we usually think of as selection and it’s natural selections but sir what Darwin was famous for it’s the basis for adaptation pretty much all adaptation we think and so it’s kind of a sexy thing to look for and people have looked for it I’m gonna focus it initially on selective sweeps which is you know it’s selection where the mutation starts out basically as a new mutation and increases in frequency I’ll add some few complications later and so the question is gonna help what’s the effect of positive selection on genetic variation how is what that means how can we detect it just by looking at genetic variation and so let’s take let’s take the out of the case where a selection doesn’t happen this is neutral evolution so suppose there’s a mutation that happens in this blue guy here and as time goes on the frequency of that allele it may increase it may decrease it kind of bops around a little bit it’s it drifts and technical to detect alarm is it drifts its genetic experiences genetic drift but on average it doesn’t actually change in frequency so very slowly it might change over time if there’s positive selection for something if this new allele this mutation provides a benefit well then over time it can rapidly increase in frequency and that’s that rapid increase is what leaves the genetic signature that we can we can look for it relieves a number of different genetic signatures and I’ll kind of describe some of them so let’s consider what the genetic situation looks like before selection happens and after so if we this is a cartoon version of some chromosomes in the population this new red mutation is beneficial and there’s a lot of diversity there there are different alleles you have different combinations of those different haplotypes present in the population because it’s just been sitting yeah look it’s just me behaving normally after selection has happened this has this allele has increased rapidly in frequency and is present in a large fraction of the population and so this signature we can look for well one signature happens suppose this selection has happened only in one geographic region like it happened in I know the Boston area something that’s I know being a Red Sox fan me or maybe even exude there’s an allele for that might make you more reproductively successful here let’s compare it to New York and you know if that’s the case then you will find that allele very at very high frequency in that region and very low frequency elsewhere so it’s unusual one signature of selection then is that there’s an unusually large difference in frequency between populations at that locus and this can indeed be seen here both I’ve just mentioned that one way of measuring a common way of measuring frequency differences in populations is a statistic called FST but there are other statistics too you can use and a classic example of this is is the Duffing olive oil it’s the Duffy protein is a blood antigen one of the mini blood antigens sits on the outside of red blood cells and does something or other not entirely clear one but one one of us because it’s not intended for it’s a is that it’s also the way by which Plasmodium vivax another cause of malaria enters red blood cells and so it’s the only way of entering the but red blood cell so it’s critical for an invasion and for infection and if you lose that protein then you are pretty much immune to by vechs and malaria and if you there is a mutation that knocks out that gene it’s called the Duffy null allele and audits shown there is the density the the frequency of that mutation around the world and see it’s a very high frequency in sub-saharan Africa and as a consequence people in sub-saharan Africa are largely immune effects of malaria so much so though there’s almost over maximal area across Africa so this was a highly successful case of natural selection in humans suggesting that there was a very large cost from the my max malaria at some point and outside of Africa it’s it’s basically not there at all and so this is sort of a classic case is in fact I believe this is how it was discovered that this allele that this protein was the invasion route for this this parasite so it’s a it can be a useful to be looking at natural selection a more recent case in terms of studying it come</p>
<p><strong>High altitude adaptation in Tibet</strong></p>
<p>came from people who are looking for adaptation for higher handling high altitude or there was very low oxygen and so what this research group did was they compared allele frequencies between a Tibetan population and sample from Tibet and a han chinese sample very closely related populations very similarly low frequencies so bottom of two axes are the Tibetan frequency and the Han Chinese frequency and you can see almost everything has this pretty much the same frequency in both populations and there are two alleles that are very high frequency in tibet and not at very high frequency in the han chinese and they’re on the low to in the lower right hand corner there are two alleles in the same gene you pass one and it turns out this doesn’t does indeed confer adaptation to handling hand up into low oxygen levels present there so this was a very easy way of finding this particular gene so it’s in principle this is a powerful way of detecting where selection has happened regionally in practice there aren’t very many low-hanging fruit like that somebody took the trouble of plotting of comparing you know how how many real outliers are there they took a whole bunch of different population pairs so each one of these dots is a pair of populations and on the x-axis is the average difference in the little frequency measured by FST between those two populations and on the y axis is y axis shows the most extreme single allele in that pair of populations and most of the time you can predict really well what the extreme is going to be just from what the averages so all you’re seeing is sort of the tail the distribution up at the high end there are a few cases here up here where there there are clearly outliers the one I was just talking about you pass one is is one of them all the other colored dots here are known pitov pigmentation genes so it’s not a there’s actually information in there but by itself it may not be a very easy way of finding out what’s been selected all right so that’s that’s that’s just the first signature I won’t focus much about all the others another signature is low diversity if everyone or a large fraction population is the same now around this new selected allele well that means if you’re the same you’re not different there’s not a lot of genetic diversity there and so if you just plot generic diversity across chromosome you’ll find regions where diversity has dips because there’s been a selective sweep there so what that means is that you know I said purifying selection produces reduced diversity around of functional elements positive selection can also reproduce to reduce diversity so they have somewhat similar signatures here which is inconvenient the signature the the loss of diversity can be more profound in the case of positive selection because of positives we if a sweep goes to all the way to fixation everybody is the same and so there’s virtually no diversity present another thing that’s a little bit easier to look for and to serve another a different way of looking at the same thing is that if everyone is the same you have the same haplotype they’re basically if you can predict if you have that red allele you can predict what other alleles everyone will have in the population up to the point where recombination is broken and down and if it’s if this election happened recently then recombination hasn’t had time to break it down yet and so you can look forward on long haplotypes that aren’t at high frequency in the population and there’s a whole series of statistical tests that have been developed for detecting that there’s an unbroken long haplotype present and this turns out to be a very powerful way of detecting selection that’s happened within the last 20,000 years roughly and here a</p>
<p><strong>Long haplotype: LCT</strong></p>
<p>very very clear signature force in a signal selection can be seen at one of the now poster children for selection humans which is lactase persistence there’s a normal state from amell’s is that as you get older you lose the ability to digest lactose because lactose is present mostly only in milk and normal normal mammals don’t drink milk as adults but in human populations where they practiced hurting for a long time you know many adults can in fact digest lactose they’re lactose tolerant most Northern Europeans I’m lactose tolerant because I’m from my ancestors of northern Europe and if you look around the lactase gene there is a an enormous Li long haplotype that’s hardly broken at all and in Northern Europe it’s around 70 percent frequency it extends for more than a megabase is result a very strong natural selection positive selection for this trait and the the mutation that gives the biscuit for digesting lactose is in fact on that half of the time it’s just plot just shows that if you plot the strength of you know this is the strength of the length of the haplotype versus the frequency like lactase sits way up here on the right this is the strongest signal selection by this test in in Western Europe okay finally one other signature that’s a little bit harder to visualize is that in regions they’ve been under undergoing recent positive selection there will be an excess of high frequency quran derived alleles normally the the derived allele the newer the newer allele is at low frequency it’s occur it tends to stay at low frequency but when this kind of sweep happens towards higher frequency it can bring any other mutations that are nearby to higher frequency and so it can bring more of these these rare mutations up to high frequency along with it so that’s a different just a different sort of thing you can look at in data and all these signatures have been looked at they’ve been known for a while and they’ve been looked at and various looking throughout in sweeps scans across the genome looking for places where selection has happened and you can do some interesting things with them it turns out that there’s independent information in each of these</p>
<p><strong>Composite of Multiple Signals (CMS)</strong></p>
<p>signatures and so one thing you can do and it’s an approach that was pioneered by pretty savetti here is combine the information of the different signals and here you know in a cartoon version we’ve got all the evidence from a long haplotype evidence from dr allele frequency and evidence from differentiation in populations and they’re they’re giving a different information but the actual selected one you know where a selection actually happened it will be you can get enhanced information about that which is important because some of these signatures tend to cover very large regions of a chromosome and not really tell you very much information about which allele or ease in which gene was actually selected for so if you want to get and downs with details then it’s better to combine information and here’s an actual case I think this is</p>
<p><strong>CMS pinpoints candidate variants</strong></p>
<p>chromosome 5 in humans and I don’t remember what data set this is it might be a few thousand genomes these are different signatures of selection the top one is a long haplotype second one is population differentiation and then the bottom one is derived alleles and a lot of these are raised or elevated so somewhere in here there’s evidence that selection happened but they’re very noisy and it’s very hard to know exactly where the selection happens but if you combine them then you might as if by magic the bottom distribution shows the the score for where you think selection happened and if you see it they’re only a handful that has elevated score indicating that right here is where selection probably happened and it turns out the biggest signal there is a nonsynonymous mutation so it was a good chance it’s it’s functional it’s sitting in a gene that’s important for skin pigmentation this in fact is an allele that that contributes to European skin color that’s one of the major alleles for that good question isn’t how strong it is you know I’d say with four really strong signals 100 is fine for as you’re getting to more subtle things thousands are better above that the problem isn’t so much sample size as knowing being able to distinguish between background stuff and what’s actually selection it’s not just not a statistical problem it’s if anytime basically we’re doing is we’re looking for the weirdest part of the genome and lots of weird stuff can be happening in the genome and there’s other various things that confound these you know for instance the long haplotype test that can be founded if there’s an inversion where that suppresses recombination in that region and I think in you know that’s one of those sorts of things that can confuse you so there’s a variety of things that can’t affect you so I said these it’s tests have been known for a while there was a big spate of genome scans about ten or twelve years ago as data you know genome-wide data became available but there is still work going on in this area this is a figure from my paper that was published two weeks ago in science which a new signature for selection for</p>
<p><strong>A new signature singleton density</strong></p>
<p>recent selection which I thought was cool enough but I will try to explain it to you the basic idea is that if so we’ve got two alleles here at a site each of the along the bottom pointer there we go well no these are all the samples and if you just construct a gene tree of which which samples are related to which at this particular locus yeah and this is the gene tree and is that this in blue here is a derived allele that the mutation happened at some point in the past and it’s been selected for and if it’s selected for its increased in frequency and what that means is that that it’s younger that on average you have more recent common ancestors and if you compare people they have a recent common ancestor that’s what it means for it to have increased in frequency recently and so if you just look at the the terminal branches the terminal lines here leading to all of these samples they tend to be longer for the one that wasn’t selected for because and any mutation that occurs on one of these is going to appear as a singleton in your sample it’s just that’s one mutation so all this test does is is count how many Singleton’s there are around each you look on the little eel from the genome just count how many Singleton’s there are nearby and look at the density of single things and then the the density will be lower fewer Singleton’s around recently selected things and one of the nice things about this test is that unlike some of the other tests I’ll say in a minute it works on not just selective sweeps but it works in some other more complicated situations and it’s according to the authors who they they’ve figured out you know what is this sensitive to if you have a reasonable sample size of like a few thousand then it’s probably sensitive to select it’s occurred in like the last 2000 years so in genetic terms that’s a very recent selection so this is this is cool and it’s worth reading that paper ok so I think there are some complications selective sweep is nice and pure if you have one but it’s making some assumptions the some assumes that the beneficial mutation occurred once or at least was so rare you treated as just there was just a one copy and then that rises to frequency and has all these effects and it as soon as there’s just that that one it’s one mutation of pretty large effect but suction can happen in other ways a suction can occur on standing variation that’s variation the variants that have been in the population for a long time you know if it’s been there for half a million years or a million years recombination has been happening all the time so that that variant is hitting on all lots of different haplotype backgrounds they’re all going to rise in frequency you’re not going to see most of these signals the same mutation can happen more than once that’s happened in the sickle cell trait if this identical mutation has occurred on multiple occasions been selected for but you’re in and I’ve again have multiple haplotypes increasing in frequency and finally there may be lots of different alleles that contribute to the same trait and each one may only shift a little bit in frequency and you can still have a substantial effect on that trait and so these signatures of selection or are pretty much hopeless there are are attempts to use sort of modifications of these tests while I’m looking at allele frequency distributions in the case of selection on the standing variation but it’s just harder than signature the the signal is isn’t isn’t as easy to find you can still do some interesting things though and find some interesting stuff if you stop thinking just about alleles but</p>
<p><strong>Focus on traits rather than alleles</strong></p>
<p>rather than think about the trait if you know what genes or what alleles that are contributing to variation in a trait then you can you know aggregate different alleles that are involved in that trade and so this is an example that was carried over studied to carry out by Joel Hirsch orange group here or here in Harvard and looking at height sure in europeans we know from aji wasps many of variants that affect stature we also know just from looking at europeans that there’s a client in height across europe from north to south Northern Europeans tend to be taller than southern Europeans and so what they did in this figure they plot they ordered they ranked the snips from the G wasps in terms of how large an effect it had each snip had hung stature and then on the y-axis they plot the difference in frequency between northern Europe and southern Europe and what they find is that the alleles that have the biggest effect on the stature are also the ones that have the biggest frequency difference in Europe and this is consistent across you know all of the you know the major alleles here they all have higher frequency in Europe this isn’t just randomly a few alleles happen to be higher all of them have higher frequency in the increased height direction in northern Europe or alternatively there all have you know the lower stature allele in southern Europe and this is pretty good evidence and it’s been substantiated by further studies by others that selection was acting on stature in Europeans in some way not it not clear exactly how but in some way in others there have been other studies this is a similar sort of study looking at a variety of traits correlated with geography in this case is finding a significant selection for stronger UV stronger damage respond stronger response to damage from ultraviolet radiation if you live near the equator if your population comes from your new equator which isn’t maybe isn’t too surprising but the different colors are different continents and on different continents the seems of selection pressure has been happening so you can indeed extract a fair bit of information about the trait that’s been selected for even if you may not necessarily know which particular allele has been selected so that leaves the question what traits have been selected for what have we actually found what’s been going on in humans in the last 20,000 years or so because that’s pretty much all the data I’m talking about or from selection that’s occurred within the last twenty or twenty to forty thousand years because that’s where that’s the easy place to look lots of interesting selection happened before like what made us anatomically modern humans but that’s a lot harder to find or to study so we’re studying all the relatively easy things so as I said you can scan the genome for these signatures of selection and many people did here’s a result of one of those scans because it was one one done here and here are a lot of places in the genome where there’s</p>
<p><strong>Results of a selection scan</strong></p>
<p>evidence that selection happened there probably some false positives in there but there are lots of actual cases of selection occurred there and of questions what do they do what was involved and the answer is mostly we have no idea something happened there somewhere and we don’t know what the trade was we don’t know what the selection pressure was and it’s just sort of going from something happened at this locus to what’s you know the phenotype is hard work that’s biology that’s not you know sitting around looking at that was genomes data you actually have to do some biology then and I’ll have one example here of</p>
<p><strong>From candidate to function: EDAR</strong></p>
<p>the kind of work that can can elucidate this sort of thing this was this is a little segment of one of those genome scans this is chromosome 2 in East Asians and this this is a test for long haplotype so it’s very high in high score here across this entire region indicating that selection happened here where exactly you know probably somewhere in here but you can’t really tell little relative to say the size of the single gene so a long run and also the sheer number of variants here is a problem if you want to see who was one of these was one of these variants had a phenotypic effect actually drove this because one of these variants was probably actually important and the rest didn’t so if you’re now knowing what to test for say do some functional work this by itself is kind of daunting this is just too many variants there to put into a model organism say and see born into a cell and see what it did but you can do the trick I mentioned before if combining information from multiple Cygnus signal signatures and that actually cleans up this particular locus very nicely there are only a handful then candidates you want to look at and one of those turns out to be a nonsynonymous change and as I said you can see that sorry here you can see that there are in fact multiple genes across this region and that’s a month it was announced anonymous gene in nonsynonymous change in a particular gene known to be involved in development of hair and sweat glands it’s known have been under selection in other organisms so that then gives you something you can you can look at in more detail you can take that variant and the study it in the lab and somebody did a postdoc at Harvard and working with a broad and I</p>
<p><strong>Mouse model EDARV370A</strong></p>
<p>said this was selection that parently happened in East Asia this increased in frequency and and she stuck it in a mouse this particular variant stuck it in a mouse and seal so what it did and well it did several different things it produced thicker hair it produced smaller and denser memory glands and it produced a higher density of eccrine sweat glands and it turns out that at least the the first and third of these the hair in the sweat glands this has the same thing as a big affected humans and the thicker hair is actually even is something you can actually see you know the East Asian hair doesn’t look the same as European hair say typically so this is this is a mutation that was selected for its illustrative in that by focusing in on something that was selected for we found something that clearly has a notable phenotypic effect on humans that distinguishes between humans it’s also illustrates one of the problems which is that mutations don’t always have one effect and so an affected hair and affected sweat glands and we don’t know why you can guess well it maybe it’s something temperature regulation and sweat glands are important or maybe hair was important we don’t know so it’s not you know a final answer but it was a major clue to finding something that’s phenotypically important and that distinguishes humans from one another if you look more broadly this is a this is from a very recent review article Sarah Tishkoff is group in science of what we’ve learned about selection in regional selection in humans and you can see on the map we’ve learned a number of different things and several of the cases I’ve talked about are on this map and I’ll just mention them selection for changes to diet the lactase persistence case up there in the Europe that’s one of the you know classic ones you can’t find lactase when you’re doing something wrong and you’re studying a selection but that’s also been studied in East Africa and the Middle East where there are other have been other hurting populations turns out to be independent mutations that in the same regulatory region that have the same phenotypic effect so it’s exactly the same pathway exactly the same mechanism but that’s occurred in multiple times in different places there are other populations around the world that have lactase persistence in South Asia in West Africa and these have not been studied in any depth at this point so exactly what the mechanisms are there I mentioned skin pigmentation there are a bunch of genes that are known to where alleles are known to contribute to the paler skin color that Europeans have and which varies with with latitude and so they’ve been well studied other genes are known to have mutations in Asia some of them independent some of them are shared so as different but partially different set of mutations other parts of the world where there’s also variation in like within Africa there’s quite a lot of variation and probably someone has been under selection but again the studies haven’t been done there there’s plenty of things still to study I mentioned the polygenic selection on on stature in Europe turns out there’s also been selection on stature elsewhere in in rainforest environments the selection for smaller stature in humans pygmies that tend to have small stature well known in the Central African rainforest and selection there has happened is is operated in a rather different way in Europe it was very polygenic selection on lots of different alleles in Central Africa it apparently was strong selection operating on a handful of low sigh for shorter stature might be because the pressure was of a different kind it may just be that’s the way it happened to work don’t really know and I think I think the last one is circle here is the one for high altitude and a patient that’s been studied in Himalayas it’s also been studied in other high places in the Ethiopian highlands and in the end these and you find in independent mutations most of them in the same pathway leading to similar phenotype so the same selection pressure different mutations but similar outcome I don’t know much about the thinking it might just be a matter of resources of you’re better off you know if you need to eat less because it’s not a very highly nourishing environment but I don’t really it’s not something I’ve looked at in any detail so that you know that those are the cases we’ve understood something about we don’t always necessarily know exactly what with exactly what the selection pressure was but least we have some idea what the phenotype might be one question that might occur to people is well how much of this is there how much has natural selection been operating in humans recently and it turns out to be kind of tricky to figure out because alright so there’s you know</p>
<p><strong>How much positive selection is there, anyway?</strong></p>
<p>differences frequency differences between populations it varies across the genome how much of that is naturally occurring and how much of that is I mean is neutrally occurring how much of that is the result of selection especially since we don’t really know the demographic history of humans in detail we can’t exactly model it and tell you what the distribution should look like so it’s kind of an open question of how much positive selection has happened and one way of trying to address this is to look for look at that reduction in diversity like this particular reduction of diversity around the genes and I said that can be caused either by selective sweeps happening repeatedly in positive selection or by background selection so you need to find a way to to distinguish them and so one group what they did was they looked at genes so we’re only looking at gene so this is presumably background selection is happening all the time and they compared cases where there’s been a nonsynonymous change in humans over the last million years or so the places where there’s only been a nonsynonymous change and the idea is if if they’ve been a lot of selective sweeps you should see reduced diversity around the nonsynonymous case and here’s their bluff from their of their paper this is Hernandez at all from 2011 since a few years ago now here’s where the substitution happened at some point and they plot diversity for synonymous in blue the substitutions and for around nonsynonymous substitutions in red and those distributions are basically the same there’s no obvious difference between the nonsynonymous and the synonymous and their conclusion was we don’t see any evidence for lower</p>
<p><strong>Conclusion 1</strong></p>
<p>diversity around functional changes classic selective sweeps have no I mean they’re sure they’ve been some but they’re not a big you know they’re not a major factor maybe it’s been you know this polygenic selection other kinds of selection are happening and basically said the conclusions you should stop wasting your time looking for these things which was a little annoying for those those who were looking for them in particularly those of us who wanted to get funding to look for them because then you know this you know the reviewers of the grant proposal say why are you looking for this Hernandez it all just showed that there aren’t any selective sweeps or you know we’ve already found them all don’t don’t waste your time and then a couple years later there was another paper from a different group Davida Nord Petrov Petrov group groups and they came to a very different conclusion I think I’ve time to go through this and they said they looked at the same data and they said no you’ve drawn the wrong conclusion because the problem is you’re you’re looking at nonsynonymous mutations well let’s think about that a little bit let’s think about you’ve got two genes gene a is highly constrained every amino acid is a precious jewel and if you change one of these is bad so anytime there’s a missense mutations deleterious there you will find very few missense mutations that happen in neutrally so if there’s any mutations in missense mutations they will have been beneficial and what that means is lots of background selection happens there’s lots of purifying selection going on all the time because all of these mutations are bad so if you look at diversity around that gene well it’s going to be very low diversity around that gene now gene be in the other hand this is a weakly constrained gene there’s a lot of them a lot of the amino acids don’t do anything so a lot of neutral changes and consequences have got to be less background selection because they’re just there’s fewer opportunities for deleterious mutations and so around those genes there’s going to be you know a smaller reduction in diversity and there’s what they say is that when you look at nonsynonymous me to pick and nonsynonymous substitutions that’s happened which is what the Hernandez a tall group did you’re picking primarily from this group from these genes on the other hand when you’re picking nonsynonymous mutations you’re picking from both of them so you’re biased yourself to finding places where there’s not a lot of change and so when if there are cases a lot of cases of positive selection going on selective sweeps you’re going to completely lose it found because of this bias and so their conclusion was that there’s actually strong</p>
<p><strong>Conclusion 2</strong></p>
<p>evidence for lots of selective sweeps in recent human history there’s been a high rate of strongly adaptive substitutions near amino acid change and has been even more sweeps and driven around regulatory changes which I think you know independently so that same data very different conclusions and a little heartening to those of us who are interested in selection I mentioned that I introduced that new test for selection is this density of Singleton’s they conclude that lots of different traits if you just accept their data at face value lots of different traits quantitative traits in humans that have been studied under by G wasps a lot of them show evidence for having been undergoing selection in the last 2000 years like more than half I don’t know if I believe that but it’s certainly a bit there’s Evans has suggested that lots of slow selection of some kind is going on all the time ok so that’s so that’s where we are in terms of studying selection from just looking at genetic variation and what its effect is but there’s one other topic that’s really interesting because it gives us a whole new way of looking at this and that’s ancient DNA because in all these things were inferring what happened in the past all these kinds of studies we look at genetic just variations in a bunch of Norwegians and you try to figure out what happened ten thousand years ago but now we can actually look at DNA from ten thousand years ago five thousand years ago and see directly what’s changed between then and now and people have done this we’ve now are at the point where we’ve sequenced enough ancient genomes we can compare our little frequencies in the past to allele frequencies now and this is data from a</p>
<p><strong>230 ancient genomes</strong></p>
<p>paper of last year David Reich’s group at Harvard and this is plots the colored dots are frequency of several particular alleles that are of interest in ancient about several ancient populations and the dashed lines are there what the frequency is in modern modern populations modern European populations is basically Europe and West Asia there are these ancient populations and I’ll point out a couple cases that the top left one is lactase again this is this classic example you can see that the frequencies in these ancient populations was at or near zero the the allele that gives you the ability at just lactose where as in its modern southern European populations it can is still fairly low but in northern European populations it’s very high so this is selection that’s happened just in the last few thousand years you can see that it’s been dramatically increased in frequency or at that time here on the other hand is of pigmentation allele one of the major ones that contributes to a European paleness and you can see that was at different levels and different populations it was they were apparently the steppe peoples these are the people of the Western steps in Asia they were apparently probably paler than than these other Europeans down here but they’re they’re all a lower frequency than and in modern Europeans so this is a selection that was ongoing at this time and has continued into the present and if you look across the genome this is the I don’t know if you’re used to Manhattan plots but the entire genome you spread out and the places where there are signature signals that suction has happened just from comparing the ancient DNA to the modern DNA you find a lot of the same things you find out that there’s skin pigmentation genes that are already known they find the dye of genes including lactase and like fatty acid dehydrogenase which is has been seem to be under selection and other populations selection for resistance to infectious disease at the HLA at Toolik receptors so the basic story we get from ancient DNA is it’s very similar which is</p>
<p><strong>Ancient DNA tells a similar selection story</strong></p>
<p>heartening because you know we were reconstructing the past based on my computer models and it’s nice to see that you actually when you actually can look at the ancient DNA it tells you we were right in a lot of these cases we were really incorrectly inferring that selection happened but you get a lot more ten get a lot more detail when you look at the ancient DNA because our model our computer models are simple so I’ll give you a few examples of the difference in the story and to end here you know I talked about this this gradient of stature in Europe and there’s this north-south gradient so we concluded selection happened well when looking at the ancient DNA they concluded a little more detail they conclude that selection happened for shorter statute in stature in southern Europe that selection for taller stature happened in West Asia in these step populations released that’s where we see it and that we see this greater height in northern Europe because those people then moved into northern Europe it wasn’t necessarily selection happening in northern Europe through a greater height it was just people having to migrate in and you know looking at modern popular modern population we have no idea we would have no idea about this it’s very hard to figure out all of these you know where all these people have been moving around we tend to assume that well if we’re looking at you know Chinese people today that’s that’s who you know their ancestors were living there 10,000 years ago no people move they move a lot turns out you’re look at ancient DNA sometimes in ways that we couldn’t tell it all from modern DNA all right a second case the pigmentation there are two main genes that contribute to European pigmentation typical color with very similar names slc24a5 and SLC four five A two and we know from the genetic evidence modern Europeans that they’ve been both in under strong selection risen there’s an allele that’s risen to high frequency in both cases turns out the two genes have somewhat different histories looking at the ancient DNA one of them actually it’s the one I just showed this skin pigmentation gene boys you can see rising in frequency within Europe as selection was occurring for lighter skin the other allele actually entered the Europe at very high frequency with farmers when the first farmers moved in from Anatolia Evelyn’s mom Turkey they largely replace the European population and they already had the lighter skin presumably as the result of earlier selection but you know you get a much more detailed picture of the history of this one and finally that the last case is the case of Adar and the one thing that gives you more sweat glands and thicker hair well you know when we were studying it the story was it seemed pretty simple this plot shows where that allele this this sucker lily is is present in East Asia and as president the new world because people from eastern Asia populated the new world and so it was clear you know you can do lots of they did very detailed modeling for this your model ahead was clearly this is estimating where this allele originated and originating in central China 30,000 years ago so it was a nice simple story there was a cell paper and it’s a great paper but this particular conclusion the problem is you look at the ancient DNA it turns out this allele was at high frequency and Swedish hunter-gatherer 6,000 years ago which is not something you would guess from looking at the modern distribution and it says I mean it’s it was stolen it was under selection in East Asia and that that story hasn’t changed but the details of what happened historically if something come more complicated happened and probably arose in Western Asia and happened to be selected for an eastern Asia so and this is the kind of information to get from it really is a tiny snapshot of ancient DNA as we get more DNA we’ll be able to learn a great deal more at least from places where there is ancient DNA a lot of the world my own tropics DNA does not preserve well I’m gonna conclude with a couple of comments so recent positive selection it’s clearly has had an impact a significant impact on human phenotypic diversity both individuals and within groups exactly how much you know some traits many traits at least two traits have you know been changed by this and many of these traits are medical interest or biological interest it’s a great way of finding out one way of finding out where these inter important phenotypic changes are so if you study natural selection you can learn you identify places where things have changed it’s not by itself it’s not a certainly not an all-purpose tool it’s it’s a clue it has to be combined with functional work with jiwa you know with association studies with all kinds of other things but it is one one tool in the toolkit else out there thanks</p>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
          // target, if specified
          link.setAttribute("target", "_blank");
          // default icon
          link.classList.add("external");
      }
    }
});
</script>
</div> <!-- /content -->



</body></html>