<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.353">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>PGC Video Textbook - Glossary</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


<link rel="stylesheet" href="styles.css">
</head>

<body class="nav-sidebar docked nav-fixed">


<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="./index.html">
    <span class="navbar-title">PGC Video Textbook</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link active" href="./index.html" rel="" target="" aria-current="page">
 <span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./about.html" rel="" target="">
 <span class="menu-text">About</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./contact.html" rel="" target="">
 <span class="menu-text">Contact us</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="https://pgc.unc.edu/" rel="" target="">
 <span class="menu-text">PGC Website</span></a>
  </li>  
</ul>
            <div class="quarto-navbar-tools ms-auto">
</div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
      <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./glossary.html">Additional Resources</a></li><li class="breadcrumb-item"><a href="./glossary.html">Glossary</a></li></ol></nav>
      <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
      </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal sidebar-navigation docked overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
      <a href="./index.html" class="sidebar-logo-link">
      <img src="./pgc_logo_website_v3.jpeg" alt="" class="sidebar-logo py-0 d-lg-inline d-none">
      </a>
      <div class="sidebar-tools-main">
    <div class="dropdown">
      <a href="" title="" id="quarto-navigation-tool-dropdown-0" class="quarto-navigation-tool dropdown-toggle px-1" data-bs-toggle="dropdown" aria-expanded="false" aria-label=""><i class="bi bi-github"></i></a>
      <ul class="dropdown-menu" aria-labelledby="quarto-navigation-tool-dropdown-0">
          <li>
            <a class="dropdown-item sidebar-tools-main-item" href="">
            Source code
            </a>
          </li>
          <li>
            <a class="dropdown-item sidebar-tools-main-item" href="">
            Report a bug
            </a>
          </li>
      </ul>
    </div>
</div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./welcome.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Welcome to the PGC Video Textbook!</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./toc.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Table of Contents</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">
 <span class="menu-text">Chapters</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter1.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 1: Introduction</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter2.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 2: The Genome</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter3.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 3: Technologies</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter4.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 4: Study Designs</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter5.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 5: GWAS Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter6.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 6: Polygenic Scores</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter7.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 7: Ancestry-Specific Analyses and Considerations</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter8.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 8: Post-GWAS Bioinformatics</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter9.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 9: Advanced Topics</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter10.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 10: Other Considerations</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true">
 <span class="menu-text">Software Tutorials</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_cnvs.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">CNVs</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_conditional.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Conditional Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_datasets.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Datasets</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_ewas.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">EWAS</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_geneset.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Gene Set Identification</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_liftover.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Genome Builds</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_gwas.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">GWAS</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_genomicSEM.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Genomic SEM</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_MR.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Mendelian Randomization</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_mtag.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">MTAG</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_prs.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">PRS</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_correlation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">SNP Heritability and Genetic Correlation</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true">
 <span class="menu-text">Additional Resources</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./glossary.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">Glossary</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./resources.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Tools/Software Resources</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="https://pgc.unc.edu/for-researchers/download-results/" class="sidebar-item-text sidebar-link">
 <span class="menu-text">PGC Summary Statistics</span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#a" id="toc-a" class="nav-link active" data-scroll-target="#a">A</a></li>
  <li><a href="#b" id="toc-b" class="nav-link" data-scroll-target="#b">B</a></li>
  <li><a href="#c" id="toc-c" class="nav-link" data-scroll-target="#c">C</a></li>
  <li><a href="#d" id="toc-d" class="nav-link" data-scroll-target="#d">D</a></li>
  <li><a href="#e" id="toc-e" class="nav-link" data-scroll-target="#e">E</a></li>
  <li><a href="#f" id="toc-f" class="nav-link" data-scroll-target="#f">F</a></li>
  <li><a href="#g" id="toc-g" class="nav-link" data-scroll-target="#g">G</a></li>
  <li><a href="#h" id="toc-h" class="nav-link" data-scroll-target="#h">H</a></li>
  <li><a href="#i" id="toc-i" class="nav-link" data-scroll-target="#i">I</a></li>
  <li><a href="#j" id="toc-j" class="nav-link" data-scroll-target="#j">J</a></li>
  <li><a href="#k" id="toc-k" class="nav-link" data-scroll-target="#k">K</a></li>
  <li><a href="#l" id="toc-l" class="nav-link" data-scroll-target="#l">L</a></li>
  <li><a href="#m" id="toc-m" class="nav-link" data-scroll-target="#m">M</a></li>
  <li><a href="#n" id="toc-n" class="nav-link" data-scroll-target="#n">N</a></li>
  <li><a href="#o" id="toc-o" class="nav-link" data-scroll-target="#o">O</a></li>
  <li><a href="#p" id="toc-p" class="nav-link" data-scroll-target="#p">P</a></li>
  <li><a href="#q" id="toc-q" class="nav-link" data-scroll-target="#q">Q</a></li>
  <li><a href="#r" id="toc-r" class="nav-link" data-scroll-target="#r">R</a></li>
  <li><a href="#s" id="toc-s" class="nav-link" data-scroll-target="#s">S</a></li>
  <li><a href="#t" id="toc-t" class="nav-link" data-scroll-target="#t">T</a></li>
  <li><a href="#u" id="toc-u" class="nav-link" data-scroll-target="#u">U</a></li>
  <li><a href="#v" id="toc-v" class="nav-link" data-scroll-target="#v">V</a></li>
  <li><a href="#w" id="toc-w" class="nav-link" data-scroll-target="#w">W</a></li>
  <li><a href="#xyz" id="toc-xyz" class="nav-link" data-scroll-target="#xyz">XYZ</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Glossary</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<p>Below are common terms and acronyms used within this book and across the field of Psychiatric Genetics.</p>
<hr>
<section id="a" class="level1">
<h1>A</h1>
<table class="table">
<colgroup>
<col style="width: 20%">
<col style="width: 70%">
<col style="width: 8%">
</colgroup>
<thead>
<tr class="header">
<th>Term</th>
<th>Definition</th>
<th>Reference chapter</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Admixture</td>
<td>Genetic ancestry that comes from distinct populations</td>
<td></td>
</tr>
<tr class="even">
<td>Allele</td>
<td>Variant forms of a gene that occupy a specific position on a chromosome.</td>
<td></td>
</tr>
<tr class="odd">
<td>Antisense</td>
<td>A molecule or strand of nucleic acid that is complementary to a specific RNA sequence, often used in gene regulation.</td>
<td></td>
</tr>
<tr class="even">
<td><strong>A</strong>rea <strong>u</strong>nder the ROC <strong>c</strong>urve (<strong>AUC</strong>)</td>
<td>A measure of a classifier’s ability to distinguish between classes in a binary classification problem. AUC is between 0.5 (no prediction) and 1 (full prediction)</td>
<td></td>
</tr>
<tr class="odd">
<td>Assembly (Genome)</td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td>Autosome</td>
<td>Any chromosome that is not a sex chromosome, responsible for carrying genetic information unrelated to sex determination.</td>
<td></td>
</tr>
</tbody>
</table>
</section>
<section id="b" class="level1">
<h1>B</h1>
<table class="table">
<colgroup>
<col style="width: 12%">
<col style="width: 85%">
<col style="width: 2%">
</colgroup>
<tbody>
<tr class="odd">
<td>Base pair</td>
<td>The fundamental unit of DNA and RNA, consisting of two nucleotides held together by hydrogen bonds.</td>
<td></td>
</tr>
<tr class="even">
<td>Biallelic (SNP)</td>
<td>Single Nucleotide Polymorphism with two possible alleles at a specific genomic position.</td>
<td></td>
</tr>
<tr class="odd">
<td>Bonferroni adjustment</td>
<td>Statistical correction to account for multiple hypothesis testing, by applying a p-value correction that corrects for the number of statistical tests performed</td>
<td></td>
</tr>
<tr class="even">
<td>Build (Genome)</td>
<td></td>
<td></td>
</tr>
</tbody>
</table>
</section>
<section id="c" class="level1">
<h1>C</h1>
<table class="table">
<colgroup>
<col style="width: 40%">
<col style="width: 56%">
<col style="width: 2%">
</colgroup>
<tbody>
<tr class="odd">
<td>Candidate Gene Study</td>
<td>Research focusing on specific genes to determine their association with a trait or disease.</td>
<td></td>
</tr>
<tr class="even">
<td>Chromatin</td>
<td>Complex of DNA, RNA, and proteins forming the chromosome’s structure.</td>
<td></td>
</tr>
<tr class="odd">
<td><strong>Ch</strong>romatin <strong>I</strong>mmunoprecipitation <strong>Seq</strong>uencing (<strong>ChIPseq</strong>)</td>
<td>Technique to identify DNA regions bound by specific proteins using antibodies.</td>
<td></td>
</tr>
<tr class="even">
<td>Chromosome</td>
<td>Thread-like structure in cells containing DNA and genes.</td>
<td></td>
</tr>
<tr class="odd">
<td>Codon</td>
<td>Three-nucleotide sequence in mRNA that encodes a specific amino acid during protein synthesis.</td>
<td></td>
</tr>
<tr class="even">
<td>Common variant</td>
<td>Frequently occurring genetic variant in a population.</td>
<td></td>
</tr>
<tr class="odd">
<td>Congenital</td>
<td>Present at birth, often referring to medical conditions or traits.</td>
<td></td>
</tr>
<tr class="even">
<td>Contig</td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td><strong>C</strong>opy <strong>DNA</strong> (<strong>cDNA</strong>)</td>
<td>DNA synthesized from an mRNA template, used in molecular biology research.</td>
<td></td>
</tr>
<tr class="even">
<td><strong>C</strong>opy <strong>N</strong>umber <strong>V</strong>ariant (CNV)</td>
<td>Variation in the number of copies of a DNA segment between individuals.</td>
<td></td>
</tr>
<tr class="odd">
<td>Crossover</td>
<td>Exchange of genetic material between homologous chromosomes during meiosis.</td>
<td></td>
</tr>
</tbody>
</table>
</section>
<section id="d" class="level1">
<h1>D</h1>
<table class="table">
<colgroup>
<col style="width: 14%">
<col style="width: 79%">
<col style="width: 5%">
</colgroup>
<tbody>
<tr class="odd">
<td>Deletion</td>
<td>Removal of a segment of DNA, leading to the loss of genetic material.</td>
<td></td>
</tr>
<tr class="even">
<td>Duplication</td>
<td>Copying of a DNA segment, resulting in extra genetic material.</td>
<td></td>
</tr>
</tbody>
</table>
</section>
<section id="e" class="level1">
<h1>E</h1>
<table class="table">
<colgroup>
<col style="width: 33%">
<col style="width: 63%">
<col style="width: 2%">
</colgroup>
<tbody>
<tr class="odd">
<td>Effect estimate</td>
<td>Statistical measurement of the impact of a factor on an outcome in research (e.g.&nbsp;odds ratio, beta value)</td>
<td></td>
</tr>
<tr class="even">
<td>Elastic Net Regression</td>
<td>Statistical learning method combining Lasso and Ridge regressions to select relevant variables</td>
<td></td>
</tr>
<tr class="odd">
<td>Enhancer</td>
<td>DNA region influencing gene transcription.</td>
<td></td>
</tr>
<tr class="even">
<td>Epidemiology</td>
<td>Study of the patterns and causes of disease and health in populations.</td>
<td></td>
</tr>
<tr class="odd">
<td>Epigenetics</td>
<td>Study of heritable changes in gene expression not involving alterations in DNA sequence.</td>
<td></td>
</tr>
<tr class="even">
<td>Epigenome</td>
<td>Overall epigenetic modifications in an organism’s DNA.</td>
<td></td>
</tr>
<tr class="odd">
<td><strong>E</strong>pigenome-<strong>W</strong>ide <strong>A</strong>ssociation <strong>S</strong>tudy (EWAS)</td>
<td>Investigation of epigenetic variations associated with traits or diseases acros the genome</td>
<td></td>
</tr>
<tr class="even">
<td>Epistasis</td>
<td>Interaction between different genetic variants affecting a trait’s expression.</td>
<td></td>
</tr>
<tr class="odd">
<td>Exome</td>
<td>The portion of the genome containing protein-coding genes.</td>
<td></td>
</tr>
<tr class="even">
<td>Exon</td>
<td>Part of a gene that will form mRNA. Some exons are coding, with information for making a protein.</td>
<td></td>
</tr>
</tbody>
</table>
</section>
<section id="f" class="level1">
<h1>F</h1>
<table class="table">
<colgroup>
<col style="width: 20%">
<col style="width: 77%">
<col style="width: 2%">
</colgroup>
<tbody>
<tr class="odd">
<td><strong>F</strong>alse <strong>D</strong>iscovery <strong>R</strong>ate (<strong>FDR</strong>)</td>
<td>Method for correction for multiple testing, by defining an acceptable false discovery rate - proportion of falsely identified significant results in multiple testing.</td>
<td></td>
</tr>
<tr class="even">
<td>FASTA file</td>
<td>Format for representing nucleotide or amino acid sequences in bioinformatics.</td>
<td></td>
</tr>
<tr class="odd">
<td>Frameshift mutation</td>
<td>Insertion or deletion altering the reading frame during translation.</td>
<td></td>
</tr>
</tbody>
</table>
</section>
<section id="g" class="level1">
<h1>G</h1>
<table class="table">
<tbody>
<tr class="odd">
<td><strong>G</strong>ene-<strong>E</strong>nvironment Interaction (<strong>GxE</strong>)</td>
<td>Joint effect of genetic and environmental factors on traits.</td>
<td></td>
</tr>
<tr class="even">
<td>Genetic Correlation</td>
<td>Proportion of variance that two traits share from common genetic causes</td>
<td></td>
</tr>
<tr class="odd">
<td>Genome</td>
<td>Complete set of genetic material in an organism.</td>
<td></td>
</tr>
<tr class="even">
<td>Genomic Inflation factor (lamda)</td>
<td>Measure of inflation in test statistics due to population stratification.</td>
<td></td>
</tr>
<tr class="odd">
<td>Genotype</td>
<td>Genetic makeup of an individual at a specific locus.</td>
<td></td>
</tr>
<tr class="even">
<td><strong>G</strong>enome-<strong>W</strong>ide <strong>A</strong>ssociation <strong>S</strong>tudy (<strong>GWAS</strong>)</td>
<td>Investigation of genetic variants across the genome to identify associations with traits.</td>
<td><a href="./chapter1.html#sec-section2">Chapter 1.2</a>; <a href="./chapter5.html">Chapter 5</a>; <a href="./software_gwas.html">Software Tutorial: GWAS</a></td>
</tr>
</tbody>
</table>
</section>
<section id="h" class="level1">
<h1>H</h1>
<table class="table">
<colgroup>
<col style="width: 36%">
<col style="width: 60%">
<col style="width: 3%">
</colgroup>
<tbody>
<tr class="odd">
<td>Haplotype</td>
<td>Set of alleles on a single chromosome inherited together.</td>
<td></td>
</tr>
<tr class="even">
<td><strong>H</strong>ardy-<strong>W</strong>einberg <strong>E</strong>quilibrium (<strong>HWE</strong>)</td>
<td>Model predicting allele frequencies remain constant in a non-evolving population.</td>
<td></td>
</tr>
<tr class="odd">
<td>Heterogeneity</td>
<td>Variability or diversity in a population or dataset.</td>
<td></td>
</tr>
<tr class="even">
<td>Heterozygosity</td>
<td>Possessing different alleles at a specific genetic locus.</td>
<td></td>
</tr>
<tr class="odd">
<td>Histone modification</td>
<td>Chemical alteration of histone proteins influencing gene expression.</td>
<td></td>
</tr>
<tr class="even">
<td>Homozygosity</td>
<td>Possessing identical alleles at a specific genetic locus.</td>
<td></td>
</tr>
<tr class="odd">
<td>Horizontal pleiotropy</td>
<td>Situation in which a genetic variant affects multiple traits independently.</td>
<td></td>
</tr>
<tr class="even">
<td>Hyperparameters</td>
<td>Parameters set before a machine learning algorithm runs.</td>
<td></td>
</tr>
</tbody>
</table>
</section>
<section id="i" class="level1">
<h1>I</h1>
<table class="table">
<colgroup>
<col style="width: 36%">
<col style="width: 59%">
<col style="width: 4%">
</colgroup>
<tbody>
<tr class="odd">
<td><strong>I</strong>dentity-<strong>b</strong>y-<strong>d</strong>ecent (<strong>IBD</strong>)</td>
<td>Shared ancestry for a specific genomic region between individuals.</td>
<td></td>
</tr>
<tr class="even">
<td>Imputation</td>
<td>Predicting missing genetic information based on known data.</td>
<td></td>
</tr>
<tr class="odd">
<td>INFO score</td>
<td>Measure of imputation quality for genetic variants</td>
<td></td>
</tr>
<tr class="even">
<td>Insertion</td>
<td>Addition of a DNA segment into a genome.</td>
<td></td>
</tr>
<tr class="odd">
<td><strong>In</strong>sertion/<strong>Del</strong>etion (<strong>Indel</strong>)</td>
<td>Genetic variation involving insertions or deletions of nucleotides.</td>
<td></td>
</tr>
<tr class="even">
<td>Intergenic</td>
<td>DNA regions between genes.</td>
<td></td>
</tr>
<tr class="odd">
<td>Intron</td>
<td>Region of DNA between the exons in a gene</td>
<td></td>
</tr>
</tbody>
</table>
</section>
<section id="j" class="level1">
<h1>J</h1>
</section>
<section id="k" class="level1">
<h1>K</h1>
<table class="table">
<colgroup>
<col style="width: 16%">
<col style="width: 77%">
<col style="width: 5%">
</colgroup>
<tbody>
<tr class="odd">
<td>Kilobase (Kb)</td>
<td>Unit of length used in molecular biology, equal to 1000 base pairs.</td>
<td></td>
</tr>
<tr class="even">
<td>Kinship</td>
<td>Genetic relatedness between individuals.</td>
<td></td>
</tr>
</tbody>
</table>
</section>
<section id="l" class="level1">
<h1>L</h1>
<table class="table">
<colgroup>
<col style="width: 32%">
<col style="width: 63%">
<col style="width: 3%">
</colgroup>
<tbody>
<tr class="odd">
<td>Lasso regression</td>
<td>Regression method promoting sparsity in feature selection.</td>
<td></td>
</tr>
<tr class="even">
<td><strong>LD Sc</strong>ore Regression (<strong>LDSC</strong>)</td>
<td>Technique to estimate genetic correlation from genome-wide summary statistics.</td>
<td></td>
</tr>
<tr class="odd">
<td>Linkage</td>
<td>Physical proximity of genetic loci on a chromosome.</td>
<td></td>
</tr>
<tr class="even">
<td><strong>L</strong>inkage <strong>d</strong>isequilibrium (<strong>LD</strong>)</td>
<td>Non-random association of alleles at two or more loci.</td>
<td></td>
</tr>
<tr class="odd">
<td>Locus</td>
<td>Specific position on a chromosome.</td>
<td></td>
</tr>
</tbody>
</table>
</section>
<section id="m" class="level1">
<h1>M</h1>
<table class="table">
<colgroup>
<col style="width: 16%">
<col style="width: 80%">
<col style="width: 3%">
</colgroup>
<tbody>
<tr class="odd">
<td>Manhattan plot</td>
<td>Graph showing results of association tests in a genome-wide analysis, where x-axis is genomic position, y-axis is -log10(p-value)</td>
<td></td>
</tr>
<tr class="even">
<td>Mendelian Randomization</td>
<td>Method using genetic variants as instrumental variables to infer causality.</td>
<td></td>
</tr>
<tr class="odd">
<td>Mendelian Trait</td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td>Messenger RNA (mRNA)</td>
<td>RNA molecule transcribed from DNA and carrying protein-coding information.</td>
<td></td>
</tr>
<tr class="odd">
<td>Methylation</td>
<td>Addition of a methyl group to DNA, often affecting gene expression.</td>
<td></td>
</tr>
<tr class="even">
<td>Microbiome</td>
<td>Community of microorganisms in a specific environment.</td>
<td></td>
</tr>
<tr class="odd">
<td>Minor allele frequency</td>
<td>Frequency of the less common allele at a genetic locus in a population.</td>
<td></td>
</tr>
<tr class="even">
<td>Missense mutation</td>
<td>Point mutation altering a codon, resulting in a different amino acid.</td>
<td></td>
</tr>
<tr class="odd">
<td>Mitochondrial DNA (mDNA)</td>
<td>DNA located in mitochondria, inherited maternally.</td>
<td></td>
</tr>
<tr class="even">
<td>Mosaicism</td>
<td>Presence of genetically distinct cell populations within an organism.</td>
<td></td>
</tr>
</tbody>
</table>
</section>
<section id="n" class="level1">
<h1>N</h1>
<table class="table">
<colgroup>
<col style="width: 23%">
<col style="width: 70%">
<col style="width: 6%">
</colgroup>
<tbody>
<tr class="odd">
<td>Nagelkerke P</td>
<td>Measure of explained variance in logistic regression.</td>
<td></td>
</tr>
<tr class="even">
<td>Nonsense mutation</td>
<td>Point mutation leading to a premature stop codon.</td>
<td></td>
</tr>
<tr class="odd">
<td>Nucleosome</td>
<td>DNA wrapped around histone proteins, forming chromatin.</td>
<td></td>
</tr>
<tr class="even">
<td>Null hypothesis</td>
<td>Hypothesis stating no significant effect or difference.</td>
<td></td>
</tr>
</tbody>
</table>
</section>
<section id="o" class="level1">
<h1>O</h1>
<table class="table">
<colgroup>
<col style="width: 24%">
<col style="width: 69%">
<col style="width: 5%">
</colgroup>
<tbody>
<tr class="odd">
<td>Observational study</td>
<td>Research investigating associations without intervention.</td>
<td></td>
</tr>
<tr class="even">
<td>Odds ratio</td>
<td>Measure of association in case-control studies.</td>
<td></td>
</tr>
<tr class="odd">
<td>Open Reading Frame</td>
<td>DNA sequence potentially encoding a protein.</td>
<td></td>
</tr>
</tbody>
</table>
</section>
<section id="p" class="level1">
<h1>P</h1>
<table class="table">
<colgroup>
<col style="width: 37%">
<col style="width: 59%">
<col style="width: 3%">
</colgroup>
<tbody>
<tr class="odd">
<td>Pedigree</td>
<td>Family tree showing genetic relationships.</td>
<td></td>
</tr>
<tr class="even">
<td>Phasing</td>
<td>Determining the parental origin of alleles in an individual.</td>
<td></td>
</tr>
<tr class="odd">
<td>Phenome</td>
<td>Complete set of an individual’s traits and characteristics.</td>
<td></td>
</tr>
<tr class="even">
<td><strong>Phe</strong>nome-<strong>w</strong>ide <strong>A</strong>ssociation <strong>S</strong>tudy (<strong>pheWAS</strong>)</td>
<td>Exploration of genetic variants across multiple traits.</td>
<td></td>
</tr>
<tr class="odd">
<td>Phenotype</td>
<td>Observable traits and characteristics of an individual.</td>
<td></td>
</tr>
<tr class="even">
<td>Pleiotropy</td>
<td>Single gene influencing multiple traits.</td>
<td></td>
</tr>
<tr class="odd">
<td>Point mutation</td>
<td>Single nucleotide change in DNA.</td>
<td></td>
</tr>
<tr class="even">
<td><strong>P</strong>olygenic <strong>R</strong>isk <strong>S</strong>core (<strong>PRS</strong>)</td>
<td>Combined effect of multiple genetic variants on a trait.</td>
<td></td>
</tr>
<tr class="odd">
<td>Polygenic trait</td>
<td>Trait influenced by multiple genes.</td>
<td></td>
</tr>
<tr class="even">
<td>Polymorphism</td>
<td>Genetic variation within a population.</td>
<td></td>
</tr>
<tr class="odd">
<td>Population stratification</td>
<td>Population substructure leading to confounding in genetic studies.</td>
<td></td>
</tr>
<tr class="even">
<td>Power (Genomic)</td>
<td>Probability of detecting an effect in a study.</td>
<td></td>
</tr>
<tr class="odd">
<td>Power analysis</td>
<td>Calculating required sample size based on desired statistical power, and effect size estimates.</td>
<td></td>
</tr>
<tr class="even">
<td>Precision Medicine</td>
<td>Tailoring medical treatment to an individual’s genetic and health characteristics.</td>
<td></td>
</tr>
<tr class="odd">
<td><strong>P</strong>rincipal <strong>C</strong>omponent <strong>A</strong>nalysis (<strong>PCA</strong>)</td>
<td>Dimensionality reduction technique, used to control for population stratification.</td>
<td></td>
</tr>
<tr class="even">
<td>Proband</td>
<td>Index individual in a genetic study.</td>
<td></td>
</tr>
<tr class="odd">
<td>Promoter</td>
<td>DNA region controlling gene transcription initiation.</td>
<td></td>
</tr>
</tbody>
</table>
</section>
<section id="q" class="level1">
<h1>Q</h1>
<table class="table">
<colgroup>
<col style="width: 31%">
<col style="width: 65%">
<col style="width: 3%">
</colgroup>
<tbody>
<tr class="odd">
<td><strong>Q</strong>uantile-<strong>q</strong>uantile <strong>plot</strong> (<strong>qqplot</strong>)</td>
<td>Plot showing how observed test statistics in a genome-wide analysis depart from expected distribution.</td>
<td></td>
</tr>
</tbody>
</table>
</section>
<section id="r" class="level1">
<h1>R</h1>
<table class="table">
<colgroup>
<col style="width: 14%">
<col style="width: 80%">
<col style="width: 5%">
</colgroup>
<tbody>
<tr class="odd">
<td>R<sup>2</sup></td>
<td>Coefficient of determination indicating the proportion of variance explained.</td>
<td></td>
</tr>
<tr class="even">
<td>Rare variant</td>
<td>Infrequently occurring genetic variant in a population.</td>
<td></td>
</tr>
<tr class="odd">
<td>REF/ALT</td>
<td>Reference and alternative alleles at a genetic locus.</td>
<td></td>
</tr>
</tbody>
</table>
</section>
<section id="s" class="level1">
<h1>S</h1>
<table class="table">
<colgroup>
<col style="width: 48%">
<col style="width: 47%">
<col style="width: 4%">
</colgroup>
<tbody>
<tr class="odd">
<td>Scaffold</td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td>Sensitivity</td>
<td>True positive rate in diagnostic testing.</td>
<td></td>
</tr>
<tr class="odd">
<td><strong>S</strong>ingle <strong>N</strong>ucleotide <strong>P</strong>olymorphism (<strong>SNP</strong>)</td>
<td>Single base-pair variation in DNA sequence.</td>
<td></td>
</tr>
<tr class="even">
<td><strong>S</strong>ingle <strong>N</strong>ucleotide <strong>V</strong>ariant (<strong>SNV</strong>)</td>
<td>Single base-pair genetic variation, including SNPs.</td>
<td></td>
</tr>
<tr class="odd">
<td>SNP Heritability (h<sup>2</sup><sub>SNP</sub>)</td>
<td>Proportion of trait variance explained by SNPs.</td>
<td></td>
</tr>
<tr class="even">
<td>Specificity</td>
<td>True negative rate in diagnostic testing.</td>
<td></td>
</tr>
<tr class="odd">
<td>Structural variant</td>
<td>Genetic variation involving larger DNA segments.</td>
<td></td>
</tr>
</tbody>
</table>
</section>
<section id="t" class="level1">
<h1>T</h1>
<table class="table">
<colgroup>
<col style="width: 28%">
<col style="width: 35%">
<col style="width: 35%">
</colgroup>
<tbody>
<tr class="odd">
<td>Transcriptome</td>
<td>Total RNA</td>
<td></td>
</tr>
<tr class="even">
<td>Transcriptomic Imputation</td>
<td>The process of predicting missing gene expression data using available information.</td>
<td></td>
</tr>
<tr class="odd">
<td><strong>T</strong>ranscriptomic <strong>R</strong>isk <strong>S</strong>core (<strong>TRS</strong>)</td>
<td>A score calculated from transcriptomic data to assess the risk of a specific outcome.</td>
<td></td>
</tr>
<tr class="even">
<td>Type I error</td>
<td>False positive result in hypothesis testing, where a null hypothesis is incorrectly rejected.</td>
<td></td>
</tr>
<tr class="odd">
<td>Type II error</td>
<td>False negative result in hypothesis testing, where a null hypothesis is incorrectly accepted. Type II error = 1-power.</td>
<td></td>
</tr>
</tbody>
</table>
</section>
<section id="u" class="level1">
<h1>U</h1>
</section>
<section id="v" class="level1">
<h1>V</h1>
<table class="table">
<colgroup>
<col style="width: 29%">
<col style="width: 67%">
<col style="width: 3%">
</colgroup>
<tbody>
<tr class="odd">
<td>Variant</td>
<td>A specific form of a genetic locus differing from the reference sequence.</td>
<td></td>
</tr>
<tr class="even">
<td>Variance explained</td>
<td>The portion of trait variability accounted for by a given factor, often a genetic variant.</td>
<td></td>
</tr>
<tr class="odd">
<td><strong>V</strong>ariant <strong>C</strong>all <strong>F</strong>ormat (<strong>VCF</strong>)</td>
<td>A standard file format for storing genetic variant information.</td>
<td></td>
</tr>
<tr class="even">
<td>Vertical pleiotropy</td>
<td>Situation in which a genetic variant affects multiple traits due to a shared biological pathway.</td>
<td></td>
</tr>
</tbody>
</table>
</section>
<section id="w" class="level1">
<h1>W</h1>
<table class="table">
<colgroup>
<col style="width: 37%">
<col style="width: 58%">
<col style="width: 4%">
</colgroup>
<tbody>
<tr class="odd">
<td><strong>W</strong>hole <strong>E</strong>xome <strong>S</strong>equencing (<strong>WES</strong>)</td>
<td>Technique for sequencing only the protein-coding regions of the genome.</td>
<td></td>
</tr>
<tr class="even">
<td><strong>W</strong>hole <strong>G</strong>enome <strong>S</strong>equencing (<strong>WGS</strong>)</td>
<td>Method for sequencing an individual’s entire genome.</td>
<td></td>
</tr>
</tbody>
</table>
</section>
<section id="xyz" class="level1">
<h1>XYZ</h1>
<table class="table">
<colgroup>
<col style="width: 9%">
<col style="width: 85%">
<col style="width: 4%">
</colgroup>
<tbody>
<tr class="odd">
<td>X-linked</td>
<td>Genetic trait located on the X chromosome, leading to sex-specific inheritance patterns.</td>
<td></td>
</tr>
</tbody>
</table>
<p>*This Glossary was constructed with the help of <a href="https://chat.openai.com/chat">ChatGPT</a>.</p>
<blockquote class="blockquote">
<p>OpenAI. (2023). <em>ChatGPT</em> (Mar 14 version) [Large language model]. <a href="https://chat.openai.com/chat" class="uri">https://chat.openai.com/chat</a></p>
</blockquote>
<p>For other glossaries of genetic terms, please see:</p>
<ul>
<li><p><a href="https://www.genome.gov/genetics-glossary">NHGRI Glossary</a></p></li>
<li><p><a href="https://www.genomicseducation.hee.nhs.uk/glossary/">Genetics Education Programme</a></p></li>
</ul>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
          // target, if specified
          link.setAttribute("target", "_blank");
          // default icon
          link.classList.add("external");
      }
    }
});
</script>
</div> <!-- /content -->



</body></html>