<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.353">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>PGC Video Textbook - Tools/Software Resources</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


<link rel="stylesheet" href="styles.css">
</head>

<body class="nav-sidebar docked nav-fixed">


<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="./index.html">
    <span class="navbar-title">PGC Video Textbook</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link active" href="./index.html" rel="" target="" aria-current="page">
 <span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./about.html" rel="" target="">
 <span class="menu-text">About</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./contact.html" rel="" target="">
 <span class="menu-text">Contact us</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="https://pgc.unc.edu/" rel="" target="">
 <span class="menu-text">PGC Website</span></a>
  </li>  
</ul>
            <div class="quarto-navbar-tools ms-auto">
</div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
      <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./glossary.html">Additional Resources</a></li><li class="breadcrumb-item"><a href="./resources.html">Tools/Software Resources</a></li></ol></nav>
      <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
      </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal sidebar-navigation docked overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
      <a href="./index.html" class="sidebar-logo-link">
      <img src="./pgc_logo_website_v3.jpeg" alt="" class="sidebar-logo py-0 d-lg-inline d-none">
      </a>
      <div class="sidebar-tools-main">
    <div class="dropdown">
      <a href="" title="" id="quarto-navigation-tool-dropdown-0" class="quarto-navigation-tool dropdown-toggle px-1" data-bs-toggle="dropdown" aria-expanded="false" aria-label=""><i class="bi bi-github"></i></a>
      <ul class="dropdown-menu" aria-labelledby="quarto-navigation-tool-dropdown-0">
          <li>
            <a class="dropdown-item sidebar-tools-main-item" href="https://github.com/PGCAnalytics/pgcvideotextbook/">
            Source code
            </a>
          </li>
      </ul>
    </div>
    <a href="https://twitter.com/PGCgenetics" title="" class="quarto-navigation-tool px-1" aria-label=""><i class="bi bi-twitter"></i></a>
</div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./welcome.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Welcome to the PGC Video Textbook!</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./toc.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Table of Contents</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">
 <span class="menu-text">Chapters</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter1.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 1: Introduction</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter2.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 2: The Genome</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter3.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 3: Technologies</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter4.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 4: Study Designs</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter5.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 5: GWAS Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter6.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 6: Polygenic Scores</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter7.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 7: Ancestry-Specific Analyses and Considerations</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter8.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 8: Post-GWAS Bioinformatics</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter9.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 9: Advanced Topics</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter10.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 10: Other Considerations</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true">
 <span class="menu-text">Software Tutorials</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_cnvs.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">CNVs</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_conditional.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Conditional Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_datasets.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Datasets</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_ewas.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">EWAS</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_geneset.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Gene Set Identification</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_liftover.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Genome Builds</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_gwas.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">GWAS</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_genomicSEM.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Genomic SEM</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_MR.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Mendelian Randomization</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_mtag.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">MTAG</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_prs.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">PRS</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_correlation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">SNP Heritability and Genetic Correlation</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true">
 <span class="menu-text">Additional Resources</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./glossary.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Glossary</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./resources.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">Tools/Software Resources</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="https://pgc.unc.edu/for-researchers/download-results/" class="sidebar-item-text sidebar-link">
 <span class="menu-text">PGC Summary Statistics</span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Tools/Software Resources</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<p>Below is a table listing various software and other tools that are commonly used in genomics analyses. Most are mentioned throughout the PGC Video Textbook. Please use the outline below to navigate through the table.</p>
<ol type="1">
<li>General Tools</li>
<li>GWAS
<ol type="1">
<li>QC</li>
<li>Imputation and haplotype phasing</li>
<li>Relatedness/Population stratification</li>
<li>Association testing</li>
<li>Power testing</li>
</ol></li>
<li>PRS methods</li>
<li>Post-GWAS analysis
<ol type="1">
<li>SNP heritability</li>
<li>Genetic correlation</li>
<li>TWAS</li>
<li>PheWAS</li>
<li>Gene/Gene set identification</li>
<li>eQTLs</li>
<li>Gene set enrichment</li>
<li>Functionally-informed risk scores</li>
<li>Fine-mapping</li>
<li>Conditional analyses</li>
<li>EWAS</li>
<li>Mendelian Randomization</li>
</ol></li>
<li>WGS and WES analysis
<ol type="1">
<li>Raw count QC</li>
<li>Preprocessing</li>
<li>Alignment</li>
<li>Post-alignment processing</li>
<li>Variant calling (SNPs/INDELs)</li>
<li>Variant calling (CNVs)</li>
<li>Variant annotation</li>
<li>Association testing</li>
</ol></li>
<li>Datasets and data access</li>
<li>Other Genomics tools</li>
<li>Genome Builds</li>
</ol>
<table class="table">
<thead>
<tr class="header">
<th style="text-align: left;">Analysis</th>
<th style="text-align: left;">Step</th>
<th style="text-align: left;">Software</th>
<th style="text-align: left;">Additional Tutorials</th>
<th style="text-align: left;">Language</th>
<th style="text-align: left;">PubMed ID</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;"><h1 id="general-tools">General Tools</h1></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;">General</td>
<td style="text-align: left;">File management/formatting</td>
<td style="text-align: left;"><a href="http://www.htslib.org/">SAMtools</a></td>
<td style="text-align: left;"><a href="http://www.htslib.org/doc/samtools.html">Manual</a></td>
<td style="text-align: left;">C; Python</td>
<td style="text-align: left;">19505943; 21903627; 21320865</td>
</tr>
<tr class="odd">
<td style="text-align: left;">General</td>
<td style="text-align: left;">File management/formatting</td>
<td style="text-align: left;"><a href="http://www.htslib.org/">BCFtools</a></td>
<td style="text-align: left;"><a href="http://samtools.github.io/bcftools/">Manual</a>; <a href="https://github.com/samtools/bcftools">GitHub</a></td>
<td style="text-align: left;">C; Python</td>
<td style="text-align: left;">19505943; 21903627; 21320865</td>
</tr>
<tr class="even">
<td style="text-align: left;">General</td>
<td style="text-align: left;">File management/formatting</td>
<td style="text-align: left;"><a href="https://bedtools.readthedocs.io/en/latest/">BEDtools</a></td>
<td style="text-align: left;"><a href="https://github.com/arq5x/bedtools2">GitHub</a></td>
<td style="text-align: left;">C; C++</td>
<td style="text-align: left;">20110278</td>
</tr>
<tr class="odd">
<td style="text-align: left;"><h1 id="gwas"><strong>GWAS</strong></h1></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;">GWAS</td>
<td style="text-align: left;">Genotype calling and Imputation</td>
<td style="text-align: left;"><a href="https://software.broadinstitute.org/software/mocha/">MoChA</a></td>
<td style="text-align: left;"><a href="https://github.com/freeseek/mocha">GitHub</a></td>
<td style="text-align: left;">C; R</td>
<td style="text-align: left;">29995854; 32581363</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GWAS</td>
<td style="text-align: left;">Rapid Imputation and Computational Pipeline for GWAS (RICOPILI)</td>
<td style="text-align: left;"><a href="https://sites.google.com/a/broadinstitute.org/ricopili/">RICOPILI pipeline</a></td>
<td style="text-align: left;"><a href="https://sites.google.com/a/broadinstitute.org/ricopili/download-installation">RICOPILI download</a>; <a href="https://docs.google.com/document/d/1ux_FbwnvSzaiBVEwgS7eWJoYlnc_o0YHFb07SPQsYjI/edit?userstoinvite=mlam@broadinstitute.org&amp;actionButton=1#heading=h.tkgxq8x9kt6n">Tutorial</a></td>
<td style="text-align: left;">Perl; R</td>
<td style="text-align: left;">31393554</td>
</tr>
<tr class="even">
<td style="text-align: left;">GWAS</td>
<td style="text-align: left;">General; QC</td>
<td style="text-align: left;"><a href="https://zzz.bwh.harvard.edu/plink/plink2.shtml">PLINK1.9</a>/<a href="https://www.cog-genomics.org/plink/2.0/">PLINK2</a></td>
<td style="text-align: left;"><a href="https://zzz.bwh.harvard.edu/plink/plink2.shtml">PLINK website</a></td>
<td style="text-align: left;">C; C++</td>
<td style="text-align: left;">17701901</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GWAS</td>
<td style="text-align: left;">Relatedness/Kinship</td>
<td style="text-align: left;"><a href="https://www.kingrelatedness.com/">KING</a></td>
<td style="text-align: left;"><a href="https://www.kingrelatedness.com/manual.shtml">Manual</a>; <a href="chrome-extension://efaidnbmnnnibpcajpcglclefindmkaj/https://bioconductor.org/packages/devel/bioc/manuals/GENESIS/man/GENESIS.pdf">GENESIS documentation</a></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">20926424</td>
</tr>
<tr class="even">
<td style="text-align: left;">GWAS</td>
<td style="text-align: left;">Population Stratification/PCA</td>
<td style="text-align: left;"><a href="https://rdrr.io/bioc/GENESIS/man/pcair.html">GENESIS/PCAiR</a></td>
<td style="text-align: left;"><a href="https://rdrr.io/bioc/GENESIS/man/pcair.html">Manual</a>; <a href="http://bioconductor.org/packages/release/bioc/vignettes/GENESIS/inst/doc/pcair.html">Tutorial</a>; <a href="https://github.com/UW-GAC/GENESIS">GENESIS GitHub</a></td>
<td style="text-align: left;">R</td>
<td style="text-align: left;">20926424; 25810074; 31329242</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GWAS</td>
<td style="text-align: left;">Population Stratification/PCA</td>
<td style="text-align: left;"><a href="https://rdrr.io/bioc/GENESIS/man/pcrelate.html">GENESIS/PC-Relate</a></td>
<td style="text-align: left;"><a href="http://bioconductor.org/packages/release/bioc/vignettes/GENESIS/inst/doc/pcair.html#relatedness-estimation-adjusted-for-principal-components-pc-relate-1">Tutorial</a></td>
<td style="text-align: left;">R</td>
<td style="text-align: left;">26748516</td>
</tr>
<tr class="even">
<td style="text-align: left;">GWAS</td>
<td style="text-align: left;">Local Ancestry and Admixture Inference</td>
<td style="text-align: left;"><a href="https://github.com/slowkoni/rfmix">RFMix</a></td>
<td style="text-align: left;"><a href="https://github.com/slowkoni/rfmix">GitHub</a></td>
<td style="text-align: left;">C++</td>
<td style="text-align: left;">23910464</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GWAS</td>
<td style="text-align: left;">Local Ancestry and Admixture Inference</td>
<td style="text-align: left;"><a href="https://github.com/Atkinson-Lab/Tractor-tutorial">TRACTOR</a></td>
<td style="text-align: left;"><a href="https://github.com/Atkinson-Lab/Tractor">GitHub</a></td>
<td style="text-align: left;">Python</td>
<td style="text-align: left;">33462486</td>
</tr>
<tr class="even">
<td style="text-align: left;">GWAS</td>
<td style="text-align: left;">Imputation panel</td>
<td style="text-align: left;"><a href="https://topmed.nhlbi.nih.gov/">TOPMed</a></td>
<td style="text-align: left;"><a href="https://imputationserver.sph.umich.edu/index.html#!">Imputation server</a></td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">33568819</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GWAS</td>
<td style="text-align: left;">Imputation panel</td>
<td style="text-align: left;"><a href="https://www.coriell.org/1/NHGRI/Collections/1000-Genomes-Project-Collection/1000-Genomes-Project?gclid=Cj0KCQjwldKmBhCCARIsAP-0rfxCrG9vQObbJBaZ9DLr1VQMejUmtf4aiMb5OXyZpbO6DckJ65ZFvdEaAsU1EALw_wcB">1000 Genomes</a></td>
<td style="text-align: left;"><a href="https://imputationserver.sph.umich.edu/index.html#!">Imputation server</a></td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">26432245</td>
</tr>
<tr class="even">
<td style="text-align: left;">GWAS</td>
<td style="text-align: left;">Imputation panel</td>
<td style="text-align: left;"><a href="https://www.sanger.ac.uk/collaboration/haplotype-reference-consortium/">HRC</a></td>
<td style="text-align: left;"><a href="https://www.sanger.ac.uk/collaboration/haplotype-reference-consortium/">Website</a></td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">27548312</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GWAS</td>
<td style="text-align: left;">Imputation algorithm</td>
<td style="text-align: left;"><a href="https://github.com/statgen/Minimac4">minimac</a></td>
<td style="text-align: left;"><a href="https://github.com/statgen/Minimac4">GitHub</a></td>
<td style="text-align: left;">C++; C</td>
<td style="text-align: left;">27571263</td>
</tr>
<tr class="even">
<td style="text-align: left;">GWAS</td>
<td style="text-align: left;">Haplotype phasing</td>
<td style="text-align: left;"><a href="https://alkesgroup.broadinstitute.org/Eagle/">EAGLE</a></td>
<td style="text-align: left;"><a href="https://alkesgroup.broadinstitute.org/Eagle/">Manual</a>; <a href="https://github.com/poruloh/Eagle">GitHub</a></td>
<td style="text-align: left;">C++</td>
<td style="text-align: left;">27694958; 27270109</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GWAS</td>
<td style="text-align: left;">Phasing and Imputation</td>
<td style="text-align: left;"><a href="https://faculty.washington.edu/browning/beagle/b4_1.html">BEAGLE</a></td>
<td style="text-align: left;"><a href="http://faculty.washington.edu/browning/beagle/beagle.html">Website</a>; <a href="https://github.com/adrianodemarino/Imputation_beagle_tutorial">Tutorial</a></td>
<td style="text-align: left;">Java</td>
<td style="text-align: left;">23535385; 30100085; 34478634</td>
</tr>
<tr class="even">
<td style="text-align: left;">GWAS</td>
<td style="text-align: left;">Haplotype phasing</td>
<td style="text-align: left;"><a href="https://mathgen.stats.ox.ac.uk/genetics_software/shapeit/shapeit.html">SHAPEIT</a></td>
<td style="text-align: left;"><a href="https://mathgen.stats.ox.ac.uk/genetics_software/shapeit/shapeit.html">Website</a>; <a href="https://mathgen.stats.ox.ac.uk/genetics_software/shapeit/shapeit.html#gettingstarted">Tutorial</a>; <a href="https://github.com/odelaneau/shapeit4">GitHub</a></td>
<td style="text-align: left;">C++</td>
<td style="text-align: left;">22138821; 23269371; 24094745; 24743097; 25653097</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GWAS</td>
<td style="text-align: left;">Power calculations</td>
<td style="text-align: left;"><a href="https://cran.r-project.org/web/packages/genpwr/vignettes/vignette.html">genpwr</a></td>
<td style="text-align: left;"><a href="https://cran.r-project.org/web/packages/genpwr/vignettes/vignette.html">Manual</a></td>
<td style="text-align: left;">R</td>
<td style="text-align: left;">32721961</td>
</tr>
<tr class="even">
<td style="text-align: left;">GWAS</td>
<td style="text-align: left;">Association Testing</td>
<td style="text-align: left;"><a href="https://saigegit.github.io/SAIGE-doc/">SAIGE</a></td>
<td style="text-align: left;"><a href="https://github.com/weizhouUMICH/SAIGE">GitHub</a>; <a href="https://github.com/saigegit/SAIGE">More documentation</a></td>
<td style="text-align: left;">R; C++</td>
<td style="text-align: left;">30104761</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GWAS</td>
<td style="text-align: left;">Association Testing</td>
<td style="text-align: left;"><a href="https://alkesgroup.broadinstitute.org/BOLT-LMM/BOLT-LMM_manual.html">BOLT-LMM</a></td>
<td style="text-align: left;"><a href="https://alkesgroup.broadinstitute.org/BOLT-LMM/downloads/">Download</a>; <a href="https://alkesgroup.broadinstitute.org/BOLT-LMM/BOLT-LMM_manual.html">Manual</a></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">25642633</td>
</tr>
<tr class="even">
<td style="text-align: left;">GWAS</td>
<td style="text-align: left;">Association Testing</td>
<td style="text-align: left;"><a href="https://rgcgithub.github.io/regenie/">REGENIE</a></td>
<td style="text-align: left;"><a href="https://github.com/rgcgithub/regenie">GitHub</a>; <a href="https://rgcgithub.github.io/regenie/options/">Tutorial</a></td>
<td style="text-align: left;">C++</td>
<td style="text-align: left;">34017140</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GWAS</td>
<td style="text-align: left;">Causal Mixture Model</td>
<td style="text-align: left;"><a href="https://github.com/precimed/mixer">MiXeR</a></td>
<td style="text-align: left;"><a href="https://github.com/precimed/mixer">GitHub</a></td>
<td style="text-align: left;">C</td>
<td style="text-align: left;">32427991; 31160569</td>
</tr>
<tr class="even">
<td style="text-align: left;">GWAS</td>
<td style="text-align: left;">Multi-Trait Analysis of GWAS</td>
<td style="text-align: left;"><a href="https://github.com/JonJala/mtag">MTAG</a></td>
<td style="text-align: left;"><a href="https://github.com/JonJala/mtag">GitHub</a></td>
<td style="text-align: left;">Python</td>
<td style="text-align: left;">29292387</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GWAS</td>
<td style="text-align: left;">qqplot and Manhattan plot</td>
<td style="text-align: left;"><a href="https://github.com/stephenturner/qqman">qqman</a></td>
<td style="text-align: left;"><a href="https://github.com/stephenturner/qqman">GitHub</a></td>
<td style="text-align: left;">R</td>
<td style="text-align: left;"><a href="https://doi.org/10.21105/joss.00731">doi link</a></td>
</tr>
<tr class="even">
<td style="text-align: left;">GWAS</td>
<td style="text-align: left;">QC of GWAS summary statistics</td>
<td style="text-align: left;"><a href="https://github.com/Yves-CHEN/DENTIST">DENTIST</a></td>
<td style="text-align: left;"><a href="https://github.com/Yves-CHEN/DENTIST">GitHub</a>; <a href="https://yanglab.westlake.edu.cn/software.html">Yang Lab Website</a></td>
<td style="text-align: left;">C++</td>
<td style="text-align: left;">34880243</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GWAS</td>
<td style="text-align: left;">GWAS Quality Score</td>
<td style="text-align: left;"><a href="https://github.com/Xswapnil/GQS/tree/main">GQS</a></td>
<td style="text-align: left;"><a href="https://github.com/Xswapnil/GQS/tree/main">GitHub</a></td>
<td style="text-align: left;">Python</td>
<td style="text-align: left;">36651666</td>
</tr>
<tr class="even">
<td style="text-align: left;"><h1 id="prs-methods"><strong>PRS Methods</strong></h1></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;">PRS</td>
<td style="text-align: left;">PRS methods</td>
<td style="text-align: left;"><a href="https://github.com/getian107/PRScs">PRS-CS</a></td>
<td style="text-align: left;"><a href="https://github.com/getian107/PRScs">GitHub</a></td>
<td style="text-align: left;">Python</td>
<td style="text-align: left;">30992449</td>
</tr>
<tr class="even">
<td style="text-align: left;">PRS</td>
<td style="text-align: left;">PRS methods</td>
<td style="text-align: left;"><a href="https://github.com/getian107/PRScsx">PRS-CSx</a></td>
<td style="text-align: left;"><a href="https://github.com/getian107/PRScsx">GitHub</a></td>
<td style="text-align: left;">Python</td>
<td style="text-align: left;">35513724</td>
</tr>
<tr class="odd">
<td style="text-align: left;">PRS</td>
<td style="text-align: left;">PRS methods</td>
<td style="text-align: left;"><a href="https://choishingwan.github.io/PRSice/">PRSice2</a></td>
<td style="text-align: left;"><a href="https://choishingwan.github.io/PRS-Tutorial/prsice/">Tutorial</a></td>
<td style="text-align: left;">R</td>
<td style="text-align: left;">31307061</td>
</tr>
<tr class="even">
<td style="text-align: left;">PRS</td>
<td style="text-align: left;">PRS methods</td>
<td style="text-align: left;"><a href="https://github.com/tshmak/lassosum/blob/master/lassosum_standalone.md#lassosum-standalone-version-for-linux">Lassosum</a></td>
<td style="text-align: left;"><a href="https://github.com/tshmak/lassosum">GitHub</a></td>
<td style="text-align: left;">R; C++</td>
<td style="text-align: left;">28480976</td>
</tr>
<tr class="odd">
<td style="text-align: left;">PRS</td>
<td style="text-align: left;">PRS methods</td>
<td style="text-align: left;"><a href="https://privefl.github.io/bigsnpr/articles/LDpred2.html">LDpred2</a></td>
<td style="text-align: left;"><a href="https://privefl.github.io/bigsnpr/articles/LDpred2.html">Manual</a></td>
<td style="text-align: left;">R</td>
<td style="text-align: left;">33326037</td>
</tr>
<tr class="even">
<td style="text-align: left;">PRS</td>
<td style="text-align: left;">PRS methods</td>
<td style="text-align: left;"><a href="https://cnsgenomics.com/software/gctb/#Overview">SBayesR</a></td>
<td style="text-align: left;"><a href="https://cnsgenomics.com/software/gctb/#SBayesRTutorial">Manual</a></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">31704910</td>
</tr>
<tr class="odd">
<td style="text-align: left;">PRS</td>
<td style="text-align: left;">PRS methods</td>
<td style="text-align: left;"><a href="https://dougspeed.com/megaprs/">MegaPRS</a></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">Python</td>
<td style="text-align: left;">34234142</td>
</tr>
<tr class="even">
<td style="text-align: left;"><h1 id="post-gwas-analyses"><strong>Post-GWAS Analyses</strong></h1></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;">SNP heritability</td>
<td style="text-align: left;"></td>
<td style="text-align: left;"><a href="https://yanglab.westlake.edu.cn/software/gcta/#GREML">GCTA-GREML</a></td>
<td style="text-align: left;"><a href="https://shiny.cnsgenomics.com/gctaPower/">Tutorial</a>; <a href="https://github.com/jianyangqt/gcta">GitHub</a></td>
<td style="text-align: left;">C++</td>
<td style="text-align: left;">27457963</td>
</tr>
<tr class="even">
<td style="text-align: left;">SNP heritability and genetic correlation</td>
<td style="text-align: left;">Linkage Disequilibrium Score Regression</td>
<td style="text-align: left;"><a href="https://github.com/bulik/ldsc/wiki/Heritability-and-Genetic-Correlation">LDSC</a></td>
<td style="text-align: left;"><a href="https://github.com/bulik/ldsc/wiki/Heritability-and-Genetic-Correlation">Tutorial</a>; <a href="https://github.com/bulik/ldsc">GitHub</a></td>
<td style="text-align: left;">Python</td>
<td style="text-align: left;">25642630</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Genetic correlation</td>
<td style="text-align: left;">Genomic Structural Equation Modeling</td>
<td style="text-align: left;"><a href="https://github.com/GenomicSEM/GenomicSEM/wiki">Genomic SEM</a></td>
<td style="text-align: left;"><a href="https://github.com/GenomicSEM/GenomicSEM/wiki/8.-Tutorials">Tutorial</a>; <a href="https://github.com/GenomicSEM/GenomicSEM">GitHub</a></td>
<td style="text-align: left;">R</td>
<td style="text-align: left;">30962613</td>
</tr>
<tr class="even">
<td style="text-align: left;">Genetic correlation</td>
<td style="text-align: left;"></td>
<td style="text-align: left;"><a href="https://github.com/josefin-werme/LAVA">LAVA</a></td>
<td style="text-align: left;"><a href="https://github.com/josefin-werme/LAVA">GitHub</a></td>
<td style="text-align: left;">R</td>
<td style="text-align: left;">35288712</td>
</tr>
<tr class="odd">
<td style="text-align: left;">TWAS</td>
<td style="text-align: left;">Functional Summary-based Imputation</td>
<td style="text-align: left;"><a href="http://gusevlab.org/projects/fusion/">FUSION</a></td>
<td style="text-align: left;"><a href="https://github.com/gusevlab/fusion_twas">GitHub</a>; <a href="http://gusevlab.org/projects/fusion/">Tutorial</a></td>
<td style="text-align: left;">R</td>
<td style="text-align: left;">26854917</td>
</tr>
<tr class="even">
<td style="text-align: left;">TWAS</td>
<td style="text-align: left;">Transcriptomic Imputation</td>
<td style="text-align: left;"><a href="https://github.com/hakyimlab/MetaXcan">PrediXcan</a></td>
<td style="text-align: left;"><a href="https://github.com/hakyimlab/MetaXcan">GitHub</a>; <a href="https://github.com/hakyimlab/MetaXcan/wiki/Individual-level-PrediXcan:-introduction,-tutorials-and-manual">Tutorial</a></td>
<td style="text-align: left;">Python</td>
<td style="text-align: left;">26258848</td>
</tr>
<tr class="odd">
<td style="text-align: left;">TWAS</td>
<td style="text-align: left;">Transcriptomic Imputation</td>
<td style="text-align: left;"><a href="https://github.com/hakyimlab/MetaXcan">S-PrediXcan</a></td>
<td style="text-align: left;"><a href="https://github.com/hakyimlab/MetaXcan">GitHub</a></td>
<td style="text-align: left;">Python</td>
<td style="text-align: left;">29739930</td>
</tr>
<tr class="even">
<td style="text-align: left;">TWAS</td>
<td style="text-align: left;">Tissue models (GTEx, CommonMind, PsychENCODE)</td>
<td style="text-align: left;"><a href="https://predictdb.org/">PredictDB</a></td>
<td style="text-align: left;"><a href="https://predictdb.org/post/2021/07/21/gtex-v8-models-on-eqtl-and-sqtl/">GTEx models</a>; <a href="https://predictdb.org/post/2020/07/19/psychencode-brain-expression-models/">PsychENCODE models</a>; <a href="https://github.com/laurahuckins/CMC_DLPFC_prediXcan">CommonMind DLPFC models</a></td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">33499903; 30545856; 30911161</td>
</tr>
<tr class="odd">
<td style="text-align: left;">TWAS/MR</td>
<td style="text-align: left;">Summary-data-based Mendelian Randomization</td>
<td style="text-align: left;"><a href="https://yanglab.westlake.edu.cn/software/smr/">SMR</a></td>
<td style="text-align: left;"><a href="https://yanglab.westlake.edu.cn/software/smr/">Software</a></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">27019110</td>
</tr>
<tr class="even">
<td style="text-align: left;">PheWAS</td>
<td style="text-align: left;">Phenome-Wide Association Studies</td>
<td style="text-align: left;"><a href="https://rdrr.io/github/PheWAS/PheWAS/man/">PheWAS R package</a></td>
<td style="text-align: left;"><a href="https://github.com/PheWAS/PheWAS">GitHub</a>; <a href="https://www.vumc.org/cpm/cpm-blog/phewas-phenome-wide-association-studies">Tutorial</a></td>
<td style="text-align: left;">R</td>
<td style="text-align: left;">20335276; 24733291; 24270849</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Gene/Gene set identification</td>
<td style="text-align: left;">Gene Set Enrichment Analysis</td>
<td style="text-align: left;"><a href="https://www.gsea-msigdb.org/gsea/index.jsp">GSEA</a></td>
<td style="text-align: left;"><a href="https://software.broadinstitute.org/cancer/software/gsea/wiki/index.php/Main_Page">Documentation</a>; <a href="https://software.broadinstitute.org/cancer/software/gsea/wiki/index.php/R-GSEA_Readme">README</a></td>
<td style="text-align: left;">Java; R</td>
<td style="text-align: left;">16199517; 12808457</td>
</tr>
<tr class="even">
<td style="text-align: left;">Gene/Gene set identification</td>
<td style="text-align: left;">Molecular Signatures Database</td>
<td style="text-align: left;"><a href="https://www.gsea-msigdb.org/gsea/msigdb/">MsigDB</a></td>
<td style="text-align: left;"><a href="https://software.broadinstitute.org/cancer/software/gsea/wiki/index.php/Main_Page#MSigDB_gene_sets">Documentation</a></td>
<td style="text-align: left;">Java</td>
<td style="text-align: left;">16199517; 21546393; 26771021</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Gene/Gene set identification</td>
<td style="text-align: left;">Generalized gene-set analysis of GWAS data</td>
<td style="text-align: left;"><a href="https://ctg.cncr.nl/software/magma">MAGMA</a></td>
<td style="text-align: left;"><a href="https://ctg.cncr.nl/software/magma">Software</a>; <a href="https://cloufield.github.io/GWASTutorial/09_Gene_based_analysis/">Tutorial</a>; <a href="chrome-extension://efaidnbmnnnibpcajpcglclefindmkaj/https://ctg.cncr.nl/software/MAGMA/doc/manual_v1.10.pdf">Manual</a></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">25885710; 27070863; 30218068</td>
</tr>
<tr class="even">
<td style="text-align: left;">Gene/Gene set identification</td>
<td style="text-align: left;">HiC-informed gene-based tests</td>
<td style="text-align: left;"><a href="https://github.com/thewonlab/H-MAGMA">H-MAGMA</a></td>
<td style="text-align: left;"><a href="https://github.com/thewonlab/H-MAGMA">GitHub</a></td>
<td style="text-align: left;">R</td>
<td style="text-align: left;">32152537</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Gene/Gene set identification</td>
<td style="text-align: left;">eQTL-informed gene-based tests</td>
<td style="text-align: left;"><a href="https://github.com/eskederks/eMAGMA-tutorial">E-MAGMA</a></td>
<td style="text-align: left;"><a href="https://github.com/eskederks/eMAGMA-tutorial">GitHub Tutorial</a></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">33624746</td>
</tr>
<tr class="even">
<td style="text-align: left;">Gene/Gene set identification</td>
<td style="text-align: left;">Functional Mapping and Annotation of Genome-Wide Association Studies</td>
<td style="text-align: left;"><a href="https://fuma.ctglab.nl/">FUMA</a></td>
<td style="text-align: left;"><a href="https://github.com/Kyoko-wtnb/FUMA-webapp">GitHub</a>; <a href="https://fuma.ctglab.nl/tutorial">Tutorial</a></td>
<td style="text-align: left;">Blade; Java</td>
<td style="text-align: left;">29184056</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Gene/Gene set identification</td>
<td style="text-align: left;">Set/Pathway-based analysis</td>
<td style="text-align: left;"><a href="https://choishingwan.github.io/PRSice/quick_start_prset/">PRSet</a></td>
<td style="text-align: left;"><a href="https://choishingwan.github.io/PRSice/quick_start_prset/">Tutorial</a></td>
<td style="text-align: left;">R</td>
<td style="text-align: left;">36749789</td>
</tr>
<tr class="even">
<td style="text-align: left;">Gene/Gene set identification</td>
<td style="text-align: left;">Gene set enrichment</td>
<td style="text-align: left;"><a href="https://biit.cs.ut.ee/gprofiler/gost">g:profiler</a></td>
<td style="text-align: left;"><a href="https://biit.cs.ut.ee/gprofiler/gost">Website</a>; <a href="https://rdrr.io/cran/gprofiler2/">R package</a>; <a href="https://cran.r-project.org/web/packages/gprofiler2/vignettes/gprofiler2.html">Documentation</a></td>
<td style="text-align: left;">R</td>
<td style="text-align: left;">31066453</td>
</tr>
<tr class="odd">
<td style="text-align: left;">eQTLs</td>
<td style="text-align: left;">eQTL and GWAS CAusal Variants Identification in Associated Regions</td>
<td style="text-align: left;"><a href="http://genetics.cs.ucla.edu/caviar/">eCAVIAR</a></td>
<td style="text-align: left;"><a href="http://genetics.cs.ucla.edu/caviar/">Website</a>; <a href="https://github.com/fhormoz/caviar">GitHub</a></td>
<td style="text-align: left;">C++</td>
<td style="text-align: left;">27866706</td>
</tr>
<tr class="even">
<td style="text-align: left;">eQTLs</td>
<td style="text-align: left;">Calling eQTLs</td>
<td style="text-align: left;"><a href="https://github.com/francois-a/fastqtl">fastQTL</a></td>
<td style="text-align: left;"><a href="https://github.com/francois-a/fastqtl">GitHub</a>; <a href="https://qtltools.github.io/qtltools/">Documentation</a></td>
<td style="text-align: left;">C++</td>
<td style="text-align: left;">26708335</td>
</tr>
<tr class="odd">
<td style="text-align: left;">eQTLs</td>
<td style="text-align: left;">Calling eQTLs</td>
<td style="text-align: left;"><a href="https://github.com/andreyshabalin/MatrixEQTL">matrixQTL</a></td>
<td style="text-align: left;"><a href="http://www.bios.unc.edu/research/genomic_software/Matrix_eQTL/runit.html">Tutorial</a>; <a href="https://github.com/andreyshabalin/MatrixEQTL">GitHub</a></td>
<td style="text-align: left;">R</td>
<td style="text-align: left;">22492648</td>
</tr>
<tr class="even">
<td style="text-align: left;">Functionally-informed Risk Scores</td>
<td style="text-align: left;">Polygenic Transcriptomic Risk Score</td>
<td style="text-align: left;"><a href="https://github.com/liangyy/ptrs-tf">PTRS</a></td>
<td style="text-align: left;"><a href="https://github.com/liangyy/ptrs-tf">GitHub</a>; <a href="https://github.com/liangyy/ptrs-ukb">Calculate PTRS for UKBB</a></td>
<td style="text-align: left;">Python</td>
<td style="text-align: left;">35027082</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Functionally-informed Risk Scores</td>
<td style="text-align: left;">Trans-ethnic polygenic risk scores</td>
<td style="text-align: left;"><a href="https://github.com/omerwe/polyfun">PolyPred/PolyPred+</a></td>
<td style="text-align: left;"><a href="https://github.com/omerwe/polyfun">GitHub</a>; <a href="https://github.com/omerwe/polyfun/wiki">Tutorial</a>; <a href="https://github.com/omerwe/polyfun/wiki/6.-Trans-ethnic-polygenic-risk-prediction-with-PolyPred">Documentation</a></td>
<td style="text-align: left;">Python</td>
<td style="text-align: left;">35393596</td>
</tr>
<tr class="even">
<td style="text-align: left;">Fine-mapping</td>
<td style="text-align: left;">“Sum of Single Effects” model</td>
<td style="text-align: left;"><a href="https://stephenslab.github.io/susie-paper/">SuSiE</a> (used with <a href="https://chr1swallace.github.io/coloc/">coloc</a>)</td>
<td style="text-align: left;"><a href="https://stephenslab.github.io/susieR/">R package</a></td>
<td style="text-align: left;">R</td>
<td style="text-align: left;">37220626; 34587156</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Fine-mapping</td>
<td style="text-align: left;">Co-localization analysis</td>
<td style="text-align: left;"><a href="https://chr1swallace.github.io/coloc/">coloc</a></td>
<td style="text-align: left;"><a href="https://github.com/chr1swallace/coloc">GitHub</a>; <a href="https://chr1swallace.github.io/coloc/">R package</a></td>
<td style="text-align: left;">R</td>
<td style="text-align: left;">24830394; 32310995; 34587156</td>
</tr>
<tr class="even">
<td style="text-align: left;">Fine-mapping</td>
<td style="text-align: left;">CAusal Variants Identification in Associated Regions</td>
<td style="text-align: left;"><a href="http://genetics.cs.ucla.edu/caviar/">CAVIAR</a></td>
<td style="text-align: left;"><a href="http://genetics.cs.ucla.edu/caviar/">Website</a>; <a href="https://github.com/fhormoz/caviar">GitHub</a></td>
<td style="text-align: left;">C++</td>
<td style="text-align: left;">25104515</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Fine-mapping</td>
<td style="text-align: left;"></td>
<td style="text-align: left;"><a href="http://christianbenner.com/">FINEMAP</a></td>
<td style="text-align: left;"><a href="http://christianbenner.com/">Benner lab website</a>; <a href="https://github.com/FINNGEN/finemapping-pipeline">FinnGen pipeline</a></td>
<td style="text-align: left;">Python; WDL</td>
<td style="text-align: left;">26773131</td>
</tr>
<tr class="even">
<td style="text-align: left;">Fine-mapping</td>
<td style="text-align: left;">Integrative Genetic Association Analysis using Deterministic Approximation of Posteriors</td>
<td style="text-align: left;"><a href="https://github.com/xqwen/dap/wiki">DAP-G</a></td>
<td style="text-align: left;"><a href="https://github.com/xqwen/dap">GitHub</a>; <a href="https://github.com/xqwen/dap/wiki">Manual</a></td>
<td style="text-align: left;">C++</td>
<td style="text-align: left;">27236919</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Conditional analyses</td>
<td style="text-align: left;">Multi-SNP-based conditional and joint association analysis using GWAS summary data</td>
<td style="text-align: left;"><a href="https://yanglab.westlake.edu.cn/software/gcta/#COJO">GCTA-COJO</a></td>
<td style="text-align: left;"><a href="https://yanglab.westlake.edu.cn/software/gcta/#COJO">Website</a>; <a href="https://github.com/jianyangqt/gcta">GCTA GitHub</a></td>
<td style="text-align: left;">C++</td>
<td style="text-align: left;">22426310</td>
</tr>
<tr class="even">
<td style="text-align: left;">Conditional analyses</td>
<td style="text-align: left;">Multi-trait-based conditional and joint analysis using GWAS summary data</td>
<td style="text-align: left;"><a href="https://yanglab.westlake.edu.cn/software/gcta/#mtCOJO">GCTA-mtCOJO</a></td>
<td style="text-align: left;"><a href="https://yanglab.westlake.edu.cn/software/gcta/#mtCOJO">Website</a>; <a href="https://github.com/jianyangqt/gcta">GCTA GitHub</a></td>
<td style="text-align: left;">C++</td>
<td style="text-align: left;">32398722</td>
</tr>
<tr class="odd">
<td style="text-align: left;">EWAS</td>
<td style="text-align: left;">Epigenome-Wide Association Study</td>
<td style="text-align: left;"><a href="https://github.com/hhhh5/ewastools">ewastools</a></td>
<td style="text-align: left;"><a href="https://github.com/hhhh5/ewastools">GitHub</a>; <a href="https://hhhh5.github.io/ewastools/articles/exemplary_ewas.html">Workflow</a></td>
<td style="text-align: left;">R</td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;">Mendelian Randomization</td>
<td style="text-align: left;">2-sample MR using MR-Base database</td>
<td style="text-align: left;"><a href="https://mrcieu.github.io/TwoSampleMR/">TwoSampleMR</a></td>
<td style="text-align: left;"><a href="https://github.com/MRCIEU/TwoSampleMR/">GitHub</a>; <a href="https://mrcieu.github.io/TwoSampleMR/">Website</a></td>
<td style="text-align: left;">R</td>
<td style="text-align: left;">29846171; 29149188</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Mendelian Randomization</td>
<td style="text-align: left;">Database and analytical platform for MR</td>
<td style="text-align: left;"><a href="https://www.mrbase.org/">MR-Base</a></td>
<td style="text-align: left;"><a href="http://app.mrbase.org/">WebApp</a>; <a href="https://github.com/MRCIEU/TwoSampleMR">R package (TwoSampleMR)</a>; <a href="https://gwas.mrcieu.ac.uk/">OpenGWAS database</a></td>
<td style="text-align: left;">R</td>
<td style="text-align: left;">29846171</td>
</tr>
<tr class="even">
<td style="text-align: left;"><h1 id="sequencing-tools"><strong>Sequencing Tools</strong></h1></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;">WGS/WES</td>
<td style="text-align: left;">Raw count QC</td>
<td style="text-align: left;"><a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a></td>
<td style="text-align: left;"><a href="https://github.com/s-andrews/FastQC">GitHub</a></td>
<td style="text-align: left;">Java</td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;">WGS/WES</td>
<td style="text-align: left;">Raw count QC</td>
<td style="text-align: left;"><a href="https://github.com/ewels/MultiQC">MultiQC</a></td>
<td style="text-align: left;"><a href="https://github.com/ewels/MultiQC">GitHub</a></td>
<td style="text-align: left;">Python</td>
<td style="text-align: left;">27312411</td>
</tr>
<tr class="odd">
<td style="text-align: left;">WGS/WES</td>
<td style="text-align: left;">Raw count QC</td>
<td style="text-align: left;"><a href="https://github.com/mjain-lab/NGSQCToolkit">NGS QC Toolkit</a></td>
<td style="text-align: left;"><a href="https://github.com/mjain-lab/NGSQCToolkit">GitHub</a></td>
<td style="text-align: left;">Perl</td>
<td style="text-align: left;">22312429</td>
</tr>
<tr class="even">
<td style="text-align: left;">WGS/WES</td>
<td style="text-align: left;">Raw count QC</td>
<td style="text-align: left;"><a href="https://github.com/genome/qc-analysis-pipeline">QC-Chain</a></td>
<td style="text-align: left;"><a href="https://github.com/genome/qc-analysis-pipeline">GitHub</a></td>
<td style="text-align: left;">WDL</td>
<td style="text-align: left;">23565205</td>
</tr>
<tr class="odd">
<td style="text-align: left;">WGS/WES</td>
<td style="text-align: left;">Preprocessing</td>
<td style="text-align: left;"><a href="https://www.bioinformatics.babraham.ac.uk/projects/trim_galore/">TrimGalore</a></td>
<td style="text-align: left;"><a href="https://github.com/FelixKrueger/TrimGalore">GitHub</a>; <a href="https://www.bioinformatics.babraham.ac.uk/projects/trim_galore/">Documentation</a></td>
<td style="text-align: left;">Perl</td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;">WGS/WES</td>
<td style="text-align: left;">Preprocessing</td>
<td style="text-align: left;"><a href="https://cutadapt.readthedocs.io/en/stable/">Cutadapt</a></td>
<td style="text-align: left;"><a href="https://github.com/marcelm/cutadapt">GitHub</a>; <a href="https://cutadapt.readthedocs.io/en/stable/">Documentation</a></td>
<td style="text-align: left;">Python</td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;">WGS/WES</td>
<td style="text-align: left;">Preprocessing</td>
<td style="text-align: left;"><a href="http://graphics.med.yale.edu/trim/">Btrim</a></td>
<td style="text-align: left;"><a href="http://graphics.med.yale.edu/trim/">Website</a></td>
<td style="text-align: left;">C</td>
<td style="text-align: left;">21651976</td>
</tr>
<tr class="even">
<td style="text-align: left;">WGS/WES</td>
<td style="text-align: left;">Preprocessing</td>
<td style="text-align: left;"><a href="https://github.com/usadellab/Trimmomatic">Trimmonatic</a></td>
<td style="text-align: left;"><a href="https://github.com/usadellab/Trimmomatic">GitHub</a></td>
<td style="text-align: left;">Java</td>
<td style="text-align: left;">24695404</td>
</tr>
<tr class="odd">
<td style="text-align: left;">WGS/WES</td>
<td style="text-align: left;">Preprocessing</td>
<td style="text-align: left;"><a href="https://grenaud.github.io/leeHom/">leeHom</a></td>
<td style="text-align: left;"><a href="https://github.com/grenaud/leeHom">GitHub</a>; <a href="https://grenaud.github.io/leeHom/">Documentation</a></td>
<td style="text-align: left;">C++</td>
<td style="text-align: left;">25100869</td>
</tr>
<tr class="even">
<td style="text-align: left;">WGS/WES</td>
<td style="text-align: left;">Alignment</td>
<td style="text-align: left;"><a href="https://github.com/lh3/bwa">BWA/BWA-mem</a></td>
<td style="text-align: left;"><a href="https://github.com/lh3/bwa">GitHub</a></td>
<td style="text-align: left;">C</td>
<td style="text-align: left;">19451168; 20080505</td>
</tr>
<tr class="odd">
<td style="text-align: left;">WGS/WES</td>
<td style="text-align: left;">Alignment</td>
<td style="text-align: left;"><a href="https://github.com/lh3/minimap2">Minimap2</a></td>
<td style="text-align: left;"><a href="https://github.com/lh3/minimap2">GitHub</a></td>
<td style="text-align: left;">C; Java; TeX</td>
<td style="text-align: left;">29750242</td>
</tr>
<tr class="even">
<td style="text-align: left;">WGS/WES</td>
<td style="text-align: left;">Alignment</td>
<td style="text-align: left;"><a href="https://github.com/BenLangmead/bowtie2">Bowtie2</a></td>
<td style="text-align: left;"><a href="https://github.com/BenLangmead/bowtie2">GitHub</a></td>
<td style="text-align: left;">C++; Perl</td>
<td style="text-align: left;">22388286</td>
</tr>
<tr class="odd">
<td style="text-align: left;">WGS/WES</td>
<td style="text-align: left;">Alignment</td>
<td style="text-align: left;"><a href="https://github.com/broadinstitute/gatk/">GATK</a></td>
<td style="text-align: left;"><a href="https://gatk.broadinstitute.org/hc/en-us">Website</a>; <a href="https://github.com/broadinstitute/gatk/">GitHub</a></td>
<td style="text-align: left;">Java</td>
<td style="text-align: left;">Book citation: Van der Auwera GA &amp; O’Connor BD. (2020). <strong><em>Genomics in the Cloud: Using Docker, GATK, and WDL in Terra (1st Edition).</em></strong> O’Reilly Media.</td>
</tr>
<tr class="even">
<td style="text-align: left;">WGS/WES</td>
<td style="text-align: left;">Alignment</td>
<td style="text-align: left;"><a href="https://github.com/ComparativeGenomicsToolkit/cactus">Cactus</a></td>
<td style="text-align: left;"><a href="https://github.com/ComparativeGenomicsToolkit/cactus">GitHub</a></td>
<td style="text-align: left;">C; Python</td>
<td style="text-align: left;">33177663</td>
</tr>
<tr class="odd">
<td style="text-align: left;">WGS/WES</td>
<td style="text-align: left;">Alignment</td>
<td style="text-align: left;"><a href="https://github.com/vgteam/vg/wiki/Mapping-short-reads-with-Giraffe">Giraffe</a></td>
<td style="text-align: left;"><a href="https://github.com/vgteam/vg/wiki/Mapping-short-reads-with-Giraffe">GitHub</a></td>
<td style="text-align: left;">C++</td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;">WGS/WES</td>
<td style="text-align: left;">Post-alignment processing</td>
<td style="text-align: left;"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/360037052812-MarkDuplicates-Picard-">Picard (MarkDuplicates)</a></td>
<td style="text-align: left;"><a href="https://github.com/broadinstitute/picard">GitHub</a>; <a href="https://gatk.broadinstitute.org/hc/en-us/articles/360037052812-MarkDuplicates-Picard-">Website</a>; <a href="http://broadinstitute.github.io/picard/">Tutorial</a></td>
<td style="text-align: left;">Java</td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;">WGS/WES</td>
<td style="text-align: left;">Post-alignment processing</td>
<td style="text-align: left;">GATK (see above links in “Alignment”)</td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;">WGS/WES</td>
<td style="text-align: left;">Variant calling (SNPs/INDELs)</td>
<td style="text-align: left;"><a href="https://github.com/Genomicsplc/wecall">WeCall</a></td>
<td style="text-align: left;"><a href="https://github.com/Genomicsplc/wecall">GitHub</a></td>
<td style="text-align: left;">C++; Python</td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;">WGS/WES</td>
<td style="text-align: left;">Variant calling (SNPs/INDELs)</td>
<td style="text-align: left;"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/360037225632-HaplotypeCaller">GATK HaplotypeCaller</a></td>
<td style="text-align: left;"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/360037225632-HaplotypeCaller">Website</a></td>
<td style="text-align: left;">Java</td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;">WGS/WES</td>
<td style="text-align: left;">Variant calling (SNPs/INDELs)</td>
<td style="text-align: left;"><a href="http://broadinstitute.github.io/picard/">Picard</a></td>
<td style="text-align: left;"><a href="http://broadinstitute.github.io/picard/">Website</a>; <a href="https://github.com/broadinstitute/picard">GitHub</a></td>
<td style="text-align: left;">Java</td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;">WGS/WES</td>
<td style="text-align: left;">Variant calling (SNPs/INDELs)</td>
<td style="text-align: left;"><a href="https://github.com/freebayes/freebayes">FreeBayes</a></td>
<td style="text-align: left;"><a href="https://github.com/freebayes/freebayes">GitHub</a></td>
<td style="text-align: left;">C++</td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;">WGS/WES</td>
<td style="text-align: left;">Variant calling (SNPs/INDELs)</td>
<td style="text-align: left;"><a href="https://github.com/google/deepvariant">DeepVariant</a></td>
<td style="text-align: left;"><a href="https://github.com/google/deepvariant">GitHub</a></td>
<td style="text-align: left;">Python; C++</td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;">WGS/WES</td>
<td style="text-align: left;">Variant calling (SNPs/INDELs)</td>
<td style="text-align: left;"><a href="https://github.com/brentp/slivar">Slivar</a></td>
<td style="text-align: left;"><a href="https://github.com/brentp/slivar">GitHub</a></td>
<td style="text-align: left;">Nim; HTML; Java</td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;">WGS/WES</td>
<td style="text-align: left;">Variant calling (CNVs)</td>
<td style="text-align: left;"><a href="https://github.com/fritzsedlazeck/parliament2">Parliament2</a></td>
<td style="text-align: left;"><a href="https://github.com/dnanexus/parliament2">GitHub</a>; <a href="https://github.com/fritzsedlazeck/parliament2">Documentation</a></td>
<td style="text-align: left;">Python; C++; C</td>
<td style="text-align: left;">33347570</td>
</tr>
<tr class="odd">
<td style="text-align: left;">WGS/WES</td>
<td style="text-align: left;">Variant calling (CNVs)</td>
<td style="text-align: left;"><a href="https://github.com/gjun/muCNV">muCNV</a></td>
<td style="text-align: left;"><a href="https://github.com/gjun/muCNV">GitHub</a></td>
<td style="text-align: left;">C++; C</td>
<td style="text-align: left;">33760063</td>
</tr>
<tr class="even">
<td style="text-align: left;">WGS/WES</td>
<td style="text-align: left;">Variant calling (CNVs)</td>
<td style="text-align: left;"><a href="https://github.com/ChaissonLab/TT-Mars">TT-Mars</a></td>
<td style="text-align: left;"><a href="https://github.com/ChaissonLab/TT-Mars">GitHub</a></td>
<td style="text-align: left;">Python</td>
<td style="text-align: left;">35524317</td>
</tr>
<tr class="odd">
<td style="text-align: left;">WGS/WES</td>
<td style="text-align: left;">Variant calling (CNVs)</td>
<td style="text-align: left;"><a href="https://github.com/ACEnglish/truvari">Truvari</a></td>
<td style="text-align: left;"><a href="https://github.com/ACEnglish/truvari">GitHub</a></td>
<td style="text-align: left;">Python</td>
<td style="text-align: left;">36575487</td>
</tr>
<tr class="even">
<td style="text-align: left;">WGS/WES</td>
<td style="text-align: left;">Variant calling (CNVs)</td>
<td style="text-align: left;"><a href="https://bitbucket.org/statgen/xhmm/src/master/">XHMM</a></td>
<td style="text-align: left;"><a href="https://bitbucket.org/statgen/xhmm/src/master/">Manual</a>; <a href="https://statgen.bitbucket.io/xhmm/">Tutorial</a></td>
<td style="text-align: left;">C++</td>
<td style="text-align: left;">24763994</td>
</tr>
<tr class="odd">
<td style="text-align: left;">WGS/WES</td>
<td style="text-align: left;">Variant calling (CNVs)</td>
<td style="text-align: left;"><a href="https://github.com/yuchaojiang/CODEX">CODEX</a></td>
<td style="text-align: left;"><a href="https://github.com/yuchaojiang/CODEX">GitHub</a></td>
<td style="text-align: left;">R</td>
<td style="text-align: left;">25618849</td>
</tr>
<tr class="even">
<td style="text-align: left;">WGS/WES</td>
<td style="text-align: left;">Variant calling (CNVs)</td>
<td style="text-align: left;"><a href="https://github.com/bigdatagenomics/deca">DECA</a></td>
<td style="text-align: left;"><a href="https://github.com/bigdatagenomics/deca">GitHub</a></td>
<td style="text-align: left;">Scala; R</td>
<td style="text-align: left;">31604420</td>
</tr>
<tr class="odd">
<td style="text-align: left;">WGS/WES</td>
<td style="text-align: left;">Variant calling (CNVs)</td>
<td style="text-align: left;"><a href="https://github.com/rgcgithub/clamms">CLAMMS</a></td>
<td style="text-align: left;"><a href="https://github.com/rgcgithub/clamms">GitHub</a></td>
<td style="text-align: left;">C</td>
<td style="text-align: left;">26382196</td>
</tr>
<tr class="even">
<td style="text-align: left;">WGS/WES</td>
<td style="text-align: left;">Variant calling (CNVs)</td>
<td style="text-align: left;"><a href="https://github.com/KarlaLG91/mk-cn.MOPS">cn.MOPS</a></td>
<td style="text-align: left;"><a href="https://github.com/KarlaLG91/mk-cn.MOPS">GitHub</a></td>
<td style="text-align: left;">R</td>
<td style="text-align: left;">22302147</td>
</tr>
<tr class="odd">
<td style="text-align: left;">WGS/WES</td>
<td style="text-align: left;">Variant calling (CNVs)</td>
<td style="text-align: left;"><a href="https://github.com/broadinstitute/gatk-sv#gcnv-training-overview">GATK-gCNV</a></td>
<td style="text-align: left;"><a href="https://github.com/broadinstitute/gatk-sv">GitHub</a>; <a href="https://github.com/broadinstitute/gatk-sv#gcnv-training-overview">Tutorial</a></td>
<td style="text-align: left;">Python; WDL</td>
<td style="text-align: left;">Babadi, et. al (2022), BioRxiv</td>
</tr>
<tr class="even">
<td style="text-align: left;">WGS/WES</td>
<td style="text-align: left;">Variant calling (CNVs)</td>
<td style="text-align: left;"><a href="https://github.com/chrisamiller/readDepth">ReadDepth</a></td>
<td style="text-align: left;"><a href="https://github.com/chrisamiller/readDepth">GitHub</a></td>
<td style="text-align: left;">R</td>
<td style="text-align: left;">21305028</td>
</tr>
<tr class="odd">
<td style="text-align: left;">WGS/WES</td>
<td style="text-align: left;">Variant calling (CNVs)</td>
<td style="text-align: left;"><a href="https://github.com/igm-team/ERDS">ERDS</a></td>
<td style="text-align: left;"><a href="https://github.com/igm-team/ERDS">GitHub</a></td>
<td style="text-align: left;">Perl; C</td>
<td style="text-align: left;">22939633</td>
</tr>
<tr class="even">
<td style="text-align: left;">WGS/WES</td>
<td style="text-align: left;">Variant annotation</td>
<td style="text-align: left;"><a href="https://gnomad.broadinstitute.org/news/2020-10-gnomad-v3-1-new-content-methods-annotations-and-data-availability/">gnomAD</a></td>
<td style="text-align: left;"><a href="https://gnomad.broadinstitute.org/news/2020-10-gnomad-v3-1-new-content-methods-annotations-and-data-availability/">Website</a></td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">32461654</td>
</tr>
<tr class="odd">
<td style="text-align: left;">WGS/WES</td>
<td style="text-align: left;">Variant annotation</td>
<td style="text-align: left;"><a href="http://pcingola.github.io/SnpEff/">SnpEff</a></td>
<td style="text-align: left;"><a href="http://pcingola.github.io/SnpEff/">GitHub</a></td>
<td style="text-align: left;">Java</td>
<td style="text-align: left;">22728672</td>
</tr>
<tr class="even">
<td style="text-align: left;">WGS/WES</td>
<td style="text-align: left;">Variant annotation</td>
<td style="text-align: left;">Hail</td>
<td style="text-align: left;"><a href="https://hail.is">Website</a></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;">WGS/WES</td>
<td style="text-align: left;">Variant annotation</td>
<td style="text-align: left;"><a href="http://pcingola.github.io/SnpEff/">SnpSift</a></td>
<td style="text-align: left;"><a href="http://pcingola.github.io/SnpEff/">GitHub</a></td>
<td style="text-align: left;">Java</td>
<td style="text-align: left;">22435069</td>
</tr>
<tr class="even">
<td style="text-align: left;">WGS/WES</td>
<td style="text-align: left;">Variant annotation</td>
<td style="text-align: left;"><a href="https://www.ncbi.nlm.nih.gov/clinvar/">ClinVar</a></td>
<td style="text-align: left;"><a href="https://www.ncbi.nlm.nih.gov/clinvar/">Website</a></td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;">WGS/WES</td>
<td style="text-align: left;">Variant annotation</td>
<td style="text-align: left;"><a href="https://www.hgmd.cf.ac.uk/ac/index.php">Human Gene Mutation Database</a> (HGMD)</td>
<td style="text-align: left;"><a href="https://www.hgmd.cf.ac.uk/ac/index.php">HGMD Website</a></td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;">WGS/WES</td>
<td style="text-align: left;">Variant annotation</td>
<td style="text-align: left;"><a href="https://www.ncbi.nlm.nih.gov/snp/">dbSNP</a></td>
<td style="text-align: left;"><a href="https://www.ncbi.nlm.nih.gov/snp/">Website</a></td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;">WGS/WES</td>
<td style="text-align: left;">Variant annotation</td>
<td style="text-align: left;"><a href="http://database.liulab.science/dbNSFP">dbNSFP</a></td>
<td style="text-align: left;"><a href="http://database.liulab.science/dbNSFP">Website</a></td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">21520341</td>
</tr>
<tr class="even">
<td style="text-align: left;">WGS/WES</td>
<td style="text-align: left;">Variant annotation</td>
<td style="text-align: left;"><a href="https://sift.bii.a-star.edu.sg/">SIFT</a></td>
<td style="text-align: left;"><a href="https://sift.bii.a-star.edu.sg/">Website</a></td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">22689647</td>
</tr>
<tr class="odd">
<td style="text-align: left;">WGS/WES</td>
<td style="text-align: left;">Variant annotation</td>
<td style="text-align: left;"><a href="http://genetics.bwh.harvard.edu/pph2/">Polyphen2</a></td>
<td style="text-align: left;"><a href="http://genetics.bwh.harvard.edu/pph2/">Website</a></td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">20354512</td>
</tr>
<tr class="even">
<td style="text-align: left;">WGS/WES</td>
<td style="text-align: left;">Variant annotation</td>
<td style="text-align: left;"><a href="https://www.mutationtaster.org/">MutationTaster</a></td>
<td style="text-align: left;"><a href="https://www.mutationtaster.org/">Website</a></td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">24681721</td>
</tr>
<tr class="odd">
<td style="text-align: left;">WGS/WES</td>
<td style="text-align: left;">Variant annotation</td>
<td style="text-align: left;"><a href="https://cadd.gs.washington.edu/">Combined Annotation Dependent Depletion</a> (CADD)</td>
<td style="text-align: left;"><a href="https://cadd.gs.washington.edu/">CADD website</a></td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">30371827</td>
</tr>
<tr class="even">
<td style="text-align: left;">WGS/WES</td>
<td style="text-align: left;">Variant annotation</td>
<td style="text-align: left;"><a href="https://bbglab.irbbarcelona.org/fannsdb/help/condel.html">CONsensus DELeteriousness</a> (CONDEL)</td>
<td style="text-align: left;"><a href="https://bbglab.irbbarcelona.org/fannsdb/help/condel.html">CONDEL website</a></td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;">WGS/WES</td>
<td style="text-align: left;">Variant annotation</td>
<td style="text-align: left;"><a href="http://mendel.stanford.edu/sidowlab/downloads/MAPP/index.html">Multivariate Analysis of Protein Polymorphism</a>(MAPP)</td>
<td style="text-align: left;"><a href="http://mendel.stanford.edu/sidowlab/downloads/MAPP/index.html">MAPP website</a></td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">15965030</td>
</tr>
<tr class="even">
<td style="text-align: left;">WGS/WES</td>
<td style="text-align: left;">Variant annotation</td>
<td style="text-align: left;"><a href="http://mutationassessor.org/r3/">MutationAssessor</a></td>
<td style="text-align: left;"><a href="http://mutationassessor.org/r3/">Website</a></td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">21727090</td>
</tr>
<tr class="odd">
<td style="text-align: left;">WGS/WES</td>
<td style="text-align: left;">Variant annotation</td>
<td style="text-align: left;"><a href="https://annovar.openbioinformatics.org/en/latest/">ANNOVAR</a></td>
<td style="text-align: left;"><a href="https://annovar.openbioinformatics.org/en/latest/">Website</a></td>
<td style="text-align: left;">Perl</td>
<td style="text-align: left;">20601685</td>
</tr>
<tr class="even">
<td style="text-align: left;">WGS/WES</td>
<td style="text-align: left;">Variant annotation</td>
<td style="text-align: left;"><a href="https://useast.ensembl.org/info/docs/tools/vep/index.html">Variant Effect Predictor</a> (VEP)</td>
<td style="text-align: left;"><a href="https://useast.ensembl.org/info/docs/tools/vep/index.html">VEP Website</a></td>
<td style="text-align: left;">Perl</td>
<td style="text-align: left;">27268795</td>
</tr>
<tr class="odd">
<td style="text-align: left;">WGS/WES</td>
<td style="text-align: left;">Variant annotation</td>
<td style="text-align: left;"><a href="https://snp.gs.washington.edu/SeattleSeqAnnotation138/">SeattleSeq</a></td>
<td style="text-align: left;"><a href="https://snp.gs.washington.edu/SeattleSeqAnnotation138/">Website</a></td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">19684571</td>
</tr>
<tr class="even">
<td style="text-align: left;">WGS/WES</td>
<td style="text-align: left;">Variant annotation</td>
<td style="text-align: left;"><a href="https://www.ncbi.nlm.nih.gov/clinvar/docs/clingen/">ClinGen</a></td>
<td style="text-align: left;"><a href="https://www.ncbi.nlm.nih.gov/clinvar/docs/clingen/">Website</a></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;">WGS/WES</td>
<td style="text-align: left;">Variant annotation</td>
<td style="text-align: left;"><a href="https://github.com/kbvstmd/XCNV">X-CNV</a></td>
<td style="text-align: left;"><a href="https://github.com/kbvstmd/XCNV">GitHub</a>; <a href="http://119.3.41.228/XCNV/index.php">Documentation</a></td>
<td style="text-align: left;">R</td>
<td style="text-align: left;">34407882</td>
</tr>
<tr class="even">
<td style="text-align: left;">WGS/WES</td>
<td style="text-align: left;">Variant annotation</td>
<td style="text-align: left;"><a href="https://lbgi.fr/AnnotSV/">AnnotSV</a></td>
<td style="text-align: left;"><a href="https://github.com/lgmgeo/AnnotSV">GitHub</a>; <a href="https://lbgi.fr/AnnotSV/">Documentation</a></td>
<td style="text-align: left;">TCL; Python</td>
<td style="text-align: left;">29669011</td>
</tr>
<tr class="odd">
<td style="text-align: left;">WGS/WES</td>
<td style="text-align: left;">Association Testing</td>
<td style="text-align: left;"><a href="https://genome.sph.umich.edu/wiki/EPACTS">EPACTS</a></td>
<td style="text-align: left;"><a href="https://github.com/statgen/EPACTS">GitHub</a>; <a href="https://genome.sph.umich.edu/wiki/EPACTS">Documentation</a></td>
<td style="text-align: left;">C++</td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;">WGS/WES</td>
<td style="text-align: left;">Association Testing</td>
<td style="text-align: left;"><a href="https://alkesgroup.broadinstitute.org/BOLT-LMM/BOLT-LMM_manual.html">BOLT-LMM</a></td>
<td style="text-align: left;"><a href="https://alkesgroup.broadinstitute.org/BOLT-LMM/downloads/">Download</a>; <a href="https://alkesgroup.broadinstitute.org/BOLT-LMM/BOLT-LMM_manual.html">Manual</a></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">25642633</td>
</tr>
<tr class="odd">
<td style="text-align: left;">WGS/WES</td>
<td style="text-align: left;">Association Testing</td>
<td style="text-align: left;"><a href="https://rgcgithub.github.io/regenie/">REGENIE</a></td>
<td style="text-align: left;"><a href="https://github.com/rgcgithub/regenie">GitHub</a>; <a href="https://rgcgithub.github.io/regenie/options/">Tutorial</a></td>
<td style="text-align: left;">C++</td>
<td style="text-align: left;">34017140</td>
</tr>
<tr class="even">
<td style="text-align: left;">WGS/WES</td>
<td style="text-align: left;">Association Testing</td>
<td style="text-align: left;"><a href="https://www.hsph.harvard.edu/skat/">SNP-Set (Sequence) Kernal Association Test</a> (SKAT)</td>
<td style="text-align: left;"><a href="chrome-extension://efaidnbmnnnibpcajpcglclefindmkaj/https://cran.r-project.org/web/packages/SKAT/SKAT.pdf">SKAT Documentation</a>; <a href="https://www.hsph.harvard.edu/skat/">Website</a></td>
<td style="text-align: left;">R</td>
<td style="text-align: left;">23684009</td>
</tr>
<tr class="odd">
<td style="text-align: left;">WGS/WES</td>
<td style="text-align: left;">Association Testing</td>
<td style="text-align: left;"><a href="https://saigegit.github.io/SAIGE-doc/">SAIGE</a></td>
<td style="text-align: left;"><a href="https://github.com/weizhouUMICH/SAIGE">GitHub</a>; <a href="https://github.com/saigegit/SAIGE">More documentation</a></td>
<td style="text-align: left;">R; C++</td>
<td style="text-align: left;">30104761</td>
</tr>
<tr class="even">
<td style="text-align: left;">WGS/WES</td>
<td style="text-align: left;">Association Testing</td>
<td style="text-align: left;"><a href="https://github.com/hanchenphd/GMMAT">Generalized linear Mixed Model Association Tests</a> (GMMAT)</td>
<td style="text-align: left;"><a href="https://github.com/hanchenphd/GMMAT">GitHub</a></td>
<td style="text-align: left;">C; R; C++</td>
<td style="text-align: left;">27018471</td>
</tr>
<tr class="odd">
<td style="text-align: left;">WGS/WES</td>
<td style="text-align: left;">Association Testing</td>
<td style="text-align: left;"><a href="https://github.com/zhanxw/rvtests">CoCoRV</a> (Rare variant tests)</td>
<td style="text-align: left;"><a href="https://github.com/zhanxw/rvtests">GitHub</a></td>
<td style="text-align: left;">C++</td>
<td style="text-align: left;">35545612</td>
</tr>
<tr class="even">
<td style="text-align: left;">WGS/WES</td>
<td style="text-align: left;">Additional tools</td>
<td style="text-align: left;"><a href="https://github.com/dnanexus-rnd/GLnexus">GLnexus</a></td>
<td style="text-align: left;"><a href="https://github.com/dnanexus-rnd/GLnexus">GitHub</a></td>
<td style="text-align: left;">C++</td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><h1 id="datasetsdata-access"><strong>Datasets/Data Access</strong></h1></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;"><a href="https://www.coriell.org/1/NHGRI/Collections/1000-Genomes-Project-Collection/1000-Genomes-Project?gclid=CjwKCAjwivemBhBhEiwAJxNWN6ibQNDxTr3NHiKE3-6O7k-Aitv4ZKFYrhYnRnPnGQa6z8y3JmP0dhoC3M0QAvD_BwE">1000 Genomes</a></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><a href="https://www.ncbi.nlm.nih.gov/gap/">dbGAP</a> (US)</td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;"><a href="https://www.finngen.fi/en">FinnGen</a></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><a href="https://gtexportal.org/home/datasets">GTEx</a></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;"><a href="https://www.ebi.ac.uk/gwas/">GWAS Catalog</a></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><a href="https://www.nimhgenetics.org/">NIMH repository</a> (US)</td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;"><a href="https://pgc.unc.edu/for-researchers/download-results/">PGC Summary Statistics</a></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><a href="https://psychencode.synapse.org/">PsychENCODE</a></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;"><a href="http://psychscreen.beta.wenglab.org/main/search">PsychSCREEN</a></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><h1 id="other-genomics-tools">Other Genomics Tools</h1></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;"><a href="ensembl.org">Ensembl</a></td>
<td style="text-align: left;">Genome browser for annotation, alignment, and other tools</td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><a href="https://useast.ensembl.org/info/data/biomart/biomart_r_package.html">BioMart R package</a></td>
<td style="text-align: left;">R tool for pulling annotation information from Ensembl and other large databases</td>
<td style="text-align: left;"></td>
<td style="text-align: left;"><a href="https://bioconductor.org/packages/release/bioc/html/biomaRt.html">Documentation</a></td>
<td style="text-align: left;">R</td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;"><a href="https://genome.ucsc.edu/cgi-bin/hgLiftOver">LiftOver</a></td>
<td style="text-align: left;">Tool for converting genome coordinates</td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><a href="https://genome.ucsc.edu/">UCSC Browser</a></td>
<td style="text-align: left;">Tools for visualizing genomic data, aligning sequences</td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;"><a href="https://www.genecards.org/">Genecards</a></td>
<td style="text-align: left;">Database for information on annotated and predicted human genes</td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><h1 id="genome-builds">Genome Builds</h1></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;"><a href="https://www.ncbi.nlm.nih.gov/datasets/genome/GCF_000001405.40/">GRCh38.p14</a></td>
<td style="text-align: left;"><p>Latest genome build from Genome Reference Consortium (GRC)</p>
<p>Build release (2022)</p></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><a href="https://www.ncbi.nlm.nih.gov/datasets/genome/GCF_000001405.26/">GRCh38/hg38</a></td>
<td style="text-align: left;">Build release (2013)</td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;"><a href="https://www.ncbi.nlm.nih.gov/datasets/genome/GCF_000001405.13/">GRCh37/hg19</a></td>
<td style="text-align: left;">Build release (2009)</td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
</tbody>
</table>



</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
          // target, if specified
          link.setAttribute("target", "_blank");
          // default icon
          link.classList.add("external");
      }
    }
});
</script>
</div> <!-- /content -->



</body></html>