<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.353">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>PGC Video Textbook - Software Tutorials: Gene Set Identification (Video Transcript)</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


<link rel="stylesheet" href="styles.css">
</head>

<body class="nav-sidebar docked nav-fixed">


<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="./index.html">
    <span class="navbar-title">PGC Video Textbook</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link active" href="./index.html" rel="" target="" aria-current="page">
 <span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./about.html" rel="" target="">
 <span class="menu-text">About</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./contact.html" rel="" target="">
 <span class="menu-text">Contact us</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="https://pgc.unc.edu/" rel="" target="">
 <span class="menu-text">PGC Website</span></a>
  </li>  
</ul>
            <div class="quarto-navbar-tools ms-auto">
</div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
      <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item">Software Tutorials: Gene Set Identification (Video Transcript)</li></ol></nav>
      <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
      </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal sidebar-navigation docked overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
      <a href="./index.html" class="sidebar-logo-link">
      <img src="./pgc_logo_website_v3.jpeg" alt="" class="sidebar-logo py-0 d-lg-inline d-none">
      </a>
      <div class="sidebar-tools-main">
    <div class="dropdown">
      <a href="" title="" id="quarto-navigation-tool-dropdown-0" class="quarto-navigation-tool dropdown-toggle px-1" data-bs-toggle="dropdown" aria-expanded="false" aria-label=""><i class="bi bi-github"></i></a>
      <ul class="dropdown-menu" aria-labelledby="quarto-navigation-tool-dropdown-0">
          <li>
            <a class="dropdown-item sidebar-tools-main-item" href="">
            Source code
            </a>
          </li>
          <li>
            <a class="dropdown-item sidebar-tools-main-item" href="">
            Report a bug
            </a>
          </li>
      </ul>
    </div>
</div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./welcome.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Welcome to the PGC Video Textbook!</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./toc.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Table of Contents</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">
 <span class="menu-text">Chapters</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter1.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 1: Introduction</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter2.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 2: The Genome</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter3.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 3: Technologies</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter4.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 4: Study designs</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter5.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 5: GWAS analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter6.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 6: Polygenic Scores</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter7.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 7: Ancestry-Specific Analyses and Considerations</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter8.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 8: Post-GWAS bioinformatics</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter9.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 9: Advanced Topics</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter10.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chapter 10: Other Considerations</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true">
 <span class="menu-text">Software Tutorials</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_cnvs.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">CNVs</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_conditional.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Conditional Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_crossdisorder.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Cross-disorder Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_ewas.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">EWAS</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_geneset.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Gene Set Identification</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_gwas.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">GWAS</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_genomicSEM.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Genomic SEM</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_imaging.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Imaging</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_MR.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Mendelian Randomization</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_prs.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">PRS</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software_correlation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">SNP Heritability and Genetic Correlation</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true">
 <span class="menu-text">Additional Resources</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./glossary.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Glossary</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./resources.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Software Resources</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./addreading.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Additional Reading</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="https://pgc.unc.edu/for-researchers/download-results/" class="sidebar-item-text sidebar-link">
 <span class="menu-text">PGC Summary Statistics</span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#sec-video1" id="toc-sec-video1" class="nav-link active" data-scroll-target="#sec-video1">MAGMA</a></li>
  <li><a href="#sec-video2" id="toc-sec-video2" class="nav-link" data-scroll-target="#sec-video2">H-MAGMA</a></li>
  <li><a href="#sec-video3" id="toc-sec-video3" class="nav-link" data-scroll-target="#sec-video3">E-MAGMA</a></li>
  <li><a href="#sec-video4" id="toc-sec-video4" class="nav-link" data-scroll-target="#sec-video4">PRSet</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Software Tutorials: Gene Set Identification (Video Transcript)</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<section id="sec-video1" class="level1">
<h1>MAGMA</h1>
<p><strong>Title</strong>: Gene- and gene-set analysis in MAGMA</p>
<p><strong>Presenter(s)</strong>: Christiaan de Leeuw</p>
<hr>
</section>
<section id="sec-video2" class="level1">
<h1>H-MAGMA</h1>
<p><strong>Title</strong>: Annotating Genetic Variants to Target Genes Using H-MAGMA</p>
<p><strong>Presenter(s)</strong>: Nancy Y.A. Sey, University of North Carolina at Chapel Hill</p>
<p>Introduction</p>
<p>hi everyone my name is Nancy and today</p>
<p>I’ll be talking to you about annotating</p>
<p>genetic variants to Target genes using</p>
<p>high C corporate magma or hmagmap for</p>
<p>short</p>
<p>Background</p>
<p>so before I get started I would just</p>
<p>like to credit doctors um Christian and</p>
<p>Danielle for developing magma Gene set</p>
<p>analysis</p>
<p>so magma is widely used in our field and</p>
<p>this is for various reasons with a few</p>
<p>being that the tool is pretty</p>
<p>revolutionary in that it’s aided in</p>
<p>making sense of guas Finance by</p>
<p>identifying the target genes of genetic</p>
<p>variants identified from guas</p>
<p>additionally magma is very easy to use</p>
<p>in that you do not have to be</p>
<p>computationally Savvy to use the tool</p>
<p>also it is very efficient compared to</p>
<p>other tools in the field so it’s it</p>
<p>takes a very short time to run magma</p>
<p>however despite all of these benefits</p>
<p>there are a few limitations to the tool</p>
<p>namely magma relies on positional</p>
<p>mapping to link variants to Target genes</p>
<p>we know from prior studies that the gene</p>
<p>regulatory landscape is complex and it’s</p>
<p>therefore possible for variants</p>
<p>especially non-coin invariants to</p>
<p>interact with and regulate the install</p>
<p>genes so additionally</p>
<p>um the regulatory landscape can be</p>
<p>tissue or cell type specific so we have</p>
<p>to take into account the tissue or cell</p>
<p>type that’s most relevant to the trait</p>
<p>or disease that we’re interested in</p>
<p>understanding</p>
<p>so these two limitations served as the</p>
<p>foundation for developing High c-coupled</p>
<p>magma or H magma for short</p>
<p>so H magma complements magma by using</p>
<p>high C data</p>
<p>Materials</p>
<p>so as mentioned previously um H magma</p>
<p>complements magma so the materials are</p>
<p>needed to run H magma are the same as</p>
<p>the ones needed to run magma where the</p>
<p>only difference being the gene2 variant</p>
<p>annotation file so in this tutorial I</p>
<p>will walk you through how to generate</p>
<p>The hmac muscativic annotation file</p>
<p>which links non-coding um variants to</p>
<p>their target genes using high C</p>
<p>interaction</p>
<p>so again I will um just walk you through</p>
<p>how to generate The annotation file for</p>
<p>hmagma</p>
<p>Required files</p>
<p>so the materials needed to run hmagma</p>
<p>are similar to the ones needed to run</p>
<p>magma with the only difference being The</p>
<p>annotation file so to generate the H</p>
<p>magma annotation file you need a few</p>
<p>other files namely you need bed files</p>
<p>for Gene Exxon and promoter coordinates</p>
<p>Additionally you need a high C data set</p>
<p>in the bedp file format you will also</p>
<p>need an ancestry reference genome and</p>
<p>for the sake of this protocol we’ll be</p>
<p>using the European ancestry</p>
<p>and once you have The annotation file</p>
<p>generated you need a Gua summary</p>
<p>statistics to run hmagmap</p>
<p>so again um the focus of this tutorial</p>
<p>will be on Steps A and B which walks you</p>
<p>through how to generate the hmag margin</p>
<p>to variant annotation file</p>
<p>Step 1 Load libraries and required data</p>
<p>so I have broken The annotation files</p>
<p>into seven different steps with code for</p>
<p>each step provided as part of this</p>
<p>recording</p>
<p>so the first step is to load your</p>
<p>libraries and all required data into a</p>
<p>work directory</p>
<p>the next step is to read in your Exon</p>
<p>and promoter batch files and create</p>
<p>genomic range objects for the Exxon and</p>
<p>promoter using the Exxon and promoter</p>
<p>coordinates from gen code version 26.</p>
<p>once you have this loaded and the</p>
<p>g-range object generated you will save</p>
<p>then as an R file</p>
<p>so similarly um you would generate a</p>
<p>genomic range object for all Snips using</p>
<p>the reference genome from the European</p>
<p>ancestry and you will save you also save</p>
<p>this as an R file in the work directory</p>
<p>for later use</p>
<p>Step 2 Overlap Snips to Gene</p>
<p>so in the following step um we will</p>
<p>overlap Snips to Gene starting with the</p>
<p>Exotic snips</p>
<p>so as a reminder the Exotic Snips are</p>
<p>assigned to your target genes using</p>
<p>positional mapping and this is because</p>
<p>they are more likely to impact the genes</p>
<p>in which they reside</p>
<p>so to assign the Exotic Snips to your</p>
<p>target genes we will overlap the gene</p>
<p>range for objects for Exon with</p>
<p>engineering objects for Snips that was</p>
<p>created in steps two and three</p>
<p>respectively</p>
<p>so similarly we will also overlap the</p>
<p>g-range object for promoters with the</p>
<p>g-range object that was created for the</p>
<p>snips</p>
<p>so once you have both g-range objects</p>
<p>created for the exons and the promoters</p>
<p>um these will be saved as an R file for</p>
<p>later use</p>
<p>Intergenic Snips</p>
<p>foreign so after um after we’ve assigned</p>
<p>the Exotic and promoter snaps to their</p>
<p>target genes there will be a subset of</p>
<p>snips that do not overlap with either so</p>
<p>these are known as the intergenic or the</p>
<p>intronic Snips so in the following steps</p>
<p>I will walk you through how to match</p>
<p>those Snips to your target jeans using</p>
<p>high C interaction and this is shown</p>
<p>here in this diagram</p>
<p>so to achieve this we will first</p>
<p>identify those in intergenic and</p>
<p>intronic Snips and save them as an R</p>
<p>file</p>
<p>then we will load our high C data so for</p>
<p>the sake of this tutorial we’ll be using</p>
<p>high C data from the adult brain as an</p>
<p>example</p>
<p>so once you load the high C data in the</p>
<p>bedpie format you will generate a</p>
<p>g-range object for it</p>
<p>so we will next find snip.physically</p>
<p>interact with excellence and with these</p>
<p>set of codes similarly we will also find</p>
<p>snip that physically interact with</p>
<p>promoter with the promoter regions</p>
<p>so next we will combine those g-range</p>
<p>outlets for the exons and the promoters</p>
<p>to retrieve a unique um with to retrieve</p>
<p>unique items</p>
<p>so lastly we will overlap the intergenic</p>
<p>and the intronic Snips with high C to</p>
<p>identify their target genes</p>
<p>so once we have all these Snips um</p>
<p>namely the exonic the promoter the</p>
<p>intergenic and the intronic Snips map to</p>
<p>your target genes</p>
<p>um the last few codes that I will show</p>
<p>you in Step seven will be used to</p>
<p>generate The annotation file that is</p>
<p>compatible with Magma</p>
<p>so to achieve this we will structure The</p>
<p>annotation file so that it has the genes</p>
<p>in The Ensemble format the chromosomal</p>
<p>location and followed by the Snips that</p>
<p>is assigned to each particular Gene</p>
<p>so this is how you you will have um so</p>
<p>following these seven steps you will</p>
<p>have your hmac vibrant to Gene</p>
<p>annotation file that can be used to run</p>
<p>each magma for any trait or disease</p>
<p>so with this um you can run hmadmav for</p>
<p>any trade um so for here I will I’ll use</p>
<p>an example using p um Parkinson’s</p>
<p>disorder as an example</p>
<p>so this is the portion of the magma um</p>
<p>file the magma code that you place your</p>
<p>hmag margin to variant annotation file</p>
<p>Output files</p>
<p>so here I am showing you one of the</p>
<p>output files that you get from running</p>
<p>hmagma so this is the dot genes.out file</p>
<p>so with this file um you can actually</p>
<p>extract the significant genes at various</p>
<p>FDR thresholds for Downstream analysis</p>
<p>Limitations</p>
<p>so as I mentioned previously in the</p>
<p>beginning of this tutorial magma has</p>
<p>transformed the field of psychiatric</p>
<p>genetics and with each magma we can</p>
<p>address some of its limitations</p>
<p>however it is important to note that</p>
<p>there are still some pendant limitations</p>
<p>that aren’t addressed using H magma</p>
<p>namely the directionality of effect so</p>
<p>even though H magma can assign Snips to</p>
<p>their target genes I’m using higher C</p>
<p>data we do not know whether the Snips</p>
<p>the variants are upregulating the gene</p>
<p>or they are doing regular down</p>
<p>regulating the gene so to address this</p>
<p>you can use various eqtl data sets to</p>
<p>to analyze the directionality of effect</p>
<p>additionally we know that not all not</p>
<p>all Snips are actually implicated in a</p>
<p>trait or disease so you need to do</p>
<p>further functional validation to further</p>
<p>prune down the list of genes</p>
<p>and then we also know that the sample</p>
<p>size of a g was greatly impacts the</p>
<p>number of Target genes that is</p>
<p>identifiable so this H magma also</p>
<p>suffers from this</p>
<p>and lastly</p>
<p>um lack of diversity in genetic studies</p>
<p>most of the analysis that we’ve done</p>
<p>with h magma has mostly been from</p>
<p>European ancestry however it’s important</p>
<p>to note that you can also run H magma</p>
<p>for other ancestries</p>
<p>Conclusion</p>
<p>so with that um in the last slide here I</p>
<p>would just like to acknowledge where we</p>
<p>currently are with h magma so previously</p>
<p>we have generated H magma for brain</p>
<p>tissues including adult and Fetal brain</p>
<p>we have also added various salty</p>
<p>specific annotation files including</p>
<p>cortical</p>
<p>ipsc derived neurons and ipsc derived</p>
<p>astrocytes as well as midbrain</p>
<p>endopliminergic neurons however H magma</p>
<p>is more versatile than just brain</p>
<p>related cells so to expand the two we</p>
<p>have created additional annotation files</p>
<p>for 28 different cells and tissue types</p>
<p>to allow other researchers to use the</p>
<p>tool</p>
<p>and these cell types um are listed here</p>
<p>and</p>
<p>most of all of these annotations files</p>
<p>can be drum can be derived from our</p>
<p>GitHub page</p>
<p>and with that I would just like to thank</p>
<p>my lab as well as my funding sources</p>
<p>additionally I’ll be hosting um a q a</p>
<p>section</p>
<p>um on October 15 2021 at 3 45 PM Eastern</p>
<p>Standard Time so if you have any</p>
<p>questions or you would just like to say</p>
<p>hi please drop by to see um and ask any</p>
<p>questions thank you</p>
<hr>
</section>
<section id="sec-video3" class="level1">
<h1>E-MAGMA</h1>
<p><strong>Title</strong>: E-MAGMA: an eQTL-informed method to identify risk genes using genome-wide association study summary statistics</p>
<p><strong>Presenter(s)</strong>: Zac Gerring, Eske Derks</p>
<p>thank you for attending my talk entitled e-magma an eqtl-informed method to identify risk genes using genome-wide association studies summary statistics magma was initially developed to extract biological insights from g-was by linking risk variants to their nearby genes</p>
<p>the method assigns single nucleotide polymorphism associations to gene level associations while correcting the confounding factors such as gene length minor allele frequency and gene density while magma is a reliable and commonly used tool there is room for alternative approaches for how snips are assigned to genes for example conventional magma assigns snips to the nearest gene which is not always the most accurate approach non-coding snips can affect the expression of distal genes known as expression quantitative trait loci eqtr magma or heat magma is a modified version of conventional magma that leverages tissue specific eqtl information to assign snips to genes the genotype tissue expression project or gtex is an ongoing effort to build a public resource to study tissue-specific gene expression and regulation the current version of gtex contains samples collected from 53 non-disease tissue sites across nearly 1 000 individuals including 13 brain tissues from around 200 individuals</p>
<p>this diagram gives an example of how gtex can be used for the functional interpretation of genome-wide association signals the expression quantitative quantitative trait losse annotation from various tissues can be used to propose one or more potential causal genes whose regulation is either tissue shared shown in green or tissue specific shown in yellow for a</p>
<p>trait associated variant these associations would not be identified assigning snips to genes based on proximity alone e magma largely adopts the magma pipeline which consists of three broad steps gene annotation to assign snips to genes gene analysis to compute gene-based p-values and gene-level analysis to test the enrichment of gene-based results in curated gene sets e-magma only modifies the annotation step this is achieved by mapping significant eqtls from gtex to nearby genes in a tissue-specific manner rather than a single annotation file containing the mappings of snipster genes e-magma uses 48 annotation files one for each tissue and gtex after running gene annotation e-magman generates all intermediary files required for the gene level analysis</p>
<p>this slide shows the basic code for each step of conventional magma with alterations for e-magma outlined by red boxes in the gene annotation step the snip location file is altered to include significant eqtls for one of 48 tissues in gtx the tissue-specific eqtail annotation files are used in the gene analysis to generate gene-based p-values in addition to generating gene-based p-values we also generate intermediary or raw files for the gene set and biological pathway analysis</p>
<p>we tested the performance of e magma against other gene-based approaches including conventional magma aspertic scan fusion and summary based mendelian randomization using a simulation experiment the simulations use snip genotype information from the qimr adult twin study we excluded non-founders snips with more than one percent missingness and snips with a minor alarm frequency of less than 0.05 and we only analyze snips on chromosome this resulted in 7138nbc with around 60 000 snips eqt information was derived from gtx whole blood which included some 650 000 eqtl gene combinations for around 8200 genes phenotypes were simulated using gcta using gender type and eqtl reference data from chromosome 1. only genes with at least one significant ukchel were included in the analysis giving 651 genes and we performed 10 simulations per gene gbos on the 6510 generated phenotypes were performed using plink and we corrected the results for the number of genes in the eqtl reference set</p>
<p>we first evaluated the type 1 error rate across methods by calculating the proportion of genes that were significant in the absence of a true association between eqtls and phenotypic values the figure on the left shows all methods had good control of the type 1 error rate we subsequently evaluated statistical power to detect associations at a gene level for varying levels of phenotypic variants explained by uqtls we assess the proportion of significant associations relative to both the total number of causal genes in figure a and when accounting for the total number of causal genes included in each method shown in figure b e magmat have performed all methods across different proportions of variants explained by gene expression after correcting for the number of genes included in each gene based method e magma is still outperformed the other approaches</p>
<p>we estimated statistical power as a function of the number of eqtls per gene with one percent phenotypic variants explained by eqtls power significantly increased with the number of eqtls per gene as shown in this figure [Music] it should be noted that there was a significant association between the number of ekgles per gene and statistical power for all methods however e magma was less sensitive to the number of eqtls compared to the other approaches</p>
<p>we compared three gene-based methods e magma conventional magma and the t-was approach aspertic scan using g-wasps for major depressive disorder shown in the red circle identified 137 genes e magma the results of which are shown on the green circle identified 99 genes and desperadic scan shown in blue identified 57 genes a total of 16 genes were identified across all three methods the figure on the right shows aspertic scan minus log term p values on the y axis and e magma p-values on the x-axis with the color of the points indicating the tissue for which the gene-based dissociation was found as you can see there is good overlap between e-magma and esperidic scan and the overlap included established risk genes such as any gr-1 and team m106b</p>
<p>in conclusion e-magnet is an equity home formed extension to conventional magma and can be applied to any trait with g1 summary statistics e-magma map maintains appropriate controls with type 1 error rate while outperforming other methods in detecting causal associations a tutorial and input files can be found using the following github repository thank you</p>
<hr>
</section>
<section id="sec-video4" class="level1">
<h1>PRSet</h1>
<p><strong>Title</strong>: How to run pathway specific Polygenic Risk Scores</p>
<p><strong>Presenter(s)</strong>: Judit García-González</p>
<p>Hello, my name is Judit. I’m a doctoral fellow at the iconic school of medicine at Mount Sinai today I will walk you through how to run pathway specific polygeneries course a tool that has been recently developed in the polygenic lab either by Paul O’Reilly so first of all uh it’s important to know what are and why it’s interesting to use pathway polygenic risk scores for that I’m going to use one example that most of you might be familiar with which is the ability to give us for the schizophrenia</p>
<p>so we know that psychiatric disorders like schizophrenia have complex ideologies where different environmental and genetic factors contribute to the liability of these sales when we talk about genetic factors we can imagine that the US is a composite of signals where each signal might represent functional roots to disease for example uh there might be some genomic reasons that Harbor genetic degrees associated with abnormal synaptic pruning there might be other regions that are associated with biological Pathways and processes related to immune activation and other regions in the genome that pick up signal associated with cannabis consumption</p>
<p>so the idea behind by the way a specific polygenic scores is that instead of aggregating effects across the entire genome we will aggregate them after those relevant pathways and because we are separating the genetic contribution of a trade accounting for the genomics and structure we hope that pathway polygenic scores might be more useful for patient stratification or to investigate the disease heterogeneity because now one single individual instead of having one single polygenic score will have as many exports as Pathways we investigate</p>
<p>so to calculate these pathway scores uh I will walk you through the tool reset which is an expansion of the precise software uh which uses a glamping and p-value responding method um and it’s important to know that the pathways that can be any type of Gene set that reflects the encoding of different biological functions so this tool is quite flexible to Define pathways uh preset as well as the newest version of precise have been developed by Dr Sam Choi and you can find a software on the manual in this website precise.info and they’re in the website you will find a quick star how to download it and also some more detail guy about what are the available commands and what is the method how the method works</p>
<p>so how to use preset um percent is very similar to precise in terms of use where we need as an input the Dual summary statistics and individual level data for genotype and phenotype but four percent uh we also require information about the pathways or they didn’t sets that we want to use um as I say the two the tool is quite flexible uh Pathways can be defined in a variety of ways including but always defined by existing canonical databases like Gene Anthology or reactone but also by experimental perturbations or some functional outputs like Ginkgo expression or protein protein interactions</p>
<p>so I’ll go in a bit more detail on the different input options that are available to Define pathways One is using EMT and gdf files using the commands mcdv and DDF the commands that are for preset will be shown in green um so MC will contain information related to pathways so that is the genes that form each pathway and you will need a file where each pathway is specified in a different row with all the units that compose that pathway file will contain the position coordinates for each gene so in general GTA files contain different features and by default preset uses the features Exon gain protein coding or CDs but this can be modified with the command feature the second option that can be used is a bad file in this case you will need to include one bad fiber pathway so if you want to include multiple Pathways then I bet files can be included separated by a comma uh in the web file three columns are required and those are chromosome name start position and end position one important thing to know when using that files is that they are indexed uh differently as being files so whereas for Blink files are one base the web files are zero base</p>
<p>so you can see in this example that we have a sequence of nucleotides I’m gonna ask for the one base indexing one nucleotide is uh one position the zero base indexing will use a range around each in nucleotide so you will need to subtract uh -1 in the zero base indexing for that values the input option 3 is an Smith list so similar as debit files each pathway will be a different file and they can be separated by comma as in this example here so when you download precise for preset there is Android data that you can use and this is the the commands that you can use like you define the keywords uh your target sample in this case I’m using the first uh option to define the pathways then you can set the number of permutations that you want to run and also the name of the output</p>
<p>so after walking you through the input uh we can see a step-by-step how preset calculates the pathway polygenic scores to do that I will compare it with the genome white parents so whereas for the genome-wide PLS you clamp in mp value stressful in for a given address for a linkage equilibrium for pi vprs this will be done for each pathway separately then similarly with us for the genome ybrs each individual will have only one polygenic score for pathway PRS each individual will have KS course 4K number of pathways and for the results whereas for genome-wide PRS uh what is reported is the association method which is like the phenotype variance explained by the genome-wide PRS in the case of the pathways course there will be the same the association of phenotype variances explained by each pathway PLS but you can also report enrichment of geost signal in the pathway which is based on the RS versus the pathways</p>
<p>so I’ll go a bit in more detail about these two different methods that preset can output so when we are talking about Association what preset will output is the self-contained p-value so it’s just the regression of the phenotype on pathway PRS and covariates so um Pathways that have many genes containing a Snips that are associated with the phenotype are highlighted here in pink and those will be significant but it’s important to note that for these results the self-contained p-value does not account for pathway sites and for example Avid pathway will be more likely to be significant because it’s easier to have a larger number of snips associated with the phenotype but the second output that we can have is this enrichment uh that will be output as competitive p-value this is resulting from a permutation procedure and this will test whether a pathway is more associated with the phenotype compared to null Pathways that have the same size so in this case with the competitive P value we’ll be accounting for uh the size of each pathway</p>
<p>uh in this plot I’ll illustrate how we calculate this competitive p-value that accounts for uh the pathway size we can imagine a pathway a that has some Snips in the unit region of the pathway and for which we will calculate an observed the value then we will calculate as many null Pathways as permutations we run and these null Pathways will have the same number of false clampede Snips as pathway a but they will have been this isn’t it will be randomly allocated so we will obtain a distribution of null P values because it’s power will give us a value for Association and the competitive p-value will be defined as the number of tests under the null that have a p-value is smaller than our observe p-value plus one divided by the number of tests performed plus one</p>
<p>so this is how the competitive competitive P values are calculated this approach and this output is similar to the type of enrichment analysis that are performed by magma and LDS Corporation but pathway PLS have the advantage that they provide individual level data that can be used for other applications</p>
<p>so let’s see how the output of preset looks like similar to precise for every preset run we will have a log file with extension.log and this will contain the commands used for the analysis and all the information regarding the filtering the fields that were selected Etc then we will have another file with extension precise and this will contain the polygenic model for each pathway across thresholds so in the in this case I was running only address for one which contains all these names but you can see that um there is information for each pathway the R square the self-contained p-value coefficient for the regression standard error a number of its names you will always see this base pathway which is the background uh pathway that will is used for the competitive p-value calculation then we have um the best file which will give us the scores for each individual and each pathway uh so here each row will be one individual and then the score for each pathway will be indicated in a separate column</p>
<p>so there will be as many columns as Pathways you are using in this case we see four examples from the care database if you wanted all the scores at all the people in response because this best file will report only the one with the best people in threshold uh you can use the command code finally we will have another file with extension summary that will have the information of the best model fit of each phenotype um pathway and in this case each pathway will be a row and then it will contain again information on the R square Ruby value coefficient and also this competitive p-value that I was talking about that will indicate the enrichment of this pathway compared to other Pathways of similar science</p>
<p>finally I would like to leave some heads up and useful commands um so for example some commands that might be useful for the user is the window size uh which is like for example if you are defining um a Snips that are within a game you can extend that window and so you can include some number and bases to the three prime region of the ranges region or to the five Prime region of his skin you can also exclude within a certain range with the common X range so for example the MHC complex or the Apple region if you want to do some Alzheimer’s analysis and you want to Absolute that one</p>
<p>if you want to improve the computational efficiency or preset you can parallelize the process using the thread command you can also speed up a little bit uh preset um at the expense of increase in the memory using um the ultra command and um you can also use the keep and the extract commands uh to exclude individuals or its names from the analysis if you are doing permutations to calculate the competitive P values this can be computational intensive so it might be useful to adjust for the covariance like sex or age beforehand on your phenotype and then use the receivers as the phenotype because this will speed up the process</p>
<p>so this was this quick introduction to preset uh thank you for listening if there is anything that is not here or you have any questions uh there is a q a session uh the 15th of October at 11 30 and this is the soon meeting ID so I’ll be happy to clarify or answer any questions thank you</p>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
          // target, if specified
          link.setAttribute("target", "_blank");
          // default icon
          link.classList.add("external");
      }
    }
});
</script>
</div> <!-- /content -->



</body></html>